<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(3)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(3) > ul > li:nth-child(3) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(3) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(3) > ul > li:nth-child(3) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(3) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(3) > ul.nav-list > li.nav-list-item:nth-child(3) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Mapping reads with HISAT2 for RNA-seq | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Mapping reads with HISAT2 for RNA-seq" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/DataProcess/HISAT2_MAUNAL/" /><meta property="og:url" content="http://localhost:4000/docs/DataProcess/HISAT2_MAUNAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Mapping reads with HISAT2 for RNA-seq" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Mapping reads with HISAT2 for RNA-seq","url":"http://localhost:4000/docs/DataProcess/HISAT2_MAUNAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/DataProcess">Data Process</a><li class="breadcrumb-nav-list-item"><span>Mapping reads with HISAT2 for RNA-seq</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="alignment-with-reference-genome-using-hisat2"> <a href="#alignment-with-reference-genome-using-hisat2" class="anchor-heading" aria-labelledby="alignment-with-reference-genome-using-hisat2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Alignment with Reference Genome using HISAT2</h1><p class="fs-4 fw-500">A first key step in RNA-seq is to align short reads to a reference genome. Several mappers have been developed according to various sample types and experimental conditions. HiSAT2 that is coded in python and a spcialized algorithm for transcriptome analysis can be fast and exactly maapped to a reference genome for whole-genome, transcriptome, and exome sequencing data. HiSAT2 can run on any computer installed on Linux or macOS and operates on python version &gt; 2.6. As the alignment process takes a very long time and spends a lot of computational resources, the minimum spec of hardware is relatively higher than other bioinformatic tools (threads &gt; 4, memory &gt; 16 Gb ). This protocol was created based on <span style="color: #7253ec;"><b>HISAT version 2.2.1</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>Python version 3.8.5, SciPy version 1.6.2, NumPy version 1.20.1, and pySam version 0.16.0.1 under macOS 12.4 environment.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="installation-hisat2"> <a href="#installation-hisat2" class="anchor-heading" aria-labelledby="installation-hisat2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation HISAT2</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <a href="https://github.com/DaehwanKimLab/hisat2"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="https://daehwankimlab.github.io/hisat2/"><img src="https://img.shields.io/badge/GitHub.io-181717?style=flat&amp;logo=github&amp;logoColor=yellow" alt="githio" /></a> <a href="https://github.com/BenLangmead/bowtie2/releases"><img src="https://img.shields.io/github/v/release/DaehwanKimLab/hisat2?labelColor=black" alt="GitHub release" /></a> <img src="https://img.shields.io/badge/status-stable-DarkSeaGreen?labelColor=black" alt="Status" /></p><p>To install HISAT2 via anaconda, use the following commands:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>conda <span class="nb">install</span> <span class="nt">-c</span> bioconda hisat2
<span class="c"># OR</span>
<span class="nv">$ </span>conda <span class="nb">install</span> <span class="nt">-c</span> bioconda/label/cf201901 hisat2
</code></pre></div></div><h2 class="fs-5 fw-400 lh-tight" id="establish-genome-builder"> <a href="#establish-genome-builder" class="anchor-heading" aria-labelledby="establish-genome-builder"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Establish Genome Builder</u></h2><ol><li>Download desired <strong>index files</strong> from official HISAT2 site.<li>Uncompress it and move desired folder.</ol><h2 class="fs-5 fw-400 lh-tight" id="prepare-genome-builder-using-hisat2-build-if-not-available-on-suitable-index-file"> <a href="#prepare-genome-builder-using-hisat2-build-if-not-available-on-suitable-index-file" class="anchor-heading" aria-labelledby="prepare-genome-builder-using-hisat2-build-if-not-available-on-suitable-index-file"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Prepare Genome Builder using `hisat2-build` if not available on suitable index file</u></h2><p><code class="language-plaintext highlighter-rouge">hisat2-build</code> builds a HISAT2 index from a set of DNA sequences. hisat2-build outputs a set of 6 files with suffixes <i>.1.ht2, .2.ht2, .3.ht2, .4.ht2, .5.ht2, .6.ht2, .7.ht2, and .8.ht2.</i> Use the following command to prepare the genome builder with <code class="language-plaintext highlighter-rouge">hisat2-build</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>hisat2-build &lt;genome sequence.fa&gt; &lt;output folder&gt;
</code></pre></div></div><p class="note">The genome sequence file downloads the fa format of the <code class="language-plaintext highlighter-rouge">toplevel</code> (e.g. species.version.dna.toplevel.fa) provided by the <strong>ENSEMBL</strong>.</p><ul><li>Here is an example command to perform hisat2-build on the reference file:<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>hisat2-build /Users/jchoi/Desktop/Build/Danio_rerio.GRCz11.dna.toplevel.fa <span class="se">\</span>
      /Users/jchoi/Desktop/Build/Danio_rerio_index
</code></pre></div></div><li><p>When hisat2-build finished typing, it prints messages processing genome builder file look like this:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>      Settings:
          Output files: "/Users/jchoi/Desktop/Build/Danio_rerio_index.*.ht2l"
          Line rate: 7 (line is 128 bytes)
          Lines per side: 1 (side is 128 bytes)
          Offset rate: 4 (one in 16)
          FTable chars: 10
          Strings: unpacked
          Local offset rate: 3 (one in 8)
          Local fTable chars: 6
          Local sequence length: 57344
          Local sequence overlap between two consecutive indexes: 1024
          Endianness: little
          Actual local endianness: little
          Sanity checking: disabled
          Assertions: disabled
          Random seed: 0
          Sizeofs: void*:8, int:4, long:8, size_t:8
      Input files DNA, FASTA:
          /Users/jchoi/Desktop/Build/Danio_rerio.GRCz11.dna.toplevel.fa
      Reading reference sizes
          Time reading reference sizes: 00:02:37
      Calculating joined length
      Writing header
      Reserving space for joined string
      Joining reference sequences
          Time to join reference sequences: 00:03:47
          Time to read SNPs and splice sites: 00:00:00
      Using parameters --bmax 313970597 --dcv 1024
      Doing ahead-of-time memory usage test
      Passed!  Constructing with these parameters: --bmax 313970597 --dcv 1024
      Constructing suffix-array element generator
      Building DifferenceCoverSample
          Building sPrime
          Building sPrimeOrder
          V-Sorting samples
          V-Sorting samples time: 00:00:35
          Allocating rank array
          Ranking v-sort output
          Ranking v-sort output time: 00:00:07
          Invoking Larsson-Sadakane on ranks
          Invoking Larsson-Sadakane on ranks time: 00:00:14
          Sanity-checking and returning
      Building samples
      Reserving space for 12 sample suffixes
      Generating random suffixes
      QSorting 12 sample offsets, eliminating duplicates
      QSorting sample offsets, eliminating duplicates time: 00:00:00
      Multikey QSorting 12 samples
          (Using difference cover)
          Multikey QSorting samples time: 00:00:00
      Calculating bucket sizes
      Splitting and merging
          Splitting and merging time: 00:00:00
      Split 1, merged 6; iterating...
      Splitting and merging
          Splitting and merging time: 00:00:00
      Split 1, merged 0; iterating...
      Splitting and merging
          Splitting and merging time: 00:00:00
      Avg bucket size: 2.09314e+08 (target: 313970596)
      Converting suffix-array elements to index image
      Allocating ftab, absorbFtab
      Entering GFM loop
      Getting block 1 of 8
          Reserving size (313970597) for bucket 1
          Calculating Z arrays for bucket 1
          Entering block accumulator loop for bucket 1:
          bucket 1: 10%
          bucket 1: 20%
          bucket 1: 30%
          bucket 1: 40%
          bucket 1: 50%
          bucket 1: 60%
          bucket 1: 70%
          bucket 1: 80%
          bucket 1: 90%
          bucket 1: 100%
          Sorting block of length 194086053 for bucket 1
          (Using difference cover)
          Sorting block time: 00:01:59
      Returning block of 194086054 for bucket 1
      ...............
      Returning block of 197484859 for bucket 8
      Exited GFM loop
      fchr[A]: 0
      fchr[C]: 531052068
      fchr[G]: 837464927
      fchr[T]: 1143883918
      fchr[$]: 1674509851
      Exiting GFM::buildToDisk()
      Returning from initFromVector
      Wrote 567251040 bytes to primary GFM file: /Users/jchoi/Desktop/Build/Danio_rerio_index.1.ht2l
      Wrote 837254932 bytes to secondary GFM file: /Users/jchoi/Desktop/Build/Danio_rerio_index.2.ht2l
      Re-opening _in1 and _in2 as input streams
      Returning from GFM constructor
      Returning from initFromVector
      Wrote 777489661 bytes to primary GFM file: /Users/jchoi/Desktop/Build/Danio_rerio_index.5.ht2l
      Wrote 426205032 bytes to secondary GFM file: /Users/jchoi/Desktop/Build/Danio_rerio_index.6.ht2l
      Re-opening _in5 and _in5 as input streams
      Returning from HGFM constructor
      Headers:
          len: 1674509851
          gbwtLen: 1674509852
          nodes: 1674509852
          sz: 418627463
          gbwtSz: 418627464
          lineRate: 7
          offRate: 4
          offMask: 0xfffffffffffffff0
          ftabChars: 10
          eftabLen: 0
          eftabSz: 0
          ftabLen: 1048577
          ftabSz: 8388616
          offsLen: 104656866
          offsSz: 837254928
          lineSz: 128
          sideSz: 128
          sideGbwtSz: 96
          sideGbwtLen: 384
          numSides: 4360703
          numLines: 4360703
          gbwtTotLen: 558169984
          gbwtTotSz: 558169984
          reverse: 0
          linearFM: Yes
      Total time for call to driver() for forward index: 00:35:21
</code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="running-hisat2"> <a href="#running-hisat2" class="anchor-heading" aria-labelledby="running-hisat2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running HISAT2</h1><p>Use the following command to perform mapping to the genome with <code class="language-plaintext highlighter-rouge">hisat2</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>hisat2 <span class="nt">-x</span> &lt;GenoemeBuilder&gt; <span class="nt">-1</span> &lt;forward&gt; <span class="nt">-2</span> &lt;reverse&gt; <span class="nt">-S</span> &lt;output.sam&gt; &lt;OptionalParameters&gt;
</code></pre></div></div><p>In these commands,</p><ul><li><strong><code class="language-plaintext highlighter-rouge">-x &lt;GenoemeBuilder&gt;</code></strong>: Specifies the genome index. Since index files consist of <code class="language-plaintext highlighter-rouge">genome.X.ht2</code>, the builder command must be typed as follows; <code class="language-plaintext highlighter-rouge">/genome/builder/path/genome</code><li><strong><code class="language-plaintext highlighter-rouge">-1, -2</code></strong>: Specifies the forward (<code class="language-plaintext highlighter-rouge">-1</code>) and reverse (<code class="language-plaintext highlighter-rouge">-2</code>) input files. Support gz commpressed fastq (fq) files.<li><code class="language-plaintext highlighter-rouge">-S &lt;output.sam&gt;</code>: Specifies the output file type and name. <br /><li><p>The <code class="language-plaintext highlighter-rouge">&lt;OptionalParameters&gt;</code> offer a wide range of variables as bellow, see the HISAT2’s manual for details.</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-5/--trim5 &lt;int</code>, <code class="language-plaintext highlighter-rouge">-3/--trim3 &lt;int&gt;</code><td>Trim <int> bases from 5' (left) or 3' (right) end of each read before alignment (default: 0).</int><tr><td><code class="language-plaintext highlighter-rouge">--mp MX,MN</code><td>Sets the maximum (MX) and minimum (MN) mismatch penalties, both integers.<tr><td><code class="language-plaintext highlighter-rouge">--sp MX,MN</code><td>Sets the maximum (MX) and minimum (MN) penalties for soft-clipping per base, both integers.<tr><td><code class="language-plaintext highlighter-rouge">--np &lt;int&gt;</code><td>Sets penalty for positions where the read, reference, or both, contain an ambiguous character such as N.<tr><td><code class="language-plaintext highlighter-rouge">--dta-cufflinks</code> option<td>Report alignments tailored specifically for Cufflinks. In addition to what HISAT2 does with the above option (–dta), With this option, HISAT2 looks for novel splice sites with three signals (GT/AG, GC/AG, AT/AC), but all user-provided splice sites are used irrespective of their signals.<tr><td><code class="language-plaintext highlighter-rouge">-k &lt;int&gt;</code><td>It searches for at most <int> distinct, primary alignments for each read. Primary alignments mean alignments whose alignment score is equal or higher than any other alignments (Default: 5).</int><tr><td><code class="language-plaintext highlighter-rouge">-p &lt;int&gt;</code><td>The number of CPU threads HISAT program will use when executing multi-processing tasks.</table></div></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code"> <a href="#example-code" class="anchor-heading" aria-labelledby="example-code"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code</u></h2><p>Here is an example command to perform alignment with the human hg19 genome on trimmed fastq files:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>hisat2 <span class="nt">-x</span> /Users/jchoi/Desktop/RNA-seq/GenomeIndex/hg19_HiSat2Builder/genome <span class="se">\</span>
    <span class="nt">-1</span> /Users/jchoi/Desktop/Trim/Trim_shCon_H2O2_1.fq.gz <span class="se">\</span>
    <span class="nt">-2</span> /Users/jchoi/Desktop/Trim/Trim_shCon_H2O2_2.fq.gz <span class="se">\</span>
    <span class="nt">-S</span> /Users/jchoi/Desktop/SAM/shCon_H2O2.sam <span class="nt">-p</span> 10 <span class="nt">--dta-cufflinks</span> <span class="nt">-k</span> 5
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h2 class="fs-5 fw-400 lh-tight" id="output"> <a href="#output" class="anchor-heading" aria-labelledby="output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Output</u></h2><p>When HISAT2 finishes running, it prints messages summarizing what happened.</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>24403217 reads; of these:
  24403217 (100.00%) were paired; of these:
    3213764 (13.17%) aligned concordantly 0 times
    18006379 (73.79%) aligned concordantly exactly 1 time
    3183074 (13.04%) aligned concordantly &gt;1 times
    ----
    3213764 pairs aligned concordantly 0 times; of these:
      316476 (9.85%) aligned discordantly 1 time
    ----
    2897288 pairs aligned 0 times concordantly or discordantly; of these:
      5794576 mates make up the pairs; of these:
        3288372 (56.75%) aligned 0 times
        1992820 (34.39%) aligned exactly 1 time
        513384 (8.86%) aligned &gt;1 times
93.26% overall alignment rate
</code></pre></div></div><ul><li>This message can also be outputted by <code class="language-plaintext highlighter-rouge">samtools</code>.</ul><p class="note">The subsequent process utilizes the resulting ‘SAM’ file.</p><h1 class="fs-7 fw-400 lh-tight" id="citation"> <a href="#citation" class="anchor-heading" aria-labelledby="citation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citation</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">HISAT2</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Kim, D.</b>, Paggi, J. M., Park, C., Bennett, C., &amp; Salzberg, S. L. (2019). Graph-based genome alignment and genotyping with HISAT2 and HISAT-genotype. <i>Nature biotechnology</i>, 37(8), 907-915. <a href="https://doi.org/10.1038/s41587-019-0201-4" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
