<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(6)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(6) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(5) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(6) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list > li.nav-list-item:nth-child(6) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Motif Enrichment Analysis | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Motif Enrichment Analysis" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/ChIPSeq/MotifAnalysis_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/ChIPSeq/MotifAnalysis_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Motif Enrichment Analysis" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Motif Enrichment Analysis","url":"http://localhost:4000/docs/ChIPSeq/MotifAnalysis_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a><li class="nav-list-item"><a href="/docs/BarGen/" class="nav-list-link">Barcode Generator</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/ChIPSeq">Data Analysis - ChIP-seq</a><li class="breadcrumb-nav-list-item"><span>Motif Enrichment Analysis</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="motif-enrichment-analysis-of-chip-seq"> <a href="#motif-enrichment-analysis-of-chip-seq" class="anchor-heading" aria-labelledby="motif-enrichment-analysis-of-chip-seq"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Motif Enrichment Analysis of ChIP-seq</h1><p class="fs-4 fw-500">In the analysis of ChIP-seq data, motif analysis plays a critical role in understanding the binding patterns of DNA-associated proteins. Tools like HOMER and MEME are pivotal in this research. HOMER is particularly effective for discovering new, short motifs (8-12 bp) within large genomic datasets, operating with a suite of command-line tools in Perl and C++. MEME, meanwhile, excels at identifying motifs across unaligned sequences using statistical models, helping to elucidate potential biological functions. Both tools are instrumental for researchers studying gene regulation and protein-DNA interactions, providing insights into the underlying mechanisms of gene expression. This protocol was created based on <span style="color: #7253ec;"><b>HOMER version 4.9.1</b></span> and <span style="color: #7253ec;"><b>MEME version 5.5.5</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>Python version 3.8.5, SciPy version 1.6.2, NumPy version 1.20.1, Xcode Command Line Tools version 2406, and perl version 5.26.2 under macOS 12.4 environment.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="installation-homer"> <a href="#installation-homer" class="anchor-heading" aria-labelledby="installation-homer"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation HOMER</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <a href="https://github.com/javrodriguez/HOMER"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="http://homer.ucsd.edu/homer/"><img src="https://img.shields.io/badge/Official-e34c26?style=flat&amp;logo=html5&amp;logoColor=white" alt="web" /></a> <img src="https://img.shields.io/badge/release-4.9.1-blue?labelColor=black" alt="version" /> <img src="https://img.shields.io/badge/status-unstable-red?labelColor=black" alt="Status" /></p><p>To install HOMER using Anaconda, use one of the following commands:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>conda <span class="nb">install</span> <span class="nt">-c</span> bioconda homer
  <span class="c"># OR</span>
  <span class="nv">$ </span>conda <span class="nb">install</span> <span class="nt">-c</span> bioconda/label/cf201901 homer
</code></pre></div></div><h2 class="fs-6 fw-400 lh-tight" id="installation-reference-genomes"> <a href="#installation-reference-genomes" class="anchor-heading" aria-labelledby="installation-reference-genomes"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Installation Reference Genomes</u></h2><p>To check the available reference genomes, enter the following commnad in terminal:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>perl /Users/jchoi/opt/anaconda3/share/homer-4.10-0/configureHomer.pl <span class="nt">-list</span>
</code></pre></div></div><p>When running <code class="language-plaintext highlighter-rouge">configureHomer.pl</code>, it will print out the progress as shown below:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  Current base directory for HOMER is /Users/jchoi/opt/anaconda3/share/homer-4.10-0/

  --2024-05-14 15:19:49--  http://homer.ucsd.edu/homer/update.txt
  Connecting to homer.ucsd.edu (homer.ucsd.edu)|169.228.63.226|:80... connected.
  HTTP request sent, awaiting response... 200 OK
  Length: 17857 (17K) [text/plain]
  Saving to: ‘/Users/jchoi/opt/anaconda3/share/homer-4.10-0//update.txt’

  /Users/jchoi/opt/anaconda3/share/homer-4.10-0//updat 100%[====================================================================================================================&gt;]  17.44K  --.-KB/s    in 0.1s

  2024-05-14 15:19:49 (122 KB/s) - ‘/Users/jchoi/opt/anaconda3/share/homer-4.10-0//update.txt’ saved [17857/17857]

    Updating Settings...
  Packages with name conflicts have a trailing -o, -p, or -g
  Version Installed	Package	Version	Description
  SOFTWARE
  v4.11.1	homer	v5.0.1	Code/Executables, ontologies, motifs for HOMER
  ORGANISMS
  =========================================(skip)=========================================
  +	zebrafinch	v6.3	Taeniopygia guttata (zebrafinch) accession and ontology information
  +	mushroom	v6.3	Agaricus bisporus (mushroom) accession and ontology information
  +	volvox	v6.3	Volvox carteri (volvox) accession and ontology information
  +	selaginella	v6.3	Selaginella moellendorffii (selaginella) accession and ontology information
  +	mouse-o	v6.3	Mus musculus (mouse) accession and ontology information
  =========================================(skip)=========================================
  +	rhesus	v6.3	Macaca mulatta (rhesus) accession and ontology information
  +	ascomycetes	v6.0	Neurospora crassa (ascomycetes) accession and ontology information
  +	fugu	v6.3	Takifugu rubripes (fugu) accession and ontology information
  +	rice	v6.3	Oryza sativa (rice) accession and ontology information
  =========================================(skip)=========================================
  PROMOTERS
  +	human-p	v5.5	human promoters (human)
  +	frog-p	v5.5	frog promoters (frog)
  +	worm-p	v5.5	worm promoters (worm)
  +	rat-p	v5.5	rat promoters (rat)
  +	fly-p	v5.5	fly promoters (fly)
  +	chicken-p	v5.5	chicken promoters (chicken)
  +	zebrafish-p	v5.5	zebrafish promoters (zebrafish)
  +	arabidopsis-p	v6.3	arabidopsis promoters (arabidopsis)
  +	mouse-p	v5.5	mouse promoters (mouse)
  +	yeast-p	v5.5	yeast promoters (yeast)
  GENOMES
  =========================================(skip)=========================================
  +	hg19	v6.4	human genome and annotation for UCSC hg19
  +	mm10	v6.4	mouse genome and annotation for UCSC mm10
  +	danRer7	v6.4	zebrafish genome and annotation for UCSC danRer7
  +	sacCer3	v6.4	yeast genome and annotation for UCSC sacCer3
  +	danRer11	v6.4	zebrafish genome and annotation for UCSC danRer11
  =========================================(skip)=========================================
  +	danRer10	v6.4	zebrafish genome and annotation for UCSC danRer10
  +	dm3	v6.0	fly genome and annotation for UCSC dm3
  +	rice.IRGSP-1.0	v5.10	rice genome and annotation (rice.IRGSP-1.0)
  +	panPan2	v6.4	human genome and annotation for UCSC panPan2
  +	gorGor4	v6.4	human genome and annotation for UCSC gorGor4
  =========================================(skip)=========================================
  +	hg38	v6.4	human genome and annotation for UCSC hg38
  +	panTro3	v6.4	human genome and annotation for UCSC panTro3
  +	galGal5	v6.4	chicken genome and annotation for UCSC galGal5
</code></pre></div></div><p>To install the desired reference genomes, promoters, or organisms, enter the following commnad in terminal:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>perl /Users/jchoi/opt/anaconda3/share/homer-4.10-0/configureHomer.pl <span class="nt">-install</span> &lt;specific-name&gt;
</code></pre></div></div><h2 class="fs-6 fw-400 lh-tight" id="accessing-the-homer-server-from-the-office-network"> <a href="#accessing-the-homer-server-from-the-office-network" class="anchor-heading" aria-labelledby="accessing-the-homer-server-from-the-office-network"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Accessing the 'HOMER' Server from the Office Network</u></h2><p>Occasionally, users may encounter difficulties accessing the ‘HOMER’ server from their office network, despite the HOMER website being reachable. This issue typically arises due to specific ports or IP addresses being blocked by our organization’s firewalls or network policies. Should you experience such connectivity issues, the following procedures are recommended:</p><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 10px; padding-bottom: 5px; margin-top: 1px; margin-bottom: 1px; margin-left: 1px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"><p><span style="color: #7253ec;"><b>Option 1. Contact the IT Department</b></span>: This is the preferred and recommended approach. IT staff are equipped to assess and rectify connectivity issues while ensuring compliance with our security protocols.</p><p><span style="color: #7253ec;"><b>Option 2. Use a Bypass Program</b></span>: Utilizing a VPN or other bypass solutions can be considered if direct assistance from IT is not feasible. <b>Please note, employing such methods must comply with your company's IT policies.</b></p><p><span style="color: #7253ec;"><b>Option 3. Manual Configuration</b></span>: This involves manually adding the reference genome. Details on how to perform this configuration will be discussed in the following section.</p><p><span style="color: #7253ec;"><b>Disclaimer</b></span>: It is crucial to emphasize that while Option 1 is generally the most effective and secure method to resolve access issues, it might not always be immediately available. As for Option 2, please be aware that I cannot assume responsibility for any security violations or other repercussions that may occur within your organization as a result of using unauthorized bypass methods. Always seek approval from relevant authorities within the company prior to proceeding with such solutions.</p></div><h2 class="fs-6 fw-400 lh-tight" id="critical-protocol-for-adding-reference-genomes-in-homer-option-3"> <a href="#critical-protocol-for-adding-reference-genomes-in-homer-option-3" class="anchor-heading" aria-labelledby="critical-protocol-for-adding-reference-genomes-in-homer-option-3"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Critical Protocol for Adding Reference Genomes in HOMER (Option 3)</u></h2><p>This protocol details the steps required to update genome files within the HOMER environment. Please follow these steps meticulously to ensure accurate data handling and integration:</p><p style="margin-left: 20px;"><span style="color: #7253ec;"><b>1. Verify Installation of HOMER</b></span>: Confirm that HOMER is properly installed on your system. You can locate the configuration file at <code>./opt/anaconda3/share/homer-XXX/config.txt</code>.</p><p style="margin-left: 20px;"><span style="color: #7253ec;"><b>2. Access Configuration File</b></span>: Navigate to the directory specified above and open the <code>config.txt</code> file using a suitable text editor.</p><p style="margin-left: 20px;"><span style="color: #7253ec;"><b>3. Locate Dataset Entry</b></span>: Within the configuration file, search for the entry corresponding to your desired dataset. This could include genome data, promoters, GO terms, or known transcription factors. Pay particular attention to the HTTP address provided in the <span style="color: #7253ec;"><b>fourth column</b></span> of the file.</p><p style="margin-left: 20px;"><span style="color: #7253ec;"><b>4. Download Dataset</b></span>: Using the HTTP address noted, manually download the dataset of interest. Ensure the source is reliable and secure.</p><p style="margin-left: 20px;"><span style="color: #7253ec;"><b>5. Extract and Transfer Data</b></span>: After downloading, unzip the file and transfer its contents to the data folder as specified in the configuration file. The path to this data folder is typically noted in the <span style="color: #7253ec;"><b>fifth column</b></span> of <code>config.txt</code>.</p><p style="margin-left: 20px;"><span style="color: #7253ec;"><b>6. Initiate Motif Analysis</b></span>: With the new data successfully integrated, you are now prepared to proceed with motif analysis using HOMER.</p><h1 class="fs-7 fw-400 lh-tight" id="running-homer"> <a href="#running-homer" class="anchor-heading" aria-labelledby="running-homer"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running HOMER</h1><p>Use the following command to find enriched motifs in genomic regions with <code class="language-plaintext highlighter-rouge">findMotifsGenome</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>findMotifsGenome.pl &lt;PeakFiles&gt; &lt;species&gt; &lt;OutputFolder&gt; <span class="nt">-size</span> &lt;int&gt; <span class="nt">-len</span> &lt;int&gt; <span class="nt">-p</span> &lt;int&gt; 
</code></pre></div></div><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">&lt;PeakFiles&gt;</code><td>Specifies the input file. Acceptable input files are peaks files, which have at minimum five columns (separated by tabs, peakID/chromosome/start/end/strand) and BED (recommended) formats.<tr><td><code class="language-plaintext highlighter-rouge">&lt;species&gt;</code><td>Specifies the reference genome. Check to the available genome in the <code class="language-plaintext highlighter-rouge">configureHomer.pl</code> file.<tr><td><code class="language-plaintext highlighter-rouge">&lt;OutputFolder&gt;</code><td>Specifies the output path.<tr><td><code class="language-plaintext highlighter-rouge">-size &lt;int&gt;</code><td>Determine region size. It is advisable to use <code class="language-plaintext highlighter-rouge">-size 200</code> option to identify both primary and co-enriched motifs in transcription factors, while regions marked by histones (methyl/acetyl) should be examined with <code class="language-plaintext highlighter-rouge">500-1000</code>. Using the option <code class="language-plaintext highlighter-rouge">-size given</code>, HOMER will use the exact regions that were used as input.<tr><td><code class="language-plaintext highlighter-rouge">-len &lt;int&gt;</code><td>Specifies the length of motifs to be found. The length of time it takes to find motifs increases greatly with increasing size. In general, it is best to try out enrichment with shorter lengths (less than <code class="language-plaintext highlighter-rouge">15</code>) before trying longer lengths.<tr><td><code class="language-plaintext highlighter-rouge">-p &lt;int&gt;</code><td>The number of CPU cores HOMER will use when executing multi-processing tasks.</table></div><h2 class="fs-5 fw-400 lh-tight" id="example-code"> <a href="#example-code" class="anchor-heading" aria-labelledby="example-code"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code</u></h2><p>Here is an example command to find significant motifs for histone ChIP-seq and TF ChIP-seq:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Human, ChIP-seq with a TF antiboty</span>
  <span class="nv">$ </span>findMotifsGenome.pl MLL1_summits.bed hg19 /Users/jchoi/Desktop/HOMER_peaks <span class="nt">-size</span> given <span class="nt">-len</span> 15 <span class="nt">-p</span> 10

<span class="c"># Mouse, ChIP-seq with a histone marker antibody</span>
  <span class="nv">$ </span>findMotifsGenome.pl H3K4me3_summits.bed mm10 /Users/jchoi/Desktop/HOMER_peaks <span class="nt">-size</span> 500 <span class="nt">-len</span> 15 <span class="nt">-p</span> 10

<span class="c"># Plant, ChIP-seq with a epitope antibody</span>
  <span class="nv">$ </span>findMotifsGenome.pl Obe-W50-G100-FDR0.05-island.bed rice.IRGSP-1.0 /Users/jchoi/Desktop/Obe_HOMER <span class="se">\</span>
  <span class="nt">-size</span> given <span class="nt">-len</span> 12 <span class="nt">-p</span> 10	
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="installation-meme"> <a href="#installation-meme" class="anchor-heading" aria-labelledby="installation-meme"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation MEME</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <a href="https://github.com/cinquin/MEME"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="https://meme-suite.org/meme/"><img src="https://img.shields.io/badge/Official-e34c26?style=flat&amp;logo=html5&amp;logoColor=white" alt="web" /></a> <img src="https://img.shields.io/badge/release-5.5.5-blue?labelColor=black" alt="version" /> <img src="https://img.shields.io/badge/status-stable-DarkSeaGreen?labelColor=black" alt="Status" /></p><p>Unlike HOMER, MEME is recommended for motif analysis in ChIP-seq data due to its flexibility in identifying multiple motifs simultaneously, its robust statistical methods that evaluate the reliability of detected motifs, and its customizable options that allow for precise adjustments to sequence variability.</p><h2 class="fs-6 fw-400 lh-tight" id="requirements"> <a href="#requirements" class="anchor-heading" aria-labelledby="requirements"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Requirements</u></h2><p>To successfully use <code class="language-plaintext highlighter-rouge">meme</code>, the following prerequisites are necessary:</p><ul><li><strong>Python Environment</strong>: Version 3.0 or higher along with the libraries <code class="language-plaintext highlighter-rouge">scipy</code> and <code class="language-plaintext highlighter-rouge">numpy</code>.<li><strong>Compiler</strong>: A <code class="language-plaintext highlighter-rouge">C compiler</code> provided by the Xcode command line tools on macOS.<li><strong>Bioinformatics Tools</strong>:<ul><li><code class="language-plaintext highlighter-rouge">bedtools</code> (version specified: 2.30.0).<li><code class="language-plaintext highlighter-rouge">gs</code> (Ghostscript, version 10.02.1), <code class="language-plaintext highlighter-rouge">autoconf</code> (version 2.71), <code class="language-plaintext highlighter-rouge">automake</code> (version 1.16.5). These tools can be installed easily using Homebrew with the command <code class="language-plaintext highlighter-rouge">brew install &lt;package name&gt;</code>.</ul></ul><h2 class="fs-6 fw-400 lh-tight" id="additional-installation-notes"> <a href="#additional-installation-notes" class="anchor-heading" aria-labelledby="additional-installation-notes"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Additional Installation Notes:</u></h2><p>If an <code class="language-plaintext highlighter-rouge">XML::Parser::Expat</code> error occurs, the necessary parser package can be installed with the following commands in the terminal:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>    <span class="nv">$ </span>cpan App::cpanminus
    <span class="nv">$ </span>cpan XML::Simple
    <span class="nv">$ </span>cpan
    <span class="nv">$ </span><span class="nb">install </span>XML::Parser
</code></pre></div></div><h2 class="fs-6 fw-400 lh-tight" id="step-by-step-installation-guide-for-meme-tools"> <a href="#step-by-step-installation-guide-for-meme-tools" class="anchor-heading" aria-labelledby="step-by-step-installation-guide-for-meme-tools"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Step-by-Step Installation Guide for MEME Tools</u></h2><h3 id="1-download-the-tools"> <a href="#1-download-the-tools" class="anchor-heading" aria-labelledby="1-download-the-tools"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Download the Tools</h3><ul><li>Obtain the appropriate version of the MEME tools from the official website.</ul><h3 id="2-unzip-and-install"> <a href="#2-unzip-and-install" class="anchor-heading" aria-labelledby="2-unzip-and-install"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Unzip and Install</h3><ul><li>Open your terminal and execute the following commands to unpack and install the tools:<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> ~/downloads
<span class="nv">$ </span><span class="nb">tar </span>zxf meme-5.5.5.tar.gz
<span class="nv">$ </span><span class="nb">cd </span>meme-5.5.5
<span class="nv">$ </span>./configure <span class="nt">--prefix</span><span class="o">=</span>/Users/jchoi/meme <span class="nt">--enable-build-libxml2</span> <span class="nt">--enable-build-libxslt</span> <span class="c"># --prefix=[/your/pathway]/meme</span>
<span class="nv">$ </span>make
<span class="nv">$ </span>make <span class="nb">test</span>  <span class="c"># Optional, to verify installation</span>
<span class="nv">$ </span>make <span class="nb">install</span>
</code></pre></div></div></ul><h3 id="3-configure-shell-environment-based-on-zsh"> <a href="#3-configure-shell-environment-based-on-zsh" class="anchor-heading" aria-labelledby="3-configure-shell-environment-based-on-zsh"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. Configure shell environment (based on <code class="language-plaintext highlighter-rouge">zsh</code>)</h3><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>nano ~/.zshrc
</code></pre></div></div><h3 id="4-add-meme-to-your-path"> <a href="#4-add-meme-to-your-path" class="anchor-heading" aria-labelledby="4-add-meme-to-your-path"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4. Add MEME to your path</h3><ul><li><p>Insert the following lines at the end of your <code class="language-plaintext highlighter-rouge">~/.zshrc</code> file to include MEME in your system PATH. Adjust the paths as necessary based on your installation location.</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># MEME Suite Configuation NOTE: CASE SENSITIVITY!</span>
  <span class="nb">export </span><span class="nv">PATH</span><span class="o">=[</span>YOUR PATH]/meme/bin:[YOUR PATH]/meme/libexec/meme-5.5.5:<span class="nv">$PATH</span>

<span class="c">## For exmaple</span>
  <span class="nb">export </span><span class="nv">PATH</span><span class="o">=</span>/Users/jchoi/meme/bin:/Users/jchoi/meme/libexec/meme-5.5.5:<span class="nv">$PATH</span>
</code></pre></div></div></ul><h3 id="5-activate-the-configuration"> <a href="#5-activate-the-configuration" class="anchor-heading" aria-labelledby="5-activate-the-configuration"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 5. Activate the configuration</h3><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span><span class="nb">source</span> ~/.zshrc
</code></pre></div></div><h1 class="fs-7 fw-400 lh-tight" id="prepare-input-files-for-meme"> <a href="#prepare-input-files-for-meme" class="anchor-heading" aria-labelledby="prepare-input-files-for-meme"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Prepare Input Files for MEME</h1><p>The MEME tool requires sequences in FASTA format as the standard input. This section describes how to convert a BED file from peak calling analysis into a FASTA format using the <code class="language-plaintext highlighter-rouge">getfasta</code> function of <code class="language-plaintext highlighter-rouge">bedtools</code>.</p><h2 class="fs-5 fw-400 lh-tight" id="convert-bed-to-fasta"> <a href="#convert-bed-to-fasta" class="anchor-heading" aria-labelledby="convert-bed-to-fasta"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Convert BED to FASTA</u></h2><p>Use the following command to convert a peak-call BED file to FASTA format:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>bedtools getfasta <span class="nt">-fi</span> &lt;referenceGenomeFiles&gt; <span class="nt">-bed</span> &lt;input.bed&gt; <span class="nt">-fo</span> &lt;output.fasta&gt; 
</code></pre></div></div><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">-fi &lt;referenceGenomeFiles&gt;</code>: specifies the reference genome. Possible types are fasta or fa.<li><code class="language-plaintext highlighter-rouge">-bed &lt;input.bed&gt;</code>: specifies input bed file.<li><code class="language-plaintext highlighter-rouge">-fo &lt;output.fasta&gt;</code>: specifies output fasta file.</ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed"> <a href="#example-code-results-are-not-printed" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code (Results are not printed)</u></h2><p>Here are example commands using <code class="language-plaintext highlighter-rouge">getfasta</code> for different BED files generated by peak calling tools like MACS3 and SICER2:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># For MACS3 bed</span>
  <span class="nv">$ </span>bedtools getfasta <span class="nt">-fi</span> RGAP_Used.fa <span class="nt">-bed</span> /Users/jchoi/Desktop/MACS/obe_IP_summits.bed <span class="se">\</span>
  <span class="nt">-fo</span> /Users/jchoi/Desktop/obe_IP.fasta

  <span class="c"># For SICER2 bed</span>
  <span class="nv">$ </span>bedtools getfasta <span class="nt">-fi</span> Mus_musculus.GRCm39.dna.toplevel.fa <span class="nt">-bed</span> /Users/jchoi/Desktop/SICER/Unt_MLL1-W30-G60-FDR0.05-island.bed <span class="se">\</span>
  <span class="nt">-fo</span> /Users/jchoi/Desktop/Unt_MLL1.fasta
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="running-meme"> <a href="#running-meme" class="anchor-heading" aria-labelledby="running-meme"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running MEME</h1><p>The meme tool is used to identify statistically significant motifs within a single set of DNA or protein sequences. It operates by analyzing how frequently each motif occurs within the sequences. Use the following command to find motif sites with <code class="language-plaintext highlighter-rouge">meme</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>meme &lt;input.fasta&gt; -&lt;alphabet&gt; <span class="nt">-mod</span> &lt;<span class="nb">type</span><span class="o">&gt;</span> <span class="nt">-pal</span> <span class="nt">-revcomp</span> <span class="nt">-minw</span> &lt;int&gt; <span class="nt">-maxw</span> &lt;int&gt; <span class="se">\</span>
  <span class="nt">-nmotifs</span> &lt;int&gt; <span class="nt">-p</span> &lt;int&gt; <span class="nt">-o</span> &lt;output&gt;
</code></pre></div></div><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">&lt;input.fasta&gt;</code><td>Specifies the input file in either fasta or fa format.<tr><td><code class="language-plaintext highlighter-rouge">-&lt;alphabet&gt;</code><td>Specifies the type of the input file. Possible types are <code class="language-plaintext highlighter-rouge">dna</code>, <code class="language-plaintext highlighter-rouge">rna</code>, and <code class="language-plaintext highlighter-rouge">protein</code>.<tr><td><code class="language-plaintext highlighter-rouge">-mod &lt;type&gt;</code><td>Defines the distribution model for motif sites, crucial for precise motif analysis. For more details, see the section <strong>Algorithm for the Distribution of Motif Sites</strong>.<tr><td><code class="language-plaintext highlighter-rouge">-pal</code> option<td>Restricts the search to palindromes in datasets that allow complementary matches.<tr><td><code class="language-plaintext highlighter-rouge">-revcomp</code> option<td>Expands the search to include both the original strand and its reverse complement in datasets with complementable alphabets.<tr><td><code class="language-plaintext highlighter-rouge">-minw &lt;int&gt;</code><td>Search for motifs with a width &gt;= <code class="language-plaintext highlighter-rouge">minw</code>.<tr><td><code class="language-plaintext highlighter-rouge">-maxw &lt;int&gt;</code><td>Search for motifs with a width =&lt; <code class="language-plaintext highlighter-rouge">maxw</code>.<tr><td><code class="language-plaintext highlighter-rouge">-nmotifs &lt;int&gt;</code><td>Specifies the number of motif sites to identify.<tr><td><code class="language-plaintext highlighter-rouge">-p &lt;int&gt;</code><td>Allocates the number of CPU cores for parallel processing, optimizing the performance of the tool.<tr><td><code class="language-plaintext highlighter-rouge">-o &lt;output&gt;</code><td>Specifies the directory for storing the output results.</table></div><h2 class="fs-5 fw-400 lh-tight" id="algorithm-for-the-distribution-of-motif-sites"> <a href="#algorithm-for-the-distribution-of-motif-sites" class="anchor-heading" aria-labelledby="algorithm-for-the-distribution-of-motif-sites"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Algorithm for the Distribution of Motif Sites</u></h2><div class="table-wrapper"><table><thead><tr><th>Value<th>Name<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">oops</code><td>One Occurrence Per Sequence<td>MEME assumes that each sequence in the dataset contains exactly one occurrence of each motif. This option is the fastest and most sensitive but the motifs returned by MEME may be “blurry” if any of the primary sequences is missing them.<tr><td><code class="language-plaintext highlighter-rouge">zoops</code><td>Zero or One Occurrence Per Sequence<td>MEME assumes that each sequence may contain at most one occurrence of each motif. This option is useful when you suspect that some motifs may be missing from some of the primary sequences. In that case, the motifs found will be more accurate than using the first option. <strong>This option takes more computer time than the first option (about twice as much)</strong> and is slightly less sensitive to weak motifs present in all of the primary sequences.<tr><td><code class="language-plaintext highlighter-rouge">anr</code><td>Any Number of Repetitions<td>MEME assumes each sequence may contain any number of non-overlapping occurrences of each motif. This option is useful when you suspect that motifs repeat multiple times within a single sequence. In that case, the motifs found will be much more accurate than using one of the other options. The <code class="language-plaintext highlighter-rouge">anr</code> option can also be used to discover repeats within a single sequence. <strong>This option takes the much more computer time than the first option (about ten times as much)</strong> and is somewhat less sensitive to weak motifs which do not repeat within a single sequence than the other two options.</table></div><h2 class="fs-5 fw-400 lh-tight" id="example-code--output"> <a href="#example-code--output" class="anchor-heading" aria-labelledby="example-code--output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code &amp; Output</u></h2><p>Here is an example command to find motif sites from ChIP-seq peaks with <code class="language-plaintext highlighter-rouge">meme</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>meme obe_IP.fasta <span class="nt">-dna</span> <span class="nt">-mod</span> zoops <span class="nt">-pal</span> <span class="nt">-minw</span> 8 <span class="nt">-maxw</span> 20 <span class="nt">-nmotifs</span> 10 <span class="nt">-p</span> 10 <span class="nt">-o</span> /Users/jchoi/Desktop/Motif
<span class="nv">$ </span>meme obe_IP.fasta <span class="nt">-dna</span> <span class="nt">-mod</span> oops <span class="nt">-pal</span> <span class="nt">-revcomp</span> <span class="nt">-minw</span> 8 <span class="nt">-maxw</span> 20 <span class="nt">-nmotifs</span> 10 <span class="nt">-p</span> 10 <span class="nt">-o</span> /Users/jchoi/Desktop/Motif_oops
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h2 class="fs-6 fw-400 lh-tight" id="understanding-meme-output"> <a href="#understanding-meme-output" class="anchor-heading" aria-labelledby="understanding-meme-output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Understanding MEME Output</u></h2><p>The result of the <code class="language-plaintext highlighter-rouge">meme</code> will be several files in the specified folder (<code class="language-plaintext highlighter-rouge">-o</code> option), according to the specific parameters. These output files are described below:</p><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 10px; padding-bottom: 5px; margin-top: 1px; margin-bottom: 1px; margin-left: 1px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"><p><code>meme.html</code>: an HTML file that provides the results in an interactive, human-readable format.</p><p><code>meme.txt</code>: a plain text file of the results for backwards compatibility with earlier versions of MEME.</p><p><code>meme.xml</code>: an XML file that provides the results in a format designed for machine processing.</p><p><code>logoN.png,.eps</code>: PNG and EPS images files containing sequence logos for each of the motifs found by MEME (where N is the motif number).</p><p><code>logo_rcN.png,.eps</code>: (complementable alphabets only) PNG and EPS images files containing reverse-complement sequence logos for each of the motifs found by MEME (where N is the motif number).</p></div><h1 class="fs-7 fw-400 lh-tight" id="running-meme-chip"> <a href="#running-meme-chip" class="anchor-heading" aria-labelledby="running-meme-chip"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running <code class="language-plaintext highlighter-rouge">meme-chip</code></h1><p><code class="language-plaintext highlighter-rouge">meme-chip</code> is an integrated web service used for searching and comparing motifs across various related sequences, including ChIP-seq data. This tool automates motif discovery and analysis using multiple MEME-Suite algorithms. Use the following command to find motif sites from peaks with <code class="language-plaintext highlighter-rouge">meme-chip</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>meme-chip &lt;input.fasta&gt; -&lt;alphabet&gt; <span class="nt">-meme-mod</span> &lt;<span class="nb">type</span><span class="o">&gt;</span> <span class="nt">-meme-pal</span> <span class="nt">-minw</span> &lt;int&gt; <span class="nt">-maxw</span> &lt;int&gt; <span class="se">\</span>
  <span class="nt">-meme-nmotifs</span> &lt;int&gt; <span class="nt">-db</span> &lt;dbpath&gt; <span class="nt">-meme-p</span> &lt;int&gt; <span class="nt">-o</span> &lt;output&gt;
</code></pre></div></div><p>The fundamental command line remains consistent with that of <code class="language-plaintext highlighter-rouge">meme</code>. In these commands:</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">&lt;input.fasta&gt;</code><td>Specifies the input file in either fasta or fa format.<tr><td><code class="language-plaintext highlighter-rouge">-&lt;alphabet&gt;</code><td>Specifies the type of the input file. Possible types are <code class="language-plaintext highlighter-rouge">dna</code>, <code class="language-plaintext highlighter-rouge">rna</code>, and <code class="language-plaintext highlighter-rouge">protein</code>.<tr><td><code class="language-plaintext highlighter-rouge">-meme-mod &lt;type&gt;</code><td>Defines the distribution model for motif sites. Same parameter the <code class="language-plaintext highlighter-rouge">-mod</code>in <code class="language-plaintext highlighter-rouge">meme</code>.<tr><td><code class="language-plaintext highlighter-rouge">-meme-pal</code> option<td>Restricts the search to palindromes in datasets that allow complementary matches. Same parameter the <code class="language-plaintext highlighter-rouge">-pal</code>in <code class="language-plaintext highlighter-rouge">meme</code>.<tr><td><code class="language-plaintext highlighter-rouge">-minw &lt;int&gt;</code><td>Search for motifs with a width &gt;= <code class="language-plaintext highlighter-rouge">minw</code>.<tr><td><code class="language-plaintext highlighter-rouge">-maxw &lt;int&gt;</code><td>Search for motifs with a width =&lt; <code class="language-plaintext highlighter-rouge">maxw</code>.<tr><td><code class="language-plaintext highlighter-rouge">-meme-nmotifs &lt;int&gt;</code><td>Specifies the number of motif sites to identify. Same parameter the <code class="language-plaintext highlighter-rouge">-nmotif</code>in <code class="language-plaintext highlighter-rouge">meme</code>.<tr><td><code class="language-plaintext highlighter-rouge">-db &lt;dbpath&gt;</code><td>(<strong>Recommended</strong>) Specifies known motifs database for comparison. Users may choose any preferred database (such as JASPAR) beyond those provided by <code class="language-plaintext highlighter-rouge">meme</code> (in <code class="language-plaintext highlighter-rouge">meme/share/meme-5.5.5/db/motif_databases/</code>). Most of these motif files are available in the <code class="language-plaintext highlighter-rouge">*.meme</code> format.<tr><td><code class="language-plaintext highlighter-rouge">-meme-p &lt;int&gt;</code><td>Allocates the number of CPU cores for parallel processing, optimizing the performance of the tool.<tr><td><code class="language-plaintext highlighter-rouge">-o &lt;output&gt;</code><td>Specifies the directory for storing the output results.</table></div><h2 class="fs-5 fw-400 lh-tight" id="example-code--output-1"> <a href="#example-code--output-1" class="anchor-heading" aria-labelledby="example-code--output-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code &amp; Output</u></h2><p>Here is an example command to find motif sites from ChIP-seq peaks with <code class="language-plaintext highlighter-rouge">meme-chip</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># Use JASPAR core database </span>
  <span class="nv">$ </span>meme-chip obe_IP.fasta <span class="nt">-meme-mod</span> zoops <span class="nt">-meme-nmotifs</span> 10 <span class="nt">-minw</span> 8 <span class="nt">-maxw</span> 20 <span class="se">\</span>
  <span class="nt">-db</span> /db/motif_databases/JASPAR/JASPAR2022_CORE_plants_non-redundant_v2.meme <span class="se">\</span>
  <span class="nt">-o</span> /Users/jchoi/Desktop/JASPAR_Motif

  <span class="c"># Use cisBP database</span>
  <span class="nv">$ </span>meme-chip obe_IP.fasta <span class="nt">-meme-mod</span> zoops <span class="nt">-meme-nmotifs</span> 10 <span class="nt">-minw</span> 6 <span class="nt">-maxw</span> 15 <span class="nt">-meme-pal</span> <span class="se">\</span>
  <span class="nt">-db</span> db/motif_databases/CIS-BP_2.00/Oryza_sativa.meme <span class="nt">-o</span> /Users/jchoi/Desktop/CIS_Motif

  <span class="c"># Use specific motif data</span>
  <span class="nv">$ </span>meme-chip obe_IP.fasta <span class="nt">-meme-mod</span> zoops <span class="nt">-meme-nmotifs</span> 10 <span class="nt">-minw</span> 6 <span class="nt">-maxw</span> 12 <span class="nt">-meme-pal</span> <span class="se">\</span>
  <span class="nt">-db</span> /Users/jchoi/Desktop/JASPAR2022_Plant_WRKY.meme <span class="nt">-o</span> /Users/jchoi/Desktop/WRKY_Motif
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="citations"> <a href="#citations" class="anchor-heading" aria-labelledby="citations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citations</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">HOMER</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Heinz, S.</b>, Benner, C., Spann, N., Bertolino, E., Lin, Y. C., Laslo, P., ... &amp; Glass, C. K. (2010). Simple combinations of lineage-determining transcription factors prime cis-regulatory elements required for macrophage and B cell identities. <i>Molecular cell</i>, 38(4), 576-589. <a href="https://doi.org/10.1016/j.molcel.2010.05.004" target="_blank">DOI</a></span></ol><span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">MEME suite</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Bailey, T. L.</b>, Boden, M., Buske, F. A., Frith, M., Grant, C. E., Clementi, L., ... &amp; Noble, W. S. (2009). MEME SUITE: tools for motif discovery and searching. <i>Nucleic acids research</i>, 37(suppl_2), W202-W208. <a href="https://doi.org/10.1093/nar/gkp335" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
