<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(2)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(3) > ul > li:nth-child(2) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(3) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(3) > ul > li:nth-child(2) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(3) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(3) > ul.nav-list > li.nav-list-item:nth-child(2) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Trimmed reads with Trimmomatic | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Trimmed reads with Trimmomatic" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/DataProcess/Trimmmomatic_manual/" /><meta property="og:url" content="http://localhost:4000/docs/DataProcess/Trimmmomatic_manual/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Trimmed reads with Trimmomatic" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Trimmed reads with Trimmomatic","url":"http://localhost:4000/docs/DataProcess/Trimmmomatic_manual/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-portfolio" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/DataProcess">Data Process</a><li class="breadcrumb-nav-list-item"><span>Trimmed reads with Trimmomatic</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="remove-adaptor-and-low-quality-reads-with-trimmomatic"> <a href="#remove-adaptor-and-low-quality-reads-with-trimmomatic" class="anchor-heading" aria-labelledby="remove-adaptor-and-low-quality-reads-with-trimmomatic"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Remove Adaptor and Low-Quality Reads with Trimmomatic</h1><p class="fs-4 fw-500">Removing adapters and low-quality sequences from sequencing data is essential for ensuring data accuracy and quality. Most sequencers employ a sequencing adapter to generate raw data; however, these artificial sequences must be removed during data processing. This not only eliminates the adapter but also improves the overall quality of the data by removing errors or low-quality read ends that may occur during the sequencing process. Trimmomatic, a multithreaded tool based on the JavaScript platform, is highly effective in performing these tasks, making it a valuable resource in bioinformatics for obtaining high-quality data. This protocol was created based on <span style="color: #7253ec;"><b>Trimmomatic version 0.39</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>Python version 3.8.5 and Java version 16.0.1 under macOS 12.4 environment.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="installation-trimmomatic"> <a href="#installation-trimmomatic" class="anchor-heading" aria-labelledby="installation-trimmomatic"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation Trimmomatic</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <a href="https://github.com/usadellab/Trimmomatic"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="http://www.usadellab.org/cms/?page=trimmomatic"><img src="https://img.shields.io/badge/Official-e34c26?style=flat&amp;logo=html5&amp;logoColor=white" alt="web" /></a> <a href="https://github.com/usadellab/Trimmomatic/releases"><img src="https://img.shields.io/github/v/release/usadellab/Trimmomatic?labelColor=black" alt="GitHub release" /></a> <img src="https://img.shields.io/badge/status-stable-DarkSeaGreen?labelColor=black" alt="Status" /></p><p>Download the file from the <a href="http://www.usadellab.org/cms/?page=trimmomatic" target="_blank">official website</a> and unzip it to a folder of your choice.</p><h3 id="requirments"> <a href="#requirments" class="anchor-heading" aria-labelledby="requirments"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Requirments</h3><ul><li>To run Trimmomatic, you need Java version 7 or higher. Use the following command to check Java version:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  $ java -version
  java version "16.0.1" 2020-04-20
  Java(TM) SE Runtime Environment (build 16.0.1+9-24)
  Java HotSpot(TM) 64-Bit Server VM (build 16.0.1+9-24, mixed mode, sharing)
</code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="running-trimmomatic"> <a href="#running-trimmomatic" class="anchor-heading" aria-labelledby="running-trimmomatic"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running Trimmomatic</h1><p>Use the following command to perform Trimmomatic via Java machine: <strong>Trimmomatic and its related command-lines must be typed in directory of trimmomatic.</strong></p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>java <span class="nt">-var</span> Trimmomatic-0.39.jar &lt;mode&gt; -&lt;score&gt; <span class="nt">-threads</span> &lt;int&gt; &lt;input1&gt; ... &lt;inputN&gt; <span class="se">\</span>
	&lt;PairedOutput1.fq.gz&gt; &lt;UnpairOutput1.fq.gz&gt; ... &lt;PairedOutputN.fq.gz&gt; &lt;UnpairOutputN.fq.gz&gt; <span class="se">\</span>
	&lt;OptionalParameters&gt;

<span class="c"># e.g. Full commands for pair-end mode</span>
<span class="nv">$ </span>java <span class="nt">-var</span> Trimmomatic-0.39.jar &lt;mode&gt; -&lt;score&gt; <span class="nt">-threads</span> &lt;int&gt; &lt;input1&gt; &lt;input2&gt; <span class="se">\</span>
	&lt;PairedOutput1.fq.gz&gt; &lt;UnpairOutput1.fq.gz&gt; &lt;PairedOutput2.fq.gz&gt; &lt;UnpairOutput2.fq.gz&gt; <span class="se">\</span>
	ILLUMINACLIP:[adaptor file.fa]:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36 

<span class="c"># e.g. Full commands for single-end mode</span>
<span class="nv">$ </span>java <span class="nt">-var</span> Trimmomatic-0.39.jar &lt;mode&gt; -&lt;score&gt; <span class="nt">-threads</span> &lt;int&gt; &lt;input1&gt; &lt;PairedOutput1.fq.gz&gt; <span class="se">\</span>
	ILLUMINACLIP:[adaptor file.fa]:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36 
</code></pre></div></div><p>In these commands:</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">java -var Trimmomatic-0.39.jar</code><td>Executes <code class="language-plaintext highlighter-rouge">trimmomatic</code> code<tr><td><code class="language-plaintext highlighter-rouge">&lt;mode&gt;</code> option<td>Specifies sequence type from either pair-end (<code class="language-plaintext highlighter-rouge">PE</code>) or single-end (<code class="language-plaintext highlighter-rouge">SE</code>)<tr><td><code class="language-plaintext highlighter-rouge">-&lt;score&gt;</code> option<td>Determines quality score whether this score is based on 33 (<code class="language-plaintext highlighter-rouge">phred33</code>) or 64 (<code class="language-plaintext highlighter-rouge">phred64</code>).<tr><td><code class="language-plaintext highlighter-rouge">-threads &lt;int&gt;</code><td>Specifies thread numbers <int> for this operation.</int><tr><td><code class="language-plaintext highlighter-rouge">&lt;input1&gt; ... &lt;inputN&gt;</code><td>Specifies forward and reverse input files, they are separated by <strong>spaces</strong>.<tr><td><code class="language-plaintext highlighter-rouge">&lt;PairedOutputN.fq.gz&gt; &lt;UnpairOutputN.fq.gz&gt;</code><td>Specifies output files. In <code class="language-plaintext highlighter-rouge">PE</code> mode, each input file will result in the creation of two output files, arranged in the specific order of <strong>“pair”</strong> and <strong>“unpair”</strong>. <strong>It is imperative to preserve this order to prevent any discrepancies.</strong> In <code class="language-plaintext highlighter-rouge">SE</code> mode, each input file will generate a single output file, irrespective of any pairing configuration.</table></div><ul><li><p>The <code class="language-plaintext highlighter-rouge">&lt;OptionalParameters&gt;</code> offer a wide range of variables as bellow, see the Trimmomatic’s manual for details (<i>For more information such as global parameter, refers to Trimmomatic’s guidelines.</i>).</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">ILLUMINACLIP:&lt;path/to/CLIP.fa&gt;</code><td>Specifies a adaptor clip file. Trimmomtic provide this universal sequence clip in <code class="language-plaintext highlighter-rouge">/adapters</code> folder. <code class="language-plaintext highlighter-rouge">ILLUMINACLIP</code> would be assigned relative path.<tr><td><code class="language-plaintext highlighter-rouge">:&lt;seed mismatches&gt;:&lt;palindrome clip threshold&gt;:&lt;simple clip threshold&gt;</code><td>Specifies the maximum mismatch, how accurate the match between the two adapter ligated reads, and how accurate the match between any adapter, respectively. This parameter should come after the <code class="language-plaintext highlighter-rouge">ILLUMINACLIP</code>, most commonly <code class="language-plaintext highlighter-rouge">:2:30:10</code>.<tr><td><code class="language-plaintext highlighter-rouge">LEADING:&lt;quality&gt;</code><td>Remove low quality bases from the beginning. Specifies the minimum quality required to keep a base, most commonly <code class="language-plaintext highlighter-rouge">LEADING:3</code>.<tr><td><code class="language-plaintext highlighter-rouge">TRAILING:&lt;quality&gt;</code><td>Remove low quality bases from the end. Specifies the minimum quality required to keep a base, most commonly <code class="language-plaintext highlighter-rouge">TRAILING:3</code><tr><td><code class="language-plaintext highlighter-rouge">SLIDINGWINDOW:&lt;size:&lt;quality&gt;</code><td>Perform a sliding window trimming, cutting once the average quality (<code class="language-plaintext highlighter-rouge">quality</code>) within the window (<code class="language-plaintext highlighter-rouge">size</code>)falls below a threshold, most commonly <code class="language-plaintext highlighter-rouge">SLIDINGWINDOW:4:15</code>.<tr><td><code class="language-plaintext highlighter-rouge">MINLEN:&lt;length&gt;</code><td>This module removes reads that fall below the specified minimal length. Specifies the minimum length of reads to be kept, most commonly <code class="language-plaintext highlighter-rouge">MINLEN:36</code>.</table></div></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-with-explanation"> <a href="#example-code-with-explanation" class="anchor-heading" aria-labelledby="example-code-with-explanation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code with explanation</u></h2><p>Here is an example command with footnote for explanation to perform Trimmomatic on QCed fastq files:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Example for PE mode</span>
java <span class="nt">-jar</span> trimmomatic-0.39.jar PE <span class="nt">-threads</span> 20 <span class="nt">-phred33</span> <span class="se">\ </span><span class="c"># Run a Trimmomatic, consist pair-end and base on ASCII 33, using 20 threads.</span>
	~/Desktop/Bone_fastq/shC_D0_1.fastq.gz ~/Desktop/Bone_fastq/shC_D0_2.fastq.gz <span class="se">\ </span><span class="c"># Speficies forward and reverse input files, respectively.		</span>
	~/Desktop/Bone_Trim/Trim_shC_D0_1.fastq.gz ~/Desktop/Bone_Trim/unTrim_shC_D0_1.fastq.gz <span class="se">\ </span><span class="c"># Output for forward</span>
	~/Desktop/Bone_Trim/Trim_shC_D0_2.fastq.gz ~/Desktop/Bone_Trim/unTrim_shC_D0_2.fastq.gz <span class="se">\ </span><span class="c"># Output for reverse</span>
	ILLUMINACLIP:adapters/TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36 <span class="c"># Specify adaptor sequence and global parameters</span>

<span class="c"># Example for SE mode</span>
java <span class="nt">-jar</span> trimmomatic-0.39.jar SE <span class="nt">-threads</span> 20 <span class="nt">-phred33</span> <span class="se">\ </span><span class="c"># Run a Trimmomatic with SE mode</span>
	~/Desktop/ChIP_seq/H3K4me3.fastq.gz TRIM_H3K4me3.fq.gz <span class="se">\ </span><span class="c"># Input and ouput</span>
	ILLUMINACLIP:adapters/TruSeq3-SE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36 <span class="c"># global parameters</span>
</code></pre></div></div><h2 class="fs-5 fw-400 lh-tight" id="output"> <a href="#output" class="anchor-heading" aria-labelledby="output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Output</u></h2><p>When Trimmomatic finishes running, it prints messages summarizing what happened.</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>	TrimmomaticPE: Started with arguments:
 	-threads 20 -phred33 /Users/jchoi/Desktop/Bone_fastq/shC_D0_1.fastq.gz /Users/jchoi/Desktop/Bone_fastq/shC_D0_2.fastq.gz /Users/jchoi/Desktop/Bone_Trim/Trim_shC_D0_1.fastq.gz /Users/jchoi/Desktop/Bone_Trim/unTrim_shC_D0_1.fastq.gz /Users/jchoi/Desktop/Bone_Trim/Trim_shC_D0_2.fastq.gz /Users/jchoi/Desktop/Bone_Trim/unTrim_shC_D0_2.fastq.gz ILLUMINACLIP:adapters/TruSeq3-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
	Using PrefixPair: 'TACACTCTTTCCCTACACGACGCTCTTCCGATCT' and 'GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT'
	ILLUMINACLIP: Using 1 prefix pairs, 0 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
	Input Read Pairs: 41429555 Both Surviving: 40573408 (97.93%) Forward Only Surviving: 646381 (1.56%) Reverse Only Surviving: 168217 (0.41%) Dropped: 41549 (0.10%)
	TrimmomaticPE: Completed successfully
</code></pre></div></div><h1 class="fs-7 fw-400 lh-tight" id="citation"> <a href="#citation" class="anchor-heading" aria-labelledby="citation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citation</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">Trimmomatic</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Bolger, A. M.</b>, Lohse, M., &amp; Usadel, B. (2014). Trimmomatic: a flexible trimmer for Illumina sequence data. <i>Bioinformatics</i>, 30(15), 2114-2120. <a href="https://doi.org/10.1093/bioinformatics/btu170" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
