<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(4)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(3) > ul > li:nth-child(4) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(3) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(3) > ul > li:nth-child(4) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(3) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(3) > ul.nav-list > li.nav-list-item:nth-child(4) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Mapping reads with Bowtie2 for ChIP-seq | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Mapping reads with Bowtie2 for ChIP-seq" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/DataProcess/Bowtie2_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/DataProcess/Bowtie2_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Mapping reads with Bowtie2 for ChIP-seq" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Mapping reads with Bowtie2 for ChIP-seq","url":"http://localhost:4000/docs/DataProcess/Bowtie2_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a><li class="nav-list-item"><a href="/docs/BarGen/" class="nav-list-link">Barcode Generator</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/DataProcess">Data Process</a><li class="breadcrumb-nav-list-item"><span>Mapping reads with Bowtie2 for ChIP-seq</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="alignment-with-reference-genome-using-bowtie2"> <a href="#alignment-with-reference-genome-using-bowtie2" class="anchor-heading" aria-labelledby="alignment-with-reference-genome-using-bowtie2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Alignment with Reference Genome using bowtie2</h1><p class="fs-4 fw-500">Bowtie2 that is an ultrafast and memory-efficient tool coded in python can be aligned sequencing reads to long reference genomes. Notably, <span style="color: #7253ec;"><b>Bowtie is suitable for ChIP-seq or ATAC-seq, but not RNA-seq.</b></span> If you have to analyze transcriptome, go to the section on RNA-seq using HISAT2 tool. Bowtie can run on any computer installed on Linux or macOS and works in python version &gt; 2.6. This protocol was created based on <span style="color: #7253ec;"><b>Bowtie2 version 2.5.1</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>Python version 3.8.5, SciPy version 1.6.2, NumPy version 1.20.1, and pySam version 0.16.0.1 under macOS 12.4 environment. This course takes very long time and spends a lot of RAM and CPU resource.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="installation-bowtie2"> <a href="#installation-bowtie2" class="anchor-heading" aria-labelledby="installation-bowtie2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation bowtie2</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <a href="https://github.com/BenLangmead/bowtie2"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <img src="https://github.com/BenLangmead/bowtie2/actions/workflows/random-tests.yml/badge.svg" alt="Random Tests" /> <img src="https://github.com/BenLangmead/bowtie2/actions/workflows/simple-tests.yml/badge.svg" alt="Simple Tests" /> <a href="https://bowtie-bio.sourceforge.net/bowtie2/index.shtml"><img src="https://img.shields.io/badge/Official-e34c26?style=flat&amp;logo=html5&amp;logoColor=white" alt="web" /></a> <a href="https://github.com/BenLangmead/bowtie2/releases"><img src="https://img.shields.io/github/v/release/BenLangmead/bowtie2?labelColor=black" alt="GitHub release" /></a> <img src="https://img.shields.io/badge/status-stable-DarkSeaGreen?labelColor=black" alt="Status" /></p><p>To install bowtie2 using homebrew, use the following command:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>brew <span class="nb">install </span>bowtie2
</code></pre></div></div><h2 class="fs-6 fw-400 lh-tight" id="establish-genome-builder"> <a href="#establish-genome-builder" class="anchor-heading" aria-labelledby="establish-genome-builder"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Establish Genome Builder</u></h2><p><code class="language-plaintext highlighter-rouge">bowtie2-build</code> builds a bowtie index from a set of DNA sequence. This builder consists of a set of 6 files with suffixes <code class="language-plaintext highlighter-rouge">.1.bt2</code>, <code class="language-plaintext highlighter-rouge">.2.bt2</code>, <code class="language-plaintext highlighter-rouge">.3.bt2</code>, <code class="language-plaintext highlighter-rouge">.4.bt2</code>, <code class="language-plaintext highlighter-rouge">.rev.1.bt2</code>, and <code class="language-plaintext highlighter-rouge">.rev.2.bt2</code>. These files together constitute the index for the alignment reads to that reference. You can download from the index section of official website all the files for a given assembly as a single zip file, or as 6 seperate bt2 files. There are two types of genome builder method; manual and automatic.</p><ol><li>Automatic method: uncompress builder files and uses ‘bowtie2-build’ or ‘make_XX_.sh’ syntax. For more information, refer to Bowtie2’s manual.<li>Manual method: uncompress builder files and moves them to the desired path.</ol><p class="note">the build step is a prerequisite for the main sequence. Unlike “HISAT2”, this tool provides index files for most species from the index zone of official site.</p><h1 class="fs-7 fw-400 lh-tight" id="running-bowtie2"> <a href="#running-bowtie2" class="anchor-heading" aria-labelledby="running-bowtie2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running bowtie2</h1><p>Use the following command to perform mapping to the genome with bowtie2:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Usage for single-end</span>
<span class="nv">$ </span>bowtie2 <span class="nt">-x</span> &lt;GenomeBuilder&gt; <span class="nt">-U</span> &lt;reads.fq&gt; <span class="nt">-S</span> &lt;output.sam&gt; <span class="nt">--</span>&lt;mode&gt; <span class="nt">--</span>&lt;preset&gt; <span class="nt">-p</span> &lt;int&gt;

<span class="c"># Usage for pair-end</span>
<span class="nv">$ </span>bowtie2 <span class="nt">-x</span> &lt;GenomeBuilder&gt; <span class="nt">-1</span> &lt;forward_reads.fq&gt; <span class="nt">-2</span> &lt;reverse_reads.fq&gt; <span class="se">\</span>
        <span class="nt">-S</span> &lt;output.sam&gt; <span class="nt">--</span>&lt;mode&gt; <span class="nt">--</span>&lt;preset&gt; <span class="nt">-p</span> &lt;int&gt;
</code></pre></div></div><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-x &lt;GenomeBuilder&gt;</code><td>Specifies the reference genome as the builder format.<tr><td><code class="language-plaintext highlighter-rouge">-U &lt;reads.fq&gt;</code> (only SE mode)<td>Specifies input read sequence. Possible file types is fastq formats (fastq or fq).<tr><td><code class="language-plaintext highlighter-rouge">-1</code> and <code class="language-plaintext highlighter-rouge">-2</code> (only PE mode)<td>Specifies forward input read and reverse input read, respectively.<tr><td><code class="language-plaintext highlighter-rouge">-S &lt;output.sam&gt;</code><td>Specifies output file.<tr><td><code class="language-plaintext highlighter-rouge">--&lt;mode&gt;&gt;</code> option<td>Bowtie2 provides two algorithms for maaping reads to the genome. The end-to-end mode (<code class="language-plaintext highlighter-rouge">--end-to-end</code>, default) aligns the entire length of reads to the reference genome, utilizing all available information. In contrast, the local mode (<code class="language-plaintext highlighter-rouge">--local</code>) flexibly aligns only the best-matching segments of reads to the reference, which is especially useful when the quality of reads is uneven. For more information, refer to Bowtie2’s manual. <strong>This manual provides the local algorithm.</strong><tr><td><code class="language-plaintext highlighter-rouge">--&lt;preset&gt;</code> option<td>Specifies preset options for the mapping algorithm. Possivle presers for the local mode are <code class="language-plaintext highlighter-rouge">--very-fast-local</code>, <code class="language-plaintext highlighter-rouge">--fast-local</code>, <code class="language-plaintext highlighter-rouge">--sensitive-local</code> (default) or <code class="language-plaintext highlighter-rouge">--very-sensitive-local</code>. For more information, refer to Bowtie2’s manual.<tr><td><code class="language-plaintext highlighter-rouge">-p &lt;int&gt;</code><td>number of processors to use in addition to main thread <int>.</int></table></div><h2 class="fs-5 fw-400 lh-tight" id="example-code"> <a href="#example-code" class="anchor-heading" aria-labelledby="example-code"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code</u></h2><p>Here is an example command to perform alignment with the mouse mm10 genome on trimmed fastq files:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>bowtie2 <span class="nt">-x</span> /Users/jchoi/Desktop/mm10/mm10 <span class="se">\</span>
    <span class="nt">-1</span> /Users/jchoi/Desktop/Trim/Trim_pair_NFIB_1.fq.gz <span class="se">\</span>
    <span class="nt">-2</span> /Users/jchoi/Desktop/Trim/Trim_pair_NFIB_2.fq.gz <span class="se">\</span>
    <span class="nt">-S</span> /Users/jchoi/Desktop/NFIB.sam <span class="nt">--local</span> <span class="nt">--very-fast-local</span> <span class="nt">-p</span> 20
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h2 class="fs-5 fw-400 lh-tight" id="output"> <a href="#output" class="anchor-heading" aria-labelledby="output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Output</u></h2><p>When bowtie2 finishes running, it prints messages summarizing what happened.</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>23156585 reads; of these:
  23156585 (100.00%) were paired; of these:
    2044536 (8.83%) aligned concordantly 0 times
    11769262 (50.82%) aligned concordantly exactly 1 time
    9342787 (40.35%) aligned concordantly &gt;1 times
    ----
    2044536 pairs aligned concordantly 0 times; of these:
      686243 (33.56%) aligned discordantly 1 time
    ----
    1358293 pairs aligned 0 times concordantly or discordantly; of these:
      2716586 mates make up the pairs; of these:
        1327258 (48.86%) aligned 0 times
        453087 (16.68%) aligned exactly 1 time
        936241 (34.46%) aligned &gt;1 times
97.13% overall alignment rate
</code></pre></div></div><ul><li>This message can also be outputted by <code class="language-plaintext highlighter-rouge">samtools</code>.</ul><p class="note">The subsequent process utilizes the resulting ‘SAM’ file.</p><h1 class="fs-7 fw-400 lh-tight" id="citations"> <a href="#citations" class="anchor-heading" aria-labelledby="citations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citations</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">Bowtie2</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Langmead, B.</b>, &amp; Salzberg, S. L. (2012). Fast gapped-read alignment with Bowtie 2. <i>Nature methods</i>, 9(4), 357-359. <a href="https://doi.org/10.1038/nmeth.1923" target="_blank">DOI</a></span><li style="margin-bottom: 10px;"> <span><b>Langmead, B.</b>, Wilks, C., Antonescu, V., &amp; Charles, R. (2019). Scaling read aligners to hundreds of threads on general-purpose processors. <i>Bioinformatics</i>, 35(3), 421-432. <a href="https://doi.org/10.1093/bioinformatics/bty648" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
