<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(1)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(4) > ul > li:nth-child(1) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(4) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(4) > ul > li:nth-child(1) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(4) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(4) > ul.nav-list > li.nav-list-item:nth-child(1) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Normalization and DEG analysis with edgeR | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Normalization and DEG analysis with edgeR" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/RNASeq/DEG_edgeR_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/RNASeq/DEG_edgeR_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Normalization and DEG analysis with edgeR" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Normalization and DEG analysis with edgeR","url":"http://localhost:4000/docs/RNASeq/DEG_edgeR_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/RNASeq">Data Analysis - RNA-seq</a><li class="breadcrumb-nav-list-item"><span>Normalization and DEG analysis with edgeR</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="advanced-rna-seq-data-analysis-with-edger-identifying-differentially-expressed-genes-and-beyond"> <a href="#advanced-rna-seq-data-analysis-with-edger-identifying-differentially-expressed-genes-and-beyond" class="anchor-heading" aria-labelledby="advanced-rna-seq-data-analysis-with-edger-identifying-differentially-expressed-genes-and-beyond"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Advanced RNA-seq Data Analysis with EdgeR: Identifying Differentially Expressed Genes and Beyond</h1><p class="fs-4 fw-500">EdgeR is a powerful tool designed for the data analysis phase of RNA-seq experiments, specifically for identifying differentially expressed genes (DEGs) across various conditions. It employs a negative binomial model to account for overdispersion in count data and uses empirical Bayes methods to stabilize variance estimates. Key features include effective normalization techniques, robust statistical testing for DEGs, and comprehensive visualization tools. While both EdgeR and DESeq are popular tools for RNA-seq data analysis, EdgeR is particularly noted for its flexibility and performance with small sample sizes due to its advanced empirical Bayes methods for dispersion estimation. This makes EdgeR especially useful in experiments where the number of biological replicates is limited. Additionally, EdgeR offers extensive options for data exploration and visualization, enhancing its utility in deriving meaningful biological insights. This protocol was created based on <span style="color: #7253ec;"><b>edgeR version 4.1.2</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>R version 4.4.0 under macOS 12.4 environment.</b></span></p><h2 class="fs-6 fw-400 lh-tight" id="the-importance-of-normalization"> <a href="#the-importance-of-normalization" class="anchor-heading" aria-labelledby="the-importance-of-normalization"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>The Importance of Normalization</u></h2><p>RNA-seq data analysis requires normalization to fairly compare samples by accounting for differences in sequencing depth and library size. For example, suppose the control group and drug-treated group have the following counts for genes A, B, C, D, and E. In the control group, the counts are 100, 200, 300, 400, and 500, respectively. In the drug-treated group, the counts are 100, 200, 300, 400, and 2000, respectively. The total counts for the control group and drug-treated group are 1500 (100+200+300+400+500) and 3000 (100+200+300+400+2000), respectively. At first glance, the gene expression levels seem identical between the groups. However, normalization reveals a different picture (<strong>see the bellow table</strong>).</p><h4 id="normalized-values"> <a href="#normalized-values" class="anchor-heading" aria-labelledby="normalized-values"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Normalized Values</h4><div class="table-wrapper"><table><thead><tr><th>Gene<th>Control Group<th>Drug-Treated Group<tbody><tr><td>A<td>100/1500 = <strong>0.067</strong><td>100/3000 = <strong>0.033</strong><tr><td>B<td>200/1500 = <strong>0.133</strong><td>200/3000 = <strong>0.067</strong><tr><td>C<td>300/1500 = <strong>0.2</strong><td>300/3000 = <strong>0.1</strong><tr><td>D<td>400/1500 = <strong>0.267</strong><td>400/3000 = <strong>0.133</strong><tr><td>E<td>500/1500 = <strong>0.333</strong><td>2000/3000 = <strong>0.667</strong></table></div><ul><li>Normalized values indicate that genes A, B, C, and D are relatively less expressed in the drug-treated group, while gene E is significantly more expressed. Although the raw counts are the same, normalization considers the total count differences between samples.<li><strong>Sometimes people try to determine gene expression trends by looking at read counts alone. This is never a scientific method and always requires normalization and statistical analysis.</strong></ul><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script id="MathJax-script" async="" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><p>In practice, EdgeR employs more sophisticated normalization methods beyond simple division by total counts. EdgeR uses the <b>TMM (Trimmed Mean of M-values)</b> method to calculate normalization factors, adjusting for sequencing depth and library size differences:</p><ol><li><p><strong>Calculate \( M \) and \( A \) values</strong>:</p><p>\[ M = \log_2 \left(\frac{X}{\bar{X}}\right) - \log_2 \left(\frac{Y}{\bar{Y}}\right) \]</p><p>\[ A = \frac{1}{2} \left(\log_2(X) + \log_2(Y)\right) \]</p><p>\( X \): Read count for a particular gene in sample 1, \(\bar{X}\): Geometric mean of read counts in sample 1</p><p>\( Y \): Read count for the same gene in sample 2, \(\bar{Y}\): Geometric mean of read counts in sample 2</p><li><p><strong>Compute the weight (\( w \))</strong>:</p><p>\[ w = \frac{\text{Trimmed mean of } M}{\log_2(\bar{Y}/\bar{X})} \]</p><p>The weight \( w \) is used to normalize the read counts, reducing technical variability and allowing for more accurate comparison of gene expression levels between samples.</p></ol><p>This complex normalization reduces technical variability and enhances the detection of genuine biological differences. EdgeR combines this advanced normalization with robust statistical techniques to accurately identify differentially expressed genes.</p><h1 class="fs-7 fw-400 lh-tight" id="installation-edger"> <a href="#installation-edger" class="anchor-heading" aria-labelledby="installation-edger"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation edgeR</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Windows-0078D6?style=flat&amp;logo=windows&amp;logoColor=white" alt="EnvTest3" /> <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html"><img src="https://img.shields.io/badge/BioC-276DC3?style=flat&amp;logo=r&amp;logoColor=white" alt="BioC" /></a> <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html"><img src="https://img.shields.io/badge/release%20version-4.2.0-DarkSeaGreen.svg?labelColor=black" alt="CRAN" /></a> <a href="https://bioconductor.org/checkResults/release/bioc-LATEST/edgeR/"><img src="https://img.shields.io/badge/status-stable-DarkSeaGreen?style=flat&amp;colorA=000000" alt="Project Status" /></a></p><ol><li><strong>Check and install edgeR</strong> Start by ensuring the BiocManager package is installed, then install edgeR using it:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">require</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w">
 </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">)</span><span class="w">

 </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"edgeR"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li><strong>Install Required Libraries</strong> Install additional libraries necessary for edgeR to function properly. You can install these packages manually using the <code class="language-plaintext highlighter-rouge">BiocManager::install("PackageName")</code>. The list of packages is shown below.</ol><p style="margin-left: 40px; margin-bottom: 1px;"><strong>Library list (<code>PackageName</code>)</strong></p><ul style="margin-top: 1px; padding-left: 80px; list-style-type: none;"><li style="margin-bottom: 0px;"> <code>limma</code>: EdgeR dependent package<li style="margin-bottom: 0px;"> <code>HTSFilter</code>: A filtering tool for a global Jaccard similarity index<li style="margin-bottom: 0px;"> <code>MASS</code>: Functions and datasets to support Venables and Ripley<li style="margin-bottom: 0px;"> <code>lattice</code>: For trellis graphics system<li style="margin-bottom: 0px;"> <code>mixOmics</code>: For multivariate methods<li style="margin-bottom: 0px;"> <code>biomaRt</code>: For annotation<li style="margin-bottom: 0px;"> <code>ggplot2</code><li style="margin-bottom: 0px;"> <code>RColorBrewer</code><li style="margin-bottom: 0px;"> <code>rtracklayer</code><li style="margin-bottom: 0px;"> <code>GenomeInfoDb</code></ul><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-running-guide-edger"> <a href="#step-by-step-running-guide-edger" class="anchor-heading" aria-labelledby="step-by-step-running-guide-edger"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Running Guide edgeR</h1><h3 id="1-load-edger-and-its-dependency-packages"> <a href="#1-load-edger-and-its-dependency-packages" class="anchor-heading" aria-labelledby="1-load-edger-and-its-dependency-packages"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Load edgeR and its dependency packages</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">edgeR</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">limma</span><span class="p">)</span><span class="w"> 
</span><span class="n">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span><span class="w"> 
</span><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span><span class="w"> 
</span><span class="n">library</span><span class="p">(</span><span class="n">stats</span><span class="p">)</span><span class="w"> </span><span class="c1"># For PCA plot</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><h3 id="2-prepare-metadata"> <a href="#2-prepare-metadata" class="anchor-heading" aria-labelledby="2-prepare-metadata"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Prepare metadata</h3><ul><li>Metadata in RNA-seq is used to organize samples, adjust statistical models, ensure reproducibility, facilitate comparisons, and manage data effectively.</ul><p class="note">the metadata must be associated with the raw count table. This file can be created in Excel, saved as a <code class="language-plaintext highlighter-rouge">csv</code> and <code class="language-plaintext highlighter-rouge">txt</code>, and imported via <code class="language-plaintext highlighter-rouge">read.csv</code> and <code class="language-plaintext highlighter-rouge">read.table</code>, or it can be created by typing directly in R as <strong>shown below</strong>.</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">File</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"shControl_Unt_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shControl_Unt_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shControl_H2O2_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shControl_H2O2_rep2"</span><span class="p">,</span><span class="w">
        </span><span class="s2">"shMLL1_Unt_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shMLL1_Unt_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shMLL1_H2O2_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shMLL1_H2O2_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shUTX_Unt_rep1"</span><span class="p">,</span><span class="w">
        </span><span class="s2">"shUTX_Unt_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shUTX_H2O2_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shUTX_H2O2_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shNRF2_Unt_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shNRF2_Unt_rep2"</span><span class="p">,</span><span class="w">
        </span><span class="s2">"shNRF2_H2O2_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shNRF2_H2O2_rep2"</span><span class="p">)</span><span class="w">
</span><span class="n">Rep</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep2"</span><span class="p">,</span><span class="w">
        </span><span class="s2">"rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rep2"</span><span class="p">)</span><span class="w">
</span><span class="n">Condition</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Unt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Unt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Unt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Unt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Unt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Unt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2"</span><span class="p">,</span><span class="w">
            </span><span class="s2">"Unt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Unt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2"</span><span class="p">)</span><span class="w">
</span><span class="n">KD</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MLL1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MLL1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MLL1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MLL1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"UTX"</span><span class="p">,</span><span class="w"> </span><span class="s2">"UTX"</span><span class="p">,</span><span class="w"> </span><span class="s2">"UTX"</span><span class="p">,</span><span class="w"> </span><span class="s2">"UTX"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NRF2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NRF2"</span><span class="p">,</span><span class="w">
        </span><span class="s2">"NRF2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NRF2"</span><span class="p">)</span><span class="w">
</span><span class="n">Group</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"CU"</span><span class="p">,</span><span class="w"> </span><span class="s2">"CU"</span><span class="p">,</span><span class="w"> </span><span class="s2">"CH"</span><span class="p">,</span><span class="w"> </span><span class="s2">"CH"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MU"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MU"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MH"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MH"</span><span class="p">,</span><span class="w"> </span><span class="s2">"UU"</span><span class="p">,</span><span class="w"> </span><span class="s2">"UU"</span><span class="p">,</span><span class="w"> </span><span class="s2">"UH"</span><span class="p">,</span><span class="w"> </span><span class="s2">"UH"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NU"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NU"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NH"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NH"</span><span class="p">)</span><span class="w">
</span><span class="n">sampleInfo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">File</span><span class="p">,</span><span class="w"> </span><span class="n">Rep</span><span class="p">,</span><span class="w"> </span><span class="n">KD</span><span class="p">,</span><span class="w"> </span><span class="n">Condition</span><span class="p">,</span><span class="w"> </span><span class="n">Group</span><span class="p">)</span><span class="w">
</span><span class="n">sampleInfo</span><span class="w"> </span><span class="c1"># Check metadata</span><span class="w">
</span></code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><p><br /></p><ul><li><strong>Output:</strong><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; sampleInfo
                  File  Rep   KD Condition Group
  1   shControl_Unt_rep1 rep1   WT       Unt    CU
  2   shControl_Unt_rep2 rep2   WT       Unt    CU
  3  shControl_H2O2_rep1 rep1   WT      H2O2    CH
  4  shControl_H2O2_rep2 rep2   WT      H2O2    CH
  5      shMLL1_Unt_rep1 rep1 MLL1       Unt    MU
  6      shMLL1_Unt_rep2 rep2 MLL1       Unt    MU
  7     shMLL1_H2O2_rep1 rep1 MLL1      H2O2    MH
  8     shMLL1_H2O2_rep2 rep2 MLL1      H2O2    MH
  9       shUTX_Unt_rep1 rep1  UTX       Unt    UU
  10      shUTX_Unt_rep2 rep2  UTX       Unt    UU
  11     shUTX_H2O2_rep1 rep1  UTX      H2O2    UH
  12     shUTX_H2O2_rep2 rep2  UTX      H2O2    UH
  13     shNRF2_Unt_rep1 rep1 NRF2       Unt    NU
  14     shNRF2_Unt_rep2 rep2 NRF2       Unt    NU
  15    shNRF2_H2O2_rep1 rep1 NRF2      H2O2    NH
  16    shNRF2_H2O2_rep2 rep2 NRF2      H2O2    NH
</code></pre></div></div></ul><h3 id="3-load-raw-count-table"> <a href="#3-load-raw-count-table" class="anchor-heading" aria-labelledby="3-load-raw-count-table"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. Load raw count table</h3><ul><li><i>This example is based on the raw count table obtained via htseq-count.</i><li>The raw count table is a tab-delimited <code class="language-plaintext highlighter-rouge">.txt</code> file where the first column contains gene or transcript IDs, and the remaining columns are the names of each sample. However, it lacks a header, making columns difficult to distinguish, and the last four rows contain summary information.<li>While you can open this file in Excel to add a header and remove the last 4 lines, I will perform these steps programmatically in R.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">rawCountTable</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="s2">"~/Desktop/H2O2.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">)</span><span class="w">
  </span><span class="n">rawCountTable</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rawCountTable</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">rawCountTable</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">
  </span><span class="n">rownames</span><span class="p">(</span><span class="n">rawCountTable</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rawCountTable</span><span class="p">[,</span><span class="m">1</span><span class="p">]</span><span class="w">
  </span><span class="n">rawCountTable</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rawCountTable</span><span class="p">[,</span><span class="m">-1</span><span class="p">]</span><span class="w">
  </span><span class="n">colnames</span><span class="p">(</span><span class="n">rawCountTable</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"shControl_Unt_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shControl_Unt_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shControl_H2O2_rep1"</span><span class="p">,</span><span class="w">
                          </span><span class="s2">"shControl_H2O2_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shMLL1_Unt_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shMLL1_Unt_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shMLL1_H2O2_rep1"</span><span class="p">,</span><span class="w">
                          </span><span class="s2">"shMLL1_H2O2_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shUTX_Unt_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shUTX_Unt_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shUTX_H2O2_rep1"</span><span class="p">,</span><span class="w">
                          </span><span class="s2">"shUTX_H2O2_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shNRF2_Unt_rep1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shNRF2_Unt_rep2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"shNRF2_H2O2_rep1"</span><span class="p">,</span><span class="w">
                          </span><span class="s2">"shNRF2_H2O2_rep2"</span><span class="p">)</span><span class="w">
  </span><span class="n">head</span><span class="p">(</span><span class="n">rawCountTable</span><span class="p">)</span><span class="w"> </span><span class="c1"># Check raw count table</span><span class="w">

  </span><span class="c1"># After processing the data in Excel, you can load it in R using the following command:</span><span class="w">
  </span><span class="n">rawCountTable</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"~/Desktop/H2O2.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
  </span><span class="n">rawCountTable</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="s2">"/path/to/the/file.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">","</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><p><br /></p><ul><li><strong>Output:</strong><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; head(rawCountTable)
                  shControl_Unt_rep1 shControl_Unt_rep2 shControl_H2O2_rep1 shControl_H2O2_rep2 shMLL1_Unt_rep1
  ENSG00000000003                  0                 12                   1                   1               1
  ENSG00000000005                119                140                 133                 160             141
  ENSG00000000419                  1                  1                   1                   1               1
  ENSG00000000457                  2                  5                   3                   5               1
  ENSG00000000460                273                196                 279                 282             242
  ENSG00000000938                 55                 69                  98                 117              76
                  shMLL1_Unt_rep2 shMLL1_H2O2_rep1 shMLL1_H2O2_rep2 shUTX_Unt_rep1 shUTX_Unt_rep2 shUTX_H2O2_rep1
  ENSG00000000003               1                1                1              0              1               0
  ENSG00000000005             182              151              166            142            109             118
  ENSG00000000419               2                2                2              2              3               1
  ENSG00000000457               1                1                1              2              3               7
  ENSG00000000460             193              218              209            214            235             203
  ENSG00000000938              57               61               77            128            117              97
                  shUTX_H2O2_rep2 shNRF2_Unt_rep1 shNRF2_Unt_rep2 shNRF2_H2O2_rep1 shNRF2_H2O2_rep2
  ENSG00000000003               1               1               2                1                0
  ENSG00000000005             129             133             147              178              170
  ENSG00000000419               0               1               1                0                0
  ENSG00000000457               0              12               9               13                4
  ENSG00000000460             204             100             171              179              180
  ENSG00000000938             108              61              52               70               61
</code></pre></div></div></ul><h3 id="4-object-setup-for-rna-seq-analysis"> <a href="#4-object-setup-for-rna-seq-analysis" class="anchor-heading" aria-labelledby="4-object-setup-for-rna-seq-analysis"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4. Object setup for RNA-seq analysis</h3><ul><li>In this step, I will set up a <code class="language-plaintext highlighter-rouge">DGEList</code> object for edgeR analysis using the previously prepared raw count data and metadata.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">dgeFull</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DGEList</span><span class="p">(</span><span class="n">rawCountTable</span><span class="p">,</span><span class="w"> </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sampleInfo</span><span class="o">$</span><span class="n">Group</span><span class="p">)</span><span class="w">
  </span><span class="n">dgeFull</span><span class="w"> </span><span class="c1"># check the object</span><span class="w">
</span></code></pre></div></div><p><strong>Key arguments</strong></p><ul><li><code class="language-plaintext highlighter-rouge">rawCountTable</code>: Specifies the object for raw count table.<li><code class="language-plaintext highlighter-rouge">group = &lt;Name&gt;</code>: Specifies the grouping of samples based on metadata, such as experimental conditions or sample types.</ul><p class="note">This grouping is crucial for DEG analysis in edgeR.</p><p><strong>Output:</strong></p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; dgeFull
  An object of class "DGEList"
  $counts
                  shControl_Unt_rep1 shControl_Unt_rep2 shControl_H2O2_rep1 shControl_H2O2_rep2 shMLL1_Unt_rep1
  ENSG00000000003                  0                 12                   1                   1               1
  ENSG00000000005                119                140                 133                 160             141
  ENSG00000000419                  1                  1                   1                   1               1
  ENSG00000000457                  2                  5                   3                   5               1
  ENSG00000000460                273                196                 279                 282             242
                  shMLL1_Unt_rep2 shMLL1_H2O2_rep1 shMLL1_H2O2_rep2 shUTX_Unt_rep1 shUTX_Unt_rep2 shUTX_H2O2_rep1
  ENSG00000000003               1                1                1              0              1               0
  ENSG00000000005             182              151              166            142            109             118
  ENSG00000000419               2                2                2              2              3               1
  ENSG00000000457               1                1                1              2              3               7
  ENSG00000000460             193              218              209            214            235             203
                  shUTX_H2O2_rep2 shNRF2_Unt_rep1 shNRF2_Unt_rep2 shNRF2_H2O2_rep1 shNRF2_H2O2_rep2
  ENSG00000000003               1               1               2                1                0
  ENSG00000000005             129             133             147              178              170
  ENSG00000000419               0               1               1                0                0
  ENSG00000000457               0              12               9               13                4
  ENSG00000000460             204             100             171              179              180
  61494 more rows ...

  $samples
                      group lib.size norm.factors
  shControl_Unt_rep1     CU 11738068            1
  shControl_Unt_rep2     CU 11905476            1
  shControl_H2O2_rep1    CH 12069056            1
  shControl_H2O2_rep2    CH 12237122            1
  shMLL1_Unt_rep1        MU 11781709            1
  11 more rows ...
</code></pre></div></div></ul><h3 id="5-check-library-size-and-distribution"> <a href="#5-check-library-size-and-distribution" class="anchor-heading" aria-labelledby="5-check-library-size-and-distribution"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 5. Check Library Size and Distribution</h3><ul><li>In this step, I will check the library size and distribution of the samples before proceeding with normalization.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">pseudoCounts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">log2</span><span class="p">(</span><span class="n">dgeFull</span><span class="o">$</span><span class="n">counts</span><span class="m">+1</span><span class="p">)</span><span class="w"> </span><span class="c1"># the values are converted to non-zero values.</span><span class="w">
  </span><span class="n">FullCol</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">factor</span><span class="p">(</span><span class="n">sampleInfo</span><span class="o">$</span><span class="n">Group</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">
  </span><span class="n">boxplot</span><span class="p">(</span><span class="n">pseudoCounts</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="n">FullCol</span><span class="p">,</span><span class="w"> </span><span class="n">las</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="w"> 
  </span><span class="n">abline</span><span class="p">(</span><span class="n">h</span><span class="o">=</span><span class="n">median</span><span class="p">(</span><span class="n">as.matrix</span><span class="p">(</span><span class="n">pseudoCounts</span><span class="p">)),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s2">"blue"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Save boxplot</span><span class="w">
</span></code></pre></div></div><p><strong>Output:</strong></p><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/beforeN.png" alt="beforeNormalization" style="width: 50vh; height: auto;" /></p></ul><h3 id="6-tmm-normalization"> <a href="#6-tmm-normalization" class="anchor-heading" aria-labelledby="6-tmm-normalization"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 6. TMM Normalization</h3><ul><li>In this process, I will perform TMM normalization.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">dgeFull</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DGEList</span><span class="p">(</span><span class="n">dgeFull</span><span class="o">$</span><span class="n">counts</span><span class="p">[</span><span class="n">apply</span><span class="p">(</span><span class="n">dgeFull</span><span class="o">$</span><span class="n">counts</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">sum</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="p">],</span><span class="w"> </span><span class="n">group</span><span class="o">=</span><span class="n">dgeFull</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span><span class="p">)</span><span class="w">
  </span><span class="n">dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">calcNormFactors</span><span class="p">(</span><span class="n">dgeFull</span><span class="p">)</span><span class="w">
  </span><span class="n">design</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">model.matrix</span><span class="p">(</span><span class="o">~</span><span class="m">0</span><span class="o">+</span><span class="n">group</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">dge</span><span class="o">$</span><span class="n">samples</span><span class="p">)</span><span class="w">
  </span><span class="n">dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">estimateDisp</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span><span class="w"> </span><span class="n">design</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">design</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>in these codes,<ul><li><strong>Line 1</strong>: Creates a DGEList object with count data, excluding genes with zero counts across all samples, and includes group information. This step improves data quality by removing noise.<li><strong>Line 2</strong>: Calculates TMM normalization factors to adjust for differences in library sizes between samples, ensuring accurate comparisons.<li><strong>Line 3</strong>: Constructs a design matrix for DEG analysis, indicating group membership of each sample. The <code class="language-plaintext highlighter-rouge">~0+group</code> command facilitates group comparisons.<li><strong>Line 4</strong>: Estimates the dispersion for each gene, crucial for reliable DEG analysis, by accounting for variability in gene expression levels.</ul><p><strong>Output:</strong></p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; dgeFull &lt;- DGEList(dgeFull$counts[apply(dgeFull$counts, 1, sum) != 0, ], group=dgeFull$samples$group)
  &gt; dgeFull$counts  # **Check the selected transcripts. I will not show results in this example because it is too long.
  &gt; dge &lt;- calcNormFactors(dgeFull)
  &gt; dge$samples # Check norm.factor
                      group lib.size norm.factors
  shControl_Unt_rep1     CU 11738068    1.0531166
  shControl_Unt_rep2     CU 11905476    0.8733873
  shControl_H2O2_rep1    CH 12069056    1.1370291
  shControl_H2O2_rep2    CH 12237122    0.8497171
  shMLL1_Unt_rep1        MU 11781709    1.1008688
  shMLL1_Unt_rep2        MU 11949117    0.8701975
  shMLL1_H2O2_rep1       MH 10394452    1.0873398
  shMLL1_H2O2_rep2       MH 10562518    0.9844330
  shUTX_Unt_rep1         UU 11606646    1.1195918
  shUTX_Unt_rep2         UU 11774054    0.8831360
  shUTX_H2O2_rep1        UH 10663731    1.1049622
  shUTX_H2O2_rep2        UH 10831797    0.9599600
  shNRF2_Unt_rep1        NU 10051201    1.0067264
  shNRF2_Unt_rep2        NU 10218609    1.0175643
  shNRF2_H2O2_rep1       NH 10533324    1.0512145
  shNRF2_H2O2_rep2       NH 10701390    0.9716580
</code></pre></div></div><p>After caculation of norm.factors:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; design &lt;- model.matrix(~0+group, data=dge$samples)
  &gt; design
                      groupCH groupCU groupMH groupMU groupNH groupNU groupUH groupUU
  shControl_Unt_rep1        0       1       0       0       0       0       0       0
  shControl_Unt_rep2        0       1       0       0       0       0       0       0
  shControl_H2O2_rep1       1       0       0       0       0       0       0       0
  shControl_H2O2_rep2       1       0       0       0       0       0       0       0
  shMLL1_Unt_rep1           0       0       0       1       0       0       0       0
  shMLL1_Unt_rep2           0       0       0       1       0       0       0       0
  shMLL1_H2O2_rep1          0       0       1       0       0       0       0       0
  shMLL1_H2O2_rep2          0       0       1       0       0       0       0       0
  shUTX_Unt_rep1            0       0       0       0       0       0       0       1
  shUTX_Unt_rep2            0       0       0       0       0       0       0       1
  shUTX_H2O2_rep1           0       0       0       0       0       0       1       0
  shUTX_H2O2_rep2           0       0       0       0       0       0       1       0
  shNRF2_Unt_rep1           0       0       0       0       0       1       0       0
  shNRF2_Unt_rep2           0       0       0       0       0       1       0       0
  shNRF2_H2O2_rep1          0       0       0       0       1       0       0       0
  shNRF2_H2O2_rep2          0       0       0       0       1       0       0       0
  attr(,"assign")
  [1] 1 1 1 1 1 1 1 1
  attr(,"contrasts")
  attr(,"contrasts")$group
  [1] "contr.treatment"
</code></pre></div></div></ul><h3 id="7-check-library-size-and-distribution-after-normalization"> <a href="#7-check-library-size-and-distribution-after-normalization" class="anchor-heading" aria-labelledby="7-check-library-size-and-distribution-after-normalization"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 7. Check Library Size and Distribution after Normalization</h3><ul><li>Next, I will now plot a boxplot based on the normalized data to observe the correlation between each sample or between groups.<li><strong>Depicting a box plot</strong><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">normCounts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cpm</span><span class="p">(</span><span class="n">dge</span><span class="p">)</span><span class="w">
  </span><span class="n">pseudoNormCounts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">log2</span><span class="p">(</span><span class="n">normCounts</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
  </span><span class="n">FullColNorm</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">factor</span><span class="p">(</span><span class="n">dge</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">
  </span><span class="n">boxplot</span><span class="p">(</span><span class="n">pseudoNormCounts</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="n">FullColNorm</span><span class="p">,</span><span class="w"> </span><span class="n">las</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="w">
  </span><span class="n">abline</span><span class="p">(</span><span class="n">h</span><span class="o">=</span><span class="n">median</span><span class="p">(</span><span class="n">as.matrix</span><span class="p">(</span><span class="n">pseudoNormCounts</span><span class="p">)),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="s2">"blue"</span><span class="p">)</span><span class="w"> </span><span class="c1"># can save a plot</span><span class="w">
</span></code></pre></div></div><li><strong>Output:</strong><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/afterN.png" alt="afterNormalization" style="width: 50vh; height: auto;" /></p><li><strong>Drwing a principal component analysis (PCA) plot</strong>: PCA visualizes the major sources of variation in the data by transforming it into principal components. It highlights the directions (principal components) where the data varies the most, helping to identify patterns and groupings among the samples. The PCA plot is useful for understanding the overall structure and key sources of variation in the dataset.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">pca_result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">prcomp</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">pseudoNormCounts</span><span class="p">),</span><span class="w"> </span><span class="n">scale.</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
  </span><span class="n">summary</span><span class="p">(</span><span class="n">pca_result</span><span class="p">)</span><span class="w">

  </span><span class="c1"># PCA Visulaize</span><span class="w">
  </span><span class="n">pca_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">Sample</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">pca_result</span><span class="o">$</span><span class="n">x</span><span class="p">),</span><span class="w"> 
                      </span><span class="n">PC1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pca_result</span><span class="o">$</span><span class="n">x</span><span class="p">[,</span><span class="m">1</span><span class="p">],</span><span class="w"> 
                      </span><span class="n">PC2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pca_result</span><span class="o">$</span><span class="n">x</span><span class="p">[,</span><span class="m">2</span><span class="p">],</span><span class="w"> 
                      </span><span class="n">Group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">dge</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span><span class="p">)</span><span class="w">
    
  </span><span class="c1"># PCA plot</span><span class="w">
  </span><span class="n">ggplot</span><span class="p">(</span><span class="n">pca_data</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">PC1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">PC2</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Group</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"PCA Plot"</span><span class="p">,</span><span class="w">
      </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Principal Component 1"</span><span class="p">,</span><span class="w">
      </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Principal Component 2"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_minimal</span><span class="p">()</span><span class="w">
</span></code></pre></div></div><li><strong>Making a multidimensional scaling (MDS) plot</strong>: MDS plots represent the pairwise distances between samples in a low-dimensional space, preserving the original distances as much as possible. This helps in visualizing the similarity or dissimilarity among samples based on their expression profiles. The MDS plot is particularly useful for detecting patterns and relationships in the data that are not immediately apparent.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Basal format</span><span class="w">
  </span><span class="n">plotMDS</span><span class="p">(</span><span class="n">pseudoNormCounts</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Advanced format</span><span class="w">
  </span><span class="n">plotMDS</span><span class="p">(</span><span class="n">pseudoNormCounts</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">FullColNorm</span><span class="p">,</span><span class="w"> </span><span class="n">pch</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">19</span><span class="p">,</span><span class="w"> </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"MDS Plot"</span><span class="p">)</span><span class="w">
  </span><span class="n">legend</span><span class="p">(</span><span class="s2">"topright"</span><span class="p">,</span><span class="w"> </span><span class="n">legend</span><span class="o">=</span><span class="n">levels</span><span class="p">(</span><span class="n">factor</span><span class="p">(</span><span class="n">dge</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span><span class="p">)),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">levels</span><span class="p">(</span><span class="n">factor</span><span class="p">(</span><span class="n">dge</span><span class="o">$</span><span class="n">samples</span><span class="o">$</span><span class="n">group</span><span class="p">))),</span><span class="w"> </span><span class="n">pch</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">19</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li><strong>Prepare log10 transformed CPM</strong><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">prior.count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w">
  </span><span class="n">cpmlog</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cpm</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span><span class="w"> </span><span class="n">log</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w"> </span><span class="n">prior.count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">prior.count</span><span class="p">)</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">cpmlog</span><span class="p">,</span><span class="w"> </span><span class="s2">"path/to/the/file.csv"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Save the CPM file</span><span class="w">
</span></code></pre></div></div></ul><h3 id="8-remove-overdispersion-and-deg-analysis"> <a href="#8-remove-overdispersion-and-deg-analysis" class="anchor-heading" aria-labelledby="8-remove-overdispersion-and-deg-analysis"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 8. Remove Overdispersion and DEG Analysis</h3><ul><li><p>In this step, I will perform DEG analysis based on the TMM normalized counts (<code class="language-plaintext highlighter-rouge">dge</code>) and the <code class="language-plaintext highlighter-rouge">design</code> matrix. <strong>Overdispersion</strong>, which is the presence of greater variability in the data than expected under a negative binomial or Poisson distribution, must be addressed to ensure accurate results.</p><li><p>This can be effectively removed using linear regression approaches, such as the <strong>Generalized Linear Model-Quasi Likelihood (GLM-QL)</strong> method. The GLM-QL approach accounts for overdispersion by modeling the extra variability and provides more reliable statistical inferences.</p><li><p>If no specific parameters are added, the GLM-QL typically calculates statistical significance by comparing the degrees of freedom between samples. Additionally, the false discovery rate (FDR) is computed using the Benjamini-Hochberg (BH) method to control for multiple testing and reduce the likelihood of false positives.</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Model fitting using the GML-QL and modeling overdispersion</span><span class="w">
  </span><span class="n">fit</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmQLFit</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span><span class="w"> </span><span class="n">design</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">design</span><span class="p">)</span><span class="w">
  </span><span class="n">glmQLFTest</span><span class="p">(</span><span class="n">fit</span><span class="p">)</span><span class="w"> </span><span class="c1"># (Optional) Test QL F-test across all samples</span><span class="w">

  </span><span class="c1"># The process of specifying what to compare based on design object</span><span class="w">
  </span><span class="n">my.contrasts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">makeContrasts</span><span class="p">(</span><span class="n">CU_CH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupCH</span><span class="o">-</span><span class="n">groupCU</span><span class="p">,</span><span class="w"> </span><span class="n">CU_MU</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupMU</span><span class="o">-</span><span class="n">groupCU</span><span class="p">,</span><span class="w"> 
                              </span><span class="n">CU_UU</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupUU</span><span class="o">-</span><span class="n">groupCU</span><span class="p">,</span><span class="w"> </span><span class="n">CU_NU</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupNU</span><span class="o">-</span><span class="n">groupCU</span><span class="p">,</span><span class="w">
                              </span><span class="n">MU_MH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupMH</span><span class="o">-</span><span class="n">groupMU</span><span class="p">,</span><span class="w"> </span><span class="n">UU_UH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupUH</span><span class="o">-</span><span class="n">groupUU</span><span class="p">,</span><span class="w">
                              </span><span class="n">NU_NH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupNH</span><span class="o">-</span><span class="n">groupNU</span><span class="p">,</span><span class="w"> </span><span class="n">CH_MH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupMH</span><span class="o">-</span><span class="n">groupCH</span><span class="p">,</span><span class="w">
                              </span><span class="n">CH_UH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupUH</span><span class="o">-</span><span class="n">groupCH</span><span class="p">,</span><span class="w"> </span><span class="n">CH_NH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupNH</span><span class="o">-</span><span class="n">groupCH</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">design</span><span class="p">)</span><span class="w"> 

  </span><span class="c1"># DEG Analysis</span><span class="w">
  </span><span class="n">H2O2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmQLFTest</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">contrast</span><span class="o">=</span><span class="n">my.contrasts</span><span class="p">[,</span><span class="s2">"CU_CH"</span><span class="p">])</span><span class="w"> </span><span class="c1"># DEG analysis using the QL F-test under the [my.contrasts] conditions.</span><span class="w">
  </span><span class="n">H2O2_FDR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">topTags</span><span class="p">(</span><span class="n">H2O2</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">Inf</span><span class="p">)</span><span class="o">$</span><span class="n">table</span><span class="w"> </span><span class="c1"># FDR calculation with BH (default) method</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">H2O2_FDR</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2_DEG.csv"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Save as csv format</span><span class="w">

  </span><span class="n">shMLL1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmQLFTest</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">contrast</span><span class="o">=</span><span class="n">my.contrasts</span><span class="p">[,</span><span class="s2">"MU_MH"</span><span class="p">])</span><span class="w">
  </span><span class="n">shMLL1_FDR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">topTags</span><span class="p">(</span><span class="n">shMLL1</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">Inf</span><span class="p">)</span><span class="o">$</span><span class="n">table</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">shMLL1_FDR</span><span class="p">,</span><span class="w"> </span><span class="s2">"shMLL1_DEG.csv"</span><span class="p">)</span><span class="w">

  </span><span class="n">shUTX</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmQLFTest</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">contrast</span><span class="o">=</span><span class="n">my.contrasts</span><span class="p">[,</span><span class="s2">"UU_UH"</span><span class="p">])</span><span class="w">
  </span><span class="n">shUTX_FDR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">topTags</span><span class="p">(</span><span class="n">shUTX</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">Inf</span><span class="p">)</span><span class="o">$</span><span class="n">table</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">shUTX_FDR</span><span class="p">,</span><span class="w"> </span><span class="s2">"shUTX_DEG.csv"</span><span class="p">)</span><span class="w">

  </span><span class="n">shNRF2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmQLFTest</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">contrast</span><span class="o">=</span><span class="n">my.contrasts</span><span class="p">[,</span><span class="s2">"NU_NH"</span><span class="p">])</span><span class="w">
  </span><span class="n">shNRF2_FDR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">topTags</span><span class="p">(</span><span class="n">shNRF2</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">Inf</span><span class="p">)</span><span class="o">$</span><span class="n">table</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">shNRF2_FDR</span><span class="p">,</span><span class="w"> </span><span class="s2">"shNRF2_DEG.csv"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><ul><li><strong>Output:</strong> <strong><code class="language-plaintext highlighter-rouge">logFC</code></strong> represents the log2-transformed value of the target group’s (H2O2) expression divided by the control group’s (Unt) expression, while <strong><code class="language-plaintext highlighter-rouge">logCPM</code></strong> is the average of log2(CPM + 1) for each sample. The <strong><code class="language-plaintext highlighter-rouge">F</code></strong> value is the result of the F-test (glmQL), and the <strong><code class="language-plaintext highlighter-rouge">PValue</code>(p-value)</strong> indicates the statistical significance of the result. <strong><code class="language-plaintext highlighter-rouge">FDR</code></strong> is the False Discovery Rate, which is a multiple testing correction of the p-value. <strong>In general, values with $|logFC|$ &gt; 1 and FDR &lt; 0.05 are considered significant DEGs</strong> (<strong>see the bellow</strong>).<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; head(H2O2_FDR)
                      logFC    logCPM         F        PValue           FDR
  ENSG00000261089 -7.753531 3.5081044 1499.6617  0.000000e+00  0.000000e+00
  ENSG00000205358 -8.403098 1.3033323  706.6869 1.355224e-155 4.167246e-151
  ENSG00000181019  3.767159 2.5203353  486.9699 7.338978e-108 1.504466e-103
  ENSG00000181625 -5.324815 1.1710369  311.2585  1.220608e-69  1.876655e-65
  ENSG00000204472  6.711876 0.8264076  227.5472  2.097907e-51  2.580384e-47
  ENSG00000155070 -1.582992 3.5280535  148.1809  4.380133e-34  4.489563e-30
</code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">IMPORTANT</span><ul style="list-style-type: none; padding-left: 20px;"><li style="margin: 0; padding-left: 20px;"> More genes with FDR &gt; 0.05 could emerge in your actual DEG analysis, despite the distinctiveness evident in the PCA and MDS plot between the samples from the two conditions under comparison.<li style="margin: 0; padding-left: 20px;"> In this case, it may be helpful to change the test for statistical significance to <b>fitness (glmLRT; Generalized Linear Model Likelihood Ratio Test function)</b> rather than <b>degrees of freedom (glmQL function).</b><li style="margin: 0; padding-left: 20px;"><p>Here is an example of R code for DEG analysis using the glmLRT:</p><figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">fit</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmQLFit</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span><span class="w"> </span><span class="n">design</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">design</span><span class="p">)</span><span class="w">
</span><span class="c1"># DEG analysis using the Likelihood Ratio Test</span><span class="w">
</span><span class="n">H2O2_LRT</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmLRT</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">contrast</span><span class="o">=</span><span class="n">my.contrasts</span><span class="p">[,</span><span class="s2">"CU_CH"</span><span class="p">])</span><span class="w">
</span><span class="n">H2O2_LRT_FDR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">topTags</span><span class="p">(</span><span class="n">H2O2_LRT</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="kc">Inf</span><span class="p">)</span><span class="o">$</span><span class="n">table</span><span class="w"> </span><span class="c1"># p.adjust.method = "[BY/holm/etc]"</span><span class="w">
</span><span class="n">write.csv</span><span class="p">(</span><span class="n">H2O2_LRT_FDR</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2_DEG_glmLRT.csv"</span><span class="p">)</span><span class="w">
            </span></code></pre></figure><li style="margin: 0; padding-left: 20px;"> There are several methods to determine the FDR value based on statistical analysis, including <code>BH</code>, <code>BY</code>, <code>holm</code>, <code>bonferroni</code>, <code>hommel</code>, <code>hochberg</code>, or <code>none</code>, but the <code>BH</code> (default) method has greater tolerance rather than other methods and most users accept <code>BH</code>. You can choose these methods from the <code>topTags</code> command as shown below:<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">shMLL1_FDR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">topTags</span><span class="p">(</span><span class="n">shMLL1</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="kc">Inf</span><span class="p">,</span><span class="w"> </span><span class="n">p.adjust.method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"&lt;BY/holm/etc&gt;"</span><span class="p">)</span><span class="o">$</span><span class="n">table</span></code></pre></figure></ul></div><h3 id="9-single-sample-deg-analysis-importance-of-biological-replicates-in-rna-seq"> <a href="#9-single-sample-deg-analysis-importance-of-biological-replicates-in-rna-seq" class="anchor-heading" aria-labelledby="9-single-sample-deg-analysis-importance-of-biological-replicates-in-rna-seq"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 9. Single Sample DEG analysis: Importance of Biological Replicates in RNA-seq</h3><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 10px; padding-bottom: 5px; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 1px; position: relative; left: -10px;">IMPORTANT</span><p style="margin-top: 2px; margin-bottom: 2px; padding-left: 23px;"><b>Biological replicates</b> in RNA-seq are crucial for ensuring reliable, reproducible results by accounting for natural biological variability. They improve statistical power, ensure reproducibility, and reduce bias. However, researchers may be constrained to single samples due to economic limitations, scarcity of samples, or ethical considerations. Despite these constraints, single-sample RNA-seq can still provide valuable insights, though results must be interpreted with caution, acknowledging the limitations imposed by the lack of biological replicates.</p></div><ul><li><p><strong>In particular</strong>, the coefficient of <strong>dispersion</strong> between samples <strong>cannot</strong> be calculated directly, necessitating the addition of an arbitrary factor. This factor is referred to as the <strong>biological coefficient of variation (BCV)</strong>. Researchers can estimate this value based on prior knowledge or data from analogous experiments.</p><li><p>It is crucial to acknowledge that the inability to calculate the dispersion coefficient implies that linear models and methods for removing overdispersion will not be effective.</p><li><p>Typically, a BCV value of <code class="language-plaintext highlighter-rouge">0.4</code> is utilized. If the dataset under analysis is anticipated to exhibit high biological variation, the BCV value can be set to <code class="language-plaintext highlighter-rouge">0.6</code> or higher. For instance, samples from different species or those collected under varying environmental conditions might necessitate a higher BCV value. Conversely, for datasets with low biological variation, a BCV value of <code class="language-plaintext highlighter-rouge">0.2</code> may be appropriate. For example, a low BCV value might be used when repeatedly collecting the same cell line under identical conditions. <strong>These values should be considered as guidelines rather than strict rules</strong>.</p><li><p>There are two methods to apply BCV to perform DEG analysis, the <code class="language-plaintext highlighter-rouge">extractTest</code> and <code class="language-plaintext highlighter-rouge">glmFit</code> methods. Until the normalization step, both methods are same. Here, I will provide how to use both methods after normalization:</p><ul><li><code class="language-plaintext highlighter-rouge">extractTest</code> method:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">calcNormFactors</span><span class="p">(</span><span class="n">dgeFull</span><span class="p">)</span><span class="w"> </span><span class="c1"># TMM normalization</span><span class="w">
  </span><span class="n">bcv</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0.4</span><span class="w">
  </span><span class="n">dgeTest_H2O2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">exactTest</span><span class="p">(</span><span class="n">dgeFull</span><span class="p">,</span><span class="w"> </span><span class="n">dispersion</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">bcv</span><span class="o">^</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">pair</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"CU"</span><span class="p">,</span><span class="s2">"CH"</span><span class="p">))</span><span class="w">
  </span><span class="n">dgeTest_H2O2_FDR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">topTags</span><span class="p">(</span><span class="n">dgeTest_H2O2</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">Inf</span><span class="p">)</span><span class="o">$</span><span class="n">table</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">dgeTest_H2O2_FDR</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2_DEG_extract,csv"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><p><strong>Output</strong></p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; dgeTest_H2O2 &lt;- exactTest(dgeFull, dispersion = bcv^2, pair = c("CU","CH"))
  &gt; dgeTest_H2O2_FDR &lt;- topTags(dgeTest_H2O2, n = Inf)$table
  &gt; dgeTest_H2O2_FDR &lt;- dgeTest_H2O2_FDR[order(rownames(dgeTest_H2O2_FDR)), ]
  &gt; head(dgeTest_H2O2_FDR)
                    logFC     logCPM    PValue FDR
  ENSG00000000003  0.00000000 -1.5072179 1.0000000   1
  ENSG00000000005  0.01620179  4.6487540 1.0000000   1
  ENSG00000000419 -3.25162822 -1.0936018 1.0000000   1
  ENSG00000000457  0.78712182 -0.4296291 0.7447837   1
  ENSG00000000460 -0.14629616  5.1399138 0.8668006   1
  ENSG00000000938  0.71112235  3.7377811 0.4122082   1
</code></pre></div></div><li><code class="language-plaintext highlighter-rouge">glmFit</code> methods:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">dge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">calcNormFactors</span><span class="p">(</span><span class="n">dgeFull</span><span class="p">)</span><span class="w"> </span><span class="c1"># TMM normalization</span><span class="w">
  </span><span class="n">design</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">model.matrix</span><span class="p">(</span><span class="o">~</span><span class="m">0</span><span class="o">+</span><span class="n">group</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">dge</span><span class="o">$</span><span class="n">samples</span><span class="p">)</span><span class="w"> </span><span class="c1"># design matrix</span><span class="w">
  </span><span class="n">bcv</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0.4</span><span class="w">  
  </span><span class="n">my.contrasts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">makeContrasts</span><span class="p">(</span><span class="n">CU_CH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupCH</span><span class="o">-</span><span class="n">groupCU</span><span class="p">,</span><span class="w"> </span><span class="n">CU_MU</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupMU</span><span class="o">-</span><span class="n">groupCU</span><span class="p">,</span><span class="w"> 
                          </span><span class="n">CU_UU</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupUU</span><span class="o">-</span><span class="n">groupCU</span><span class="p">,</span><span class="w"> </span><span class="n">CU_NU</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupNU</span><span class="o">-</span><span class="n">groupCU</span><span class="p">,</span><span class="w">
                          </span><span class="n">MU_MH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupMH</span><span class="o">-</span><span class="n">groupMU</span><span class="p">,</span><span class="w"> </span><span class="n">UU_UH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupUH</span><span class="o">-</span><span class="n">groupUU</span><span class="p">,</span><span class="w">
                          </span><span class="n">NU_NH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupNH</span><span class="o">-</span><span class="n">groupNU</span><span class="p">,</span><span class="w"> </span><span class="n">CH_MH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupMH</span><span class="o">-</span><span class="n">groupCH</span><span class="p">,</span><span class="w">
                          </span><span class="n">CH_UH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupUH</span><span class="o">-</span><span class="n">groupCH</span><span class="p">,</span><span class="w"> </span><span class="n">CH_NH</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">groupNH</span><span class="o">-</span><span class="n">groupCH</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">design</span><span class="p">)</span><span class="w"> 
  </span><span class="n">fit</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmFit</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span><span class="w"> </span><span class="n">design</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">design</span><span class="p">,</span><span class="w"> </span><span class="n">dispersion</span><span class="o">=</span><span class="n">bcv</span><span class="o">^</span><span class="m">2</span><span class="p">)</span><span class="w">
  </span><span class="n">glmLRT</span><span class="p">(</span><span class="n">fit</span><span class="p">)</span><span class="w">
  </span><span class="n">IND_H2O2_LRT</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">glmLRT</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">contrast</span><span class="o">=</span><span class="n">my.contrasts</span><span class="p">[,</span><span class="s2">"CU_CH"</span><span class="p">])</span><span class="w">
  </span><span class="n">IND_H2O2_LRT_FDR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">topTags</span><span class="p">(</span><span class="n">IND_H2O2_LRT</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="kc">Inf</span><span class="p">)</span><span class="o">$</span><span class="n">table</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">IND_H2O2_LRT_FDR</span><span class="p">,</span><span class="w"> </span><span class="s2">"H2O2_DEG_LRT.csv"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><p><strong>Output</strong></p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; IND_H2O2_LRT &lt;- glmLRT(fit, contrast=my.contrasts[,"CU_CH"])
  &gt; IND_H2O2_LRT_FDR &lt;- topTags(IND_H2O2_LRT, n=Inf)$table
  &gt; IND_H2O2_LRT_FDR &lt;- IND_H2O2_LRT_FDR[order(rownames(IND_H2O2_LRT_FDR)), ]
  &gt; head(IND_H2O2_LRT_FDR)
                   logFC     logCPM           LR    PValue       FDR
  ENSG00000000003  0.0000000 -1.5072179 0.0000000000 1.0000000 1.0000000
  ENSG00000000005  0.0162005  4.6487540 0.0003788167 0.9844716 1.0000000
  ENSG00000000419 -3.1350554 -1.0936018 1.4297184098 0.2318104 0.9934706
  ENSG00000000457  0.7839430 -0.4296291 0.3116571808 0.5766650 1.0000000
  ENSG00000000460 -0.1462899  5.1399138 0.0314383257 0.8592660 1.0000000
  ENSG00000000938  0.7110305  3.7377811 0.7053500976 0.4009918 1.0000000
</code></pre></div></div></ul></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h3 id="10-annotation"> <a href="#10-annotation" class="anchor-heading" aria-labelledby="10-annotation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 10. Annotation</h3><p>After performing DEG analysis, the results may be displayed using gene (or transcript) IDs instead of gene symbols. To convert these IDs into various information such as gene symbols and biotypes, annotation is required. Two main methods for this are using the <code class="language-plaintext highlighter-rouge">biomaRt</code> R package or extracting information directly from a GTF file.</p><ul><li><p><strong>Biomart</strong> (<code class="language-plaintext highlighter-rouge">biomaRt</code>) offers access to a vast array of biological data through the Ensembl database, facilitating the easy conversion of gene IDs to gene symbols. However, this conversion can be more challenging and complex for species other than humans and mice. Therefore, it is recommended to use Biomart for gene conversion in the cases of human and mouse data.</p><li><p>Alternatively, the <strong>GTF file method</strong> entails extracting information from a locally stored gene annotation file. By parsing the GTF file, one can map gene IDs to gene symbols. This approach is versatile and can be applied to any species, provided that a GTF file is available.</p><li><p>the biomaRt method for human and mouse data:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">mart</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">useDataset</span><span class="p">(</span><span class="s2">"hsapiens_gene_ensembl"</span><span class="p">,</span><span class="w"> </span><span class="n">useMart</span><span class="p">(</span><span class="s2">"ensembl"</span><span class="p">))</span><span class="w"> 
  </span><span class="n">anno</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">getBM</span><span class="p">(</span><span class="n">filters</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"ensembl_gene_id"</span><span class="p">,</span><span class="w">
      </span><span class="n">attributes</span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"ensembl_gene_id"</span><span class="p">,</span><span class="w"> </span><span class="s2">"gene_biotype"</span><span class="p">,</span><span class="w"> </span><span class="s2">"external_gene_name"</span><span class="p">),</span><span class="w">
      </span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">row.names</span><span class="p">(</span><span class="n">IND_H2O2_LRT_FDR</span><span class="p">),</span><span class="w">
      </span><span class="n">mart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mart</span><span class="p">)</span><span class="w"> 
  </span><span class="n">Compare</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">setDT</span><span class="p">(</span><span class="n">IND_H2O2_LRT_FDR</span><span class="p">,</span><span class="w"> </span><span class="n">keep.rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"ensembl_gene_id"</span><span class="p">)[]</span><span class="w">
  </span><span class="n">MergeDEG</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">merge</span><span class="p">(</span><span class="n">IND_H2O2_LRT_FDR</span><span class="p">,</span><span class="w"> </span><span class="n">anno</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="s2">"ensembl_gene_id"</span><span class="p">,</span><span class="w"> </span><span class="n">all</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">MergeDEG</span><span class="p">,</span><span class="w"> </span><span class="s2">"IND_H2O2_LRT_Annotated.csv"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>in these codes,<ul><li><strong>Line 1</strong>: Connects to the Ensembl database (<code class="language-plaintext highlighter-rouge">useMart("ensembl")</code>) and specifies the dataset to use (<code class="language-plaintext highlighter-rouge">useDataset</code>), in this case, the human gene dataset from Ensembl (mouse: <code class="language-plaintext highlighter-rouge">"mmusculus_gene_ensembl"</code>). The available <code class="language-plaintext highlighter-rouge">useDataset</code> can be found via the <code class="language-plaintext highlighter-rouge">listDatasets(mart)</code> command, note the case of the lists.<li><strong>Line 2</strong>: <code class="language-plaintext highlighter-rouge">filters</code> specifies that the filter for the query is the Ensembl gene ID. <code class="language-plaintext highlighter-rouge">attributes</code> defines the attributes to retrieve, including the Ensembl gene ID, gene biotype, and external gene name (gene symbol). <code class="language-plaintext highlighter-rouge">values</code> provides the list of Ensembl gene IDs to be annotated, which are the row names of the <code class="language-plaintext highlighter-rouge">IND_H2O2_LRT_FDR</code> object. The available <code class="language-plaintext highlighter-rouge">attributes</code> can be found via the <code class="language-plaintext highlighter-rouge">listAttributes(mart)</code> command, note the case of the lists.<li><strong>Line 3</strong>: Converts <code class="language-plaintext highlighter-rouge">IND_H2O2_LRT_FDR</code> to a data.table and keeps the row names as <strong>a column named <code class="language-plaintext highlighter-rouge">ensembl_gene_id</code></strong>. <code class="language-plaintext highlighter-rouge">[]</code> ensures the operation is executed immediately and the result is assigned to Compare.<li><strong>Line 4</strong>: Merges the DEG results (<code class="language-plaintext highlighter-rouge">IND_H2O2_LRT_FDR</code>) with the annotation information (<code class="language-plaintext highlighter-rouge">anno</code>) based on the <code class="language-plaintext highlighter-rouge">by = "ensembl_gene_id"</code> column. The <code class="language-plaintext highlighter-rouge">all = TRUE</code> argument ensures that all entries from both data frames are included in the result, performing a full outer join.</ul><p><strong>Output</strong></p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; mart &lt;- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
  &gt; anno &lt;- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id", "gene_biotype", "external_gene_name"), values = row.names(IND_H2O2_LRT_FDR),mart= mart)
  &gt; head(anno)
  ensembl_gene_id   gene_biotype external_gene_name
  1 ENSG00000000003 protein_coding             TSPAN6
  2 ENSG00000000005 protein_coding               TNMD
  3 ENSG00000000419 protein_coding               DPM1
  4 ENSG00000000457 protein_coding              SCYL3
  5 ENSG00000000460 protein_coding              FIRRM
  6 ENSG00000000938 protein_coding                FGR

  &gt; Compare &lt;- setDT(IND_H2O2_LRT_FDR, keep.rownames = "ensembl_gene_id")[]
  &gt; head(Compare)
  ensembl_gene_id      logFC     logCPM           LR    PValue       FDR
              &lt;char&gt;      &lt;num&gt;      &lt;num&gt;        &lt;num&gt;     &lt;num&gt;     &lt;num&gt;
  1: ENSG00000000003  0.0000000 -1.5072179 0.0000000000 1.0000000 1.0000000
  2: ENSG00000000005  0.0162005  4.6487540 0.0003788167 0.9844716 1.0000000
  3: ENSG00000000419 -3.1350554 -1.0936018 1.4297184098 0.2318104 0.9934706
  4: ENSG00000000457  0.7839430 -0.4296291 0.3116571808 0.5766650 1.0000000
  5: ENSG00000000460 -0.1462899  5.1399138 0.0314383257 0.8592660 1.0000000
  6: ENSG00000000938  0.7110305  3.7377811 0.7053500976 0.4009918 1.0000000

  &gt; MergeDEG &lt;- merge(IND_H2O2_LRT_FDR, anno, by="ensembl_gene_id", all=TRUE)
  &gt; head(MergeDEG)
  Key: &lt;ensembl_gene_id&gt;
  ensembl_gene_id      logFC     logCPM           LR    PValue       FDR   gene_biotype external_gene_name
              &lt;char&gt;      &lt;num&gt;      &lt;num&gt;        &lt;num&gt;     &lt;num&gt;     &lt;num&gt;         &lt;char&gt;             &lt;char&gt;
  1: ENSG00000000003  0.0000000 -1.5072179 0.0000000000 1.0000000 1.0000000 protein_coding             TSPAN6
  2: ENSG00000000005  0.0162005  4.6487540 0.0003788167 0.9844716 1.0000000 protein_coding               TNMD
  3: ENSG00000000419 -3.1350554 -1.0936018 1.4297184098 0.2318104 0.9934706 protein_coding               DPM1
  4: ENSG00000000457  0.7839430 -0.4296291 0.3116571808 0.5766650 1.0000000 protein_coding              SCYL3
  5: ENSG00000000460 -0.1462899  5.1399138 0.0314383257 0.8592660 1.0000000 protein_coding              FIRRM
  6: ENSG00000000938  0.7110305  3.7377811 0.7053500976 0.4009918 1.0000000 protein_coding                FGR
</code></pre></div></div></ul></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><ul><li>the GTF file method for zebrafish:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">gtf_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"/Users/jchoi/Desktop/Danio_rerio.GRCz11.111.gtf"</span><span class="w"> 
  </span><span class="n">gtf_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rtracklayer</span><span class="o">::</span><span class="n">import</span><span class="p">(</span><span class="n">gtf_path</span><span class="p">)</span><span class="w">
  </span><span class="n">gene_info</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">subset</span><span class="p">(</span><span class="n">gtf_data</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"gene"</span><span class="p">)</span><span class="w">
  </span><span class="n">all_attributes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">elementMetadata</span><span class="p">(</span><span class="n">gene_info</span><span class="p">)</span><span class="w">
  </span><span class="n">gene_attributes_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">all_attributes</span><span class="p">,</span><span class="w"> </span><span class="n">stringsAsFactors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">gene_attributes_df</span><span class="p">,</span><span class="w"> </span><span class="s2">"annot.csv"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>in these codes,<ul><li><strong>Line 1</strong>: Specify the GTF file path.<li><strong>Line 2</strong>: Import the information from the GTF file using <code class="language-plaintext highlighter-rouge">rtracklayer</code> package.<li><strong>Line 3</strong>: Creates a subset of <code class="language-plaintext highlighter-rouge">gtf_data</code> that includes only the entries where the type is <code class="language-plaintext highlighter-rouge">"gene"</code>. The <code class="language-plaintext highlighter-rouge">gene_info</code> object will now contain only the gene-level annotations.<li><strong>Line 4</strong>: Extracts the metadata (<code class="language-plaintext highlighter-rouge">attributes</code>) of the <code class="language-plaintext highlighter-rouge">gene_info</code> object using the <code class="language-plaintext highlighter-rouge">elementMetadata</code> function. The <code class="language-plaintext highlighter-rouge">all_attributes</code> object will contain all the <code class="language-plaintext highlighter-rouge">attributes</code> of the gene annotations.<li><strong>Line 5</strong>: Converts the <code class="language-plaintext highlighter-rouge">all_attributes</code> object into a data frame named <code class="language-plaintext highlighter-rouge">gene_attributes_df</code> using the <code class="language-plaintext highlighter-rouge">data.frame</code> function. The <code class="language-plaintext highlighter-rouge">stringsAsFactors = FALSE</code> argument ensures that string columns are not converted to factors.</ul><p><strong>Output</strong></p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; gene_attributes_df &lt;- data.frame(all_attributes, stringsAsFactors = FALSE)
  &gt; head(gene_attributes_df)
          source type score phase            gene_id gene_version        gene_name    gene_source         gene_biotype
  1         havana gene    NA    NA ENSDARG00000103202            2       CR383668.1         havana              lincRNA
  2 ensembl_havana gene    NA    NA ENSDARG00000009657            8         fgfr1op2 ensembl_havana       protein_coding
  3         havana gene    NA    NA ENSDARG00000096472            2       AL845295.2         havana processed_transcript
  4         havana gene    NA    NA ENSDARG00000096156            3 si:dkey-21h14.12         havana       protein_coding
  5         havana gene    NA    NA ENSDARG00000076160            6 si:dkey-285e18.2         havana       protein_coding
  6         havana gene    NA    NA ENSDARG00000117163            1          znf1114         havana       protein_coding
  transcript_id transcript_version transcript_name transcript_source transcript_biotype  tag exon_number exon_id
  1          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt; &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
  2          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt; &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
  3          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt; &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
  4          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt; &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
  5          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt; &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
  6          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;               &lt;NA&gt; &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
  exon_version protein_id protein_version
  1         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;
  2         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;
  3         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;
  4         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;
  5         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;
  6         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;
</code></pre></div></div></ul><li>For some other species (e.g. rice), there are also symbol conversion packages:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">library</span><span class="p">(</span><span class="n">riceidconverter</span><span class="p">)</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">org.Osativa.eg.db</span><span class="p">)</span><span class="w">
  </span><span class="n">RAP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">RiceIDConvert</span><span class="p">(</span><span class="n">myID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">Name</span><span class="p">,</span><span class="w"> </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"MSU"</span><span class="p">,</span><span class="w"> </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"RAP"</span><span class="p">)</span><span class="w">
  </span><span class="n">Symbol</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">RiceIDConvert</span><span class="p">(</span><span class="n">myID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">MSU</span><span class="p">,</span><span class="w"> </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"MSU"</span><span class="p">,</span><span class="w"> </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"SYMBOL"</span><span class="p">)</span><span class="w">
  </span><span class="c1">## Symbol indicates Gene ID (LOCXXXXXXX format)</span><span class="w">
</span></code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="citations"> <a href="#citations" class="anchor-heading" aria-labelledby="citations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citations</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">edgeR/TMM</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Robinson, M. D.</b>, McCarthy, D. J., &amp; Smyth, G. K. (2010). edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. <i>bioinformatics</i>, 26(1), 139-140. <a href="https://doi.org/10.1093/bioinformatics/btp616" target="_blank">DOI</a></span><li style="margin-bottom: 10px;"> <span><b>McCarthy, D. J.</b>, Chen, Y., &amp; Smyth, G. K. (2012). Differential expression analysis of multifactor RNA-Seq experiments with respect to biological variation. <i>Nucleic acids research</i>, 40(10), 4288-4297. <a href="https://doi.org/10.1093/nar/gks042" target="_blank">DOI</a></span></ol><span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">limma</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Ritchie, M. E.</b>, Phipson, B., Wu, D. I., Hu, Y., Law, C. W., Shi, W., &amp; Smyth, G. K. (2015). limma powers differential expression analyses for RNA-sequencing and microarray studies. <i>Nucleic acids research</i>, 43(7), e47-e47. <a href="https://doi.org/10.1093/nar/gkv007" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
