<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(1)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(1) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(5) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(1) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list > li.nav-list-item:nth-child(1) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Exploring Sequencing Data with deepTools | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Exploring Sequencing Data with deepTools" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/ChIPSeq/deepTools_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/ChIPSeq/deepTools_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Exploring Sequencing Data with deepTools" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Exploring Sequencing Data with deepTools","url":"http://localhost:4000/docs/ChIPSeq/deepTools_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-portfolio" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/ChIPSeq">Data Analysis - ChIP-seq</a><li class="breadcrumb-nav-list-item"><span>Exploring Sequencing Data with deepTools</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="exploring-deep-sequencing-data-with-deeptools"> <a href="#exploring-deep-sequencing-data-with-deeptools" class="anchor-heading" aria-labelledby="exploring-deep-sequencing-data-with-deeptools"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Exploring Deep-sequencing Data with deepTools</h1><p class="fs-4 fw-500">The deepTools effectively manages the vast data volumes routinely produced from sequencing platforms. This tool suite includes versatile modules designed to process mapped reads, performing numerous quality assessments. It also generates normalized coverage files in standard bedGraph and bigWig formats, facilitating comparisons between various datasets, such as treatment and control groups. This protocol was created based on <span style="color: #7253ec;"><b>deepTools version 3.5.1</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>Python version 3.8.5, SciPy version 1.6.2, NumPy version 1.20.1, and conda version 22.9.0 under macOS 12.4 environment.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="installation-deeptools"> <a href="#installation-deeptools" class="anchor-heading" aria-labelledby="installation-deeptools"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation deepTools</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <a href="https://github.com/deeptools/deepTools"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="https://deeptools.readthedocs.io/en/latest/index.html"><img src="https://img.shields.io/badge/GitHub.io-181717?style=flat&amp;logo=github&amp;logoColor=yellow" alt="githio" /></a> <img src="https://github.com/deeptools/deepTools/actions/workflows/test.yml/badge.svg?labelColor=black" alt="test" /> <a href="https://pypi.org/project/deepTools/"><img src="https://img.shields.io/pypi/v/deeptools.svg?style=flat&amp;labelColor=black" alt="PyPI Version" /></a></p><p>To install deepTools using Anaconda, use the following command:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>conda <span class="nb">install</span> <span class="nt">-c</span> conda-forge <span class="nt">-c</span> bioconda deeptools
</code></pre></div></div><h2 class="fs-6 fw-400 lh-tight" id="troubleshooting-in-linux-environment"> <a href="#troubleshooting-in-linux-environment" class="anchor-heading" aria-labelledby="troubleshooting-in-linux-environment"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Troubleshooting in Linux Environment</u></h2><ol><li>If error occurs for freeze environment, update conda as follow;<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code> <span class="nv">$ </span><span class="nb">sudo </span>apt-get update
 <span class="nv">$ </span>conda update xz
</code></pre></div></div><li>If error occurs for repository, remove it as follow;<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code> <span class="nv">$ </span><span class="nb">sudo </span>add-apt-repository <span class="nt">--remove</span> ppa:[Dir]/[Sub-Dir]

 <span class="c"># For example</span>
 <span class="nv">$ </span><span class="nb">sudo </span>add-apt-repository <span class="nt">--remove</span> ppa:webupd8team/sublime-text-3
</code></pre></div></div><li>If deeptools cannot be run due to <code class="language-plaintext highlighter-rouge">libcryto.so.1.0.0</code> source error, refer to document of ‘phyton setting’</ol><h1 class="fs-7 fw-400 lh-tight" id="bamcoverage-convert-bam-into-bigwig"> <a href="#bamcoverage-convert-bam-into-bigwig" class="anchor-heading" aria-labelledby="bamcoverage-convert-bam-into-bigwig"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> BamCoverage: Convert BAM into bigWig</h1><ul><li><p>BamCoverage module takes an alignment of reads or fragments as input (BAM file) and generates a coverage track (bigWig or bedGraph) as output. Use the following command to convert BAM to BigWig with BamCoverage:</p><p class="note">This module requires a BAM file sorted by position and a bai file in the same folder.</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>bamCoverage <span class="nt">-b</span> &lt;input.bam&gt; <span class="nt">-o</span> &lt;output.bigwig&gt; <span class="nt">-of</span> &lt;outputType&gt; <span class="nt">-bs</span> &lt;int&gt; <span class="se">\</span>
      <span class="nt">--normalizeUsing</span> &lt;Norm&gt; <span class="nt">--effectiveGenomeSize</span> &lt;int&gt; <span class="nt">--smoothLength</span> &lt;int&gt; <span class="se">\</span>
      <span class="nt">--region</span> &lt;region&gt; <span class="nt">-p</span> &lt;int&gt; <span class="nt">-e</span>
</code></pre></div></div><li><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-b &lt;input.bam&gt;</code><td>Specifies the input data.<tr><td><code class="language-plaintext highlighter-rouge">-o &lt;output.bigwig&gt;</code><td>Specifies output file.<tr><td><code class="language-plaintext highlighter-rouge">-of &lt;outputType&gt;</code><td>Specifies format of the output data. Possible values are bigWig or bdg.<tr><td><code class="language-plaintext highlighter-rouge">-bs &lt;int&gt;</code><td>Size of the bins, in bases, for the output of the bigwig/bedgraph file (Default: 50). <br /><strong><code class="language-plaintext highlighter-rouge">-bs 1</code> setting is recommended for publication.</strong> The smaller the value of <code class="language-plaintext highlighter-rouge">-bs</code>, the more computation.<tr><td><code class="language-plaintext highlighter-rouge">--normalizeUsing &lt;Norm&gt;</code><td>Specifies whether to apply a method to normalize the number of reads per bin.<br />Possible values for <code class="language-plaintext highlighter-rouge">&lt;Norm&gt;</code> are <code class="language-plaintext highlighter-rouge">RPCG</code>, <code class="language-plaintext highlighter-rouge">RPKM</code>, <code class="language-plaintext highlighter-rouge">CPM</code>, or <code class="language-plaintext highlighter-rouge">none</code> (default: <code class="language-plaintext highlighter-rouge">none</code>).<tr><td><code class="language-plaintext highlighter-rouge">--effectiveGenomeSize &lt;int&gt;</code><td>The effective genome size (EGS) is the portion of the genome that is mappable. this parameter is couple with <code class="language-plaintext highlighter-rouge">RPGC</code> normalization. <br />human: 2.7e9 (2700000000), mouse: 1.87e9 (1870000000).<tr><td><code class="language-plaintext highlighter-rouge">--smoothLength &lt;int&gt;</code><td>The smooth length defines a window, larger than the binSize, to average the number of reads. For example, if the –binSize is set to 20 and the –smoothLength is set to 60, then, for each bin, the average of the bin and its left and right neighbors is considered. <br /><strong>For Publication, <code class="language-plaintext highlighter-rouge">-bs 1, --smoothLength 10</code>settting is recommended.</strong> <br />The larger the value of <code class="language-plaintext highlighter-rouge">---smoothLength</code>, the more computation.<tr><td><code class="language-plaintext highlighter-rouge">--region &lt;region&gt;</code><td>Region of the genome to limit the operation to - <strong>this is useful when testing parameters to reduce the computing time.</strong><br />The format is chr:start:end, for example <code class="language-plaintext highlighter-rouge">--region chr2:178104147:178154424</code>.<tr><td><code class="language-plaintext highlighter-rouge">-p &lt;int&gt;</code><td>Specifies thread numbers <code class="language-plaintext highlighter-rouge">&lt;int&gt;</code><tr><td><code class="language-plaintext highlighter-rouge">-e</code> option<td>This parameter allows the extension of reads to fragment size. If set, each read is extended, without exception. <strong>Can not apply to RNA-seq data.</strong></table></div></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code"> <a href="#example-code" class="anchor-heading" aria-labelledby="example-code"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code</u></h2><ul><li><p>Here is an example command to perform converting a bigWig file from the BAM file:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># Coded to quickly identify track patterns.</span>
  <span class="nv">$ </span>bamcoverage <span class="nt">-b</span> Final.sort.Unt_IPT.bam <span class="nt">-o</span> /Users/jchoi/desktop/Unt_IPT.bigwig <span class="se">\</span>
      <span class="nt">-bs</span> 100 <span class="nt">--normalizeUsing</span> RPGC <span class="nt">--effectiveGenomeSize</span> 2700000000 <span class="nt">-e</span> <span class="nt">-p</span> 10

  <span class="c"># For GEO submission, it takes a very long time.</span>
  <span class="nv">$ </span>bamcoverage <span class="nt">-b</span> Final.sort.Unt_IPT.bam <span class="nt">-o</span> /Users/jchoi/desktop/Unt_IPT.bigwig <span class="nt">-bs</span> 1 <span class="se">\</span>
      <span class="nt">--normalizeUsing</span> RPGC <span class="nt">--effectiveGenomeSize</span> 2700000000 <span class="nt">--smoothLength</span> 10 <span class="nt">-e</span> <span class="nt">-p</span> 10
  <span class="nv">$ </span>bamcoverage <span class="nt">-b</span> Final.sort.Unt_IPT.bam <span class="nt">-o</span> /Users/jchoi/desktop/Unt_IPT.bigwig <span class="nt">-bs</span> 5 <span class="se">\</span>
      <span class="nt">--normalizeUsing</span> RPGC <span class="nt">--effectiveGenomeSize</span> 2700000000 <span class="nt">--smoothLength</span> 20 <span class="nt">-e</span> <span class="nt">-p</span> 10

  <span class="c"># This code makes the bigWig file by specifying a target gene region as wide, which can be a time saver.</span>
  <span class="nv">$ </span>bamcoverage <span class="nt">-b</span> Final.sort.Unt_MLL1.bam <span class="nt">-o</span> /Users/jchoi/desktop/Unt_MLL1_NRF2.bigwig <span class="nt">-bs</span> 1 <span class="se">\</span>
      <span class="nt">--normalizeUsing</span> RPGC <span class="nt">--effectiveGenomeSize</span> 2700000000 <span class="nt">--region</span> chr2:178104147:178154424 <span class="se">\ </span>
      <span class="nt">--smoothLength</span> 20 <span class="nt">-e</span>
</code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h2 class="fs-5 fw-400 lh-tight" id="output"> <a href="#output" class="anchor-heading" aria-labelledby="output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Output</u></h2><ul><li>When <code class="language-plaintext highlighter-rouge">BamCoverage</code> running, it prints messages printing as the follow:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>      bamFilesList: ['Final.sort.Unt_MLL1.bam']
              binLength: 1
              numberOfSamples: None
              blackListFileName: None
              skipZeroOverZero: False
              bed_and_bin: False
              genomeChunkSize: None
              defaultFragmentLength: 254
              numberOfProcessors: 2
              verbose: False
              region: None
              bedFile: None
              minMappingQuality: None
              ignoreDuplicates: False
              chrsToSkip: []
              stepSize: 1
              center_read: False
              samFlag_include: None
              samFlag_exclude: None
              minFragmentLength: 0
              maxFragmentLength: 0
              zerosToNans: False
              smoothLength: None
              save_data: False
              out_file_for_raw_data: None
              maxPairedFragmentLength: 1016
</code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="bigwigcompare-and-bamcompare"> <a href="#bigwigcompare-and-bamcompare" class="anchor-heading" aria-labelledby="bigwigcompare-and-bamcompare"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> bigwigCompare and bamCompare</h1><ul><li><p><code class="language-plaintext highlighter-rouge">bigwigCompare</code> compares two bigWig files based on the number of mapped reads. To compare the bigWig files, the genome is partitioned into bins of equal size, then the number of reads found in each BAM file are counted per bin and finally a summary value is reported. Similarly, <code class="language-plaintext highlighter-rouge">bamCompare</code> can be used to generate a bigWig or bedGraph file based on two BAM files that are compared to each other while being simultaneously normalized for sequencing depth. Use the following command to compare two bigWigs or two BAMs:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># bigwigCompare</span>
  <span class="nv">$ </span>bigwigCompare <span class="nt">-b1</span> &lt;target.bigwig&gt; <span class="nt">-b2</span> &lt;control.bigwig&gt; <span class="nt">--operation</span> &lt;ratio&gt; <span class="se">\</span>
      <span class="nt">-o</span> &lt;output.bigwig&gt; <span class="nt">-p</span> &lt;INT&gt;

  <span class="c"># bamCompare</span>
  <span class="nv">$ </span>bamCompare <span class="nt">-b1</span> &lt;target.bam&gt; <span class="nt">-b2</span> &lt;control.bam&gt; <span class="nt">--operation</span> &lt;ratio&gt; <span class="se">\</span>
      <span class="nt">-o</span> &lt;output.bigwig&gt; <span class="nt">-p</span> &lt;INT&gt;
</code></pre></div></div><li><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-b1 &lt;target.bigwig&gt;</code><td>Specifies the bigwig or the sorted BAM file for <strong>the target/treatment.</strong><tr><td><code class="language-plaintext highlighter-rouge">-b2 &lt;control.bigwig&gt;</code><td>Specifies the bigwig or the sorted BAM file for <strong>the control.</strong><tr><td><code class="language-plaintext highlighter-rouge">---operation &lt;ratio&gt;</code><td>Specifies how to determine two values are compared.<br />Possible values for <code class="language-plaintext highlighter-rouge">&lt;ratio&gt;</code> are <code class="language-plaintext highlighter-rouge">log2</code>, <code class="language-plaintext highlighter-rouge">ratio</code>, <code class="language-plaintext highlighter-rouge">substrate</code>, <code class="language-plaintext highlighter-rouge">add</code>, or <code class="language-plaintext highlighter-rouge">mean</code> (default: <code class="language-plaintext highlighter-rouge">log2</code>).<tr><td><code class="language-plaintext highlighter-rouge">-o &lt;output.bigwig&gt;</code><td>Specifies output file.<tr><td><code class="language-plaintext highlighter-rouge">-p &lt;int&gt;</code><td>Specifies thread numbers <code class="language-plaintext highlighter-rouge">&lt;int&gt;</code></table></div></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed"> <a href="#example-code-results-are-not-printed" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code (Results are not printed)</u></h2><ul><li><p>Here is an example command to perform <code class="language-plaintext highlighter-rouge">bigwigCompare</code> and <code class="language-plaintext highlighter-rouge">bamCompare</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># bigwigCompare</span>
  <span class="nv">$ </span>bigwigCompare <span class="nt">-b1</span> /Users/jchoi/desktop/Unt_H3K4me3.bigwig <span class="se">\</span>
      <span class="nt">-b2</span> /Users/jchoi/desktop/Unt_IPT.bigwig <span class="nt">--operation</span> log2 <span class="se">\</span>
      <span class="nt">-o</span> /Users/jchoi/desktop/Matrix/Unt_IPT_H3K4me3.bigwig <span class="nt">-p</span> 10
  <span class="nv">$ </span>bigwigCompare <span class="nt">-b1</span> /Users/jchoi/desktop/Unt_H3K27me3.bigwig <span class="se">\</span>
      <span class="nt">-b2</span> /Users/jchoi/desktop/Unt_IPT.bigwig <span class="nt">--operation</span> log2 <span class="se">\</span>
      <span class="nt">-o</span> /Users/jchoi/desktop/Matrix/Unt_IPT_H3K27me3.bigwig <span class="nt">-p</span> 10

  <span class="c"># bamCompare</span>
  <span class="nv">$ </span>bamCompare <span class="nt">-b1</span> /Users/jchoi/desktop/BAM/Sorted.Unt_NFIB.bam <span class="se">\</span>
      <span class="nt">-b2</span> /Users/jchoi/desktop/BAM/Sorted.Unt_IPT.bam <span class="nt">--operation</span> log2 <span class="se">\</span>
      <span class="nt">-o</span> /Users/jchoi/desktop/BAM/Matrix/Unt_IPT_NFIB.bigwig <span class="nt">-p</span> 10
  <span class="nv">$ </span>bamCompare <span class="nt">-b1</span> /Users/jchoi/desktop/BAM/Sorted.Unt_MLL1.bam <span class="se">\</span>
      <span class="nt">-b2</span> /Users/jchoi/desktop/BAM/Sorted.Unt_IPT.bam <span class="nt">--operation</span> log2 <span class="se">\</span>
      <span class="nt">-o</span> /Users/jchoi/desktop/BAM/Matrix/Unt_IPT_MLL1.bigwig <span class="nt">-p</span> 10
</code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="computematrix-prepare-matrix-data"> <a href="#computematrix-prepare-matrix-data" class="anchor-heading" aria-labelledby="computematrix-prepare-matrix-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> computeMatrix: Prepare matrix data</h1><ul><li><p><code class="language-plaintext highlighter-rouge">computeMatrix</code> calculates scores per genome regions and prepares an intermediate file that can be used with <code class="language-plaintext highlighter-rouge">plotHeatmap</code> and <code class="language-plaintext highlighter-rouge">plotProfiles</code>. <code class="language-plaintext highlighter-rouge">computeMatrix</code> accepts multiple score files (bigWig format) and multiple regions files (BED format). This tool can also be used to filter and sort regions according to their score. Use the following command to make a matrix data using bigWig files generated from <code class="language-plaintext highlighter-rouge">bigwigCompare</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>computeMatrix &lt;mode&gt; <span class="nt">--referencePoint</span> &lt;point&gt; <span class="nt">-b</span> &lt;int&gt; <span class="nt">-a</span> &lt;int&gt; <span class="nt">-R</span> &lt;referenceGenome&gt; <span class="se">\</span>
      <span class="nt">-S</span> &lt;bigwigCompared_1.bigwig&gt;...&lt;bigwigCompared_N.bigwig&gt; <span class="se">\</span>
      <span class="nt">--skipZeros</span> <span class="nt">--missingDataAsZero</span> <span class="nt">-o</span> &lt;matrix file.gz&gt; <span class="nt">-p</span> &lt;int&gt;
</code></pre></div></div><li><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">&lt;mode&gt;</code><td>Possible values are <code class="language-plaintext highlighter-rouge">reference-point</code> or <code class="language-plaintext highlighter-rouge">scale-regions</code>. <b>See the bellow</b><tr><td><code class="language-plaintext highlighter-rouge">--referencePoint &lt;point&gt;</code><td>Specifies a reference point. The reference point for the plotting could be either the region start (<code class="language-plaintext highlighter-rouge">TSS</code>), the region end (<code class="language-plaintext highlighter-rouge">TES</code>) or the <code class="language-plaintext highlighter-rouge">center</code> of the region.<tr><td><code class="language-plaintext highlighter-rouge">-b &lt;int&gt;</code><td>Distance upstream of the reference-point selected.<tr><td><code class="language-plaintext highlighter-rouge">-a &lt;int&gt;</code><td>Distance downstream of the reference-point selected.<tr><td><code class="language-plaintext highlighter-rouge">-R &lt;referenceGenome&gt;</code><td>File name or names, in BED or GTF format, containing the regions to plot. It is recommend to use the GTF file obtained in the htseeq-count step.<tr><td><code class="language-plaintext highlighter-rouge">-S &lt;bigwigCompared_1.bigwig&gt;...&lt;bigwigCompared_N.bigwig&gt;</code><td>specifies input data, it is separated by spaces. <strong>These files should be obtained by using the <code class="language-plaintext highlighter-rouge">bamCoverage</code> or <code class="language-plaintext highlighter-rouge">bamCompare</code> modules.</strong><tr><td><code class="language-plaintext highlighter-rouge">--skipZeros</code> option<td>Whether regions with only scores of zero should be included or not.<tr><td><code class="language-plaintext highlighter-rouge">--missingDataAsZero</code> option<td>If set, missing data (NAs) will be treated as zeros.<tr><td><code class="language-plaintext highlighter-rouge">-o &lt;matrix file.gz&gt;</code><td>Specifies output file. The matrix file must have a ‘gz’ extension.<tr><td><code class="language-plaintext highlighter-rouge">-p &lt;int&gt;</code><td>Specifies thread numbers <code class="language-plaintext highlighter-rouge">&lt;int&gt;</code></table></div><p class="note">The <code class="language-plaintext highlighter-rouge">reference-point</code> mode plots genomic positions upstream and downstream from a specified position within a genome region, such as the starting point. Conversely, the <code class="language-plaintext highlighter-rouge">scale-regions</code> mode adjusts all regions in the genome file to a user-defined length, stretching or shrinking them as necessary.</p></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-1"> <a href="#example-code-1" class="anchor-heading" aria-labelledby="example-code-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code</u></h2><ul><li><p>Here is an example command to prepare a matrix data with <code class="language-plaintext highlighter-rouge">computeMatrix</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># Mouse, reference-point mode, TSS, TF ChIP-seq</span>
  <span class="nv">$ </span>computeMatrix reference-point <span class="nt">--referencePoint</span> TSS <span class="nt">-b</span> 3000 <span class="nt">-a</span> 3000 <span class="se">\</span>
      <span class="nt">-R</span> /Users/jchoi/desktop/RefGene/mm10.refGene.gtf
      <span class="nt">-S</span> IPT_shCtrl_rep1.bigwig IPT_shCtrl_rep2.bigwig IPT_shNFIB_rep1.bigwig IPT_shNFIB_rep2.bigwig <span class="se">\</span>
      <span class="nt">--skipZeros</span> <span class="nt">--missingDataAsZero</span> <span class="nt">-o</span> /Users/jchoi/desktop/bigWig/Matrix/matrix_TSS.gz <span class="nt">-p</span> 10

  <span class="c"># Rice, reference-point mode, TES, histone ChIP-seq</span>
  <span class="nv">$ </span>computeMatrix reference-point <span class="nt">--referencePoint</span> TES <span class="nt">-b</span> 3000 <span class="nt">-a</span> 3000 <span class="se">\</span>
      <span class="nt">-R</span> /Users/jchoi/Desktop/ProfYoon/O_Sativa_Genome/RGAP_Used/all.gtf <span class="se">\</span>
      <span class="nt">-S</span> /Users/jchoi/Desktop/ProfYoon/ChIPSeq_Track/LowBin_WTe.bigwig /Users/jchoi/Desktop/ProfYoon/ChIPSeq_Track/LowBin_ehd1e.bigwig <span class="se">\</span>
      <span class="nt">--skipZeros</span> <span class="nt">--missingDataAsZero</span> <span class="nt">-o</span> /Users/jchoi/Desktop/matrix_TES.gz <span class="nt">-p</span> 10
</code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h2 class="fs-5 fw-400 lh-tight" id="output-1"> <a href="#output-1" class="anchor-heading" aria-labelledby="output-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Output</u></h2><ul><li>This code <strong>does not output any results or progress</strong>, but sometimes you will see a message like the one below:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  Skipping XX_XXXXXX_XX, due to being absent in the computeMatrix output.
  Skipping NR_164356_11, due to being absent in the computeMatrix output. 
</code></pre></div></div><li>Ignore these mentions because they mean missing data and has been processed by <code class="language-plaintext highlighter-rouge">--skipZeros</code> and <code class="language-plaintext highlighter-rouge">--missingDataAsZero</code> commands.</ul><h1 class="fs-7 fw-400 lh-tight" id="plotheatmap"> <a href="#plotheatmap" class="anchor-heading" aria-labelledby="plotheatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> plotHeatmap</h1><ul><li><p><code class="language-plaintext highlighter-rouge">plotHeatmap</code> creates a heatmap for scores associated with genomic regions. The program requires a matrix file generated by the <code class="language-plaintext highlighter-rouge">computeMatrix</code>. Use the following command to make a heatmap:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>plotHeatmap <span class="nt">-m</span> &lt;matrixFile.gz&gt; <span class="nt">-out</span> &lt;output.extention&gt; <span class="nt">--colorMap</span> &lt;colorCode&gt; <span class="nt">--colorList</span> &lt;<span class="s2">"colorCode"</span><span class="o">&gt;</span> <span class="se">\</span>
      <span class="nt">--whatToShow</span> &lt;<span class="s1">'type'</span><span class="o">&gt;</span> <span class="nt">--zMin</span> &lt;group1 ... groupN&gt; <span class="nt">--zMax</span> &lt;group1 ... groupN&gt; <span class="se">\</span>
      <span class="nt">--yMin</span> &lt;group1 ... groupN&gt; <span class="nt">--yMax</span> &lt;group1 ... groupN&gt; <span class="se">\</span>
      <span class="nt">--kmeans</span> &lt;int&gt; <span class="nt">--perGroup</span> <span class="nt">--dpi</span> &lt;int&gt; 
</code></pre></div></div><li><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-m &lt;matrixFile.gz&gt;</code><td>Specifies matrix file.<tr><td><code class="language-plaintext highlighter-rouge">-out &lt;output.extention&gt;</code><td>Specifies output file. Possible values for file extention are <code class="language-plaintext highlighter-rouge">png</code>, <code class="language-plaintext highlighter-rouge">eps</code>, <code class="language-plaintext highlighter-rouge">pdf</code>, and <code class="language-plaintext highlighter-rouge">svg</code>.<tr><td><code class="language-plaintext highlighter-rouge">--colorMap &lt;colorCode&gt;</code><td>Color map to use for the heatmap. If more than one heatmap is being plotted the color of each heatmap can be enter individually (e.g. –colorMap Reds Blues). <a href="http://matplotlib.org/users/colormaps.html" target="_blank">Available color codes</a><tr><td><code class="language-plaintext highlighter-rouge">--colorList &lt;colorCode&gt;</code><td>List of colors to use to create a colormap. For example, if –colorList black,yellow,blue is set (colors separated by comas) then a color map that starts with black, continues to yellow and finishes in blue is created. You can also use hexagonal color codes (#000000).<tr><td><code class="language-plaintext highlighter-rouge">--whatToShow &lt;'type'&gt;</code><td>The default is to include a summary or profile plot on top of the heatmap and a heatmap colorbar. Other options are <code class="language-plaintext highlighter-rouge">'plot and heatmap'</code>, <code class="language-plaintext highlighter-rouge">'heatmap only'</code>, and <code class="language-plaintext highlighter-rouge">'heatmap and colorbar'</code>. Not recommended for use.<tr><td><code class="language-plaintext highlighter-rouge">--zMin/--zMax</code> option<td>adjust heatmap intensities. If more than one heatmap is being plotted the intensity of each heatmap can be enter individually.<tr><td><code class="language-plaintext highlighter-rouge">--yMin/--yMax</code> option<td>adjust plot range of the y-axis. If more than one heatmap is being plotted the intensity of each heatmap can be enter individually.<tr><td><code class="language-plaintext highlighter-rouge">--kmeans &lt;int&gt;</code><td>Number of clusters to compute. When this option is set, the matrix is split into clusters using the k-means algorithm. <code class="language-plaintext highlighter-rouge">--hclust</code>option is also possible for Hierarchical clustering.<tr><td><code class="language-plaintext highlighter-rouge">--perGroup</code> option<td>The default is to plot all groups of regions by sample. Using this option instead plots all samples by group of regions.</table></div></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed-1"> <a href="#example-code-results-are-not-printed-1" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code (Results are not printed)</u></h2><ul><li><p>Here is an example command to plot heatmap with <code class="language-plaintext highlighter-rouge">plotHeatmap</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>plotHeatmap <span class="nt">-m</span> /Users/jchoi/desktop/bigWig/Matrix/matrix_TSS.gz <span class="se">\</span>
      <span class="nt">-out</span> /Users/jchoi/desktop/Distribution.svg <span class="nt">--colorMap</span> RdBu
      <span class="nt">--zMin</span> <span class="nt">-1</span> <span class="nt">--zMax</span> 6 <span class="nt">--yMin</span> 0 <span class="nt">--yMax</span> 4 <span class="nt">--dpi</span> 300

  <span class="nv">$ </span>plotHeatmap <span class="nt">-m</span> /Users/jchoi/desktop/bigWig/Matrix/matrix_TSS.gz <span class="se">\</span>
      <span class="nt">-out</span> /Users/jchoi/desktop/Distribution_heatmap.svg <span class="nt">--colorMap</span> RdBu
      <span class="nt">--whatToShow</span> <span class="s1">'heatmap only'</span> <span class="nt">--zMin</span> <span class="nt">-1</span> <span class="nt">--zMax</span> 6
      <span class="nt">--kmeans</span> 3 <span class="nt">--dpi</span> 300

  <span class="nv">$ </span>plotHeatmap <span class="nt">-m</span> /Users/jchoi/Desktop/matrix_TSS.gz 
      <span class="nt">-out</span> /Users/jchoi/Desktop/Rice_H3ac.svg <span class="nt">--colorList</span> <span class="s2">"white,#bd1829"</span> 
      <span class="nt">--yMin</span> 20 <span class="nt">--yMax</span> 50 <span class="nt">--zMin</span> 40 
</code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="plotprofile"> <a href="#plotprofile" class="anchor-heading" aria-labelledby="plotprofile"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> plotProfile</h1><ul><li><p><code class="language-plaintext highlighter-rouge">plotProfile</code> creates a profile plot for scores over sets of genomic regions. This is a simplified version of the <code class="language-plaintext highlighter-rouge">plotHeatmap</code> provided above that only plots individual line graphs. The parameters for this function are quite similar to others, so I will keep the explanation brief without example code.</p><p class="note">Unfortunately, <code class="language-plaintext highlighter-rouge">plotProfile</code> is currently not used by many users.</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>plotProfile <span class="nt">-m</span> &lt;matrix file.gz&gt; <span class="nt">-out</span> &lt;output.extention&gt; <span class="nt">--plotType</span><span class="o">=</span>&lt;<span class="k">select</span><span class="o">&gt;</span>
      <span class="nt">--color</span> &lt;<span class="s1">'color code'</span><span class="o">&gt;</span> <span class="nt">--numPlotsPerRow</span> &lt;int&gt; <span class="nt">--plotTitle</span> &lt;<span class="s2">"Test data profile"</span><span class="o">&gt;</span>
      <span class="nt">--kmeans</span> &lt;int&gt; <span class="nt">--perGroup</span> <span class="nt">--dpi</span> &lt;int&gt;
</code></pre></div></div><li><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">--plotType=&lt;select&gt;</code>: Possible values are <code class="language-plaintext highlighter-rouge">lines</code>, <code class="language-plaintext highlighter-rouge">fill</code>, <code class="language-plaintext highlighter-rouge">se</code>, <code class="language-plaintext highlighter-rouge">std</code>, <code class="language-plaintext highlighter-rouge">overlapped_lines</code>, and <code class="language-plaintext highlighter-rouge">heatmap</code>.<li><code class="language-plaintext highlighter-rouge">--color &lt;'color code'&gt;</code>: List of colors to use for the plotted lines (color names or hex strings). For example <code class="language-plaintext highlighter-rouge">--colors '#e89b81' '#959dc8' '#f6f0a8'</code><li><code class="language-plaintext highlighter-rouge">--numPlotsPerRow &lt;int&gt;</code>: Number of plots per row</ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 10px; padding-bottom: 5px; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 1px; position: relative; left: -10px;">Critical Note</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span>Overall, the <code>plotHeatmap</code> and <code>plotProfile</code> functions, powered by <code>computeMatrix</code>, are commonly used to visualize the tag distribution of peaks and are featured prominently in numerous publications.</span><li style="margin-bottom: 10px;"> <span>However, for more detailed analysis, I recommend using <u>Python-based <code>ngs.plot.r</code></u> or the <u><code>ChIPseeker</code> R package </u>, which offer enhanced detail and additional analytical capabilities.</span></ol></div></ul><h1 class="fs-7 fw-400 lh-tight" id="multibigwigsummary-and-multibamsummary"> <a href="#multibigwigsummary-and-multibamsummary" class="anchor-heading" aria-labelledby="multibigwigsummary-and-multibamsummary"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> multiBigwigSummary and multiBamSummary</h1><ul><li><p><code class="language-plaintext highlighter-rouge">multiBigwigSummary</code> and <code class="language-plaintext highlighter-rouge">multiBamSummary</code> compute the average score/read coverages for genomic regions for two or more bigWig and BAM files, respectively. This analysis is performed for the entire genome by running the program in bins mode, or for certain user selected regions in BED-file mode. Most commonly, the default output of <code class="language-plaintext highlighter-rouge">multiBigwigSummary</code> (a compressed numpy array, <strong>.npz</strong>) is used by other tools such as <code class="language-plaintext highlighter-rouge">plotCorrelation</code> or <code class="language-plaintext highlighter-rouge">plotPCA</code> for visualization and diagnostic purposes. Use the following command to make a summary matrix from bigWig files with <code class="language-plaintext highlighter-rouge">multiBigwigSummary</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>multiBigwigSummary bins <span class="nt">-b</span> &lt;file1.bigwig&gt;...&lt;fileN.bigwig&gt; <span class="nt">-l</span> &lt;file1&gt;...&lt;fileN&gt; <span class="nt">-o</span> &lt;results.npz&gt;
  <span class="nv">$ </span>multiBamSummary bins <span class="nt">-b</span> &lt;file1.bam&gt;...&lt;fileN.bam&gt; <span class="nt">-l</span> &lt;file1&gt;...&lt;fileN&gt; <span class="nt">-e</span> <span class="nt">-o</span> &lt;results.npz&gt;
</code></pre></div></div><li><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">-b &lt;file1.bigwig&gt;...&lt;fileN.bigwig&gt;</code>: Specifies input data, it is separated by spaces.<li><code class="language-plaintext highlighter-rouge">-l &lt;file1&gt;...&lt;fileN&gt;</code>: User defined labels instead of default labels from file names. Multiple labels have to be separated by a space.<li><code class="language-plaintext highlighter-rouge">-o &lt;results.npz&gt;</code>: Specifies output data as npz format.<li><code class="language-plaintext highlighter-rouge">-e</code>option: <strong>only <code class="language-plaintext highlighter-rouge">multiBamSummary</code></strong>, allows the extension of reads to fragment size.</ul></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed-2"> <a href="#example-code-results-are-not-printed-2" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed-2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code (Results are not printed)</u></h2><ul><li><p>Here is an example command to make a coverage matrix with <code class="language-plaintext highlighter-rouge">multiBigwigSummary</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>multiBigwigSummary bins <span class="nt">-b</span> shControl_Input.bigwig shControl_H3K4me3.bigwig shNFIB_H3K4me3.bigwig shMLL1_H3K4me3.bigwig <span class="se">\</span>
      <span class="nt">-l</span> Input shControl shNFIB shMLL1 <span class="nt">-o</span> /Users/jchoi/Desktop/CorrMatrix.npz
</code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="plotcorrelation"> <a href="#plotcorrelation" class="anchor-heading" aria-labelledby="plotcorrelation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> plotCorrelation</h1><ul><li><p><code class="language-plaintext highlighter-rouge">plotCorrelation</code>analyzes and visualizes sample correlations based on the output of <code class="language-plaintext highlighter-rouge">multiBamSummary</code> or <code class="language-plaintext highlighter-rouge">multiBigwigSummary</code>. Pearson or Spearman methods are available to compute correlation coefficients. Use the following command to perform plotting for correlation with <code class="language-plaintext highlighter-rouge">plotCorrelation</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>plotCorrelation <span class="nt">-in</span> &lt;CorrMatrix.npz&gt; <span class="nt">--corMethod</span> &lt;<span class="k">select</span><span class="o">&gt;</span> <span class="nt">--skipZeros</span> <span class="nt">--whatToPlot</span> &lt;<span class="nb">type</span><span class="o">&gt;</span> <span class="se">\</span>
      <span class="nt">--removeOutliers</span> <span class="nt">--colorMap</span> &lt;color&gt; <span class="nt">--plotNumbers</span> <span class="nt">-o</span> &lt;output.extention&gt; <span class="nt">--outFileCorMatrix</span> &lt;matrix_name.tab&gt;
</code></pre></div></div><li><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-in &lt;CorrMatrix.npz&gt;</code><td>Specifies a coverage matrix file.<tr><td><code class="language-plaintext highlighter-rouge">---corMethod &lt;select&gt;</code><td>Specifies a correlation method. Possible values for <code class="language-plaintext highlighter-rouge">&lt;select&gt;</code> are <code class="language-plaintext highlighter-rouge">spearman</code> and <code class="language-plaintext highlighter-rouge">pearson</code>.<tr><td><code class="language-plaintext highlighter-rouge">--skipZeros</code> option<td>By setting this option, genomic regions that have zero or missing (NaN) values in all samples are excluded.<tr><td><code class="language-plaintext highlighter-rouge">--whatToPlot &lt;type&gt;</code><td>Choose between a heatmap or pairwise scatter plots. Possible values for <code class="language-plaintext highlighter-rouge">&lt;type&gt;</code> are <code class="language-plaintext highlighter-rouge">heatmap</code> and <code class="language-plaintext highlighter-rouge">scatterplot</code>.<tr><td><code class="language-plaintext highlighter-rouge">--removeOutliers</code> option<td>If set, bins with very large counts are removed (only <code class="language-plaintext highlighter-rouge">pearson</code>).<tr><td><code class="language-plaintext highlighter-rouge">--colorMap &lt;color&gt;</code><td>Color map to use for the heatmap.<tr><td><code class="language-plaintext highlighter-rouge">--zMin/--zMax</code> option<td>adjust heatmap intensities. If not specified, the value is set automatically.<tr><td><code class="language-plaintext highlighter-rouge">--xRange/--yRange</code> option<td>adjust plot range of the x/y-axis. If not specified, the value is set automatically.<tr><td><code class="language-plaintext highlighter-rouge">--plotNumbers</code> option<td>If set, then the correlation number is plotted on top of the heatmap (only <code class="language-plaintext highlighter-rouge">heatmap</code>).<tr><td><code class="language-plaintext highlighter-rouge">-o &lt;output.extention&gt;</code><td>Specifies output file. Possible values for file extention are <code class="language-plaintext highlighter-rouge">png</code>, <code class="language-plaintext highlighter-rouge">eps</code>, <code class="language-plaintext highlighter-rouge">pdf</code>, and <code class="language-plaintext highlighter-rouge">svg</code> (recommended).<tr><td><code class="language-plaintext highlighter-rouge">--outFileCorMatrix &lt;matrix_name.tab&gt;</code><td>Save matrix with pairwise correlation values to a tab-separated file.</table></div></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed-3"> <a href="#example-code-results-are-not-printed-3" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed-3"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code (Results are not printed)</u></h2><ul><li>Here is an example command to make a correlation plot with <code class="language-plaintext highlighter-rouge">plotCorrelation</code>:<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># Heatmap, Pearson r</span>
  <span class="nv">$ </span>plotCorrelation <span class="nt">-in</span> /Users/jchoi/Desktop/CorrMatrix.npz <span class="nt">--corMethod</span> pearson <span class="nt">--skipZeros</span> <span class="se">\</span>
      <span class="nt">--whatToPlot</span> heatmap <span class="nt">--removeOutliers</span> <span class="nt">--colorMap</span> RdYlBu <span class="nt">--plotNumbers</span> <span class="se">\</span>
      <span class="nt">-o</span> /Users/jchoi/Desktop/heatmap_PearsonCorr.svg <span class="nt">--outFileCorMatrix</span> /Users/jchoi/Desktop/heatmap_PearsonCorr.tab

  <span class="c"># Scatter plot, Spearman r</span>
  <span class="nv">$ </span>plotCorrelation <span class="nt">-in</span> /Users/jchoi/Desktop/CorrMatrix.npz <span class="nt">--corMethod</span> spearman <span class="nt">--skipZeros</span> <span class="se">\</span>
      <span class="nt">--whatToPlot</span> scatterplot <span class="nt">--colorMap</span> RdYlBu <span class="nt">-o</span> /Users/jchoi/Desktop/heatmap_SpearmanCorr.svg <span class="se">\</span>
      <span class="nt">--outFileCorMatrix</span> /Users/jchoi/Desktop/heatmap_SpearmanCorr.tab
</code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="citations"> <a href="#citations" class="anchor-heading" aria-labelledby="citations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citations</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">deeptools</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Ramírez, F.</b>, Dündar, F., Diehl, S., Grüning, B. A., &amp; Manke, T. (2014). deepTools: a flexible platform for exploring deep-sequencing data. <i>Nucleic acids research</i>, 42(W1), W187-W191. <a href="https://doi.org/10.1093/nar/gku365" target="_blank">DOI</a></span><li style="margin-bottom: 10px;"> <span><b>Ramírez, F.</b>, Ryan, D. P., Grüning, B., Bhardwaj, V., Kilpert, F., Richter, A. S., ... &amp; Manke, T. (2016). deepTools2: a next generation web server for deep-sequencing data analysis. <i>Nucleic acids research</i>, 44(Web Server issue), W160. <a href="https://doi.org/10.1093/nar/gkw257" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
