<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(5)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(3) > ul > li:nth-child(5) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(3) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(3) > ul > li:nth-child(5) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(3) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(3) > ul.nav-list > li.nav-list-item:nth-child(5) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>File Conversion and Management with SAMtools | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="File Conversion and Management with SAMtools" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/DataProcess/SAMtools_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/DataProcess/SAMtools_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="File Conversion and Management with SAMtools" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"File Conversion and Management with SAMtools","url":"http://localhost:4000/docs/DataProcess/SAMtools_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a><li class="nav-list-item"><a href="/docs/BarGen/" class="nav-list-link">Barcode Generator</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/DataProcess">Data Process</a><li class="breadcrumb-nav-list-item"><span>File Conversion and Management with SAMtools</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="file-conversion-and-management-with-samtools"> <a href="#file-conversion-and-management-with-samtools" class="anchor-heading" aria-labelledby="file-conversion-and-management-with-samtools"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> File Conversion and Management with SAMtools</h1><p class="fs-4 fw-500">The reads mapped onto the reference genome generate substantial data files in SAM format, ranging from 20 to 30 GB per sample, which could result in prolonged further analysis durations. <code class="language-plaintext highlighter-rouge">SAMtools</code> that is coded in python can be converted SAM into BAM to fast access the sequence data and analyze the sequence depth. This protocol was created based on <span style="color: #7253ec;"><b>SAMtools version 1.12</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>Python version 3.8.5, SciPy version 1.6.2, NumPy version 1.20.1, and pySam version 0.16.0.1 under macOS 12.4 OS environment.** The tool offers a variety of functions to analyze sequencing data, but this part will cover only the most commonly used features.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="installation-samtools"> <a href="#installation-samtools" class="anchor-heading" aria-labelledby="installation-samtools"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation samtools</h1><p><img src="https://img.shields.io/badge/TestPlatform-linux_64,osx_64-orange" alt="Platform" /> <a href="https://github.com/samtools/samtools"><img src="https://img.shields.io/badge/Original-samtools-DarkSeaGreen" alt="Original" /></a> <a href="https://github.com/samtools/samtools/releases"><img src="https://img.shields.io/github/v/release/samtools/samtools" alt="GitHub release" /></a> <a href="https://ci.appveyor.com/project/samtools/samtools/branch/develop"><img src="https://ci.appveyor.com/api/projects/status/enujqi06jlqw493t/branch/develop?svg=true" alt="Build status" /></a></p><p>To install HISAT2 via homebrew, type the following commands:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>brew <span class="nb">install </span>Samtools
</code></pre></div></div><h1 class="fs-7 fw-400 lh-tight" id="stats---check-the-mapping-rate"> <a href="#stats---check-the-mapping-rate" class="anchor-heading" aria-labelledby="stats---check-the-mapping-rate"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Stats - Check the mapping rate</h1><p>SAMtools <code class="language-plaintext highlighter-rouge">stats</code> collects statistics from SAM/BAM files and outputs in a text format. The output can be visualized graphically using <code class="language-plaintext highlighter-rouge">multiqc</code>. Use the following command to perform the <code class="language-plaintext highlighter-rouge">stats</code> function of samtools:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools stats -@ &lt;int&gt; &lt;input sam file.sam&gt; <span class="o">&gt;</span> &lt;output file.stats&gt;
</code></pre></div></div><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">-@ &lt;int&gt;</code>: specifies core numbers used in this work.<li><code class="language-plaintext highlighter-rouge">&lt;input.sam&gt;</code>: specifies input file as a SAM format.<li><code class="language-plaintext highlighter-rouge">&gt; &lt;output.stats&gt;</code>: specifies output file as a stats format. <code class="language-plaintext highlighter-rouge">&gt;</code> symbol must appear before the <code class="language-plaintext highlighter-rouge">&lt;output file.txt&gt;</code> syntax.</ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed"> <a href="#example-code-results-are-not-printed" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code (Results are not printed)</u></h2><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools stats -@ 10 /Users/jchoi/Desktop/SAM/Unt_UTX.sam <span class="o">&gt;</span> /Users/jchoi/Desktop/SAM/Unt_UTX.sam.stats 
</code></pre></div></div><ul><li>The resulting <code class="language-plaintext highlighter-rouge">stats</code> file can be opened text editors and it consists of alignment information:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  # This file was produced by samtools stats (1.12+htslib-1.12) and can be plotted using plot-bamstats
  # This file contains statistics for all reads.
  # The command line was:  stats -@ 10 /Users/jchoi/Desktop/SAM/Unt_UTX.sam
  # CHK, Checksum	[2]Read Names	[3]Sequences	[4]Qualities
  # CHK, CRC32 of reads which passed filtering followed by addition (32bit overflow)
  CHK	6b2cf2a8	0ec11f5c	a187d603
  # Summary Numbers. Use `grep ^SN | cut -f 2-` to extract this part.
  SN	raw total sequences:	45639340	# excluding supplementary and secondary reads
  SN	filtered sequences:	0
  SN	sequences:	45639340
  SN	is sorted:	0
  SN	1st fragments:	22819670
  SN	last fragments:	22819670
  SN	reads mapped:	40370690
  SN	reads mapped and paired:	40243620	# paired-end technology bit set + both mates mapped
  SN	reads unmapped:	5268650
  SN	reads properly paired:	36659322	# proper-pair bit set
  SN	reads paired:	45639340	# paired-end technology bit set
  SN	reads duplicated:	0	# PCR or optical duplicate bit set
  SN	reads MQ0:	227872	# mapped and MQ=0
  SN	reads QC failed:	0
  SN	non-primary alignments:	0
  SN	supplementary alignments:	0
  SN	total length:	4562450957	# ignores clipping
  SN	total first fragment length:	2281908536	# ignores clipping
  SN	total last fragment length:	2280542421	# ignores clipping
  SN	bases mapped:	4038619347	# ignores clipping
  SN	bases mapped (cigar):	3942644569	# more accurate
  ...................
</code></pre></div></div></ul><h2 class="fs-5 fw-400 lh-tight" id="export-stats-report-via-multiqc"> <a href="#export-stats-report-via-multiqc" class="anchor-heading" aria-labelledby="export-stats-report-via-multiqc"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Export stats report via multiqc</u></h2><p>The resulting stats files under the presence of the same folder can be compared and analyzed collectively with <code class="language-plaintext highlighter-rouge">multiqc</code>, use the following command:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>multiqc ./
</code></pre></div></div><h1 class="fs-7 fw-400 lh-tight" id="view---convert-to-bam-file-or-view-sam-or-bam-file"> <a href="#view---convert-to-bam-file-or-view-sam-or-bam-file" class="anchor-heading" aria-labelledby="view---convert-to-bam-file-or-view-sam-or-bam-file"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> View - Convert to BAM file or View SAM or BAM file</h1><p>SAMtools <code class="language-plaintext highlighter-rouge">view</code>, with no options or regions specified, prints all alignments in the specified input alignment file (in SAM, BAM, or CRAM format) to standard output in SAM format (with no header). <strong>Ultimately,</strong> the <code class="language-plaintext highlighter-rouge">view</code> function can convert text-format SAM files into binary BAM files and vice versa, enabling faster and more detailed manipulation of sequencing reads. Use the following command to perform the <code class="language-plaintext highlighter-rouge">view</code> function of samtools:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># For explore the strucure of them.</span>
<span class="nv">$ </span>samtools view &lt;SAM or BAM files&gt; | <span class="nb">head</span> -&lt;int&gt; <span class="c">## Usage is the same as the 'gzip' and 'cat' commands.</span>
 
 <span class="c"># For converting BAM file.</span>
<span class="nv">$ </span>samtools view <span class="nt">-b</span> <span class="nt">-q</span> &lt;int&gt; <span class="nt">-f</span> &lt;FLAG&gt; <span class="nt">-F</span> &lt;FLAG&gt; -@ &lt;int&gt; &lt;input.sam&gt; <span class="o">&gt;</span> &lt;output.bam&gt;
</code></pre></div></div><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-b</code> option<td>specifies output file format as a SAM.<tr><td><code class="language-plaintext highlighter-rouge">-q &lt;int&gt;</code> option<td>skip alignments with MAPQ smaller than <int>. <br />**Critical option,** A MAPQ &gt; 30 is considered the highest quality.</int><tr><td><code class="language-plaintext highlighter-rouge">-f &lt;FLAG&gt;</code> option<td>only output alignments with all bits set in <code class="language-plaintext highlighter-rouge">&lt;FLAG&gt;</code>.<tr><td><code class="language-plaintext highlighter-rouge">-F &lt;FLAG&gt;</code> option<td>Do not include output alignments with any bits set in <code class="language-plaintext highlighter-rouge">&lt;FLAG&gt;</code>. <br /><strong>FLAG field refers as bellow table.</strong> <br /> <code class="language-plaintext highlighter-rouge">-f</code> and <code class="language-plaintext highlighter-rouge">-F</code> options cannot be used at the same time (mutually exclusive).<tr><td><code class="language-plaintext highlighter-rouge">-@ &lt;int&gt;</code> option<td>specifies thread numbers <code class="language-plaintext highlighter-rouge">&lt;int&gt;</code>.<tr><td><code class="language-plaintext highlighter-rouge">&lt;input.sam&gt;</code><td>specifies input sam file.<tr><td><code class="language-plaintext highlighter-rouge">&gt; &lt;output.bam&gt;</code><td>specifies output bam file. ‘&gt;’ symbol must appear before the <code class="language-plaintext highlighter-rouge">&lt;output.bam&gt;</code> syntax.</table></div><blockquote class="note"><p>There are no parameters when viewing or converting a specific region, only the following format would be used: <code class="language-plaintext highlighter-rouge">chr:start-end</code>.</p><p>For more information such as global parameter, refers to samtools’ guidelines.</p></blockquote><h2 class="fs-5 fw-400 lh-tight" id="flag-field-table"> <a href="#flag-field-table" class="anchor-heading" aria-labelledby="flag-field-table"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>FLAG field table</u></h2><div class="table-wrapper"><table><thead><tr><th>Bit<th>Flag<th>Description<tbody><tr><td>0<td>0x1<td>read paired<tr><td>1<td>0x2<td>read mapped in proper pair<tr><td>2<td>0x4<td>read unmapped<tr><td>3<td>0x8<td>mate unmapped<tr><td>4<td>0x10<td>read reverse strand<tr><td>5<td>0x20<td>mate reverse strand<tr><td>6<td>0x40<td>first in pair<tr><td>7<td>0x80<td>second in pair<tr><td>8<td>0x100<td>not primary alignment<tr><td>9<td>0x200<td>read fails platform/vendor quality checks<tr><td>10<td>0x400<td>read is PCR or optical duplicate<tr><td>11<td>0x800<td>supplementary alignment</table></div><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed-1"> <a href="#example-code-results-are-not-printed-1" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code (Results are not printed)</u></h2><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools view <span class="nt">-b</span> -@ 20 shCon_Unt.sam <span class="o">&gt;</span> ~/Desktop/BAM/shCon_Unt.bam
<span class="nv">$ </span>samtools view <span class="nt">-b</span> -@ 20 shCon_H2O2.sam <span class="o">&gt;</span> ~/Desktop/BAM/shCon_H2O2.bam
<span class="nv">$ </span>samtools view <span class="nt">-b</span> -@ 20 shCon_NP.sam <span class="o">&gt;</span> ~/Desktop/BAM/shCon_NP.bam

<span class="c"># Convert to BAM file, which has a MAPQ score &gt; 40 and marked FLAG field corresponded to 0x1 and 0x2.</span>
<span class="nv">$ </span>samtools view <span class="nt">-b</span> <span class="nt">-q</span> 40 <span class="nt">-f</span> 0x1,0x2 -@ 20 shControl.sam <span class="o">&gt;</span> shControl.bam 

<span class="c"># Convert to BAM file from SAM file, which has a MAPQ score &gt; 30 and excepts FLAG field corresponded to 0x8.</span>
<span class="nv">$ </span>samtools view <span class="nt">-b</span> <span class="nt">-q</span> 30 <span class="nt">-F</span> 0x8 -@ 20 shNFIB.sam <span class="o">&gt;</span> shNFIB.bam 
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="rmdup---remove-duplicative-reads"> <a href="#rmdup---remove-duplicative-reads" class="anchor-heading" aria-labelledby="rmdup---remove-duplicative-reads"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Rmdup - Remove duplicative reads</h1><p>SAMtools <code class="language-plaintext highlighter-rouge">rmdup</code> removes potential PCR duplicates (only retain the pair with highest mapping quality). The official manual describes <strong>“This command is obsolete. Use markdup instead.”</strong>, so it probably not operate in lastly versions. Use the following command to perform the <code class="language-plaintext highlighter-rouge">rmdup</code> function of samtools:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools rmdup <span class="nt">-s</span> &lt;input.bam&gt; &lt;output.bam&gt;
</code></pre></div></div><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">-s/-S</code> option: <code class="language-plaintext highlighter-rouge">-s</code> is responsible for pair-end mode, whereas <code class="language-plaintext highlighter-rouge">-S</code> is for single-end mode.<li><code class="language-plaintext highlighter-rouge">&lt;input.bam&gt;</code>: specifies input data.<li><code class="language-plaintext highlighter-rouge">&lt;output.bam&gt;</code>: specifies output data.</ul><p class="note">the <code class="language-plaintext highlighter-rouge">rmdup</code>function does not provide multi-threads processing.</p><h2 class="fs-5 fw-400 lh-tight" id="example-code-and-output"> <a href="#example-code-and-output" class="anchor-heading" aria-labelledby="example-code-and-output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code and output</u></h2><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools rmdup <span class="nt">-s</span> shCon_Unt.bam rmdup.shCon_Unt.bam
<span class="nv">$ </span>samtools rmdup <span class="nt">-s</span> shCon_H2O2.bam rmdup.shCon_H2O2.bam
<span class="nv">$ </span>samtools rmdup <span class="nt">-s</span> shCon_NP.bam rmdup.shCon_NP.bam
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><ul><li>When <code class="language-plaintext highlighter-rouge">rmdup</code> finishes running, it prints messages summarizing what happened.<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  After operation of rmdup, terminal appears to as follows,
      [bam_rmdupse_core] 590 / 46146253 = 0.0000 in library
</code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="sort---sort-reads-by-position-or-name"> <a href="#sort---sort-reads-by-position-or-name" class="anchor-heading" aria-labelledby="sort---sort-reads-by-position-or-name"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Sort - Sort reads by position or name</h1><p>SAMtools <code class="language-plaintext highlighter-rouge">sort</code> function sorts BAM files by leftmost coordinates of references (chromosome position, no option) or by read name (<code class="language-plaintext highlighter-rouge">-n</code>option). In general, RNA-seq for feature counting (<code class="language-plaintext highlighter-rouge">htseq-count</code>) uses the <code class="language-plaintext highlighter-rouge">-n</code> option to sort by read name, while ChIP-seq for visualization or a BED file conversion does not use the <code class="language-plaintext highlighter-rouge">-n</code> option to sort by position. Use the following command to perform the <code class="language-plaintext highlighter-rouge">sort</code> function of samtools:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools <span class="nb">sort</span> <span class="nt">-n</span> -@ &lt;int&gt; &lt;input.bam&gt; <span class="nt">-o</span> &lt;output.bam&gt;
</code></pre></div></div><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">-n</code> option: sort by name, if want to sort by position, remove <code class="language-plaintext highlighter-rouge">-n</code>.<li><code class="language-plaintext highlighter-rouge">-@</code> option: specifies thread numbers <code class="language-plaintext highlighter-rouge">&lt;int&gt;</code>.<li><code class="language-plaintext highlighter-rouge">&lt;input.bam&gt;</code>: specifies input file.<li><code class="language-plaintext highlighter-rouge">-o &lt;output.bam&gt;</code>: specifies output file.</ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-and-output-1"> <a href="#example-code-and-output-1" class="anchor-heading" aria-labelledby="example-code-and-output-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code and output</u></h2><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># with -n option</span>
<span class="nv">$ </span>samtools <span class="nb">sort</span> <span class="nt">-n</span> -@ 20 rmdup.shCon_Unt.bam <span class="nt">-o</span> sort.shCon_Unt.bam
<span class="nv">$ </span>samtools <span class="nb">sort</span> <span class="nt">-n</span> -@ 20 rmdup.shCon_H2O2.bam <span class="nt">-o</span> sort.shCon_H2O2.bam
<span class="nv">$ </span>samtools <span class="nb">sort</span> <span class="nt">-n</span> -@ 20 rmdup.shCon_NP.bam <span class="nt">-o</span> sort.shCon_NP.bam

<span class="c"># with no option</span>
<span class="nv">$ </span>samtools <span class="nb">sort</span> -@ 20 /home/jh/Desktop/Bam/Rmdup_shControl.bam <span class="nt">-o</span> /home/jh/Desktop/Bam/Sort_shControl.bam
<span class="nv">$ </span>samtools <span class="nb">sort</span> -@ 20 /home/jh/Desktop/Bam/Rmdup_shNFIB.bam <span class="nt">-o</span> /home/jh/Desktop/Bam/Sort_shNFIB.bam
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><ul><li>When <code class="language-plaintext highlighter-rouge">sort</code> finishes running, it prints messages summarizing what happened.<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  After operation of sort, terminal appears to as follows,
      [bam_sort_core] merging from 20 files and 1 in-memory blocks...
</code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="index---for-chip-seq-analysis"> <a href="#index---for-chip-seq-analysis" class="anchor-heading" aria-labelledby="index---for-chip-seq-analysis"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Index - for ChIP-seq analysis</h1><p>SAMtools <code class="language-plaintext highlighter-rouge">index</code> function Indexes a coordinate-sorted BGZIP-compressed SAM, BAM, or CRAM file for fast random access. <strong>This function requires a BAM file sorted by position.</strong> Use the following command to perform the <code class="language-plaintext highlighter-rouge">index</code> function of samtools:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools index &lt;input.bam&gt;
</code></pre></div></div><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">&lt;input.bam&gt;</code>: specifies input file.</ul><p>This command has no parameters that specifically define the output. The resulting bai file is gennerated in the input folder.</p><p class="note">Preferably, the sorted BAM file and the bai file should exist in the same directory for further analysis.</p><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed-2"> <a href="#example-code-results-are-not-printed-2" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed-2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code (Results are not printed)</u></h2><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools index /home/jh/Desktop/Bam/Sort_input.bam
<span class="nv">$ </span>samtools index /home/jh/Desktop/Bam/Sort_shControl.bam
</code></pre></div></div><h1 class="fs-7 fw-400 lh-tight" id="coverage---check-the-sequencing-depth"> <a href="#coverage---check-the-sequencing-depth" class="anchor-heading" aria-labelledby="coverage---check-the-sequencing-depth"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Coverage - Check the sequencing depth</h1><p>SAMtools <code class="language-plaintext highlighter-rouge">coverage</code> function computes the coverage at each position or region and draws an ASCII-art histogram or tabulated text. The coverage is defined as the percentage of positions within each bin with at least one base aligned against it. <strong>This function requires a BAM file sorted by position and a bai file.</strong> Use the following command to perform the <code class="language-plaintext highlighter-rouge">coverage</code> function of samtools:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools coverage <span class="nt">-A</span> <span class="nt">-r</span> &lt;region&gt; &lt;input.bam&gt;
</code></pre></div></div><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">-A</code> option: Show only ASCII characters in histogram using colon and fullstop for full and half height characters.<li><code class="language-plaintext highlighter-rouge">-r &lt;region&gt;</code>: Show specified region. Format: chr:start-end.<li><code class="language-plaintext highlighter-rouge">&lt;input.bam&gt;</code>: specifies input file.</ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-and-output-2"> <a href="#example-code-and-output-2" class="anchor-heading" aria-labelledby="example-code-and-output-2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code and output</u></h2><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Coverage across all chromosome without ASCII histogram</span>
<span class="nv">$ </span>samtools coverage sort_MA_coverage.bam

<span class="c"># Coverage with ASCII histogram on chr1</span>
<span class="nv">$ </span>samtools coverage <span class="nt">-A</span> <span class="nt">-r</span> chr1 sort_MA_coverage.bam
</code></pre></div></div><ul><li>When <code class="language-plaintext highlighter-rouge">coverage</code> with no option finishes running, it prints messages summarizing what happened.<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  #rname	numreads	covbases	coverage	meandepth	meanbaseq	meanmapq
  1	1582614	6662248	11.1823	2.66237	36.3	53.9
  2	1847826	7551970	12.6625	3.10498	36.3	56.5
  3	2100124	8084882	12.9093	3.36035	36.3	54.9
  4	1195390	9339824	11.9598	1.5328	36.3	54.2
  5	2054896	8817714	12.1623	2.84217	36.3	45
  6	1548488	6936500	11.509	2.57578	36.3	53
  7	1837930	8069184	10.8628	2.48051	36.3	54.7
  8	1513826	6812219	12.5444	2.79332	36.3	54.4
  9	1194217	6236630	11.0461	2.1184	36.3	53.5
  10	1347613	5541625	12.2006	2.972	36.3	52.1
  11	1352685	5446330	11.9739	2.97988	36.3	55.7
  12	1125210	5623379	11.4336	2.29375	36.3	52.8
  13	1427718	6164305	11.8122	2.74218	36.3	54.2
  14	1581018	5558226	10.5549	3.00721	36.3	51.7
  15	1194286	5929981	12.3437	2.49079	36.3	55.8
  16	1721170	6981746	12.6329	3.12217	36.3	56.2
  17	1239184	6120904	11.4493	2.32304	36.3	54.6
  18	1010858	5168646	10.1299	1.98516	36.3	51.8
  19	2085292	6031878	12.4498	4.31301	36.3	57.3
  20	1839414	6841016	12.3928	3.33999	36.3	57.2
  21	1744680	6044533	13.1592	3.80718	36.3	57.4
  22	1028090	5238163	13.3855	2.63309	36.3	54.9
  23	1436699	5811855	12.5734	3.11603	36.3	51.5
  24	911126	4395211	10.4219	2.1645	36.3	55.8
  25	1158715	4763344	12.7016	3.09503	36.3	55.1
  MT	3248278	16596	100	19679.7	36.3	60
</code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><ul><li>When <code class="language-plaintext highlighter-rouge">coverage</code> with <code class="language-plaintext highlighter-rouge">-A</code> option finishes running, it prints messages summarizing what happened.<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  1 (59.58Mbp)
  &gt;  25.52% |.                                :                                                                       :                       :          | Number of reads: 1582614
  &gt;  22.69% |:                                :                                 :                                     : :  .                  :          |     (288307 filtered)
  &gt;  19.85% |:                                :                               . :                                     : : .:                 .:       .  | Covered bases:   6.66Mbp
  &gt;  17.02% |::   .                     :     ::      .  .                 .  : :                                     : : ::        :.     :.::      .:::| Percent covered: 11.18%
  &gt;  14.18% |::   :        :     .      :.    ::      :  : .       .       :  : ::                        .:         .::: ::     :. ::. : :::::      ::::| Mean coverage:   2.66x
  &gt;  11.34% |::  ::        :  .:.:  .   ::    ::   .  :  :.:       :     :::  : ::    ..   .              ::  :.    ::::: ::     ::.:::.: :::::      ::::| Mean baseQ:      36.3
  &gt;   8.51% |::  ::      : :  ::::.::.: ::    ::   :  :  :::  ..: ::    ::::  ::::: : ::   :    ..  .     ::  ::    :::::.::     :::::::: ::::: :  . ::::| Mean mapQ:       53.9
  &gt;   5.67% |::  ::      :.:: ::::::::: ::   .:::  :. : .:::  ::: ::... ::::  :::::.::::.  :.:: ::.::  :. :::.::   :::::::::::   :::::::: ::::: :. ::::::|
  &gt;   2.84% |:::.:: . .. ::::.:::::::::.:::..::::.::: : ::::..::::::::: ::::: :::::::::::: :::::::::: ::: ::::::   :::::::::::...::::::::::::::.:: ::::::| Histo bin width: 425.6Kbp
  &gt;   0.00% |:::::::::::.:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::.::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::| Histo max bin:   28.359%
          1       4.26M     8.51M     12.77M    17.02M    21.28M    25.53M    29.79M    34.04M    38.30M    42.56M    46.81M    51.07M    55.32M     59.58M
</code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="merge"> <a href="#merge" class="anchor-heading" aria-labelledby="merge"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Merge</h1><p><code class="language-plaintext highlighter-rouge">merge</code> function combines multiple sorted alignment files, producing a single sorted output file that contains all the input records and maintains the existing sort order. Use the following command to perform the <code class="language-plaintext highlighter-rouge">merge</code> function of samtools:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools merge -@ &lt;int&gt; &lt;output.bam&gt; &lt;input-1.bam&gt;...&lt;input-N.bam&gt;
</code></pre></div></div><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">-@</code> option: specifies thread numbers <code class="language-plaintext highlighter-rouge">&lt;int&gt;</code>.<li><code class="language-plaintext highlighter-rouge">&lt;output.bam&gt;</code>: specifies output file.<li><code class="language-plaintext highlighter-rouge">&lt;input.bam&gt;...&lt;input-N.bam&gt;</code>: specifies input file.</ul><h2 class="fs-5 fw-400 lh-tight" id="example-code-results-are-not-printed-3"> <a href="#example-code-results-are-not-printed-3" class="anchor-heading" aria-labelledby="example-code-results-are-not-printed-3"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example code (Results are not printed)</u></h2><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>samtools merge -@ 20 merged_H3K4me3.bam sort.H3K4me3_rep1.bam sort.H3K4me3_rep2.bam sort.H3K4me3_rep3.bam
<span class="c"># After merge, performs sort and index.</span>
<span class="nv">$ </span>samtools <span class="nb">sort</span> -@ 20 merged_H3K4me3.bam <span class="nt">-o</span> sort.merged_H3K4me3.bam
<span class="nv">$ </span>samtools index sort.merged_H3K4me3.bam
</code></pre></div></div><h1 class="fs-7 fw-400 lh-tight" id="citation"> <a href="#citation" class="anchor-heading" aria-labelledby="citation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citation</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">Samtools</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Li, H.</b>, Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., ... &amp; 1000 Genome Project Data Processing Subgroup. (2009). The sequence alignment/map format and SAMtools. <i>bioinformatics</i>, 25(16), 2078-2079. <a href="https://doi.org/10.1093/bioinformatics/btp352" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
