<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(2)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(2) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(5) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(2) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list > li.nav-list-item:nth-child(2) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Efficient Analysis and Visualization with ngs.plot | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Efficient Analysis and Visualization with ngs.plot" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/ChIPSeq/ngsplot_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/ChIPSeq/ngsplot_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Efficient Analysis and Visualization with ngs.plot" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Efficient Analysis and Visualization with ngs.plot","url":"http://localhost:4000/docs/ChIPSeq/ngsplot_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/ChIPSeq">Data Analysis - ChIP-seq</a><li class="breadcrumb-nav-list-item"><span>Efficient Analysis and Visualization with ngs.plot</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="efficient-analysis-and-visualization-of-chip-seq-data-via-genomic-databases"> <a href="#efficient-analysis-and-visualization-of-chip-seq-data-via-genomic-databases" class="anchor-heading" aria-labelledby="efficient-analysis-and-visualization-of-chip-seq-data-via-genomic-databases"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Efficient Analysis and Visualization of ChIP-seq Data via Genomic Databases</h1><p class="fs-4 fw-500">The ‘ngs.plot’ tool is a specialized resource designed to enhance the visualization and analysis of next-generation sequencing (NGS) data. This tool is particularly adept at quickly accessing and integrating large genomic databases to plot dense NGS data against a backdrop of functional genomic elements. One of the primary advantages of ‘ngs.plot’ is its ability to manage and visualize complex datasets with ease and precision. Users can rapidly generate high-quality plots of genomic regions of interest, which is critical for interpreting interactions and expression levels in various genomic studies, such as ChIP-seq and RNA-seq analyses. This protocol was created based on <span style="color: #7253ec;"><b>ngs.plot version 2.61</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>Python version 3.8.5, Perl version 5.26.2, and R version 4.4.0 under macOS 12.4 environment.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-installation"> <a href="#step-by-step-installation" class="anchor-heading" aria-labelledby="step-by-step-installation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Installation</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <a href="https://github.com/shenlab-sinai/ngsplot"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <img src="https://img.shields.io/badge/release-2.61-blue?labelColor=black" alt="version" /> <img src="https://img.shields.io/badge/status-unstable-red?labelColor=black" alt="Status" /></p><h3 id="1-download-ngsplot-package-from-gdrive-official-to-a-desired-folder-and-exract-it-as-follow"> <a href="#1-download-ngsplot-package-from-gdrive-official-to-a-desired-folder-and-exract-it-as-follow" class="anchor-heading" aria-labelledby="1-download-ngsplot-package-from-gdrive-official-to-a-desired-folder-and-exract-it-as-follow"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Download ngs.plot package from <a href="https://drive.google.com/folderview?id=0B1PVLadG_dCKN1liNFY0MVM1Ulk&amp;usp=sharing" target="_blank">GDrive</a> (Official) to a desired folder and exract it as follow.</h3><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd </span>downloads
<span class="nv">$ </span><span class="nb">tar </span>xvfz ngsplot-2.61.tar.gz
<span class="nv">$ </span><span class="nb">mv </span>ngsplot-2.61 /Users/jchoi/ngsplot
</code></pre></div></div><h3 id="2-add-ngsplot-executables-to-your-path-under-zshrc"> <a href="#2-add-ngsplot-executables-to-your-path-under-zshrc" class="anchor-heading" aria-labelledby="2-add-ngsplot-executables-to-your-path-under-zshrc"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Add ngsplot executables to your path under zshrc.</h3><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>nano ~/.zshrc

<span class="c"># Add syntax as follows;</span>
<span class="nb">export </span><span class="nv">PATH</span><span class="o">=</span><span class="s2">"/your/path/to/ngsplot/bin:</span><span class="nv">$PATH</span><span class="s2">"</span>
<span class="nb">export </span><span class="nv">NGSPLOT</span><span class="o">=</span><span class="s2">"/your/path/to/ngsplot"</span>

<span class="c"># For example</span>
<span class="nb">export </span><span class="nv">PATH</span><span class="o">=</span><span class="s2">"/Users/Desktop/ChIP_seq/ngsplot/bin:</span><span class="nv">$PATH</span><span class="s2">"</span>
<span class="nb">export </span><span class="nv">NGSPLOT</span><span class="o">=</span><span class="s2">"/Users/Desktop/ChIP_seq/ngsplot"</span>
</code></pre></div></div><h3 id="3-save-and-reload-shell-environment-as-follow"> <a href="#3-save-and-reload-shell-environment-as-follow" class="anchor-heading" aria-labelledby="3-save-and-reload-shell-environment-as-follow"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. save and reload shell environment as follow.</h3><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">source</span> ~/.zshrc
</code></pre></div></div><h3 id="4-install-ngsplot-dependent-libraries-in-r"> <a href="#4-install-ngsplot-dependent-libraries-in-r" class="anchor-heading" aria-labelledby="4-install-ngsplot-dependent-libraries-in-r"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4. Install ngs.plot dependent libraries in R.</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">install.packages</span><span class="p">(</span><span class="s2">"doMC"</span><span class="p">,</span><span class="w"> </span><span class="n">dep</span><span class="o">=</span><span class="nb">T</span><span class="p">)</span><span class="w">
</span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"caTools"</span><span class="p">,</span><span class="w"> </span><span class="n">dep</span><span class="o">=</span><span class="nb">T</span><span class="p">)</span><span class="w">
</span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"utils"</span><span class="p">,</span><span class="w"> </span><span class="n">dep</span><span class="o">=</span><span class="nb">T</span><span class="p">)</span><span class="w">
</span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"BSgenome"</span><span class="p">)</span><span class="w">
</span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"Rsamtools"</span><span class="p">)</span><span class="w">
</span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"ShortRead"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><h3 id="5-prepare-genome-database"> <a href="#5-prepare-genome-database" class="anchor-heading" aria-labelledby="5-prepare-genome-database"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 5. Prepare Genome Database.</h3><ul><li>Download desired genome database from <a href="https://drive.google.com/drive/folders/0B1PVLadG_dCKNEsybkh5TE9XZ1E?resourcekey=0-ux6XMoJr0qaS6lemqPJBpQ" target="_blank">GDrive</a> (Official).<li>Install this file as follows:</ul><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>ngsplotdb.py list <span class="c"># check current list</span>
ID       Assembly     Species      EnsVer   NPVer    InstalledFeatures
hg19	GRCh37	homo_sapiens	75.0	3.0	cgi,exon,genebody,tss,tes
IRGSP-1	IRGSP-1.0	oryza_sativa	21.0	3.0	exon,genebody,tss,tes

<span class="nv">$ </span>ngsplotdb.py <span class="nb">install</span> /Users/jchoi/Downloads/ngsplotdb_mm10_75_3.00.tar.gz <span class="c"># install databse</span>

<span class="nv">$ </span>ngsplotdb.py list <span class="c"># check the installed database</span>
ID       Assembly     Species      EnsVer   NPVer    InstalledFeatures
hg19	GRCh37	homo_sapiens	75.0	3.0	cgi,exon,genebody,tss,tes
IRGSP-1	IRGSP-1.0	oryza_sativa	21.0	3.0	exon,genebody,tss,tes
mm10	GRCm38	mus_musculus	75.0	3.0	cgi,exon,genebody,tss,tes
</code></pre></div></div><p class="note">This tool may occasionally experience malfunctions. These are predominantly attributed to issues with Python syntax, and the author swiftly addresses and resolves these concerns by issuing updated source code. Should the tool cease to function abruptly, it is advisable to consult the creator’s GitHub repository. <a href="https://github.com/shenlab-sinai/ngsplot"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a></p><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="running-ngsplot"> <a href="#running-ngsplot" class="anchor-heading" aria-labelledby="running-ngsplot"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running ngs.plot</h1><ul><li><p>Use the following command to visulize enriched peaks from ChIP-seq data with <code class="language-plaintext highlighter-rouge">ngs.plot</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>ngs.plot.r <span class="nt">-G</span> &lt;genome&gt; <span class="nt">-R</span> &lt;region&gt; <span class="nt">-C</span> &lt;IndexedInput.bam&gt; <span class="nt">-O</span> &lt;outputFolder&gt; &lt;OptionalParameters&gt;
</code></pre></div></div><li><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-G &lt;Genome&gt;</code><td>Specifies the genome name. Use <code class="language-plaintext highlighter-rouge">ngsplotdb.py</code> list to show available genomes. See the Step-by-step Installation section.<tr><td><code class="language-plaintext highlighter-rouge">-R &lt;region&gt;</code><td>Specifies genomic regions to plot. Possible values for region are <code class="language-plaintext highlighter-rouge">tss</code>, <code class="language-plaintext highlighter-rouge">tes</code>, <code class="language-plaintext highlighter-rouge">genebody</code>, <code class="language-plaintext highlighter-rouge">exon</code>, <code class="language-plaintext highlighter-rouge">cgi</code>, and <code class="language-plaintext highlighter-rouge">enhancer</code>.<tr><td><code class="language-plaintext highlighter-rouge">-C &lt;IndexedInput.bam&gt;</code><td>Specifies the input file. This file reqires an <strong>indexed</strong> BAM file for single plot or a specific configuration file for multiplot (See the below).<tr><td><code class="language-plaintext highlighter-rouge">-O &lt;outputFolder&gt;</code><td>Specifies the output file. Several files will be generated.</table></div><li><p>Prepare a configuration file and save as plaintext (<code class="language-plaintext highlighter-rouge">name.txt</code>) as follow:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># If you want to specify the gene list as "genome", use "-1".</span>
  <span class="c"># Use TAB to separate the three columns: coverage file&lt;TAB&gt;gene list&lt;TAB&gt;title</span>
  <span class="c"># "title" will be shown in the figure's legend.</span>
  hesc.H3k4me3.rmdup.sort.bam     high_expressed_genes.txt         <span class="s2">"High"</span>
  hesc.H3k4me3.rmdup.sort.bam     medium_expressed_genes.txt       <span class="s2">"Med"</span>
  hesc.H3k4me3.rmdup.sort.bam     low_expressed_genes.txt          <span class="s2">"Low"</span>
</code></pre></div></div><li><p>Major Optional Parameters (<code class="language-plaintext highlighter-rouge">&lt;OptionalParameter&gt;</code>)</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-T &lt;Title&gt;</code><td>Specifies image title (Default: NoName).<tr><td><code class="language-plaintext highlighter-rouge">-L &lt;int&gt;</code><td>Specifies size in bps. By default, when <code class="language-plaintext highlighter-rouge">-R tss, tes, genebody</code>, <code class="language-plaintext highlighter-rouge">-L 2000</code>; when <code class="language-plaintext highlighter-rouge">-R exon, cgi</code>, <code class="language-plaintext highlighter-rouge">-L 500</code>.<tr><td><code class="language-plaintext highlighter-rouge">-P &lt;int&gt;</code><td>Specifies <code class="language-plaintext highlighter-rouge">0</code> to use all CPUs that are detected on your machine.<tr><td><code class="language-plaintext highlighter-rouge">-FL &lt;int&gt;</code><td>This parameter calculates physical instead of read coverage. This will produce figures that contain more accurate representation of ChIP enrichment. <br /><strong>You should set this value equal to the average fragment length in your sequencing library</strong> (Default: 150).<tr><td><code class="language-plaintext highlighter-rouge">-LEG &lt;int&gt;</code><td>Control the display of legend: 1 (default) or 0 (No legend).<tr><td><code class="language-plaintext highlighter-rouge">-VLN &lt;int&gt;</code><td>Control the display of vertical lines: 1 (default) or 0 (No line).</table></div><li><p>For more detailed parameter options, see the <a href="https://github.com/shenlab-sinai/ngsplot/wiki/ProgramArguments101" target="_blank">ProgramArguments101</a>.</p></ul><h2 class="fs-5 fw-400 lh-tight" id="example-code"> <a href="#example-code" class="anchor-heading" aria-labelledby="example-code"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code</u></h2><ul><li><p>Here is an example command to depict peaks distribution:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="c"># Single plot</span>
  <span class="nv">$ </span>ngs.plot.R <span class="nt">-G</span> mm10 <span class="nt">-R</span> genebody <span class="nt">-C</span> /Users/jchoi/Desktop/R_FINAL.bam <span class="nt">-O</span> /Users/jchoi/Desktop/NFIB <span class="se">\</span>
  <span class="nt">-T</span> NFIB <span class="nt">-L</span> 2000 <span class="nt">-SE</span> 0 <span class="nt">-FL</span> 100 <span class="nt">-LEG</span> 0 <span class="nt">-VLN</span> 0

  <span class="c"># Multiplot</span>
  <span class="c">## Configuration file, H3K4_K27.txt</span>
  Final.sort.H2O2_H3K27me3.bam:Final.sort.Unt_IPT.bam	<span class="nt">-1</span>	<span class="s2">"H3K27me3"</span>
  Final.sort.H2O2_H3K4me3.bam:Final.sort.Unt_IPT.bam	<span class="nt">-1</span>	<span class="s2">"H3K4me3"</span>
  <span class="c">## Command</span>
  <span class="nv">$ </span>ngs.plot.r <span class="nt">-G</span> hg19 <span class="nt">-R</span> genebody <span class="nt">-C</span> H3K4_K27.txt <span class="nt">-O</span> /Users/jchoi/Desktop/Bivalency <span class="se">\</span>
  <span class="nt">-L</span> 3000 <span class="nt">-SE</span> 0 <span class="nt">-FL</span> 100 <span class="nt">-LEG</span> 0 <span class="nt">-VLN</span> 0 <span class="nt">-YAS</span> <span class="nt">-0</span>.4,1.5 <span class="nt">-SC</span> <span class="nt">-2</span>,4
</code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h2 class="fs-5 fw-400 lh-tight" id="example-plot"> <a href="#example-plot" class="anchor-heading" aria-labelledby="example-plot"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Plot</u></h2><p>The graph created using the second (multiplot) of the example codes above, refined using Illustrator 2024.</p><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_ChIPSeq/Images/ngsPlot.png" alt="ngsplot" style="width: 40vh; height: auto;" /></p><h1 class="fs-7 fw-400 lh-tight" id="citation"> <a href="#citation" class="anchor-heading" aria-labelledby="citation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citation</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">ngsplot</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Shen, L.</b>, Shao, N., Liu, X., &amp; Nestler, E. (2014). ngs. plot: Quick mining and visualization of next-generation sequencing data by integrating genomic databases. <i>BMC genomics</i>, 15, 1-14. <a href="https://doi.org/10.1186/1471-2164-15-284" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
