<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(5)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(5) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(5) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(5) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list > li.nav-list-item:nth-child(5) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Exploring ChIP-seq Data with ChIPseeker | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Exploring ChIP-seq Data with ChIPseeker" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/ChIPSeq/PeakAnnotation_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/ChIPSeq/PeakAnnotation_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Exploring ChIP-seq Data with ChIPseeker" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Exploring ChIP-seq Data with ChIPseeker","url":"http://localhost:4000/docs/ChIPSeq/PeakAnnotation_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-portfolio" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/ChIPSeq">Data Analysis - ChIP-seq</a><li class="breadcrumb-nav-list-item"><span>Exploring ChIP-seq Data with ChIPseeker</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="exploring-chip-seq-data-with-chipseeker-an-essential-tool-for-peak-annotation"> <a href="#exploring-chip-seq-data-with-chipseeker-an-essential-tool-for-peak-annotation" class="anchor-heading" aria-labelledby="exploring-chip-seq-data-with-chipseeker-an-essential-tool-for-peak-annotation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Exploring ChIP-seq Data with ChIPseeker: An Essential Tool for Peak Annotation</h1><p class="fs-4 fw-500">Peak annotation in ChIP-seq data analysis involves identifying genomic features associated with regions of significant enrichment (peaks), such as genes, promoters, enhancers, or other regulatory elements. This process is crucial for understanding the biological significance of the data by linking DNA-protein interactions to specific functional genomic regions. ChIPseeker is a popular R package designed for the annotation and visualization of ChIP-seq data. It supports multiple genome annotations and can be easily integrated with other bioinformatics tools, enhancing its utility. A key strength of ChIPseeker is its ability to provide detailed annotation of peaks with respect to nearest genes and genomic features, along with comprehensive visualization options like pie charts and histograms that summarize the genomic distribution of the peaks. This protocol was created based on <span style="color: #7253ec;"><b>ChIPseeker version 1.32.1</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>R version 4.4.0 under macOS 12.4 environment.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="installation-chipseeker"> <a href="#installation-chipseeker" class="anchor-heading" aria-labelledby="installation-chipseeker"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation ChIPseeker</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Windows-0078D6?style=flat&amp;logo=windows&amp;logoColor=white" alt="EnvTest3" /> <a href="https://github.com/YuLab-SMU/ChIPseeker"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="https://bioconductor.org/packages/release/bioc/html/ChIPseeker.html"><img src="https://img.shields.io/badge/BioC-276DC3?style=flat&amp;logo=r&amp;logoColor=white" alt="BioC" /></a> <a href="https://bioconductor.org/packages/release/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html"><img src="https://img.shields.io/badge/Official-e34c26?style=flat&amp;logo=html5&amp;logoColor=white" alt="web" /></a> <a href="https://www.bioconductor.org/packages/ChIPseeker"><img src="https://img.shields.io/badge/release%20version-1.32.1-DarkSeaGreen.svg?labelColor=black" alt="" /></a> <a href="http://www.repostatus.org/#active"><img src="https://img.shields.io/badge/repo%20Status-Active-DarkSeaGreen?style=flat&amp;colorA=000000" alt="Project Status" /></a></p><h3 id="1-check-and-install-chipseeker"> <a href="#1-check-and-install-chipseeker" class="anchor-heading" aria-labelledby="1-check-and-install-chipseeker"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Check and install ChIPseeker</h3><ul><li><p>Start by ensuring the BiocManager package is installed, then install ChIPseeker using it:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">requireNamespace</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))</span><span class="w">
      </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">)</span><span class="w">
  </span><span class="c1">## BiocManager::install("BiocUpgrade") ## you may need this</span><span class="w">
  </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"ChIPseeker"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><h3 id="2-install-required-libraries"> <a href="#2-install-required-libraries" class="anchor-heading" aria-labelledby="2-install-required-libraries"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Install Required Libraries</h3><ul><li><p>Install additional libraries necessary for ChIPseeker to function properly. These include packages for genomic data manipulation and annotation. You can install these packages manually using the <code class="language-plaintext highlighter-rouge">BiocManager::install("&lt;PackageName&gt;")</code>. The list of packages is shown below.</p><p><strong>Dependency list</strong></p><ul><li><code class="language-plaintext highlighter-rouge">GenomeInfoDb</code>: Utilities for manipulating chromosome names<li><code class="language-plaintext highlighter-rouge">GenomicRanges</code>: Representation and manipulation of genomic intervals<li><code class="language-plaintext highlighter-rouge">GenomicFeatures</code>: Conveniently import and query gene models<li><code class="language-plaintext highlighter-rouge">rtracklayer</code>: R interface to genome annotation files and the UCSC genome browser<li><code class="language-plaintext highlighter-rouge">clusterProfiler</code>: statistical analysis and visualization of functional profiles for genes and gene clusters<li><code class="language-plaintext highlighter-rouge">AnnotationDbi</code>: Manipulation of SQLite-based annotations</ul><p><strong>Genome data list</strong> (<strong>Note,</strong> it depends on the species you want to analyze. In this manual, I only provide the analysis of human and mouse genome.)</p><ul><li><code class="language-plaintext highlighter-rouge">TxDb.Mmusculus.UCSC.mm10.knownGene</code>: Annotation package for TxDb objects for mouse<li><code class="language-plaintext highlighter-rouge">TxDb.Hsapiens.UCSC.hg19.knownGene</code>: Annotation package for TxDb objects for human<li><code class="language-plaintext highlighter-rouge">EnsDb.Mmusculus.v79</code>: Ensembl based annotation package for mouse<li><code class="language-plaintext highlighter-rouge">EnsDb.Hsapiens.v86</code>: Ensembl based annotation package for human<li><code class="language-plaintext highlighter-rouge">org.Mm.eg.db</code>: Genome wide annotation for Mouse<li><code class="language-plaintext highlighter-rouge">org.Hs.eg.db</code>: Genome wide annotation for human</ul></ul><p><span class="label label-purple" style="display: inline-block; margin-bottom: -10px; position: relative; left: -10px;">Critical Note</span></p><ul><li><p>Before installing each package, check if it is already installed and only install those that are missing. This can save time and avoid unnecessary installations. Here is how you can check and install a package only if it’s not already installed:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">install_if_missing</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">package_name</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
      </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">requireNamespace</span><span class="p">(</span><span class="n">package_name</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="p">{</span><span class="w">
          </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="n">package_name</span><span class="p">)</span><span class="w">
      </span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="w"> </span><span class="p">{</span><span class="w">
          </span><span class="n">message</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="n">package_name</span><span class="p">,</span><span class="w"> </span><span class="s2">"is already installed."</span><span class="p">))</span><span class="w">
      </span><span class="p">}</span><span class="w">
  </span><span class="p">}</span><span class="w">

  </span><span class="c1"># List of dependencies</span><span class="w">
  </span><span class="n">dependencies</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"GenomeInfoDb"</span><span class="p">,</span><span class="w"> </span><span class="s2">"GenomicRanges"</span><span class="p">,</span><span class="w"> </span><span class="s2">"GenomicFeatures"</span><span class="p">,</span><span class="w"> 
                  </span><span class="s2">"rtracklayer"</span><span class="p">,</span><span class="w"> </span><span class="s2">"clusterProfiler"</span><span class="p">,</span><span class="w"> </span><span class="s2">"AnnotationDbi"</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Install dependencies one by one</span><span class="w">
  </span><span class="n">lapply</span><span class="p">(</span><span class="n">dependencies</span><span class="p">,</span><span class="w"> </span><span class="n">install_if_missing</span><span class="p">)</span><span class="w">

  </span><span class="c1"># List of genome data packages (Optional)</span><span class="w">
  </span><span class="n">genome_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"TxDb.Mmusculus.UCSC.mm10.knownGene"</span><span class="p">,</span><span class="w"> 
                  </span><span class="s2">"TxDb.Hsapiens.UCSC.hg19.knownGene"</span><span class="p">,</span><span class="w"> 
                  </span><span class="s2">"EnsDb.Mmusculus.v79"</span><span class="p">,</span><span class="w"> </span><span class="s2">"EnsDb.Hsapiens.v86"</span><span class="p">,</span><span class="w">
                  </span><span class="s2">"org.Mm.eg.db"</span><span class="p">,</span><span class="w"> </span><span class="s2">"org.Hs.eg.db"</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Install genome data packages one by one</span><span class="w">
  </span><span class="n">lapply</span><span class="p">(</span><span class="n">genome_data</span><span class="p">,</span><span class="w"> </span><span class="n">install_if_missing</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-running-guide-chipseeker"> <a href="#step-by-step-running-guide-chipseeker" class="anchor-heading" aria-labelledby="step-by-step-running-guide-chipseeker"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-step Running Guide ChIPseeker</h1><h3 id="1-load-chipseeker-and-its-dependency-packages"> <a href="#1-load-chipseeker-and-its-dependency-packages" class="anchor-heading" aria-labelledby="1-load-chipseeker-and-its-dependency-packages"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Load ChIPseeker and its dependency packages</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ChIPseeker</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">EnsDb.Hsapiens.v86</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">AnnotationDbi</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggupset</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggplotify</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggimage</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><h3 id="2-load-peak-data"> <a href="#2-load-peak-data" class="anchor-heading" aria-labelledby="2-load-peak-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Load peak data</h3><ul><li><p>In this course, I will use peak data acquired with SICER. When using data acquired with MACS, you can load <code class="language-plaintext highlighter-rouge">narrow</code>, <code class="language-plaintext highlighter-rouge">broad</code>, or <code class="language-plaintext highlighter-rouge">bed</code> files as you see fit for your analysis.</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">Sicer_H3K4me3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readPeakFile</span><span class="p">(</span><span class="s2">"/Users/jchoi/Desktop/Unt_H3K4me3-W50-G100-islands-summary"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">
  </span><span class="n">Sicer_H3K27me3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readPeakFile</span><span class="p">(</span><span class="s2">"/Users/jchoi/Desktop/Unt_H3K27me3-W50-G100-islands-summary"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">
  </span><span class="n">Sicer_MLL1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readPeakFile</span><span class="p">(</span><span class="s2">"/Users/jchoi/Desktop/Unt_MLL1-W50-G100-islands-summary"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">
  </span><span class="n">Sicer_UTX</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">readPeakFile</span><span class="p">(</span><span class="s2">"/Users/jchoi/Desktop/Unt_UTX-W50-G100-islands-summary"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">)</span><span class="w">
  </span><span class="n">peaks</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">H3K4me3</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Sicer_H3K4me3</span><span class="p">,</span><span class="w"> </span><span class="n">H3K27me3</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Sicer_H3K27me3</span><span class="p">,</span><span class="w"> </span><span class="n">MLL1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Sicer_MLL1</span><span class="p">,</span><span class="w"> </span><span class="n">UTX</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Sicer_UTX</span><span class="p">)</span><span class="w">
  </span><span class="n">peaks</span><span class="w">
</span></code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><ul><li><strong>Output:</strong><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; peaks
  $H3K4me3
  GRanges object with 59583 ranges and 5 metadata columns:
          seqnames            ranges strand |        V4        V5           V6         V7           V8
              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;    &lt;numeric&gt;  &lt;numeric&gt;    &lt;numeric&gt;
      [1]     chr1       10051-10599      * |        53        68    1.0000000   0.924306    1.0000000
      [2]     chr1       11601-14399      * |       171       181    0.0653982   1.120382    0.0695043
      [3]     chr1       14551-15249      * |        34        28    0.0166897   1.440023    0.0182155
      [4]     chr1       15401-17099      * |        73        77    0.1441852   1.124296    0.1513844
      [5]     chr1       17251-18049      * |        38        64    1.0000000   0.704129    1.0000000
      ...      ...               ...    ... .       ...       ...          ...        ...          ...
  [59579]     chrY 59011501-59026949      * |      1413       778 7.55580e-145  2.1538291 4.09978e-144
  [59580]     chrY 59027151-59033299      * |       552       225 1.81527e-101  2.9094120 8.81781e-101
  [59581]     chrY 59213451-59214949      * |        75         5  2.58202e-66 17.7885247  1.10346e-65
  [59582]     chrY 59362851-59363399      * |        25        46  1.00000e+00  0.6445118  1.00000e+00
  [59583]     chrM       12351-13249      * |        39      1856  1.00000e+00  0.0249193  1.00000e+00
  -------
  seqinfo: 25 sequences from an unspecified genome; no seqlengths

  $H3K27me3
  GRanges object with 93824 ranges and 5 metadata columns:
          seqnames      ranges strand |        V4        V5          V6        V7          V8
              &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;   &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
      [1]     chr1 31101-31849      * |        41        11 7.52852e-14   4.02644 2.02150e-13
      [2]     chr1 32601-33349      * |        32        13 4.73028e-07   2.65911 7.00814e-07
      [3]     chr1 33701-34499      * |        54        23 8.04691e-10   2.53627 1.53203e-09
      [4]     chr1 34651-35399      * |        40        11 3.12778e-13   3.92823 7.96170e-13
      [5]     chr1 35651-36349      * |        54        11 9.84610e-23   5.30312 4.90046e-22
      ...      ...         ...    ... .       ...       ...         ...       ...         ...
  [93820]     chrM   4751-5899      * |        62      2070           1 0.0323557           1
  [93821]     chrM   6051-6999      * |        46      1885           1 0.0263619           1
  [93822]     chrM  7151-10949      * |       187      6820           1 0.0296202           1
  [93823]     chrM 11101-14399      * |       202      6535           1 0.0333915           1
  [93824]     chrM 14551-16549      * |       131      3655           1 0.0387181           1
  -------
  seqinfo: 25 sequences from an unspecified genome; no seqlengths

  $MLL1
  GRanges object with 34003 ranges and 5 metadata columns:
          seqnames            ranges strand |        V4        V5          V6        V7          V8
              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;   &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
      [1]     chr1       10001-10499      * |        51        63   1.0000000  0.780916   1.0000000
      [2]     chr1       11901-13049      * |        60        92   1.0000000  0.629126   1.0000000
      [3]     chr1       16901-17299      * |        31        22   0.0397828  1.359294   0.0557065
      [4]     chr1       17451-18099      * |        36        49   1.0000000  0.708730   1.0000000
      [5]     chr1       19851-20799      * |        39        44   1.0000000  0.855040   1.0000000
      ...      ...               ...    ... .       ...       ...         ...       ...         ...
  [33999]     chrY 59029651-59030999      * |        91        52 1.48881e-06  1.688156 5.32648e-06
  [34000]     chrY 59031151-59032099      * |        68        30 3.24751e-09  2.186563 2.14925e-08
  [34001]     chrY 59032401-59033199      * |        36        70 1.00000e+00  0.496111 1.00000e+00
  [34002]     chrY 59362951-59363399      * |        85        44 6.37554e-08  1.863548 3.07182e-07
  [34003]     chrM           1-16549      * |      7206     29637 1.00000e+00  0.234549 1.00000e+00
  -------
  seqinfo: 25 sequences from an unspecified genome; no seqlengths

  $UTX
  GRanges object with 44845 ranges and 5 metadata columns:
          seqnames            ranges strand |        V4        V5          V6        V7          V8
              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;   &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
      [1]     chr1       10051-10449      * |        58        59 0.161707386  1.127780 0.178211140
      [2]     chr1       13301-13899      * |        32        35 0.349485362  1.048891 0.373021173
      [3]     chr1       14551-15249      * |        26        28 0.325894887  1.065279 0.348971602
      [4]     chr1       16851-17799      * |        47        76 1.000000000  0.709467 1.000000000
      [5]     chr1       19101-20649      * |        70        53 0.000421408  1.515202 0.000653707
      ...      ...               ...    ... .       ...       ...         ...       ...         ...
  [44841]     chrY 59028701-59029749      * |        40        32 1.17852e-02  1.434030 1.48488e-02
  [44842]     chrY 59030051-59030599      * |        42        24 1.55131e-05  2.007642 3.07036e-05
  [44843]     chrY 59030851-59032599      * |        85        59 6.65690e-06  1.652780 1.41208e-05
  [44844]     chrY 59362901-59363399      * |        45        45 1.57970e-01  1.147224 1.74366e-01
  [44845]     chrM           1-16549      * |      4103     29637 1.00000e+00  0.158824 1.00000e+00
  -------
  seqinfo: 25 sequences from an unspecified genome; no seqlengths
</code></pre></div></div></ul><h3 id="3-make-coverage-plot"> <a href="#3-make-coverage-plot" class="anchor-heading" aria-labelledby="3-make-coverage-plot"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. Make coverage plot</h3><ul><li><p>Before depict coverage plot, it is important to understand the structure of the <code class="language-plaintext highlighter-rouge">peaks</code> objects: make sure you know what each column means. The output of ‘sicer’ is chromosome (<strong>V1; seqnames</strong>), start (<strong>V2; ranges</strong>), end (<strong>V2; ranges</strong>), strand (<strong>V3; strand</strong>), ChIP_signal (<strong>V4</strong>), Input_signal (<strong>V5</strong>), p-val (<strong>V6</strong>), fold change (<strong>V7</strong>), fdr (<strong>V8</strong>), in that order. This means that to draw a coverage plot, you will need values from <strong>V4</strong> or <strong>V7</strong>.</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">covplot</span><span class="p">(</span><span class="n">peaks</span><span class="p">[[</span><span class="s2">"H3K4me3"</span><span class="p">]],</span><span class="w"> </span><span class="n">weightCol</span><span class="o">=</span><span class="s2">"V4"</span><span class="p">)</span><span class="w"> </span><span class="c1"># To save the plot, use 'Export' in the 'Plots' section on the right.</span><span class="w">
  </span><span class="n">covplot</span><span class="p">(</span><span class="n">peaks</span><span class="p">[[</span><span class="s2">"H3K27me3"</span><span class="p">]],</span><span class="w"> </span><span class="n">weightCol</span><span class="o">=</span><span class="s2">"V4"</span><span class="p">)</span><span class="w">
  </span><span class="n">covplot</span><span class="p">(</span><span class="n">peaks</span><span class="p">[[</span><span class="s2">"MLL1"</span><span class="p">]],</span><span class="w"> </span><span class="n">weightCol</span><span class="o">=</span><span class="s2">"V4"</span><span class="p">)</span><span class="w">
  </span><span class="n">covplot</span><span class="p">(</span><span class="n">peaks</span><span class="p">[[</span><span class="s2">"UTX"</span><span class="p">]],</span><span class="w"> </span><span class="n">weightCol</span><span class="o">=</span><span class="s2">"V4"</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Can be selected specific chromosomes as follow:</span><span class="w">
  </span><span class="n">covplot</span><span class="p">(</span><span class="n">peaks</span><span class="p">[[</span><span class="s2">"H3K4me3"</span><span class="p">]],</span><span class="w"> </span><span class="n">weightCol</span><span class="o">=</span><span class="s2">"V4"</span><span class="p">,</span><span class="w"> </span><span class="n">chrs</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">"chr3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"chr4"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li><p><strong>Output:</strong></p></ul><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_ChIPSeq/Images/CoveragePlot.png" alt="CoveragePlot" style="width: 60vh; height: auto;" /></p><h3 id="4-profile-of-chip-peaks-binding-to-tss-regions"> <a href="#4-profile-of-chip-peaks-binding-to-tss-regions" class="anchor-heading" aria-labelledby="4-profile-of-chip-peaks-binding-to-tss-regions"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4. Profile of ChIP peaks binding to TSS regions</h3><ul><li><p>For calculating the profile of ChIP peaks binding to TSS regions, prepare the TSS regions, which are defined as the flanking sequence of the TSS sites. Then align the peaks that are mapping to these regions, and generate the tagMatrix:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">txdb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="w">
  </span><span class="n">promoter</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">getPromoters</span><span class="p">(</span><span class="n">TxDb</span><span class="o">=</span><span class="n">txdb</span><span class="p">,</span><span class="w"> </span><span class="n">upstream</span><span class="o">=</span><span class="m">3000</span><span class="p">,</span><span class="w"> </span><span class="n">downstream</span><span class="o">=</span><span class="m">3000</span><span class="p">)</span><span class="w">
  </span><span class="n">tagMatrixList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">as.list</span><span class="p">(</span><span class="n">peaks</span><span class="p">),</span><span class="w"> </span><span class="n">getTagMatrix</span><span class="p">,</span><span class="w"> </span><span class="n">windows</span><span class="o">=</span><span class="n">promoter</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li><strong>Output:</strong><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; txdb &lt;- TxDb.Hsapiens.UCSC.hg19.knownGene
  &gt; promoter &lt;- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
  &gt; tagMatrixList &lt;- lapply(as.list(peaks), getTagMatrix, windows=promoter)
  &gt;&gt; preparing start_site regions by gene... 2024-05-21 12:41:47
  &gt;&gt; preparing tag matrix...  2024-05-21 12:41:47 
  &gt;&gt; preparing start_site regions by gene... 2024-05-21 12:41:56
  &gt;&gt; preparing tag matrix...  2024-05-21 12:41:56 
  &gt;&gt; preparing start_site regions by gene... 2024-05-21 12:42:01
  &gt;&gt; preparing tag matrix...  2024-05-21 12:42:01 
  &gt;&gt; preparing start_site regions by gene... 2024-05-21 12:42:06
  &gt;&gt; preparing tag matrix...  2024-05-21 12:42:06
</code></pre></div></div><li><u>Heatmap of ChIP binding to TSS regions</u><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Multiplot</span><span class="w">
  </span><span class="n">peakHeatmap</span><span class="p">(</span><span class="n">peaks</span><span class="p">,</span><span class="w"> </span><span class="n">TxDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">txdb</span><span class="p">,</span><span class="w"> </span><span class="n">nbin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">upstream</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3000</span><span class="p">,</span><span class="w"> </span><span class="n">downstream</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3000</span><span class="p">,</span><span class="w"> 
              </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gene"</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"start_site"</span><span class="p">,</span><span class="w"> </span><span class="n">palette</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Reds"</span><span class="p">)</span><span class="w">
    
  </span><span class="c1"># Individual plot</span><span class="w">
  </span><span class="n">peakHeatmap</span><span class="p">(</span><span class="n">peaks</span><span class="p">[[</span><span class="s2">"PeakName"</span><span class="p">]],</span><span class="w"> </span><span class="n">TxDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">txdb</span><span class="p">,</span><span class="w"> </span><span class="n">nbin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">upstream</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3000</span><span class="p">,</span><span class="w"> </span><span class="n">downstream</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3000</span><span class="p">,</span><span class="w"> 
              </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gene"</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"start_site"</span><span class="p">,</span><span class="w"> </span><span class="n">palette</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Reds"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li><p>In these arguments,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">peaks</code><td>Path to the peak file or a GRanges object containing peak information.<tr><td><code class="language-plaintext highlighter-rouge">TxDb</code><td>Transcript Database (TxDb) object.<tr><td><code class="language-plaintext highlighter-rouge">nbin</code><td>Number of bins to use for drawing the heatmap. Determines the resolution of the heatmap. The lower the <code class="language-plaintext highlighter-rouge">nbin</code> value, the faster the processing, but the lower the image quality.<tr><td><code class="language-plaintext highlighter-rouge">upstream</code>, <code class="language-plaintext highlighter-rouge">downstream</code><td>Distance upstream and downstream of the TSS, respectively.<tr><td><code class="language-plaintext highlighter-rouge">by</code><td>Category to base the peak heatmap on. Possible values are <code class="language-plaintext highlighter-rouge">gene</code>, <code class="language-plaintext highlighter-rouge">transcript</code>, <code class="language-plaintext highlighter-rouge">exon</code>, <code class="language-plaintext highlighter-rouge">intron</code>, <code class="language-plaintext highlighter-rouge">3UTR</code>, <code class="language-plaintext highlighter-rouge">5UTR</code>, and <code class="language-plaintext highlighter-rouge">UTR</code>.<tr><td><code class="language-plaintext highlighter-rouge">type</code><td>Specifies genomic regions to heatmap. Possible values are <code class="language-plaintext highlighter-rouge">start_site</code>(tss), <code class="language-plaintext highlighter-rouge">end_site</code>(tes), <code class="language-plaintext highlighter-rouge">body</code>(gene body).<tr><td><code class="language-plaintext highlighter-rouge">palette</code><td>Color palette for the heatmap.</table></div></ul><li><strong>Output:</strong><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; peakHeatmap(peaks, TxDb = txdb, nbin = 100, upstream = 3000, downstream = 3000, 
              by = "gene", type = "start_site", palette = "Reds")
  &gt;&gt; preparing promoter regions...	 2024-05-21 14:38:23 
  &gt;&gt; preparing tag matrix...		 2024-05-21 14:38:23 
  &gt;&gt; binning method is used...2024-05-21 14:38:23
  &gt;&gt; preparing start_site regions by gene... 2024-05-21 14:38:23
  &gt;&gt; preparing tag matrix by binning...  2024-05-21 14:38:23 
  &gt;&gt; binning method is used...2024-05-21 14:38:40
  &gt;&gt; preparing start_site regions by gene... 2024-05-21 14:38:40
  &gt;&gt; preparing tag matrix by binning...  2024-05-21 14:38:40 
  &gt;&gt; binning method is used...2024-05-21 14:38:47
  &gt;&gt; preparing start_site regions by gene... 2024-05-21 14:38:47
  &gt;&gt; preparing tag matrix by binning...  2024-05-21 14:38:47 
  &gt;&gt; binning method is used...2024-05-21 14:38:51
  &gt;&gt; preparing start_site regions by gene... 2024-05-21 14:38:51
  &gt;&gt; preparing tag matrix by binning...  2024-05-21 14:38:51 
  &gt;&gt; generating figure...		 2024-05-21 14:38:54 
  &gt;&gt; done...			 2024-05-21 14:38:54 
</code></pre></div></div><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_ChIPSeq/Images/peakHeatmap.png" alt="peakHeatmap" style="width: 50vh; height: auto;" /></p><li><p><u>Average Profile of ChIP peaks binding to TSS region</u></p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Multiplot</span><span class="w">
  </span><span class="n">plotAvgProf</span><span class="p">(</span><span class="n">tagMatrixList</span><span class="p">,</span><span class="w"> </span><span class="n">xlim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="w"> </span><span class="m">3000</span><span class="p">),</span><span class="w">
              </span><span class="n">xlab</span><span class="o">=</span><span class="s2">"Genomic Region (5'-&gt;3')"</span><span class="p">,</span><span class="w">
              </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Read Count Frequency"</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Selected plot</span><span class="w">
  </span><span class="n">plotAvgProf</span><span class="p">(</span><span class="n">tagMatrixList</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s2">"H3K4me3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MLL1"</span><span class="p">)],</span><span class="w"> </span><span class="n">xlim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="w"> </span><span class="m">3000</span><span class="p">),</span><span class="w">
              </span><span class="n">xlab</span><span class="o">=</span><span class="s2">"Genomic Region (5'-&gt;3')"</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Read Count Frequency"</span><span class="p">)</span><span class="w">
                
  </span><span class="c1"># Selected plot plus confidence interval estimated by bootstrap method</span><span class="w">
  </span><span class="n">plotAvgProf</span><span class="p">(</span><span class="n">tagMatrixList</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s2">"H3K4me3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"MLL1"</span><span class="p">)],</span><span class="w"> </span><span class="n">xlim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="w"> </span><span class="m">3000</span><span class="p">),</span><span class="w"> 
              </span><span class="n">xlab</span><span class="o">=</span><span class="s2">"Genomic Region (5'-&gt;3')"</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Read Count Frequency"</span><span class="p">,</span><span class="w">
              </span><span class="n">conf</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.95</span><span class="p">,</span><span class="w"> </span><span class="n">resample</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">)</span><span class="w">

  </span><span class="c1"># facet</span><span class="w">
  </span><span class="n">plotAvgProf</span><span class="p">(</span><span class="n">tagMatrixList</span><span class="p">,</span><span class="w"> </span><span class="n">xlim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="w"> </span><span class="m">3000</span><span class="p">),</span><span class="w"> 
              </span><span class="n">xlab</span><span class="o">=</span><span class="s2">"Genomic Region (5'-&gt;3')"</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Read Count Frequency"</span><span class="p">,</span><span class="w">
              </span><span class="n">conf</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.95</span><span class="p">,</span><span class="w"> </span><span class="n">resample</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">500</span><span class="p">,</span><span class="w"> </span><span class="n">facet</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li><strong>Output:</strong><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; plotAvgProf(tagMatrixList[c("H3K4me3", "MLL1")], xlim=c(-3000, 3000), xlab="Genomic Region (5'-&gt;3')", ylab = "Read Count Frequency",conf = 0.95, resample = 1000)
  &gt;&gt; plotting figure...			 2024-05-21 16:15:42 
  &gt;&gt; Running bootstrapping for tag matrix...		 2024-05-21 16:17:15 
  &gt;&gt; Running bootstrapping for tag matrix...		 2024-05-21 16:17:34 
</code></pre></div></div><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_ChIPSeq/Images/AveragePlot.png" alt="AveragePlot" style="width: 50vh; height: auto;" /></p></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h4 class="fs-5 fw-300 lh-tight" id="adding-your-own-transcript-databasetxdb"> <a href="#adding-your-own-transcript-databasetxdb" class="anchor-heading" aria-labelledby="adding-your-own-transcript-databasetxdb"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Adding your own Transcript Database(Txdb)</u></h4><ul><li><p>Species that are less commonly studied often do not have associated database packages in R. In such cases, it is possible to extract the Txdb simply using <code class="language-plaintext highlighter-rouge">makeTxDbFromGFF</code> function from the GTF file. In this context, I will use the rice genome as an example to demonstrate the extraction method.</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">gtf_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"/Users/jchoi/Desktop/IRGSP.gtf"</span><span class="w">
  </span><span class="n">txdb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">makeTxDbFromGFF</span><span class="p">(</span><span class="n">gtf_file</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><h3 id="5-peak-annotation"> <a href="#5-peak-annotation" class="anchor-heading" aria-labelledby="5-peak-annotation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 5. Peak Annotation</h3><ul><li>The <code class="language-plaintext highlighter-rouge">annotatePeak</code> function in the ChIPseeker package is used to annotate peaks in ChIP-seq data analysis. The function identifies and annotates where peaks map to functional elements of the genome, such as <strong>promoters, UTRs (5 and 3), exons, introns, intergenic, and downstream regions</strong>.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">peakAnnoList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">lapply</span><span class="p">(</span><span class="n">peaks</span><span class="p">,</span><span class="w"> </span><span class="n">annotatePeak</span><span class="p">,</span><span class="w"> </span><span class="n">TxDb</span><span class="o">=</span><span class="n">txdb</span><span class="p">,</span><span class="w"> </span><span class="n">tssRegion</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-3000</span><span class="p">,</span><span class="w"> </span><span class="m">3000</span><span class="p">),</span><span class="w">
                  </span><span class="n">addFlankGeneInfo</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">flankDistance</span><span class="o">=</span><span class="m">5000</span><span class="p">,</span><span class="w"> </span><span class="n">annoDb</span><span class="o">=</span><span class="s2">"org.Hs.eg.db"</span><span class="p">)</span><span class="w">
  </span><span class="n">peakAnnoList</span><span class="w"> </span><span class="c1"># Check results</span><span class="w">
</span></code></pre></div></div><ul><li><p>In these arguments,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">peaks</code><td>The list of peak files provided as the first argument to the <code class="language-plaintext highlighter-rouge">lapply</code> function.<tr><td><code class="language-plaintext highlighter-rouge">TxDb</code><td>Transcript Database (TxDb) object.<tr><td><code class="language-plaintext highlighter-rouge">tssRegion</code><td>The distance range from the TSS. The default is <code class="language-plaintext highlighter-rouge">c(-3000, 3000)</code>, which means peaks will be annotated within 3000 base pairs upstream and downstream of the TSS.<tr><td><code class="language-plaintext highlighter-rouge">addFlankGeneInfo</code><td>Specifies whether to add information about genes flanking the peaks. Setting this to <code class="language-plaintext highlighter-rouge">TRUE</code> will include information about genes located upstream and downstream of the peaks.<tr><td><code class="language-plaintext highlighter-rouge">flankDistance</code><td>Specifies the distance upstream and downstream to include when adding flanking gene information. The default is 5000 base pairs.<tr><td><code class="language-plaintext highlighter-rouge">annoDb</code><td>Specifies annotation database.<tr><td><code class="language-plaintext highlighter-rouge">palette</code><td>Color palette for the heatmap.</table></div></ul></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><ul><li><p><strong>Output:</strong></p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; peakAnnoList &lt;- lapply(peaks, annotatePeak, TxDb=txdb, tssRegion=c(-3000, 3000),
  +                        addFlankGeneInfo=TRUE, flankDistance=5000, annoDb="org.Hs.eg.db")
      &gt;&gt; preparing features information...		 2024-05-21 16:51:05 
      &gt;&gt; identifying nearest features...		 2024-05-21 16:51:06 
      &gt;&gt; calculating distance from peak to TSS...	 2024-05-21 16:51:06 
      &gt;&gt; assigning genomic annotation...		 2024-05-21 16:51:06 
      &gt;&gt; adding gene annotation...			 2024-05-21 16:51:17 
      'select()' returned 1:many mapping between keys and columns
      &gt;&gt; adding flank feature information from peaks...	 2024-05-21 16:51:18 
      &gt;&gt; assigning chromosome lengths			 2024-05-21 16:51:26 
      &gt;&gt; done...					 2024-05-21 16:51:26 

  &gt; peakAnnoList
  $H3K4me3
  Annotated peaks generated by ChIPseeker
  59582/59583  peaks were annotated
  Genomic Annotation Summary:
              Feature   Frequency
  9    Promoter (&lt;=1kb) 26.79332684
  10   Promoter (1-2kb)  3.96092780
  11   Promoter (2-3kb)  3.49434393
  4              5' UTR  0.47665402
  3              3' UTR  1.22855896
  1            1st Exon  0.64113323
  7          Other Exon  1.96032359
  2          1st Intron  8.64355007
  8        Other Intron 14.83501729
  6  Downstream (&lt;=300)  0.07384781
  5   Distal Intergenic 37.89231647

  ====================(skip)========================
    
  $UTX
  Annotated peaks generated by ChIPseeker
  44844/44845  peaks were annotated
  Genomic Annotation Summary:
              Feature  Frequency
  9    Promoter (&lt;=1kb)  3.1375435
  10   Promoter (1-2kb)  3.3850682
  11   Promoter (2-3kb)  3.1643029
  4              5' UTR  0.4147712
  3              3' UTR  1.2576933
  1            1st Exon  0.3255731
  7          Other Exon  3.0171260
  2          1st Intron 10.2600125
  8        Other Intron 27.2277228
  6  Downstream (&lt;=300)  0.1204174
  5   Distal Intergenic 47.6897690
</code></pre></div></div><li><p>Save annotated peaks as the csv format as follow:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"H3K4me3"</span><span class="p">]],</span><span class="w"> </span><span class="s2">"Peaked_H3K4me3.csv"</span><span class="p">)</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"H3K27me3"</span><span class="p">]],</span><span class="w"> </span><span class="s2">"Peaked_H3K27me3.csv"</span><span class="p">)</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"MLL1"</span><span class="p">]],</span><span class="w"> </span><span class="s2">"Peaked_MLL1.csv"</span><span class="p">)</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"UTX"</span><span class="p">]],</span><span class="w"> </span><span class="s2">"Peaked_UTX.csv"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><h3 id="6-visulize-genomic-annotation"> <a href="#6-visulize-genomic-annotation" class="anchor-heading" aria-labelledby="6-visulize-genomic-annotation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 6. Visulize genomic annotation</h3><ul><li>Finally, I will discuss and visualize where the annotated peaks are located in the genomic regions, including <strong>promoters, UTRs (5 and 3), exons, introns, intergenic, and downstream regions</strong>.<li><strong>Pie plot</strong>: This funtion does not provide drawing multiplot.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">plotAnnoPie</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"PeakName"</span><span class="p">]])</span><span class="w"> </span><span class="c1"># PeakName is an assigned peak name using the list function</span><span class="w">

  </span><span class="c1"># For example</span><span class="w">
  </span><span class="n">plotAnnoPie</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"H3K4me3"</span><span class="p">]])</span><span class="w"> </span><span class="c1"># Ouput</span><span class="w">
</span></code></pre></div></div><p><strong>Output:</strong></p><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_ChIPSeq/Images/Pieplot.png" alt="Pieplot" style="width: 40vh; height: auto;" /></p><li><strong>Bar plot</strong><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">plotAnnoBar</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">)</span><span class="w"> </span><span class="c1"># Multiplot, output</span><span class="w">
  </span><span class="n">plotAnnoBar</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"PeakName"</span><span class="p">]])</span><span class="w"> </span><span class="c1"># Indivisual plot</span><span class="w">
</span></code></pre></div></div><p><strong>Output:</strong></p><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_ChIPSeq/Images/Barplot.png" alt="Barplot" style="width: 40vh; height: auto;" /></p><li><strong>Vennpie plot</strong>: This funtion does not provide drawing multiplot.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">vennpie</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"PeakName"</span><span class="p">]])</span><span class="w">
</span></code></pre></div></div><li><strong>UpSet plot</strong>: This funtion does not provide drawing multiplot.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">upsetplot</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"PeakName"</span><span class="p">]],</span><span class="w"> </span><span class="n">vennpie</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1"># TRUE/FALSE: with/without vennpie, output</span><span class="w">
</span></code></pre></div></div><p><strong>Output:</strong></p><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_ChIPSeq/Images/UpsetWithVenn.png" alt="UpsetWithVenn" style="width: 40vh; height: auto;" /></p><li><strong>Distribution of TF-binding loci relative to TSS</strong>: The distance from the peak (binding site) to the TSS of the nearest gene is calculated by annotatePeak and reported in the output.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">plotDistToTSS</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">)</span><span class="w"> </span><span class="c1"># output</span><span class="w">
  </span><span class="n">plotDistToTSS</span><span class="p">(</span><span class="n">peakAnnoList</span><span class="p">[[</span><span class="s2">"PeakName"</span><span class="p">]])</span><span class="w">
</span></code></pre></div></div><p><strong>Output:</strong></p><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_ChIPSeq/Images/Dist2TSS.png" alt="Dist2TSS" style="width: 40vh; height: auto;" /></p></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><p><br /></p><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 10px; padding-bottom: 5px; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 1px; position: relative; left: -10px;">Comments</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span>The functions <code>bdgdiff</code> in MACS3 and <code>sicer-df</code> in SICER2 produce results using the same annotation method (<code>annotatePeak</code>).</span><li style="margin-bottom: 10px;"> <span>Although HOMER's <code>annotatePeaks.pl</code> can be used for annotation, it is rarely employed currently.</span><li style="margin-bottom: 10px;"> <span>Peak distribution analysis, including heatmaps and average plots, can be effectively performed with both <code>deeptools</code> and <code>ngsplot</code>. I recommend <code>ngsplot</code> for optimal results.</span><li style="margin-bottom: 10px;"> <span>Peak feature analysis (<b>Section 6, visualization</b>) is seldom used unless investigating a novel transcription factor via ChIP-seq.</span><li style="margin-bottom: 10px;"> <span><b>Despite detailed peak analysis</b>, the current trend focuses on identifying meaningful genes from the detected peaks.</span></ol></div><h1 class="fs-7 fw-400 lh-tight" id="citation"> <a href="#citation" class="anchor-heading" aria-labelledby="citation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citation</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">ngsplot</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Yu, G.</b>, Wang, L. G., &amp; He, Q. Y. (2015). ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization. <i>Bioinformatics</i>, 31(14), 2382-2383. <a href="https://doi.org/10.1093/bioinformatics/btv145" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
