<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(3)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(3) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(5) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(5) > ul > li:nth-child(3) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(5) > ul.nav-list > li.nav-list-item:nth-child(3) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>Peak Detection - MACS | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="Peak Detection - MACS" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/ChIPSeq/MACS_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/ChIPSeq/MACS_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Peak Detection - MACS" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"Peak Detection - MACS","url":"http://localhost:4000/docs/ChIPSeq/MACS_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/ChIPSeq">Data Analysis - ChIP-seq</a><li class="breadcrumb-nav-list-item"><span>Peak Detection - MACS</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="model-based-analysis-for-chip-seq-macs"> <a href="#model-based-analysis-for-chip-seq-macs" class="anchor-heading" aria-labelledby="model-based-analysis-for-chip-seq-macs"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Model-based Analysis for ChIP-Seq (MACS)</h1><p class="fs-4 fw-500">MACS3 is a computational tool used for the analysis of ChIP-Seq data. This tool is designed to identify transcription factor binding sites and regions of chromatin modifications across the genome with high precision and accuracy. The main features of MACS3 include the ability to handle different sequencing depths and conditions without user-defined parameters, an enhanced model-building strategy to better capture the nuances of peak distributions (<code class="language-plaintext highlighter-rouge">callpeak</code>). MACS3 also supports differential binding analysis (<code class="language-plaintext highlighter-rouge">bdgdiff</code>), facilitating the comparison between different samples to identify changes in protein-DNA interaction patterns. This protocol was created based on <span style="color: #7253ec;"><b>MACS3 version 3.0.0a6</b></span> running on a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>Python version 3.8.5, SciPy version 1.6.2, NumPy version 1.20.1, and Cython version 0.29.30 under macOS 12.4 environment.</b></span></p><h1 class="fs-7 fw-400 lh-tight" id="installation-macs3"> <a href="#installation-macs3" class="anchor-heading" aria-labelledby="installation-macs3"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Installation MACS3</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <a href="https://github.com/macs3-project/MACS"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="https://github.com/macs3-project/MACS/releases"><img src="https://img.shields.io/github/v/release/macs3-project/MACS?labelColor=black" alt="Github Release" /></a> <img src="https://img.shields.io/pypi/status/macs3.svg?labelColor=black" alt="Status" /></p><p>To install MACS3 via Pypi, use the following commands:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>pip3 <span class="nb">install </span>macs3
</code></pre></div></div><h2 class="fs-6 fw-400 lh-tight" id="requirements-for-operating-macs3-version-301-lastly-released"> <a href="#requirements-for-operating-macs3-version-301-lastly-released" class="anchor-heading" aria-labelledby="requirements-for-operating-macs3-version-301-lastly-released"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Requirements for operating MACS3 version 3.0.1 (lastly released)</u></h2><p>To use <code class="language-plaintext highlighter-rouge">macs3</code> you need:</p><ul><li><code class="language-plaintext highlighter-rouge">Python &gt; 3.0</code> and its libraries <code class="language-plaintext highlighter-rouge">scipy &gt;= 1.12</code>, <code class="language-plaintext highlighter-rouge">numpy &gt;= 1.25</code>, <code class="language-plaintext highlighter-rouge">pytest &gt;= 7.0</code>, and <code class="language-plaintext highlighter-rouge">scikit-learn&gt;=1.3</code><li><code class="language-plaintext highlighter-rouge">Cython ~= 3.0</code> and its wrapper <code class="language-plaintext highlighter-rouge">cykhash &gt;= 2.0, &lt; 3.0</code></ul><blockquote class="note-title"><p>Disclaimer</p><p>The requirements for running the latest version have changed significantly from the original specifications, incorporating numerous additional packages and updates to existing packages to newer versions. I have not tested MACS3 with this version, so I cannot ensure complete compatibility or functionality. Consequently, I advise beginners to manually install version 3.0.0a6 (alpha 6 released) from the source code.</p></blockquote><h1 class="fs-7 fw-400 lh-tight" id="running-macs3"> <a href="#running-macs3" class="anchor-heading" aria-labelledby="running-macs3"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running MACS3</h1><p>Use the following command to perform peak-call analysis with <code class="language-plaintext highlighter-rouge">macs3</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>macs3 callpeak <span class="nt">-t</span> &lt;TargetFile&gt; <span class="nt">-c</span> &lt;ControlFile&gt; <span class="nt">-f</span> &lt;<span class="nb">type</span><span class="o">&gt;</span> <span class="nt">-g</span> &lt;genomeSize&gt; <span class="nt">-n</span> &lt;NAME&gt; <span class="se">\</span>
	<span class="nt">-B</span> <span class="nt">--nomodel</span> <span class="nt">--extsize</span> &lt;int&gt; <span class="nt">--shift</span> &lt;int&gt; <span class="nt">-q</span> &lt;int&gt; <span class="nt">--broad</span> <span class="nt">--broad-cutoff</span> &lt;int&gt; 
</code></pre></div></div><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">-t &lt;TargetFile&gt;</code><td>Specifies the target data.<tr><td><code class="language-plaintext highlighter-rouge">-c &lt;ControlFile&gt;</code><td>Specifies the control data (input).<tr><td><code class="language-plaintext highlighter-rouge">-f &lt;type&gt;</code><td>Specifies the file type. Possible data type is BAM (recommended), BED, ELANDMULTI, BAMPE, or BEDPE.<tr><td><code class="language-plaintext highlighter-rouge">-g &lt;genomeSize&gt;</code><td>It is the mappable genome size or effective genome size which is defined as the genome size which can be sequenced. <br />Possible values for size are <code class="language-plaintext highlighter-rouge">hs</code> (2913022398/GRCh38), <code class="language-plaintext highlighter-rouge">mm</code> (2652783500/GRCm38), <code class="language-plaintext highlighter-rouge">dm</code>(142573017/dm6), or <code class="language-plaintext highlighter-rouge">&lt;int&gt;</code>. Genome sizes for other species are available from <code class="language-plaintext highlighter-rouge">deeptools</code>.<tr><td><code class="language-plaintext highlighter-rouge">-n &lt;NAME&gt;</code><td>Determine the prefix to apply to the output files.<tr><td><code class="language-plaintext highlighter-rouge">-B</code> option<td>Specifies that store the fragment pileup, control lambda in bedGraph files.<tr><td><code class="language-plaintext highlighter-rouge">--nomodel</code> option<td>While on, MACS3 will bypass building the shifting model. Please combine the usage of <code class="language-plaintext highlighter-rouge">--extsize</code> and <code class="language-plaintext highlighter-rouge">--shift</code> to achieve the effect you expect.<tr><td><code class="language-plaintext highlighter-rouge">--extsize &lt;int&gt;</code><td>While this option is set, MACS3 uses this parameter to extend reads in forward direction to fix-sized fragments.<tr><td><code class="language-plaintext highlighter-rouge">--shift &lt;int&gt;</code><td>Specifies a starting point for reading in reads. <strong>Start reading at a position <code class="language-plaintext highlighter-rouge">&lt;int&gt;</code> away from the 5’ read. The size of the read is determined by <code class="language-plaintext highlighter-rouge">&lt;--extsize&gt;</code>.</strong> When this value is negative, the cutting ends (5’) will be moved toward 3’-&gt;5’ direction, otherwise 5’-&gt;3’ direction.<tr><td><code class="language-plaintext highlighter-rouge">-q &lt;int&gt;</code><td>Specifies an FDR cutoff to call significant regions (Default: 0.05).<tr><td><code class="language-plaintext highlighter-rouge">--broad</code> option<td>This option, along with the bdgbroadcall command, facilitates broad peak calling, producing results in the UCSC gappedPeak format which encapsulates a nested structure of peaks.<tr><td><code class="language-plaintext highlighter-rouge">--broad-cutoff &lt;int&gt;</code><td>Specifies cutoff for the broad region. This option is not available unless <code class="language-plaintext highlighter-rouge">--broad</code> is set (Default: 0.1).</table></div><blockquote class="important-title"><p>Critical Note</p><ol><li><p>It is important to understand that <code class="language-plaintext highlighter-rouge">-g</code>option means the size of the genome, not the species.</p><li><p>The <code class="language-plaintext highlighter-rouge">--extsize</code> option is particularly useful when the binding size of a specific transcription factor is known. For example, if the size of the binding region for target TF is 200 bp, this parameter can be set as 200.</p><li><p>The <code class="language-plaintext highlighter-rouge">--extsize</code> and <code class="language-plaintext highlighter-rouge">--shift</code>options are a very useful for DNase and ATAC-seq. <strong>To find enriched cutting sites in DNAse-seq</strong>, all 5’ ends of sequenced reads should be extended in both directions (approx. 200 bp) to smooth the pileup signals as following commands: <code class="language-plaintext highlighter-rouge">--nomodel --shift -100 --extsize 200</code> (shift = -1 * [a half of extsize]).</p><li><p><strong>For nucleosome-seq data</strong>, need to pile up the centers of nucleosomes using a half-nucleosome size for NPS algorithm as the following command (DNA wrapped on nucleosome is about 147 bp): <code class="language-plaintext highlighter-rouge">--nomodel --shift 37 --extsize 73</code>.</p></ol></blockquote><h2 class="fs-5 fw-400 lh-tight" id="example-code"> <a href="#example-code" class="anchor-heading" aria-labelledby="example-code"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Example Code</u></h2><p>Here is an example command to find significant peaks for histone ChIP-seq and TF ChIP-seq:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Histone ChIP-seq</span>
<span class="nv">$ </span>macs3 callpeak <span class="nt">-t</span> Sort_shControl.bam <span class="nt">-c</span> Sort_Input.bam <span class="nt">-f</span> BAM <span class="nt">-g</span> mm <span class="se">\</span>
<span class="nt">-n</span> /home/jh/Desktop/MACS/shControl <span class="nt">--broad</span> <span class="nt">--broad-cutoff</span> 0.1

<span class="c"># TF ChIP-seq, with basal parameters</span>
<span class="nv">$ </span>macs3 callpeak <span class="nt">-t</span> Sort_shNFIB_MLL1.bam <span class="nt">-c</span> Sort_Input.bam <span class="nt">-f</span> BAM <span class="nt">-g</span> mm <span class="se">\</span>
<span class="nt">-n</span> /home/jh/Desktop/MACS/shNFIB_MLL1 <span class="nt">-B</span> <span class="nt">-q</span> 0.05

<span class="c"># TF ChIP-seq, with advanced parameters</span>
<span class="nv">$ </span>macs3 callpeak <span class="nt">-t</span> Sort_Unt_Control.bam <span class="nt">-c</span> Sort_Input_Control.bam <span class="nt">-f</span> BAM <span class="nt">-g</span> mm <span class="se">\</span>
<span class="nt">-n</span> /home/jh/Desktop/MACS/HaCaT_Unt_MLL1 <span class="nt">-B</span> <span class="nt">--nomodel</span> <span class="nt">--extsize</span> 150 <span class="nt">-q</span> 0.05
</code></pre></div></div><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h2 class="fs-5 fw-400 lh-tight" id="output"> <a href="#output" class="anchor-heading" aria-labelledby="output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Output</u></h2><p>When running <code class="language-plaintext highlighter-rouge">callpeak</code>, it will print out the progress as shown below:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>	INFO  @ Sun, 02 May 2021 11:55:38: 
	# Command line: callpeak -t /home/jh/Desktop/Bam/Sort_shNFIB.bam -c /home/jh/Desktop/Bam/Sort_Input.bam --broad -B -g mm --broad-cutoff 0.1 -n /home/jh/Desktop/MACS/shNFIB
	# ARGUMENTS LIST:
	# name = /home/jh/Desktop/MACS/shNFIB
	# format = AUTO
	# ChIP-seq file = ['/home/jh/Desktop/Bam/Sort_shNFIB.bam']
	# control file = ['/home/jh/Desktop/Bam/Sort_Input.bam']
	# effective genome size = 1.87e+09
	# band width = 300
	# model fold = [5, 50]
	# qvalue cutoff for narrow/strong regions = 5.00e-02
	# qvalue cutoff for broad/weak regions = 1.00e-01
	# The maximum gap between significant sites is assigned as the read length/tag size.
	# The minimum length of peaks is assigned as the predicted fragment length "d".
	# Larger dataset will be scaled towards smaller dataset.
	# Range for calculating regional lambda is: 1000 bps and 10000 bps
	# Broad region calling is on
	# Paired-End mode is off
	
	INFO  @ Sun, 02 May 2021 11:55:38: #1 read tag files... 
	INFO  @ Sun, 02 May 2021 11:55:38: #1 read treatment tags... 
	INFO  @ Sun, 02 May 2021 11:55:38: Detected format is: BAM 
	INFO  @ Sun, 02 May 2021 11:55:38: * Input file is gzipped. 
	INFO  @ Sun, 02 May 2021 11:55:40:  1000000 reads parsed 
	INFO  @ Sun, 02 May 2021 11:56:09:  13000000 reads parsed 
	INFO  @ Sun, 02 May 2021 11:56:12: 13369442 reads have been read. 
	INFO  @ Sun, 02 May 2021 11:56:12: #1.2 read input tags... 
	INFO  @ Sun, 02 May 2021 11:56:12: Detected format is: BAM 
	INFO  @ Sun, 02 May 2021 11:56:12: * Input file is gzipped. 
	INFO  @ Sun, 02 May 2021 11:56:16:  1000000 reads parsed 
	INFO  @ Sun, 02 May 2021 11:57:03:  13000000 reads parsed 
	INFO  @ Sun, 02 May 2021 11:57:07: 13700548 reads have been read. 
	INFO  @ Sun, 02 May 2021 11:57:07: #1 tag size is determined as 101 bps 
	INFO  @ Sun, 02 May 2021 11:57:07: #1 tag size = 101.0 
	INFO  @ Sun, 02 May 2021 11:57:07: #1  total tags in treatment: 13369442 
	INFO  @ Sun, 02 May 2021 11:57:07: #1 user defined the maximum tags... 
	INFO  @ Sun, 02 May 2021 11:57:07: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) 
	INFO  @ Sun, 02 May 2021 11:57:08: #1  tags after filtering in treatment: 12297927 
	INFO  @ Sun, 02 May 2021 11:57:08: #1  Redundant rate of treatment: 0.08 
	INFO  @ Sun, 02 May 2021 11:57:08: #1  total tags in control: 13700548 
	INFO  @ Sun, 02 May 2021 11:57:08: #1 user defined the maximum tags... 
	INFO  @ Sun, 02 May 2021 11:57:08: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) 
	INFO  @ Sun, 02 May 2021 11:57:08: #1  tags after filtering in control: 11744178 
	INFO  @ Sun, 02 May 2021 11:57:08: #1  Redundant rate of control: 0.14 
	INFO  @ Sun, 02 May 2021 11:57:08: #1 finished! 
	INFO  @ Sun, 02 May 2021 11:57:08: #2 Build Peak Model... 
	INFO  @ Sun, 02 May 2021 11:57:08: #2 looking for paired plus/minus strand peaks... 
	INFO  @ Sun, 02 May 2021 11:57:12: #2 Total number of paired peaks: 10769 
	INFO  @ Sun, 02 May 2021 11:57:12: #2 Model building with cross-correlation: Done 
	INFO  @ Sun, 02 May 2021 11:57:12: #2 finished! 
	INFO  @ Sun, 02 May 2021 11:57:12: #2 predicted fragment length is 207 bps 
	INFO  @ Sun, 02 May 2021 11:57:12: #2 alternative fragment length(s) may be 207 bps 
	INFO  @ Sun, 02 May 2021 11:57:12: #2.2 Generate R script for model : /home/jh/Desktop/MACS/shNFIB_model.r 
	INFO  @ Sun, 02 May 2021 11:57:12: #3 Call peaks... 
	INFO  @ Sun, 02 May 2021 11:57:12: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... 
	INFO  @ Sun, 02 May 2021 11:57:12: #3 Pre-compute pvalue-qvalue table... 
	INFO  @ Sun, 02 May 2021 11:58:19: #3 In the peak calling step, the following will be performed simultaneously: 
	INFO  @ Sun, 02 May 2021 11:58:19: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /home/jh/Desktop/MACS/shNFIB_treat_pileup.bdg 
	INFO  @ Sun, 02 May 2021 11:58:19: #3   Write bedGraph files for control lambda (after scaling if necessary)... /home/jh/Desktop/MACS/shNFIB_control_lambda.bdg 
	INFO  @ Sun, 02 May 2021 11:58:19: #3 Call peaks for each chromosome... 
	INFO  @ Sun, 02 May 2021 11:59:01: #4 Write output xls file... /home/jh/Desktop/MACS/shNFIB_peaks.xls 
	INFO  @ Sun, 02 May 2021 11:59:01: #4 Write broad peak in broadPeak format file... /home/jh/Desktop/MACS/shNFIB_peaks.broadPeak 
	INFO  @ Sun, 02 May 2021 11:59:01: #4 Write broad peak in bed12/gappedPeak format file... /home/jh/Desktop/MACS/shNFIB_peaks.gappedPeak 
	INFO  @ Sun, 02 May 2021 11:59:01: Done! 
</code></pre></div></div><p><span class="label label-purple" style="display: inline-block; margin-bottom: -10px; position: relative; left: -10px;">CRITICAL COMMENTS</span></p><p>Similar to SICER2, MACS3 also makes it important for users to adjust the settings of the algorithm to suit their particular experiment. For specific conditions below, I recommend applying the following parameters to increase efficiency and save the time:</p><ol><li><p>Typically, use the <code class="language-plaintext highlighter-rouge">--broad</code> option for detecting broad peaks associated with histone activity markers. For identifying more variable peaks, such as histone repression markers and TFs, it is advisable not to use this option.</p><li><p>For a more detailed analysis, use <code class="language-plaintext highlighter-rouge">--nomodel</code> mode, where the crucial factor is the estimated read size (<code class="language-plaintext highlighter-rouge">--extsize</code>). You can determine this using the <code class="language-plaintext highlighter-rouge">predictd</code> subcommand of MACS3. Below is the method to utilize this subcommand.</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code> <span class="nv">$ </span>macs3 predictd <span class="nt">-i</span> &lt;InputFile.bam&gt; <span class="nt">-g</span> &lt;genomeSize&gt; <span class="nt">--rfile</span> &lt;output.R&gt;
</code></pre></div></div><p>In these commands,</p><ul><li><code class="language-plaintext highlighter-rouge">-i &lt;InputFile.bam&gt;</code>: Specifies input file. Possible data type is BAM (recommended), BED, ELANDMULTI, BAMPE, or BEDPE.<li><code class="language-plaintext highlighter-rouge">-g &lt;genomeSize&gt;</code>: Specifies the genome size. This parameter is the same as the <code class="language-plaintext highlighter-rouge">callpeak</code> function.<li><code class="language-plaintext highlighter-rouge">--rfile &lt;output.R&gt;</code>: Specifies output file as peak model plot.</ul><p>Here is an example command to perform <code class="language-plaintext highlighter-rouge">predictd</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code> <span class="nv">$ </span>macs3 predictd <span class="nt">-i</span> /Users/jchoi/Desktop/Final.sort.Unt_MLL1.bam <span class="nt">-g</span> hs <span class="se">\</span>
     <span class="nt">--rfile</span> /Users/jchoi/Desktop/predict_model_MLL1.R
</code></pre></div></div><p>When running <code class="language-plaintext highlighter-rouge">predictd</code>, it will print out the progress as shown below:</p><div style="background-color: #F5F6FA; padding: 5px; border-radius: 5px;"><pre>
 <code>
 INFO  @ Fri, 10 May 2024 14:18:31: # read alignment files...
 INFO  @ Fri, 10 May 2024 14:18:31: # read treatment tags...
 INFO  @ Fri, 10 May 2024 14:18:31: Detected format is: BAM
 INFO  @ Fri, 10 May 2024 14:18:31:  * Input file is gzipped.*
 INFO  @ Fri, 10 May 2024 14:18:34:  1000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:36:  2000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:38:  3000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:40:  4000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:43:  5000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:45:  6000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:47:  7000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:49:  8000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:51:  9000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:54:  10000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:56:  11000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:18:58:  12000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:00:  13000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:02:  14000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:05:  15000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:07:  16000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:09:  17000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:11:  18000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:13:  19000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:16:  20000000 reads parsed
 INFO  @ Fri, 10 May 2024 14:19:20: 20958780 reads have been read.
 INFO  @ Fri, 10 May 2024 14:19:20: tag size is determined as 93 bps
 INFO  @ Fri, 10 May 2024 14:19:20: # tag size = 93
 INFO  @ Fri, 10 May 2024 14:19:20: # total tags in alignment file: 20958780
 INFO  @ Fri, 10 May 2024 14:19:20: # Build Peak Model...
 INFO  @ Fri, 10 May 2024 14:19:20: #2 looking for paired plus/minus strand peaks...
 INFO  @ Fri, 10 May 2024 14:19:24: #2 Total number of paired peaks: 1407
 INFO  @ Fri, 10 May 2024 14:19:25: #2 Model building with cross-correlation: Done
 INFO  @ Fri, 10 May 2024 14:19:25: # finished!
 <span style="background-color: pink;">INFO  @ Fri, 10 May 2024 14:19:25: # predicted fragment length is 195 bps</span>
 INFO  @ Fri, 10 May 2024 14:19:25: # alternative fragment length(s) may be 175,195 bps
 INFO  @ Fri, 10 May 2024 14:19:25: # Generate R script for model : predict_model_MLL1.R
 </code>
 </pre></div><ul><li>You can use the number of base pair printed in <strong>predicted fragment length</strong> on <span style="background-color: pink;"><b>line 34, pink</b></span> as the <code class="language-plaintext highlighter-rouge">--extsize</code> option.<li>The output model file can be output as a PDF image in the terminal with the following command:<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>Rscript &lt;output.R&gt; 
</code></pre></div></div></ul></ol><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="understanding-macs-output"> <a href="#understanding-macs-output" class="anchor-heading" aria-labelledby="understanding-macs-output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Understanding MACS Output</h1><ul><li><p>The result of the <code class="language-plaintext highlighter-rouge">callpeak</code> will be 3 – 6 files in the specified folder (<code class="language-plaintext highlighter-rouge">-n</code> option), according to the specific parameters. These output files are described below:</p><ol><li><p><span style="color: #7253ec;"><b><code>name</code>_peaks.xls</b></span> <i>(e.g. shCtrl_MLL1_peaks.xls)</i>: This file is a tabular file which contains information about called peaks. It is in “chromosome name, start, end, length of peak region, absolute peal summit, pileup height, -log10(pvalue), fold enrichment, -log10(qvalue)” format.</p><li><p><span style="color: #7253ec;"><b><code>name</code>_peaks.narrowPeak</b></span> <i>(e.g. shCtrl_MLL1_peaks.narrowPeak)</i>: This file is BED6+4 format file which contains the peak locations together with peak summit, p-value, and q-value. <b>It is can be loaded directly to the IGV or UCSC brower.</b></p><li><p><span style="color: #7253ec;"><b><code>name</code>_summits.bed</b></span> <i>(e.g. shCtrl_MLL1_summits.bed)</i>: This file is in BED format, which contains the peak summits locations for every peak. The 5th column in this file is the same as what is in the narrowPeak file. If you want to find the motifs at the binding sites, this file is recommended. <b>It is can be loaded directly to the IGV or UCSC brower.</b></p><li><p><span style="color: #7253ec;"><b><code>name</code>_peaks.broadPeak</b></span> <i>(e.g. shCtrl_MLL1_peaks.broadPeak)</i>: This file is in BED6+3 format which is similar to narrowPeak file, except for missing the 10th column for annotating peak summits. It is only generated <u>when the <code>--broad</code> option is on</u>. For more information on each column, refer to MACS3’s github. <b>It is can be loaded directly to the IGV or UCSC brower.</b></p><li><p><span style="color: #7253ec;"><b><code>name</code>_peaks.gappedPeak</b></span> <i>(e.g. shCtrl_MLL1_peaks.gappedPeak)</i>: This file is in BED12+3 format which contains both the broad region and narrow peaks. It is only generated <u>when the <code>--broad</code> option is on</u>. For more information on each column, refer to MACS3’s github. <b>It is can be loaded directly to the IGV or UCSC brower.</b></p><li><p><span style="color: #7253ec;"><b><code>name</code>_treat_pileup.bdg and <code>name</code>_control_lambda.bdg</b></span>: The <code>name</code>_treat_pileup.bdg contains the pileup signals from <b>ChIP/treatment samples</b>. The <code>name</code>_control_lambda.bdg contains local biases estimated for each genomic location from <b>the control/input samples</b>, or from treatment sample when the control sample is absent. <u>when the <code>-B</code> option is on.</u></p></ol></ul><h1 class="fs-7 fw-400 lh-tight" id="running-bdgdiff-for-differential-peak-calling"> <a href="#running-bdgdiff-for-differential-peak-calling" class="anchor-heading" aria-labelledby="running-bdgdiff-for-differential-peak-calling"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running <code class="language-plaintext highlighter-rouge">bdgdiff</code> for Differential Peak Calling</h1><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script id="MathJax-script" async="" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><p>The <code>bdgdiff</code> command takes <b>four</b> input bedGraph files (two targets and two control files) and produces three output files with differential peaks called. Users should provide paired four bedGraph files: <span style="color: #7253ec;"><b>for each condition, a treatment pileup signal track in bedGraph format, and a control lambda track in bedGraph format</b></span>.<br />This differential calling can only handle one replicate per condition. The method I use to define the differential peaks is based on multiple likelihood tests, based on the poisson distribution. The likelihood function I used while comparing two conditions: ChIP (enrichment) or control (chromatin bias) is:</p><p>\[ ln(LR) = x*(ln(x)-ln(y)) + y - x \]</p><p>Here \(LR\) is the likelihood ratio, \(x\) is the signal (fragment pileup) observed in condition 1, and \(y\) is the signal in condition 2. And \(ln\) is the natural logarithm.</p><p>Use the following command to perform differential peak calling analysis with <code class="language-plaintext highlighter-rouge">bdgdiff</code>:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>macs3 bdgdiff <span class="nt">--t1</span> &lt;cond1_treat_pileup.bdg&gt; <span class="nt">--c1</span> &lt;cond1_control_lambda.bdg&gt; <span class="se">\</span>
	<span class="nt">--t1</span> &lt;cond2_treat_pileup.bdg&gt; <span class="nt">--c1</span> &lt;cond2_control_lambda.bdg&gt; <span class="se">\</span>
	<span class="nt">-C</span> &lt;int&gt; <span class="nt">-l</span> &lt;int&gt; <span class="nt">-g</span> &lt;int&gt; <span class="nt">--d1</span> &lt;cond1_depth&gt; <span class="nt">--d2</span> &lt;cond2_depth&gt; <span class="se">\</span>
	<span class="nt">--outdir</span> &lt;output_path&gt; <span class="nt">--o-prefix</span> &lt;name&gt;
</code></pre></div></div><p>In these commands,</p><div class="table-wrapper"><table><thead><tr><th>Parameter<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">--t1 &lt;cond1_treat_pileup.bdg&gt;</code><td>Specifies the MACS pileup bdg file for condition 1.<tr><td><code class="language-plaintext highlighter-rouge">--c1 &lt;cond1_control_lambda.bdg&gt;</code><td>Specifies the MACS control lambda bdg file for condition 1.<tr><td><code class="language-plaintext highlighter-rouge">--t2 &lt;cond2_treat_pileup.bdg&gt;</code><td>Specifies the MACS pileup bdg file for condition 2.<tr><td><code class="language-plaintext highlighter-rouge">--c2 &lt;cond2_control_lambda.bdg&gt;</code><td>Specifies the MACS control lambda bdg file for condition 2.<tr><td><code class="language-plaintext highlighter-rouge">-C &lt;int&gt;</code><td>Determine log10LR cutoff. Regions with signals lower than the cutoff will not be considered as enriched regions (DEFAULT: 3, likelihood ratio = 1000).<tr><td><code class="language-plaintext highlighter-rouge">-l &lt;int&gt;</code><td>Specifies minimum length of the differential region. Try a bigger value to remove small regions (DEFAULT: 200).<tr><td><code class="language-plaintext highlighter-rouge">-g &lt;int&gt;</code><td>Specifies maximum gap to merge nearby differential regions. Consider a wider gap for broad marks (DEFAULT: 100). The maximum gap should be smaller than the minimum length (<code class="language-plaintext highlighter-rouge">-l</code>).<tr><td><code class="language-plaintext highlighter-rouge">--d1 &lt;cond1_depth&gt;</code><td>Specifies sequencing depth for condition 1.<tr><td><code class="language-plaintext highlighter-rouge">--d2 &lt;cond1_depth&gt;</code><td>Specifies sequencing depth for condition 2.<tr><td><code class="language-plaintext highlighter-rouge">--outdir &lt;output_path&gt;</code><td>Specifies output folder. All output files will be written to that directory.<tr><td><code class="language-plaintext highlighter-rouge">---o-prefix &lt;name&gt;</code><td>Specifies output file prefix. Actual files will be named as PREFIX_cond1.bed, PREFIX_cond2.bed, and PREFIX_common.bed.</table></div><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-manual-example-code-and-output"> <a href="#step-by-step-manual-example-code-and-output" class="anchor-heading" aria-labelledby="step-by-step-manual-example-code-and-output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-step Manual (Example Code and Output)</h1><h3 id="step-1-determine-fix-sized-fragment-using-the-predictd-funtion"> <a href="#step-1-determine-fix-sized-fragment-using-the-predictd-funtion" class="anchor-heading" aria-labelledby="step-1-determine-fix-sized-fragment-using-the-predictd-funtion"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step 1. Determine fix-sized fragment using the <code class="language-plaintext highlighter-rouge">predictd</code> funtion.</h3><ul><li><p>To get a uniform size for running <code class="language-plaintext highlighter-rouge">callpeak</code>, run <code class="language-plaintext highlighter-rouge">predictd</code> as following example:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>macs3 predictd <span class="nt">-i</span> Unt_H3K4me3.bam
  <span class="nv">$ </span>macs3 predictd <span class="nt">-i</span> H2O2_H3K4me3.bam
</code></pre></div></div></ul><p class="note">An easy solution is to use the average of two ‘fragment size’ predicted in callpeak, however any reasonable value will work. For example, you can use <strong>200</strong> for most ChIP-seq datasets for transcription factors, or <strong>147</strong> for most histone modification ChIP-seq.</p><ul><li><p>Anyway, I got the result below with the above commands:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  # For Unt_H3K4me3.bam
      INFO  @ Tue, 14 May 2024 09:36:56: 17993682 reads have been read.
      INFO  @ Tue, 14 May 2024 09:36:56: tag size is determined as 85 bps
      INFO  @ Tue, 14 May 2024 09:36:56: # tag size = 85
      INFO  @ Tue, 14 May 2024 09:36:56: # total tags in alignment file: 17993682
      INFO  @ Tue, 14 May 2024 09:36:56: # Build Peak Model...
      INFO  @ Tue, 14 May 2024 09:36:56: #2 looking for paired plus/minus strand peaks...
      INFO  @ Tue, 14 May 2024 09:37:00: #2 Total number of paired peaks: 14653
      INFO  @ Tue, 14 May 2024 09:37:01: #2 Model building with cross-correlation: Done
      INFO  @ Tue, 14 May 2024 09:37:01: # finished!
      **INFO  @ Tue, 14 May 2024 09:37:01: # predicted fragment length is 219 bps**
      INFO  @ Tue, 14 May 2024 09:37:01: # alternative fragment length(s) may be 219 bps
      INFO  @ Tue, 14 May 2024 09:37:01: # Generate R script for model : predictd

  # For H2O2_H3K4me3.bam
      INFO  @ Tue, 14 May 2024 09:39:00: 18880466 reads have been read.
      INFO  @ Tue, 14 May 2024 09:39:00: tag size is determined as 85 bps
      INFO  @ Tue, 14 May 2024 09:39:00: # tag size = 85
      INFO  @ Tue, 14 May 2024 09:39:00: # total tags in alignment file: 18880466
      INFO  @ Tue, 14 May 2024 09:39:00: # Build Peak Model...
      INFO  @ Tue, 14 May 2024 09:39:00: #2 looking for paired plus/minus strand peaks...
      INFO  @ Tue, 14 May 2024 09:39:05: #2 Total number of paired peaks: 15129
      INFO  @ Tue, 14 May 2024 09:39:05: #2 Model building with cross-correlation: Done
      INFO  @ Tue, 14 May 2024 09:39:05: # finished!
      **INFO  @ Tue, 14 May 2024 09:39:05: # predicted fragment length is 224 bps**
      INFO  @ Tue, 14 May 2024 09:39:05: # alternative fragment length(s) may be 224 bps
      INFO  @ Tue, 14 May 2024 09:39:05: # Generate R script for model : predictd
</code></pre></div></div></ul><h3 id="step-2-prepare-bdg-files-corresponding-to-each-sample-using-the-callpeak-funtion"> <a href="#step-2-prepare-bdg-files-corresponding-to-each-sample-using-the-callpeak-funtion" class="anchor-heading" aria-labelledby="step-2-prepare-bdg-files-corresponding-to-each-sample-using-the-callpeak-funtion"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step 2. Prepare bdg files corresponding to each sample using the <code class="language-plaintext highlighter-rouge">callpeak</code> funtion.</h3><ul><li><p>Using the size of the fragment calculated above for the <code class="language-plaintext highlighter-rouge">--extsize</code> option, perform the <code class="language-plaintext highlighter-rouge">callpeak</code> function as shown below. I will not describe the output (refer to the section of basal MACS3).</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>macs3 macs3 callpeak <span class="nt">-B</span> <span class="nt">-t</span> Unt_H3K4me3.bam <span class="nt">-c</span> IPT.bam <span class="nt">-g</span> hs <span class="nt">-n</span> Unt <span class="nt">--nomodel</span> <span class="nt">--extsize</span> 219
  <span class="nv">$ </span>macs3 callpeak <span class="nt">-B</span> <span class="nt">-t</span> H2O2_H3K4me3.bam <span class="nt">-c</span> IPT.bam <span class="nt">-g</span> hs <span class="nt">-n</span> H2O2 <span class="nt">--nomodel</span> <span class="nt">--extsize</span> 224
</code></pre></div></div></ul><h3 id="step-3-determine-seqeuncing-depth"> <a href="#step-3-determine-seqeuncing-depth" class="anchor-heading" aria-labelledby="step-3-determine-seqeuncing-depth"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step 3. Determine seqeuncing depth.</h3><ul><li>Extract the “tags after filtering in treatment” and “tags after filtering in control” lines from the <code class="language-plaintext highlighter-rouge">&lt;name&gt;</code>_peaks.xls to see the effective sequencing depths for both conditions.<li><p>The effective sequencing depth is the smaller number of treatment and control. For example, use the <code class="language-plaintext highlighter-rouge">egrep</code> function in the terminal as the following commands:</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>egrep <span class="s2">"tags after filtering in treatment|tags after filtering in control"</span> Unt_peaks.xls
  <span class="c"># tags after filtering in treatment: 14060436</span>
  <span class="c"># tags after filtering in control: 16211598</span>

  <span class="nv">$ </span>egrep <span class="s2">"tags after filtering in treatment|tags after filtering in control"</span> H2O2_peaks.xls
  <span class="c"># tags after filtering in treatment: 14486767</span>
  <span class="c"># tags after filtering in control: 16211598</span>
</code></pre></div></div><li>Then actual effective depths of condition 1 (Unt) and 2 (H2O2) are <strong>14060436</strong> and <strong>14486767</strong>, respectively.</ul><h3 id="step-4-running-the-bdgdiff-function"> <a href="#step-4-running-the-bdgdiff-function" class="anchor-heading" aria-labelledby="step-4-running-the-bdgdiff-function"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step 4. Running the <code class="language-plaintext highlighter-rouge">bdgdiff</code> function.</h3><ul><li>The purpose of this step is to do a three ways comparisons to find out where in the genome has differential enrichment between two conditions. A log10 likelihood ratio cutoff (<code class="language-plaintext highlighter-rouge">-C</code> option) will be applied in this step.<li><p>Three types of differential regions will be reported: <u>PREFIX_cond1.bed (cond1 &gt; cond2), PREFIX_cond2.bed (cond1 &lt; cond2), and PREFIX_common.bed (cond1 = cond2)</u>.</p><div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="nv">$ </span>macs3 bdgdiff <span class="nt">--t1</span> Unt_treat_pileup.bdg <span class="nt">--c1</span> Unt_control_lambda.bdg <span class="se">\</span>
      <span class="nt">--t2</span> H2O2_treat_pileup.bdg <span class="nt">--c2</span> H2O2_control_lambda.bdg <span class="se">\</span>
      <span class="nt">--d1</span> 14060436 <span class="nt">--d2</span> 14486767 <span class="nt">-g</span> 60 <span class="nt">-l</span> 120 <span class="nt">--o-prefix</span> diffPeaks
</code></pre></div></div><li><p>When running <code class="language-plaintext highlighter-rouge">bdgdiff</code>, it will print out the progress as shown below:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>      INFO  @ Tue, 14 May 2024 10:03:32: Read and build treatment 1 bedGraph...
      INFO  @ Tue, 14 May 2024 10:03:52: Read and build control 1 bedGraph...
      INFO  @ Tue, 14 May 2024 10:04:18: Read and build treatment 2 bedGraph...
      INFO  @ Tue, 14 May 2024 10:04:38: Read and build control 2 bedGraph...
      INFO  @ Tue, 14 May 2024 10:06:51: Write peaks...
      INFO  @ Tue, 14 May 2024 10:06:51: Done
</code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="citation"> <a href="#citation" class="anchor-heading" aria-labelledby="citation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citation</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">MACS</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Zhang, Y.</b>, Liu, T., Meyer, C. A., Eeckhoute, J., Johnson, D. S., Bernstein, B. E., ... &amp; Liu, X. S. (2008).Model-based analysis of ChIP-Seq (MACS). <i>Genome biology</i>, 9, 1-9. <a href="https://doi.org/10.1186/gb-2008-9-9-r137" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
