<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(5)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(4) > ul > li:nth-child(5) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(4) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(4) > ul > li:nth-child(5) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(4) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(4) > ul.nav-list > li.nav-list-item:nth-child(5) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>DEG - Heatmap | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="DEG - Heatmap" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/RNASeq/HEATMAP_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/RNASeq/HEATMAP_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="DEG - Heatmap" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"DEG - Heatmap","url":"http://localhost:4000/docs/RNASeq/HEATMAP_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/RNASeq">Data Analysis - RNA-seq</a><li class="breadcrumb-nav-list-item"><span>DEG - Heatmap</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="essential-tools-for-heatmap-visualization-in-sequencing-data-pheatmap-and-complexheatmap"> <a href="#essential-tools-for-heatmap-visualization-in-sequencing-data-pheatmap-and-complexheatmap" class="anchor-heading" aria-labelledby="essential-tools-for-heatmap-visualization-in-sequencing-data-pheatmap-and-complexheatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Essential Tools for Heatmap Visualization in Sequencing Data: pheatmap and ComplexHeatmap</h1><p class="fs-4 fw-500">In the analysis of sequencing data, drawing heatmaps is crucial for visualizing complex information in a comprehensible manner, allowing researchers to detect patterns, correlations, and outliers within large datasets. Heatmaps represent matrix data where individual values contained in a matrix are represented as colors, facilitating the quick absorption of insights about the underlying biological processes. Among the tools available for this purpose in R, the pheatmap and ComplexHeatmap packages are particularly noteworthy. The pheatmap package provides a simple and highly customizable interface for generating heatmaps, offering features such as clustering, annotation, and adjustments to color scales. On the other hand, ComplexHeatmap extends these functionalities by enabling the integration of multiple heatmaps, adding complex annotations, and supporting the arrangement of heatmaps and other graphical elements in a highly flexible manner. This protocol was developed using a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>R version 4.4.0 running on macOS 14.4.1.</b></span></p><h2 class="fs-6 fw-400 lh-tight" id="pheatmap-versus-complexheatmap"> <a href="#pheatmap-versus-complexheatmap" class="anchor-heading" aria-labelledby="pheatmap-versus-complexheatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>pheatmap versus ComplexHeatmap</u></h2><p>The packages <code class="language-plaintext highlighter-rouge">pheatmap</code> and <code class="language-plaintext highlighter-rouge">complexheatmap</code> are both apt for the generation of heatmaps, albeit with distinct differences. Refer to the following table, which outlines the respective advantages and disadvantages of each package, and select the corresponding manuals for further guidance. Additionally, unlike other plotting tutorials (Volcano, scatter, correlation, and buble chart), this guide incorporates complex parameters and fundamental logical functions of R, necessitating a foundational understanding of the R programming language. Specifically, the <code class="language-plaintext highlighter-rouge">complexheatmap</code> package is designed for users with an intermediate level of expertise.</p><div class="table-wrapper"><table><thead><tr><th>Aspect<th>pheatmap<th>ComplexHeatmap<tbody><tr><td><strong>Ease of Use</strong><td>More effective for beginners and quick implementations<td>Less straightforward initially, better for experienced users<tr><td><strong>Customization and Flexibility</strong><td>Limited in customization, suitable for basic heatmaps<td>Highly customizable, ideal for detailed visualizations<tr><td><strong>Integration with Other Data Types and Visuals</strong><td>Less effective for complex data integration<td>Highly effective for integrating multiple data types and visuals<tr><td><strong>Performance and Scalability</strong><td>Good for smaller datasets and simpler analyses<td>Superior for large datasets and complex analyses</table></div><h1 class="fs-7 fw-400 lh-tight" id="pheatmap"> <a href="#pheatmap" class="anchor-heading" aria-labelledby="pheatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> pheatmap</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Windows-0078D6?style=flat&amp;logo=windows&amp;logoColor=white" alt="EnvTest3" /> <a href="https://github.com/raivokolde/pheatmap"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="https://cran.r-project.org/web/packages/pheatmap/index.html"><img src="https://img.shields.io/badge/BioC-276DC3?style=flat&amp;logo=r&amp;logoColor=white" alt="BioC" /></a> <a href="https://cran.r-project.org/web/checks/check_results_pheatmap.html"><img src="https://img.shields.io/badge/release%20version-1.0.12-DarkSeaGreen.svg?labelColor=black" alt="CRAN" /></a> <a href="https://github.com/raivokolde/pheatmap"><img src="https://img.shields.io/badge/status-stable-DarkSeaGreen?style=flat&amp;colorA=000000" alt="Project Status" /></a></p><p>In this section, the user is introduced to the techniques for visualizing basic RNA-seq data through heatmaps, using the <code class="language-plaintext highlighter-rouge">pheatmap</code> package. The dataset featured originates from CPM values, which were collected following RNA-seq assays on duplicate samples of both wild-type rice and genetically modified rice with certain genes knockouted. Please be advised that the <code class="language-plaintext highlighter-rouge">pheatmap</code> package supports only fundamental heatmap visualization. For an in-depth examination of advanced features, one is directed to consult the section dedicated to the <code class="language-plaintext highlighter-rouge">complexheatmap</code> package, which follows.</p><p><strong>Required Packages:</strong></p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">pheatmap</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">viridis</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">viridisLite</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-pheatmap-guide"> <a href="#step-by-step-pheatmap-guide" class="anchor-heading" aria-labelledby="step-by-step-pheatmap-guide"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step pheatmap Guide</h1><h3 id="1-load-cpm-data-and-prepare-data-frame"> <a href="#1-load-cpm-data-and-prepare-data-frame" class="anchor-heading" aria-labelledby="1-load-cpm-data-and-prepare-data-frame"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Load CPM Data and Prepare Data Frame</h3><p>Load your data. In this guide, I will load CPM data without using <code class="language-plaintext highlighter-rouge">row.names = 1</code>, <strong>but this is optional</strong>:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"heatmap.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>This dataset is organized as follows:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; head(data)
            MSU  WT_rep1  WT_rep2    KO_rep1    KO_Rep2  Cluster
1 LOC_Os02g57000 6.475709 6.216468 -3.0875734 -3.3857658 cluster1
2 LOC_Os02g33740 5.597263 5.432519 -2.3418637 -3.0058172 cluster1
3 LOC_Os03g59300 6.656864 6.558811  0.3320713  0.3533387 cluster1
4 LOC_Os01g21580 4.679595 4.487096 -2.7940856 -2.7053595 cluster1
5 LOC_Os02g03640 4.724022 4.939755 -2.1597539 -1.8963395 cluster1
6 LOC_Os09g29050 5.663938 5.379374 -0.4579156 -0.2427508 cluster1
</code></pre></div></div><p>Select the values in the data that will be analyzed and specify their row names as gene numbers (<code class="language-plaintext highlighter-rouge">data$MSU</code>):</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">MSU</span><span class="w">
</span><span class="n">expression_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="o">:</span><span class="m">5</span><span class="p">]</span><span class="w">
</span></code></pre></div></div><li>The data frame of <code class="language-plaintext highlighter-rouge">expression_data</code> is organized as follows:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; head(expression_data)
                WT_rep1  WT_rep2    KO_rep1    KO_Rep2
LOC_Os02g57000 6.475709 6.216468 -3.0875734 -3.3857658
LOC_Os02g33740 5.597263 5.432519 -2.3418637 -3.0058172
LOC_Os03g59300 6.656864 6.558811  0.3320713  0.3533387
LOC_Os01g21580 4.679595 4.487096 -2.7940856 -2.7053595
LOC_Os02g03640 4.724022 4.939755 -2.1597539 -1.8963395
LOC_Os09g29050 5.663938 5.379374 -0.4579156 -0.2427508
</code></pre></div></div></ul><h3 id="2-create-annotation-frames-for-rows-and-columns"> <a href="#2-create-annotation-frames-for-rows-and-columns" class="anchor-heading" aria-labelledby="2-create-annotation-frames-for-rows-and-columns"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Create Annotation Frames for Rows and Columns</h3><p>The rows of the dataset are divided by the <code class="language-plaintext highlighter-rouge">Cluster</code> of <code class="language-plaintext highlighter-rouge">data</code> and the columns are divided by the head of the data to create the annotation:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># For row</span><span class="w">
</span><span class="n">annotation</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">Cluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">Cluster</span><span class="p">))</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">annotation</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w">

</span><span class="c1">#For column</span><span class="w">
</span><span class="n">annotation_col</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KO"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KO"</span><span class="p">)))</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">annotation_col</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">expression_data</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>The structure of the <code class="language-plaintext highlighter-rouge">annotation</code> and <code class="language-plaintext highlighter-rouge">annotation_col</code> data frames is shown below:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; head(annotation)
                Cluster
LOC_Os02g57000 cluster1
LOC_Os02g33740 cluster1
LOC_Os03g59300 cluster1
LOC_Os01g21580 cluster1
LOC_Os02g03640 cluster1
LOC_Os09g29050 cluster1

&gt; annotation_col
        Type
WT_rep1   WT
WT_rep2   WT
KO_rep1   KO
KO_Rep2   KO
</code></pre></div></div></ul><h3 id="3-prepare-color-palette-for-the-heatmap"> <a href="#3-prepare-color-palette-for-the-heatmap" class="anchor-heading" aria-labelledby="3-prepare-color-palette-for-the-heatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. Prepare Color Palette for the Heatmap</h3><p>Choose a color scheme for your heatmap visualization. This can be done by specifying either the name or hex codes of colors:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">myCol</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s1">'dodgerblue'</span><span class="p">,</span><span class="w"> </span><span class="s1">'black'</span><span class="p">,</span><span class="w"> </span><span class="s1">'yellow'</span><span class="p">))(</span><span class="m">100</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>The number <code class="language-plaintext highlighter-rouge">100</code> represents the granularity of the heatmap, affecting the differentiation between data points. More colors provide finer distinctions, helpful for detailed data visualization, whereas fewer colors might emphasize broader trends.</ul><p>If you need a broader range of color palettes, the <code class="language-plaintext highlighter-rouge">RColorBrewer</code> package offers a variety of user-favorite choices:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">myCol</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">11</span><span class="p">,</span><span class="w"> </span><span class="s2">"RdYlBu"</span><span class="p">))(</span><span class="m">100</span><span class="p">)</span><span class="w">
</span><span class="n">myCol_reverse</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="n">rev</span><span class="p">(</span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">11</span><span class="p">,</span><span class="w"> </span><span class="s2">"RdYlBu"</span><span class="p">)))(</span><span class="m">100</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>In these codes, <code class="language-plaintext highlighter-rouge">"RdYlBu"</code> indicates a specific color palette, the number <code class="language-plaintext highlighter-rouge">11</code> specifically refers to the number of different colors that you want to extract from the “RdYlBu” palette provided by the RColorBrewer package. The number of <code class="language-plaintext highlighter-rouge">100</code> means the granularity in the heatmap. The <code class="language-plaintext highlighter-rouge">rev</code> argument is used to reverse the color order. The available color codes and numbers are shown below image.</ul><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/rcolorbrw.png" alt="Rcolor" style="width: 50vh; height: auto;" /></p><p>The <code class="language-plaintext highlighter-rouge">viridis</code> package is also a good one that provides a palette for color selection.</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">myCol</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">viridis</span><span class="o">::</span><span class="n">viridis</span><span class="p">(</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">option</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"magma"</span><span class="p">)</span><span class="w">
</span><span class="n">myCol_magma_rev</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">viridis</span><span class="o">::</span><span class="n">viridis</span><span class="p">(</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">option</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"magma"</span><span class="p">,</span><span class="w"> </span><span class="n">direction</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-1</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><p>The following images display the available color palettes from Viridis:</p><div style="text-align: center;"> <img src="https://search.r-project.org/CRAN/refmans/viridisLite/help/figures/viridis-scales.png" width="40%" height="auto" /></div><h3 id="4-plotting"> <a href="#4-plotting" class="anchor-heading" aria-labelledby="4-plotting"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4. Plotting</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">myCol</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">viridis</span><span class="o">::</span><span class="n">viridis</span><span class="p">(</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="n">option</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"viridis"</span><span class="p">)</span><span class="w">
</span><span class="n">pheatmap_result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pheatmap</span><span class="p">(</span><span class="n">expression_data</span><span class="p">,</span><span class="w"> 
         </span><span class="n">annotation_row</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation</span><span class="p">,</span><span class="w">
         </span><span class="n">annotation_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation_col</span><span class="p">,</span><span class="w">
         </span><span class="n">cluster_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
         </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">show_colnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
         </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">,</span><span class="w"> 
         </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">myCol</span><span class="p">,</span><span class="w">
         </span><span class="n">clustering_distance_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"euclidean"</span><span class="p">,</span><span class="w">
         </span><span class="n">clustering_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"average"</span><span class="p">,</span><span class="w">
         </span><span class="n">cellheight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.06</span><span class="p">,</span><span class="w"> </span><span class="n">cellwidth</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><p><strong>Important:</strong> Typically, heatmap data is normalized and presented as a z-score to ensure comparability across data points. In the provided example, normalization is achieved by computing a z-score with the <code class="language-plaintext highlighter-rouge">scale = "row"</code> option directly in the heatmap function, thus eliminating the need for using the <code class="language-plaintext highlighter-rouge">t(scale(t()))</code> function.</p><ul><li>These codes utilize <code class="language-plaintext highlighter-rouge">Euclidean</code> distance for clustering and employs the <code class="language-plaintext highlighter-rouge">average</code> algorithm.<li>To modify the clustering distance, add the <code class="language-plaintext highlighter-rouge">clustering_distance_rows = "&lt;distance&gt;"</code> parameter. <strong>Available <code class="language-plaintext highlighter-rouge">distance</code> metrics</strong> include <code class="language-plaintext highlighter-rouge">euclidean</code>, <code class="language-plaintext highlighter-rouge">maximum</code>, <code class="language-plaintext highlighter-rouge">manhattan</code>, <code class="language-plaintext highlighter-rouge">canberra</code>, and <code class="language-plaintext highlighter-rouge">binary</code>.<li><p>Additionally, to change the clustering algorithm, adjust the <code class="language-plaintext highlighter-rouge">clustering_method = "&lt;algorithm&gt;"</code> parameter. <strong>Possible algorithms</strong> are <code class="language-plaintext highlighter-rouge">single</code> (minimum linkage), <code class="language-plaintext highlighter-rouge">complete</code> (maximum linkage), <code class="language-plaintext highlighter-rouge">average</code> (UPGMA), <code class="language-plaintext highlighter-rouge">ward.D</code> (Ward’s minimum variance method), <code class="language-plaintext highlighter-rouge">ward.D2</code> (Modified Ward’s method), <code class="language-plaintext highlighter-rouge">mcquitty</code> (McQuitty’s method, WPGMA), <code class="language-plaintext highlighter-rouge">median</code> (Median linkage, WPGMC), and <code class="language-plaintext highlighter-rouge">centroid</code> (Centroid linkage, UPGMC).</p><li>If you want to adjust the range of the heatmap, you can do so using the <code class="language-plaintext highlighter-rouge">breaks</code> argument:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">my_breaks</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">seq</span><span class="p">(</span><span class="m">-1</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">length.out</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">myCol</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="c1"># Set the breaks</span><span class="w">

</span><span class="n">pheatmap_result_Rev</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pheatmap</span><span class="p">(</span><span class="n">expression_data</span><span class="p">,</span><span class="w"> 
        </span><span class="n">annotation_row</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation</span><span class="p">,</span><span class="w">
        </span><span class="n">annotation_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation_col</span><span class="p">,</span><span class="w">
        </span><span class="n">cluster_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
        </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">show_colnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
        </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">,</span><span class="w"> 
        </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">myCol</span><span class="p">,</span><span class="w">
        </span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">my_breaks</span><span class="p">,</span><span class="w">
        </span><span class="n">clustering_distance_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"euclidean"</span><span class="p">,</span><span class="w">
        </span><span class="n">clustering_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"average"</span><span class="p">,</span><span class="w">
        </span><span class="n">cellheight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.06</span><span class="p">,</span><span class="w"> </span><span class="n">cellwidth</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>You can also output a list of genes in clustering order:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">clustered_gene_order</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">expression_data</span><span class="p">)[</span><span class="n">pheatmap_result</span><span class="o">$</span><span class="n">tree_row</span><span class="o">$</span><span class="n">order</span><span class="p">]</span><span class="w">
</span><span class="n">clustered_genes_with_clusters</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="w">
  </span><span class="n">Gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">clustered_gene_order</span><span class="p">,</span><span class="w">
  </span><span class="n">OriginalCluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation</span><span class="p">[</span><span class="n">clustered_gene_order</span><span class="p">,</span><span class="w"> </span><span class="s2">"Cluster"</span><span class="p">]</span><span class="w">
</span><span class="p">)</span><span class="w">
</span><span class="c1"># Save the list as a csv format</span><span class="w">
</span><span class="n">write.csv</span><span class="p">(</span><span class="n">clustered_genes_with_clusters</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"clustered_genes_with_clusters.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><h3 id="5-optional-represent-specific-gene-symbols-in-the-heatmap"> <a href="#5-optional-represent-specific-gene-symbols-in-the-heatmap" class="anchor-heading" aria-labelledby="5-optional-represent-specific-gene-symbols-in-the-heatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 5. (Optional) Represent Specific Gene Symbols in the Heatmap</h3><p>Although this functionality is not fully implemented in the current setup, it is recommended to use the <code class="language-plaintext highlighter-rouge">ComplexHeatmap</code> package for a robust implementation of this featur</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Specify the genes whose names you want to be displayed</span><span class="w">
</span><span class="n">selected_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"LOC_Os06g49240"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LOC_Os01g49710"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LOC_Os10g03220"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LOC_Os04g41680"</span><span class="p">)</span><span class="w">
</span><span class="n">labels_row</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ifelse</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">expression_data</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">selected_genes</span><span class="p">,</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">expression_data</span><span class="p">),</span><span class="w"> </span><span class="s2">""</span><span class="p">)</span><span class="w">

</span><span class="c1"># Plotting</span><span class="w">
</span><span class="n">pheatmap_result_Name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pheatmap</span><span class="p">(</span><span class="n">expression_data</span><span class="p">,</span><span class="w"> 
         </span><span class="n">annotation_row</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation</span><span class="p">,</span><span class="w">
         </span><span class="n">annotation_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation_col</span><span class="p">,</span><span class="w">
         </span><span class="n">cluster_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
         </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">show_colnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
         </span><span class="n">labels_row</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">labels_row</span><span class="p">,</span><span class="w">
         </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">,</span><span class="w"> 
         </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">myCol</span><span class="p">,</span><span class="w">
         </span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">my_breaks</span><span class="p">,</span><span class="w">
         </span><span class="n">clustering_distance_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"euclidean"</span><span class="p">,</span><span class="w">
         </span><span class="n">clustering_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"average"</span><span class="p">,</span><span class="w">
         </span><span class="n">cellheight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.06</span><span class="p">,</span><span class="w"> </span><span class="n">cellwidth</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li><strong>Output (Left; Step 4, Right; Step 5):</strong> <i>(Clicking on the image will open it in a larger view.)</i></ul><link rel="stylesheet" href="/assets/css/modal.css" /><p align="center" style="display: flex; justify-content: center;"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Pheat1.png" alt="pheat1" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Pheat1.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Pheat1.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Pheat2.png'])" /> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Pheat2.png" alt="pheat2" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Pheat2.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Pheat1.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Pheat2.png'])" /></p><div id="myModal" class="modal"> <span class="close" onclick="closeModal()">&times;</span> <span class="arrow left" onclick="changeImage(-1)">&#10094;</span> <img class="modal-content" id="img01" /> <span class="arrow right" onclick="changeImage(1)">&#10095;</span><div class="modal-list" id="imgList"></div></div><script src="/assets/js/modal.js"></script><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="compexheatmap"> <a href="#compexheatmap" class="anchor-heading" aria-labelledby="compexheatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> CompexHeatmap</h1><p><img src="https://img.shields.io/badge/macOS-000000?style=flat&amp;logo=apple&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Ubuntu-E95420?style=flat&amp;logo=ubuntu&amp;logoColor=white" alt="EnvTest" /> <img src="https://img.shields.io/badge/Windows-0078D6?style=flat&amp;logo=windows&amp;logoColor=white" alt="EnvTest3" /> <a href="https://github.com/jokergoo/ComplexHeatmap"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a> <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/"><img src="https://img.shields.io/badge/GitHub.io-181717?style=flat&amp;logo=github&amp;logoColor=yellow" alt="githio" /></a> <a href="https://bioconductor.org/packages/release/bioc/html/ComplexHeatmap.html"><img src="https://img.shields.io/badge/BioC-276DC3?style=flat&amp;logo=r&amp;logoColor=white" alt="BioC" /></a> <a href="https://github.com/jokergoo/ComplexHeatmap/releases"><img src="https://img.shields.io/github/v/release/jokergoo/ComplexHeatmap?labelColor=black" alt="GitHub release" /></a> <a href="https://github.com/jokergoo/ComplexHeatmap"><img src="https://img.shields.io/badge/status-stable-DarkSeaGreen?style=flat&amp;colorA=000000" alt="Project Status" /></a></p><p>In this section, the <code class="language-plaintext highlighter-rouge">complexheatmap</code> package will be elucidated through the analysis of three distinct case studies. <strong>Initially</strong>, data from a heatmap generated using the ‘pheatmap’ package will be implemented employing the ‘complexheatmap’ package. <strong>Subsequently</strong>, the integration of RNA-seq and ChIP-seq data will be demonstrated, utilizing the data integration parameters that constitute a principal feature of this package. <strong>Lastly</strong>, the process for integrating various RNA-seq datasets will be discussed.</p><p><strong>Required Packages:</strong></p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ComplexHeatmap</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">circlize</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">grid</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">viridis</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">viridisLite</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>If you do not possess specialized knowledge, the use of the <code class="language-plaintext highlighter-rouge">ht_opt</code> function in the Complexheatmap may not be advisable. It is important to note that this function’s settings depend permanently on the previously saved value, even after re-invoking the library. If a reset is necessary, please utilize the command provided below.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ht_opt</span><span class="p">(</span><span class="n">Reset</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span><span class="n">print</span><span class="p">(</span><span class="n">ht_opt</span><span class="p">)</span><span class="w"> </span><span class="c1"># Check</span><span class="w">
</span></code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-guide-case-i---rna-seq-heatmap"> <a href="#step-by-step-guide-case-i---rna-seq-heatmap" class="anchor-heading" aria-labelledby="step-by-step-guide-case-i---rna-seq-heatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Guide: Case I - RNA-seq Heatmap</h1><p>In this case, I will visualize the CPM values of wild-type rice and rice with a specific gene has been knocked out, through the application of biological replicates. The data were subjected to DEG analysis using <code class="language-plaintext highlighter-rouge">edgeR</code>. Additionally, I have generated a <code class="language-plaintext highlighter-rouge">Cluster</code> row labeled <code class="language-plaintext highlighter-rouge">cluster1</code> for genes that are upregulated and <code class="language-plaintext highlighter-rouge">cluster2</code> for genes that are downregulated.</p><h3 id="1-load-data-and-normalize-with-z-score"> <a href="#1-load-data-and-normalize-with-z-score" class="anchor-heading" aria-labelledby="1-load-data-and-normalize-with-z-score"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Load Data and Normalize with z-score</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Load data without the 'row.names' option</span><span class="w">
</span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"heatmap.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

</span><span class="c1"># Specify the range of the data to be analyzed</span><span class="w">
</span><span class="n">expression_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">data</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="o">:</span><span class="m">5</span><span class="p">])</span><span class="w">

</span><span class="c1"># Normalize to z-score</span><span class="w">
</span><span class="n">z_score</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">scale</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">expression_data</span><span class="p">)))</span><span class="w">
</span><span class="n">z_min</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">min</span><span class="p">(</span><span class="n">z_score</span><span class="p">)</span><span class="w">
</span><span class="n">z_max</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">z_score</span><span class="p">)</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">z_score</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">MSU</span><span class="w">
</span></code></pre></div></div><ul><li><strong>Output</strong>:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; head(data)
            MSU  WT_rep1  WT_rep2    KO_rep1    KO_Rep2  Cluster
1 LOC_Os02g57000 6.475709 6.216468 -3.0875734 -3.3857658 cluster1
2 LOC_Os02g33740 5.597263 5.432519 -2.3418637 -3.0058172 cluster1
3 LOC_Os03g59300 6.656864 6.558811  0.3320713  0.3533387 cluster1
4 LOC_Os01g21580 4.679595 4.487096 -2.7940856 -2.7053595 cluster1
5 LOC_Os02g03640 4.724022 4.939755 -2.1597539 -1.8963395 cluster1
6 LOC_Os09g29050 5.663938 5.379374 -0.4579156 -0.2427508 cluster1

&gt; head(z_score)
                WT_rep1   WT_rep2    KO_rep1    KO_Rep2
LOC_Os02g57000 0.8890761 0.8422390 -0.8387204 -0.8925947
LOC_Os02g33740 0.8819111 0.8471257 -0.7944220 -0.9346147
LOC_Os03g59300 0.8795228 0.8524170 -0.8689095 -0.8630303
LOC_Os01g21580 0.8885737 0.8431155 -0.8763208 -0.8553683
LOC_Os02g03640 0.8382743 0.8927108 -0.8987265 -0.8322585
LOC_Os09g29050 0.9071574 0.8232973 -0.8969315 -0.8335232
</code></pre></div></div></ul><h3 id="2-prepare-cluster-annotations"> <a href="#2-prepare-cluster-annotations" class="anchor-heading" aria-labelledby="2-prepare-cluster-annotations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Prepare Cluster Annotations</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Create a factor for the cluster data and setup the annotation dataframe</span><span class="w">
</span><span class="n">cluster_factor</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">Cluster</span><span class="p">)</span><span class="w">
</span><span class="n">annotation</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">Cluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cluster_factor</span><span class="p">)</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">annotation</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">MSU</span><span class="w">

</span><span class="c1"># Define a color palette</span><span class="w">
</span><span class="n">set2_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="s2">"Set2"</span><span class="p">)</span><span class="w">

</span><span class="c1"># Create row annotations</span><span class="w">
</span><span class="n">row_anno</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rowAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation</span><span class="p">,</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Cluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">setNames</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">set2_palette</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">1</span><span class="p">]),</span><span class="w"> </span><span class="n">levels</span><span class="p">(</span><span class="n">cluster_factor</span><span class="p">))),</span><span class="w">
  </span><span class="n">annotation_name_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w">
  </span><span class="n">simple_anno_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Create column annotations</span><span class="w">
</span><span class="n">col_anno</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">HeatmapAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KO"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KO"</span><span class="p">)),</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">WT</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">3</span><span class="p">],</span><span class="w"> </span><span class="n">KO</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">4</span><span class="p">])),</span><span class="w">
  </span><span class="n">show_annotation_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">  
  </span><span class="n">simple_anno_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li><strong>Output</strong>:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; row_anno
A HeatmapAnnotation object with 1 annotation
  name: heatmap_annotation_0 
  position: row 
  items: 5981 
  width: 3mm 
  height: 1npc 
  this object is subsettable
  14.4069666666667mm extension on the top 

    name annotation_type color_mapping width
Cluster discrete vector  user-defined   3mm
&gt; col_anno
A HeatmapAnnotation object with 1 annotation
  name: heatmap_annotation_1 
  position: column 
  items: 4 
  width: 1npc 
  height: 3mm 
  this object is subsettable

name annotation_type color_mapping height
Type discrete vector  user-defined    3mm
</code></pre></div></div></ul><h3 id="3-plotting"> <a href="#3-plotting" class="anchor-heading" aria-labelledby="3-plotting"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. Plotting</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">main_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="w">
  </span><span class="n">z_score</span><span class="p">,</span><span class="w">
  </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Expression Data"</span><span class="p">,</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colorRamp2</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">z_min</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">z_max</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'dodgerblue'</span><span class="p">,</span><span class="w"> </span><span class="s1">'black'</span><span class="p">,</span><span class="w"> </span><span class="s1">'yellow'</span><span class="p">)),</span><span class="w">
  </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
  </span><span class="n">column_title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Expression Data"</span><span class="p">,</span><span class="w">
  </span><span class="n">clustering_method_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"average"</span><span class="p">,</span><span class="w">
  </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">  
  </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w">  
  </span><span class="n">column_split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"WT"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KO"</span><span class="p">,</span><span class="w"> </span><span class="s2">"KO"</span><span class="p">)),</span><span class="w">
  </span><span class="n">top_annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">col_anno</span><span class="p">,</span><span class="w">  
  </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cluster_factor</span><span class="p">,</span><span class="w">
  </span><span class="n">row_title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w">
  </span><span class="n">left_annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">row_anno</span><span class="p">,</span><span class="w">
  </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
  </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">draw</span><span class="p">(</span><span class="n">main_heatmap</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>These codes utilize <code class="language-plaintext highlighter-rouge">Euclidean</code> distance for clustering and employs the <code class="language-plaintext highlighter-rouge">complete</code> algorithm (Default).<li>To modify the clustering distance, add the <code class="language-plaintext highlighter-rouge">clustering_distance_rows = "&lt;distance&gt;"</code> parameter. Available <code class="language-plaintext highlighter-rouge">distance</code> metrics include <code class="language-plaintext highlighter-rouge">euclidean</code>, <code class="language-plaintext highlighter-rouge">maximum</code>, <code class="language-plaintext highlighter-rouge">manhattan</code>, <code class="language-plaintext highlighter-rouge">canberra</code>, and <code class="language-plaintext highlighter-rouge">binary</code>.<li>Additionally, to change the clustering algorithm, adjust the <code class="language-plaintext highlighter-rouge">clustering_method_rows = "&lt;algorithm&gt;"</code> parameter. Possible algorithms are <code class="language-plaintext highlighter-rouge">single</code> (minimum linkage), <code class="language-plaintext highlighter-rouge">complete</code> (maximum linkage), <code class="language-plaintext highlighter-rouge">average</code> (UPGMA), <code class="language-plaintext highlighter-rouge">ward.D</code> (Ward’s minimum variance method), <code class="language-plaintext highlighter-rouge">ward.D2</code> (Modified Ward’s method), <code class="language-plaintext highlighter-rouge">mcquitty</code> (McQuitty’s method, WPGMA), <code class="language-plaintext highlighter-rouge">median</code> (Median linkage, WPGMC), and <code class="language-plaintext highlighter-rouge">centroid</code> (Centroid linkage, UPGMC).</ul><h3 id="4-optional-represent-specific-gene-symbols-in-the-heatmap"> <a href="#4-optional-represent-specific-gene-symbols-in-the-heatmap" class="anchor-heading" aria-labelledby="4-optional-represent-specific-gene-symbols-in-the-heatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4. (Optional) Represent specific gene symbols in the heatmap</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Specify the genes whose names you want to be displayed</span><span class="w">
</span><span class="n">selected_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"LOC_Os06g49240"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LOC_Os01g49710"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LOC_Os10g03220"</span><span class="p">,</span><span class="w"> </span><span class="s2">"LOC_Os04g41680"</span><span class="p">)</span><span class="w">
</span><span class="n">row_indices</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">MSU</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">selected_genes</span><span class="p">)</span><span class="w">

</span><span class="c1"># Plotting</span><span class="w">
</span><span class="n">main_heatmap2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">main_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">rowAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">foo</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">anno_mark</span><span class="p">(</span><span class="w">
    </span><span class="n">at</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">row_indices</span><span class="p">,</span><span class="w">  
    </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">MSU</span><span class="p">[</span><span class="n">row_indices</span><span class="p">],</span><span class="w">
    </span><span class="n">which</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">,</span><span class="w">  
    </span><span class="n">labels_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">))</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Save</span><span class="w">
</span><span class="n">draw</span><span class="p">(</span><span class="n">main_heatmap2</span><span class="p">,</span><span class="w"> </span><span class="n">merge_legends</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">gap</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><ul><li><strong>Output (Left; Step 3, Right; Step 4):</strong> <i>(Clicking on the image will open it in a larger view.)</i></ul><link rel="stylesheet" href="/assets/css/modal.css" /><p align="center" style="display: flex; justify-content: center;"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase1_1.png" alt="com1" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase1_1.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase1_1.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase1_2.png'])" /> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase1_2.png" alt="com2" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase1_2.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase1_1.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase1_2.png'])" /></p><div id="myModal" class="modal"> <span class="close" onclick="closeModal()">&times;</span> <span class="arrow left" onclick="changeImage(-1)">&#10094;</span> <img class="modal-content" id="img01" /> <span class="arrow right" onclick="changeImage(1)">&#10095;</span><div class="modal-list" id="imgList"></div></div><script src="/assets/js/modal.js"></script><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-guide-case-ii---integration-of-rna-seq-with-simpler-structured-dataset"> <a href="#step-by-step-guide-case-ii---integration-of-rna-seq-with-simpler-structured-dataset" class="anchor-heading" aria-labelledby="step-by-step-guide-case-ii---integration-of-rna-seq-with-simpler-structured-dataset"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Guide: Case II - Integration of RNA-seq with Simpler Structured Dataset</h1><p>In this dataset, genes that respond to stress in human keratinocytes and three derived knockdown cell types are categorized by their fold change values, using normal cell expression levels as a baseline. Then, these genes were manually grouped into four categories (<code class="language-plaintext highlighter-rouge">Cluster</code>row) based on their expression patterns. Additionally, genes associated with ROS are enumerated in a separate row (<code class="language-plaintext highlighter-rouge">ROS.hallmark</code>) and identified as <code class="language-plaintext highlighter-rouge">ROS</code>. Moreover, the presence of MLL1, H3K4me3, UTX, and H3K27me3 peaks, as detected in normal cells under stress conditions from ChIP-seq results, is indicated by <code class="language-plaintext highlighter-rouge">0</code> (NO) and <code class="language-plaintext highlighter-rouge">1</code> (Yes) values.</p><h3 id="1-normalize-data-and-prepare-matrixannotation"> <a href="#1-normalize-data-and-prepare-matrixannotation" class="anchor-heading" aria-labelledby="1-normalize-data-and-prepare-matrixannotation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Normalize data and prepare matrix/annotation</h3><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Load data</span><span class="w">
</span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"ROS_heatmap.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

</span><span class="c1"># Matrix the data as main heatmap and Normalize to Z-score.</span><span class="w">
</span><span class="n">expression_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">data</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="o">:</span><span class="m">5</span><span class="p">])</span><span class="w">
</span><span class="n">z_score</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">scale</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">expression_data</span><span class="p">)))</span><span class="w">
</span><span class="n">z_min</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">min</span><span class="p">(</span><span class="n">z_score</span><span class="p">)</span><span class="w">
</span><span class="n">z_max</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">z_score</span><span class="p">)</span><span class="w">

</span><span class="c1"># Create the annotation dataframe and set annotation</span><span class="w">
</span><span class="n">cluster_factor</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">Cluster</span><span class="p">)</span><span class="w">
</span><span class="n">annotation</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">Cluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cluster_factor</span><span class="p">)</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">annotation</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w">
</span><span class="n">set2_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="s2">"Set2"</span><span class="p">)</span><span class="w">
</span><span class="n">row_anno</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rowAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation</span><span class="p">,</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Cluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">setNames</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">set2_palette</span><span class="p">[</span><span class="m">4</span><span class="p">],</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">3</span><span class="p">],</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">1</span><span class="p">]),</span><span class="w"> </span><span class="n">levels</span><span class="p">(</span><span class="n">cluster_factor</span><span class="p">))),</span><span class="w">
  </span><span class="n">annotation_name_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w">
  </span><span class="n">simple_anno_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li>This dataset used in this case is organized as follows:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; head(data)
  SYMBOL shControl   shNRF2     shMLL1      shUTX  Cluster ROS.hallmark MLL1 H3K4 UTX H3K27
1  HHATL  1.191316 1.473315 -3.0637972  0.1852907 cluster2                 0    0   0     0
2    AHR  1.478816 1.352787 -3.0637972  0.0000000 cluster2                 1    0   0     0
3   NQO1  3.706885 1.283004  0.0000000 -1.2765334 cluster2          ROS    0    1   0     0
4   SSPN  1.076450 1.185496 -0.3337084 -3.2310447 cluster2                 0    0   0     0
5  G6PC1  1.386497 2.442824 -0.1576905 -0.1286908 cluster2                 0    0   0     1
6 PPP6R2  4.561280 4.742004 -0.1064526  3.1118796 cluster2                 0    0   0     0
</code></pre></div></div><li>Next, matrix the dataset (ChIP-seq) that will be used as the secondary heatmap.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">mll1_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">MLL1</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">dimnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">),</span><span class="w"> </span><span class="s2">"MLL1"</span><span class="p">))</span><span class="w">
</span><span class="n">h3k4_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">H3K4</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">dimnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">),</span><span class="w"> </span><span class="s2">"H3K4"</span><span class="p">))</span><span class="w">
</span><span class="n">utx_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">UTX</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">dimnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">),</span><span class="w"> </span><span class="s2">"UTX"</span><span class="p">))</span><span class="w">
</span><span class="n">h3k27_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">H3K27</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">dimnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">),</span><span class="w"> </span><span class="s2">"H3K27"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div></ul><h3 id="2-plotting"> <a href="#2-plotting" class="anchor-heading" aria-labelledby="2-plotting"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Plotting</h3><ul><li>Create color palettes for the secondary heatmap.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">oranges_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="s2">"Oranges"</span><span class="p">))</span><span class="w">
</span><span class="n">reds_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="s2">"Reds"</span><span class="p">))</span><span class="w">
</span><span class="n">greens_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="s2">"Greens"</span><span class="p">))</span><span class="w">
</span><span class="n">purples_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="s2">"Purples"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li>Draw the secondary heatmap.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">mll1_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="n">mll1_data</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colorRamp2</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">oranges_palette</span><span class="p">[</span><span class="m">5</span><span class="p">])),</span><span class="w"> 
                        </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"MLL1"</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                        </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">))</span><span class="w">
</span><span class="n">h3k4_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="n">h3k4_data</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colorRamp2</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">reds_palette</span><span class="p">[</span><span class="m">9</span><span class="p">])),</span><span class="w"> 
                        </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"H3K4"</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                        </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">))</span><span class="w">
</span><span class="n">utx_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="n">utx_data</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colorRamp2</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">greens_palette</span><span class="p">[</span><span class="m">9</span><span class="p">])),</span><span class="w"> 
                      </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UTX"</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">))</span><span class="w">
</span><span class="n">h3k27_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="n">h3k27_data</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colorRamp2</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">purples_palette</span><span class="p">[</span><span class="m">8</span><span class="p">])),</span><span class="w"> 
                        </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"H3K27"</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                        </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li>Draw a main heatmap.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">main_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="n">z_score</span><span class="p">,</span><span class="w"> </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Expression Data"</span><span class="p">,</span><span class="w">
                        </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colorRamp2</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">z_min</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">z_max</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s1">'dodgerblue'</span><span class="p">,</span><span class="w"> </span><span class="s1">'black'</span><span class="p">,</span><span class="w"> </span><span class="s1">'yellow'</span><span class="p">)),</span><span class="w">
                        </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
                        </span><span class="n">column_title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Expression Data"</span><span class="p">,</span><span class="w">
                        </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">  
                        </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w">  
                        </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cluster_factor</span><span class="p">,</span><span class="w"> 
                        </span><span class="n">row_title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w">
                        </span><span class="n">left_annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">row_anno</span><span class="p">,</span><span class="w">
                        </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">  
                        </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li>Combine the main heatmap with the secondary heatmap.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">heatmap_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">main_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">mll1_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">h3k4_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">utx_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">h3k27_heatmap</span><span class="w">

</span><span class="c1"># Draw without margin</span><span class="w">
</span><span class="n">draw</span><span class="p">(</span><span class="n">heatmap_list</span><span class="p">,</span><span class="w"> </span><span class="n">merge_legends</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">gap</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h3 id="3-optional-showing-specific-gene-symbols-from-a-large-list-in-a-heatmap"> <a href="#3-optional-showing-specific-gene-symbols-from-a-large-list-in-a-heatmap" class="anchor-heading" aria-labelledby="3-optional-showing-specific-gene-symbols-from-a-large-list-in-a-heatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. (Optional) Showing specific gene symbols from a large list in a heatmap</h3><p>If the <code class="language-plaintext highlighter-rouge">ROS.hallmark</code> row is marked <code class="language-plaintext highlighter-rouge">ROS</code>, display the corresponding value from the <code class="language-plaintext highlighter-rouge">SYMBOL</code> row and annotate it accordingly:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Specify ROS markers </span><span class="w">
</span><span class="n">ros_indices</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">ROS.hallmark</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"ROS"</span><span class="p">)</span><span class="w">

</span><span class="c1"># Reflect these markers on the last heatmap among the secondary heatmaps</span><span class="w">
</span><span class="n">h3k27_heatmap_Rev</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">h3k27_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">rowAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">foo</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">anno_mark</span><span class="p">(</span><span class="w">
    </span><span class="n">at</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ros_indices</span><span class="p">,</span><span class="w">  
    </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">SYMBOL</span><span class="p">[</span><span class="n">ros_indices</span><span class="p">],</span><span class="w">
    </span><span class="n">which</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">,</span><span class="w">  
    </span><span class="n">labels_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">))</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Simple version</span><span class="w">
</span><span class="n">h3k27_heatmap_Rev</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">h3k27_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">rowAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">foo</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">anno_mark</span><span class="p">(</span><span class="w">
    </span><span class="n">at</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ros_indices</span><span class="p">,</span><span class="w"> 
    </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">SYMBOL</span><span class="p">[</span><span class="n">ros_indices</span><span class="p">],</span><span class="w"> 
    </span><span class="n">which</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Advanced version</span><span class="w">
</span><span class="n">h3k27_heatmap_Rev</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">h3k27_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">rowAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">foo</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">anno_mark</span><span class="p">(</span><span class="w">
    </span><span class="n">at</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ros_indices</span><span class="p">,</span><span class="w">  
    </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">SYMBOL</span><span class="p">[</span><span class="n">ros_indices</span><span class="p">],</span><span class="w">
    </span><span class="n">which</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">,</span><span class="w">  
    </span><span class="n">labels_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">),</span><span class="w">  
    </span><span class="n">padding</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">)</span><span class="w">  
  </span><span class="p">),</span><span class="w">
  </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">5.0</span><span class="p">,</span><span class="w"> </span><span class="s1">'cm'</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">max_text_width</span><span class="p">(</span><span class="n">ros_indices</span><span class="p">,</span><span class="w"> </span><span class="n">gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">))</span><span class="w"> 
</span><span class="p">)</span><span class="w">  
</span></code></pre></div></div><p>I typically apply these markers to the auxiliary heatmap rather than the primary one, and their inclusion is not obligatory. Please designate these markers accordingly and subsequently integrate your heatmaps:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Combine</span><span class="w">
</span><span class="n">heatmap_final</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">main_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">mll1_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">h3k4_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">utx_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">h3k27_heatmap_Rev</span><span class="w">

</span><span class="c1"># Draw</span><span class="w">
</span><span class="n">draw</span><span class="p">(</span><span class="n">heatmap_final</span><span class="p">,</span><span class="w"> </span><span class="n">merge_legends</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">gap</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><ul><li><strong>Output (Left; Step 2, Right; Step 3):</strong> <i>(Clicking on the image will open it in a larger view.)</i></ul><link rel="stylesheet" href="/assets/css/modal.css" /><p align="center" style="display: flex; justify-content: center;"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase2_1.png" alt="com1" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase2_1.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase2_1.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase2_2.png'])" /> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase2_2.png" alt="com2" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase2_2.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase2_1.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase2_2.png'])" /></p><div id="myModal" class="modal"> <span class="close" onclick="closeModal()">&times;</span> <span class="arrow left" onclick="changeImage(-1)">&#10094;</span> <img class="modal-content" id="img01" /> <span class="arrow right" onclick="changeImage(1)">&#10095;</span><div class="modal-list" id="imgList"></div></div><script src="/assets/js/modal.js"></script><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-guide-case-iii---multiple-rna-seq-combined-with-chip-seq"> <a href="#step-by-step-guide-case-iii---multiple-rna-seq-combined-with-chip-seq" class="anchor-heading" aria-labelledby="step-by-step-guide-case-iii---multiple-rna-seq-combined-with-chip-seq"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Guide: Case III - Multiple RNA-seq Combined with ChIP-seq</h1><p>This example involves big data set derived from mouse mesenchymal stem cells (MSCs) and three different knockdown cell lines following their differentiation into osteoblasts and adipocytes, respectively. The primary data set comprises RNA-seq results from these cell lines undergone each differential stimuli. Additionally, as secondary data, ChIP-seq results are utilized for the genome-wide identification of alterations in the occupancy and enrichment of NFIB and H3K4me3 following the induction of bone differentiation in intact MSCs. Lastly, I present a heatmap to show whether the list of genes analyzed are transcription factors and regulators of each differentiation process.</p><h3 id="1-normalize-data-and-prepare-matrixannotation-1"> <a href="#1-normalize-data-and-prepare-matrixannotation-1" class="anchor-heading" aria-labelledby="1-normalize-data-and-prepare-matrixannotation-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Normalize data and prepare matrix/annotation</h3><ul><li>Load data:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"ComplexSeq.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>This dataset used in this case is organized as follows:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; head(data)
        Symbol     O_Con    O_NFIB     O_MLL1    A_Con    A_NFIB     A_MLL1     NFIB  H3K4me3 TF Adipo Osteo  Cluster
1 1500009L16Rik 1.7141525  1.650957 -8.2522356 5.884279 -2.097777 -5.0292486 2.296628 1.787946  0     0     0 cluster2
2 A530016L24Rik 0.7738813  1.030212 -0.5199491 4.063868 -3.585915 -1.7736988 1.149132 1.370758  0     0     0 cluster3
3         Abca1 1.8617470  6.888385  0.5757785 3.776418 -2.151568 -1.5198181 2.175035 1.841612  0     0     0 cluster3
4        Abhd15 1.4146485 -0.256023 -0.3941591 3.219146 -3.508747 -3.6085345 2.350038 0.000000  0     0     0 cluster3
5         Acacb 2.4237469  2.502078 -1.0830766 2.121201 -1.460384 -0.6849265 1.762528 0.000000  0     0     0 cluster1
6         Acadm 0.7387791  6.690138 -0.1115445 1.582898 -1.675392 -1.2446482 1.935325 1.584191  0     0     0 cluster3
</code></pre></div></div><li>Create and normalize data frames by experiment type and sample type:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># For RNA-seq</span><span class="w">
</span><span class="n">RNA_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">data</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="o">:</span><span class="m">7</span><span class="p">])</span><span class="w">
</span><span class="n">RNA_z_score</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">scale</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">RNA_data</span><span class="p">)))</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">RNA_z_score</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w">
</span><span class="n">Osteo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">RNA_z_score</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">])</span><span class="w">
</span><span class="n">Adipo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">RNA_z_score</span><span class="p">[,</span><span class="w"> </span><span class="m">4</span><span class="o">:</span><span class="m">6</span><span class="p">])</span><span class="w">

</span><span class="c1"># For ChIP-seq</span><span class="w">
</span><span class="n">ChIP_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">data</span><span class="p">[,</span><span class="w"> </span><span class="m">8</span><span class="o">:</span><span class="m">9</span><span class="p">])</span><span class="w">
</span><span class="n">ChIP_z_score</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">scale</span><span class="p">(</span><span class="n">t</span><span class="p">(</span><span class="n">ChIP_data</span><span class="p">)))</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">ChIP_z_score</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>Create annotation:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Specifies color code and sets annotation for the row</span><span class="w">
</span><span class="n">set2_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="s2">"Set2"</span><span class="p">)</span><span class="w">
</span><span class="n">cluster_factor</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">Cluster</span><span class="p">)</span><span class="w">
</span><span class="n">annotation</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">Cluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cluster_factor</span><span class="p">)</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">annotation</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">)</span><span class="w">

</span><span class="c1"># Create row annotation</span><span class="w">
</span><span class="n">row_anno</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rowAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotation</span><span class="p">,</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Cluster</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">setNames</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">set2_palette</span><span class="p">[</span><span class="m">4</span><span class="p">],</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">3</span><span class="p">],</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">2</span><span class="p">]),</span><span class="w"> </span><span class="n">levels</span><span class="p">(</span><span class="n">cluster_factor</span><span class="p">))),</span><span class="w">
  </span><span class="n">annotation_name_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w">
  </span><span class="n">simple_anno_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">
</span><span class="n">ordered_levels</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"cluster3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"cluster2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"cluster1"</span><span class="p">)</span><span class="w">
</span><span class="n">cluster_factor_ordered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="n">cluster_factor</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ordered_levels</span><span class="p">)</span><span class="w">

</span><span class="c1"># Create column annotation</span><span class="w">
</span><span class="n">col_osteo_anno</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">HeatmapAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Osteogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Osteogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Osteogenesis"</span><span class="p">)),</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">Osteogenesis</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">5</span><span class="p">])),</span><span class="w">
  </span><span class="n">show_annotation_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">  
  </span><span class="n">simple_anno_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">col_adipo_anno</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">HeatmapAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Adipogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Adipogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Adipogenesis"</span><span class="p">)),</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">Adipogenesis</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">6</span><span class="p">])),</span><span class="w">
  </span><span class="n">show_annotation_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">  
  </span><span class="n">simple_anno_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">col_chip_anno</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">HeatmapAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"ChIPseq"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ChIPseq"</span><span class="p">)),</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">ChIPseq</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">set2_palette</span><span class="p">[</span><span class="m">7</span><span class="p">])),</span><span class="w">
  </span><span class="n">show_annotation_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">  
  </span><span class="n">simple_anno_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="n">column_split_factor</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="w">
    </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Osteogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Osteogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Osteogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Adipogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Adipogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Adipogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ChIPseq"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ChIPseq"</span><span class="p">),</span><span class="w">
    </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Osteogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Adipogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ChIPseq"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><h3 id="2-plotting-1"> <a href="#2-plotting-1" class="anchor-heading" aria-labelledby="2-plotting-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Plotting</h3><ul><li>Create color palettes and plot the main heatmaps:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Specifies colocr pallettes</span><span class="w">
</span><span class="n">osteo_color</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="n">rev</span><span class="p">(</span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="s2">"RdBu"</span><span class="p">)))</span><span class="w">
</span><span class="n">osteo_color_function</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">osteo_color</span><span class="p">(</span><span class="m">256</span><span class="p">)</span><span class="w">
</span><span class="n">adipo_color</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="n">rev</span><span class="p">(</span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="s2">"PiYG"</span><span class="p">)))</span><span class="w">
</span><span class="n">adipo_color_function</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">adipo_color</span><span class="p">(</span><span class="m">256</span><span class="p">)</span><span class="w">
</span><span class="n">chip_color</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="n">rev</span><span class="p">(</span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="s2">"YlGn"</span><span class="p">)))</span><span class="w">
</span><span class="n">chip_color_function</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">chip_color</span><span class="p">(</span><span class="m">256</span><span class="p">)</span><span class="w">

</span><span class="c1"># Osteogenic RNA-seq heatmap (as a main heatmap)</span><span class="w">
</span><span class="n">osteo_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="w">
  </span><span class="n">Osteo</span><span class="p">,</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">osteo_color_function</span><span class="p">,</span><span class="w">
  </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
  </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Osteo(z-score)"</span><span class="p">,</span><span class="w">
  </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">  
  </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w">  
  </span><span class="n">column_split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Osteogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Osteogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Osteogenesis"</span><span class="p">)),</span><span class="w">
  </span><span class="n">top_annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">col_osteo_anno</span><span class="p">,</span><span class="w">  
  </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">cluster_factor_ordered</span><span class="p">,</span><span class="w">
  </span><span class="n">row_title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w">
  </span><span class="n">left_annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">row_anno</span><span class="p">,</span><span class="w">
  </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
  </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Adipogenic RNA-seq heatmap</span><span class="w">
</span><span class="n">adipo_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="w">
  </span><span class="n">Adipo</span><span class="p">,</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">adipo_color_function</span><span class="p">,</span><span class="w">
  </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
  </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Adipo(z-score)"</span><span class="p">,</span><span class="w">
  </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w">  
  </span><span class="n">column_split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Adipogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Adipogenesis"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Adipogenesis"</span><span class="p">)),</span><span class="w">
  </span><span class="n">top_annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">col_adipo_anno</span><span class="p">,</span><span class="w">  
  </span><span class="n">row_title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w">
  </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
  </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">
</span><span class="c1"># ChIP-seq heatmap</span><span class="w">
</span><span class="n">chip_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="w">
  </span><span class="n">ChIP_z_score</span><span class="p">,</span><span class="w">
  </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">chip_color_function</span><span class="p">,</span><span class="w">
  </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
  </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Peak intensity (z-score)"</span><span class="p">,</span><span class="w">
  </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
  </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w">  
  </span><span class="n">column_split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"ChIPseq"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ChIPseq"</span><span class="p">)),</span><span class="w">
  </span><span class="n">top_annotation</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">col_chip_anno</span><span class="p">,</span><span class="w">  
  </span><span class="n">row_title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w">
  </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">1.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
  </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># combine and draw</span><span class="w">
</span><span class="n">comb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">osteo_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">adipo_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">chip_heatmap</span><span class="w">
</span><span class="n">draw</span><span class="p">(</span><span class="n">comb</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>Crate a data frame and matrix it for the secondary heatmaps:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">adipo_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">Adipo</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">dimnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">),</span><span class="w"> </span><span class="s2">"AdipoMarker"</span><span class="p">))</span><span class="w">
</span><span class="n">osteo_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">Osteo</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">dimnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">),</span><span class="w"> </span><span class="s2">"OsteoMarker"</span><span class="p">))</span><span class="w">
</span><span class="n">TF_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">TF</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">dimnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">),</span><span class="w"> </span><span class="s2">"TFs"</span><span class="p">))</span><span class="w">

</span><span class="c1"># Make sure you have NA in your data, in this example, NA is converted to '0' as shown below.</span><span class="w">
</span><span class="n">TF_data</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">TF_data</span><span class="p">)]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0</span><span class="w">
</span></code></pre></div></div><li>Create color palettes and plot the secondary heatmaps:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">reds_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="s2">"Reds"</span><span class="p">))</span><span class="w">
</span><span class="n">blues_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="s2">"Blues"</span><span class="p">))</span><span class="w">
</span><span class="n">purples_palette</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">brewer.pal</span><span class="p">(</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="s2">"Purples"</span><span class="p">))</span><span class="w">

</span><span class="c1"># Osteogenic markers heatmap with box legend</span><span class="w">
</span><span class="n">osteo_marker_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="n">osteo_data</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">reds_palette</span><span class="p">[</span><span class="m">7</span><span class="p">]),</span><span class="w"> 
                      </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Osteogenic Factor"</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
                      </span><span class="n">heatmap_legend_param</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="w">
                          </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Osteogenic"</span><span class="p">,</span><span class="w">
                          </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"No"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Yes"</span><span class="p">),</span><span class="w"> 
                          </span><span class="n">at</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w">                
                          </span><span class="n">legend_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">reds_palette</span><span class="p">[</span><span class="m">8</span><span class="p">]),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> 
                          </span><span class="n">labels_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">),</span><span class="w">
                          </span><span class="n">border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="w">
                      </span><span class="p">))</span><span class="w">

</span><span class="c1"># Adipogenic markers heatmap with box legend</span><span class="w">
</span><span class="n">adipo_marker_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="n">adipo_data</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">blues_palette</span><span class="p">[</span><span class="m">8</span><span class="p">]),</span><span class="w"> 
                      </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Adipogenic Factor"</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
                      </span><span class="n">heatmap_legend_param</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="w">
                          </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Adipogenic"</span><span class="p">,</span><span class="w">
                          </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"No"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Yes"</span><span class="p">),</span><span class="w"> 
                          </span><span class="n">at</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w">                
                          </span><span class="n">legend_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">blues_palette</span><span class="p">[</span><span class="m">8</span><span class="p">]),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> 
                          </span><span class="n">labels_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">),</span><span class="w">
                          </span><span class="n">border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="w">
                      </span><span class="p">))</span><span class="w">

</span><span class="c1"># TF heatmap with box legend</span><span class="w">
</span><span class="n">TF_heatmap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Heatmap</span><span class="p">(</span><span class="n">TF_data</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">purples_palette</span><span class="p">[</span><span class="m">8</span><span class="p">]),</span><span class="w"> 
                      </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Transcription Factor"</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">cluster_columns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">show_row_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">show_column_names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">column_names_side</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"top"</span><span class="p">,</span><span class="w"> 
                      </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"cm"</span><span class="p">),</span><span class="w">
                      </span><span class="n">heatmap_legend_param</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="w">
                          </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"TF"</span><span class="p">,</span><span class="w">
                          </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"non-TF"</span><span class="p">,</span><span class="w"> </span><span class="s2">"TF"</span><span class="p">),</span><span class="w"> 
                          </span><span class="n">at</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w">                
                          </span><span class="n">legend_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"white"</span><span class="p">,</span><span class="w"> </span><span class="n">purples_palette</span><span class="p">[</span><span class="m">8</span><span class="p">]),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> 
                          </span><span class="n">labels_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">),</span><span class="w">
                          </span><span class="n">border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="w">
                      </span><span class="p">))</span><span class="w">

</span><span class="n">comb2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">osteo_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">adipo_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">chip_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">osteo_marker_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">adipo_marker_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TF_heatmap</span><span class="w">
</span><span class="n">draw</span><span class="p">(</span><span class="n">comb2</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h3 id="3-showing-specific-gene-symbols-in-the-heatmap"> <a href="#3-showing-specific-gene-symbols-in-the-heatmap" class="anchor-heading" aria-labelledby="3-showing-specific-gene-symbols-in-the-heatmap"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. Showing Specific Gene Symbols in the Heatmap</h3><ul><li><p>Select a specific gene list and create logic to display it in the heatmap:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Specific gene list</span><span class="w">
</span><span class="n">selected_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Cebpa"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Dlx5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Ehf"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Epas1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Hic1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Maf"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Nkd2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Nr1h3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Pparg"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Runx2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Satb2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Thrsp"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Trp63"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Twist2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Zfp467"</span><span class="p">)</span><span class="w">

</span><span class="c1"># Find the specific gene among input data index</span><span class="w">
</span><span class="n">row_indices</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">data</span><span class="o">$</span><span class="n">Symbol</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">selected_genes</span><span class="p">)</span><span class="w">

</span><span class="c1"># Apply the logic on the heatmap</span><span class="w">
</span><span class="n">TF_heatmap_Rev</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">TF_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">rowAnnotation</span><span class="p">(</span><span class="w">
  </span><span class="n">foo</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">anno_mark</span><span class="p">(</span><span class="w">
    </span><span class="n">at</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">row_indices</span><span class="p">,</span><span class="w">  
    </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">data</span><span class="o">$</span><span class="n">Symbol</span><span class="p">[</span><span class="n">row_indices</span><span class="p">],</span><span class="w">
    </span><span class="n">which</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"row"</span><span class="p">,</span><span class="w">  
    </span><span class="n">labels_gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">))</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Draw</span><span class="w">
</span><span class="n">comb3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">osteo_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">adipo_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">chip_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">osteo_marker_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">adipo_marker_heatmap</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TF_heatmap_Rev</span><span class="w">
</span><span class="n">draw</span><span class="p">(</span><span class="n">comb3</span><span class="p">,</span><span class="w"> </span><span class="n">merge_legends</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">gap</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li><p><strong>Output:</strong></p></ul><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/complexCase3.png" alt="complexHeat" style="width: 60vh; height: auto;" /></p><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="citations"> <a href="#citations" class="anchor-heading" aria-labelledby="citations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citations</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">pheatmap</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Kolde, R.</b>, &amp; Kolde, M. R. (2015). Package ‘pheatmap’. <i>R package</i>, 1(7), 790. <a href="https://cran.ms.unimelb.edu.au/web/packages/pheatmap/pheatmap.pdf" target="_blank">PDF</a></span></ol><span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">complexheatmap</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Gu, Z.</b>, &amp; Hübschmann, D. (2022). Make interactive complex heatmaps in R, <i>Bioinformatics</i>, 38(5), 1460-1462. <a href="https://doi.org/10.1093/bioinformatics/btab806" target="_blank">DOI</a></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
