<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li > a, .site-nav > ul.nav-list:first-child > li > ul > li:not(:nth-child(4)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(4) > ul > li:nth-child(4) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(4) > button svg, .site-nav > ul.nav-list:first-child > li:nth-child(4) > ul > li:nth-child(4) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(4) > ul.nav-list, .site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(4) > ul.nav-list > li.nav-list-item:nth-child(4) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>DEG - Gene ontology and KEGG pathway analysis | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="DEG - Gene ontology and KEGG pathway analysis" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/RNASeq/GO_KEGG_MANUAL/" /><meta property="og:url" content="http://localhost:4000/docs/RNASeq/GO_KEGG_MANUAL/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="DEG - Gene ontology and KEGG pathway analysis" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"DEG - Gene ontology and KEGG pathway analysis","url":"http://localhost:4000/docs/RNASeq/GO_KEGG_MANUAL/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/docs/RNASeq">Data Analysis - RNA-seq</a><li class="breadcrumb-nav-list-item"><span>DEG - Gene ontology and KEGG pathway analysis</span></ol></nav><div id="main-content" class="main-content"><main><p>© 2022 Janghyun Choi<br /> <img src="https://img.shields.io/badge/Revised%20version-5%2E15%2E24-green?style=flat&amp;logo=Markdown&amp;colorA=black" alt="Docer" /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a></p><h1 class="fs-8 lh-tight" id="understanding-gene-functions-an-overview-of-gene-ontology-and-kegg-pathway-analysis"> <a href="#understanding-gene-functions-an-overview-of-gene-ontology-and-kegg-pathway-analysis" class="anchor-heading" aria-labelledby="understanding-gene-functions-an-overview-of-gene-ontology-and-kegg-pathway-analysis"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Understanding Gene Functions: An Overview of Gene Ontology and KEGG Pathway Analysis</h1><p class="fs-4 fw-500">Gene Ontology (GO) and KEGG Pathway analysis are essential bioinformatics tools for understanding gene functions and their roles within biological systems. GO provides a structured framework for classifying gene functions across different species, organizing them into three main categories: Biological Process, Molecular Function, and Cellular Component. This classification helps researchers comprehend the biological context and interactions of genes. On the other hand, KEGG Pathway analysis maps genes to known biological pathways, allowing scientists to visualize and interpret complex biochemical processes and molecular interactions within the cell. Together, these tools facilitate a deeper understanding of gene functions and their contributions to various biological phenomena. This protocol was developed using a system equipped with an Intel 10th generation i9-10910 processor and 48GB of memory. The test environment includes <span style="color: #7253ec;"><b>R version 4.4.0 running on macOS 14.4.1.</b></span><br /> Currently, GO/KEGG analysis is generally categorized into <span style="color: #7253ec;"><b>web-based and terminal-based formats.</b></span> In this section, I will elaborate on the functional analysis of genes using the widely-used web-based tool, DAVID, as well as the terminal-based tools, enrichGO and enrichKEGG, which are part of the clusterProfile R package.</p><h1 class="fs-7 fw-400 lh-tight" id="web-based-tool-david"> <a href="#web-based-tool-david" class="anchor-heading" aria-labelledby="web-based-tool-david"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Web-based Tool: DAVID</h1><p>DAVID (Database for Annotation, Visualization, and Integrated Discovery) is a comprehensive web-based tool that offers functional annotation tools for interpreting the biological significance of large gene lists. Integrates multiple annotation resources, including GO and KEGG, for functional annotation, gene ontology enrichment, pathway mapping, and visualization.</p><div class="table-wrapper"><table><thead><tr><th>Advantage<th>Disadvantage<tbody><tr><td>- User-friendly interface with integrated resources. <br /> - Allows easy upload of gene lists and quick retrieval of functional annotations. <br /> - Provides a variety of visualization tools to aid interpretation.<td>- Limited customization options compared to command-line tools. <br /> - Dependent on internet access and server availability. <br /> - Can be slower with very large datasets.</table></div><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-david-guide"> <a href="#step-by-step-david-guide" class="anchor-heading" aria-labelledby="step-by-step-david-guide"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step DAVID Guide</h1><ol><li>Go to the <a href="https://david.ncifcrf.gov/" target="_blank">DAVID website</a>.<li>Click <strong>Start Analysis</strong> on the website.<li>Paste your gene list in the step 1 box.<ul><li>In this example, I selected 1962 upragulated DEGs included by osteogenic stimuli and uploaded them. These genes designated as Ensembl IDs rather than gene symbols.</ul><li>Select identifier.<ul><li>Choose your gene type. In this example, I selected “ENSEMBL_GENE_ID”.<li>Note that Ensembl IDs have different prefixes for different species, so specifying a species is not necessary, but some identifiers may require it.</ul><li>Select the list type as either <strong>gene list</strong> or ‘background’.<ul><li><strong>Gene List</strong>: This is the primary list of genes you are interested in analyzing. It is the focus of your enrichment analysis.<li>Background (Optional): This is an optional list used to define the background set of genes against which your gene list is compared. If not provided, DAVID uses a default background set based on the genome of the organism.</ul><li>Click <strong>Submit List</strong>.<li>Check the <strong>Gene List Manager</strong> in the left panel.<ul><li>The number of genes that DAVID recognized and those it did not (Unknown) will appear. If there are many unknown genes, consider re-uploading using a conversion tool or another identifier.</ul><li>Click <strong>Functional Annotation Tool</strong> in the center panel.<li>DAVID will display the <strong>Annotation Summary Results</strong>.<li>Use the <strong>Gene Ontology</strong> section and the <strong>Pathway</strong> section in the results to explore the functional annotation and enrichment of your gene list.<ul><li>All data is available for download as tab-delimited text.</ul></ol><h2 class="fs-6 fw-400 lh-tight" id="understanding-gene-ontology-section"> <a href="#understanding-gene-ontology-section" class="anchor-heading" aria-labelledby="understanding-gene-ontology-section"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Understanding Gene Ontology section</u></h2><p>In DAVID functional annotation results, when you click on the GO section, you encounter several subsections labeled as <strong>BP (Biological Process)</strong>, <strong>MF (Molecular Function)</strong>, <strong>CC (Cellular Component)</strong>, followed by labels such as <code class="language-plaintext highlighter-rouge">_1</code> to <code class="language-plaintext highlighter-rouge">_5</code>, <code class="language-plaintext highlighter-rouge">ALL</code>, and <code class="language-plaintext highlighter-rouge">DIRECT</code>. Here is what each of these terms signifies:</p><div style="padding-left: 20px;"><ol><li><b>BP (Biological Process)</b>: This section pertains to biological objectives or roles completed by sets of molecular activities. Examples include cellular processes and metabolic functions.<li><b>MF (Molecular Function)</b>: This section describes the biochemical activity of genes, such as binding or catalysis, independent of the gene products (e.g., protein) performing them.<li><b>CC (Cellular Component)</b>: This part describes where gene products are located in the cell, such as membranes or organelles.<li><b>ALL</b>: This category provides a consolidated view of the results across all three GO domains (BP, MF, CC), offering a comprehensive overview without the segregation into specific domains.<li><b>DIRECT</b>: This indicates the annotation derived directly from the primary literature and not inferred from other sources or computational predictions. It is typically more specific and can be considered more reliable as it is directly tied to experimental evidence.</ol></div><ul><li>Lower numbers (like <code class="language-plaintext highlighter-rouge">_1</code> or <code class="language-plaintext highlighter-rouge">_2</code>) generally indicate groups of GO terms that are more general and broadly applicable to a variety of biological contexts. As the numbers increase (towards <code class="language-plaintext highlighter-rouge">_3</code>, <code class="language-plaintext highlighter-rouge">_4</code>, and <code class="language-plaintext highlighter-rouge">_5</code>), the GO terms become more specific and detailed. These higher numbers represent more niche and precise annotations that are closely related to the specific functions or locations of the genes in your list.</ul><h2 class="fs-6 fw-400 lh-tight" id="understanding-gokegg-analysis-output"> <a href="#understanding-gokegg-analysis-output" class="anchor-heading" aria-labelledby="understanding-gokegg-analysis-output"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Understanding GO/KEGG Analysis Output</u></h2><ul><li>Open GO result with R:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; GO_DAVID &lt;- read.table("Up_GO.txt", sep = "\t", header = TRUE)
  &gt; GO_DAVID
          Category                                  Term Count   Percent       PValue              Genes ListTotal PopHits
  1 GOTERM_BP_DIRECT    GO:0006629~lipid metabolic process   119 6.0994362 6.263319e-12 ENSMUSG00000025350      1695     741
  2 GOTERM_BP_DIRECT        GO:0001666~response to hypoxia    43 2.2039979 1.888764e-07 ENSMUSG00000034205      1695     215
  3 GOTERM_BP_DIRECT              GO:0006811~ion transport    84 4.3054844 5.390971e-06 ENSMUSG00000031075      1695     605
  4 GOTERM_BP_DIRECT      GO:0006956~complement activation    12 0.6150692 7.838427e-06 ENSMUSG00000022018      1695      28
  5 GOTERM_BP_DIRECT GO:0001649~osteoblast differentiation    25 1.2813942 1.118882e-05 ENSMUSG00000008136      1695     110
  PopTotal FoldEnrichment   Bonferroni    Benjamini          FDR
  1    20199       1.913766 3.367162e-08 3.367160e-08 3.352128e-08
  2    20199       2.383363 1.014884e-03 5.076997e-04 5.054332e-04
  3    20199       1.654566 2.856599e-02 9.660621e-03 9.617493e-03
  4    20199       5.107206 4.126402e-02 9.976222e-03 9.931685e-03
  5    20199       2.708367 5.837805e-02 9.976222e-03 9.931685e-03
</code></pre></div></div><li>Open KEGG result with R:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; KEGG_DAVID &lt;- read.table("Up_KEGG.txt", sep = "\t", header = TRUE)
  &gt; KEGG_DAVID
      Category                                                       Term Count   Percent       PValue
  1 KEGG_PATHWAY                                mmu01100:Metabolic pathways   214 10.968734 3.927550e-11
  2 KEGG_PATHWAY                 mmu00982:Drug metabolism - cytochrome P450    21  1.076371 1.615860e-06
  3 KEGG_PATHWAY            mmu00760:Nicotinate and nicotinamide metabolism    13  0.666325 1.704787e-04
  4 KEGG_PATHWAY mmu05208:Chemical carcinogenesis - reactive oxygen species    37  1.896463 2.304140e-04
  5 KEGG_PATHWAY                             mmu04014:Ras signaling pathway    36  1.845208 1.168008e-03
                  Genes ListTotal PopHits PopTotal FoldEnrichment   Bonferroni    Benjamini          FDR
  1 ENSMUSG00000031231827      1625    9463     9463       1.506898 1.327511e-08 1.327512e-08 1.284309e-08
  2    ENSMUSG00000074183       827      71     9463       3.384420 5.460121e-04 2.730804e-04 2.641932e-04
  3    ENSMUSG00000115338       827      42     9463       3.541746 5.599775e-02 1.920727e-02 1.858218e-02
  4    ENSMUSG00000074183       827     224     9463       1.890066 7.493281e-02 1.946998e-02 1.883635e-02
  5    ENSMUSG00000031453       827     235     9463       1.752903 3.263318e-01 6.597445e-02 6.382735e-02
</code></pre></div></div></ul><p>When interpreting the results of a GO or KEGG analysis, several key metrics are presented:</p><div style="padding-left: 20px;"><ol><li><b>Count</b>: This metric indicates the number of genes associated with a particular GO and KEGG term within the list of genes being analyzed.<li><b>List Total</b>: Represents the total number of genes analyzed. This value serves as the denominator for calculating proportions of genes associated with specific GO/KEGG terms.<li><b>Pop Hits</b>: Refers to the number of genes associated with the same GO term in the reference gene set, which typically includes all genes in the organism’s genome.<li><b>Pop Total</b>: The total number of genes in the reference gene set. This number is used to normalize the data and helps in calculating how representative the term is in the context of the entire genome.<li><b>Fold Enrichment</b>: This is a crucial statistic that indicates how much more often genes associated with a particular GO term are observed in the list of analyzed genes compared to what would be expected based on their representation in the genome.</ol></div><h2 class="fs-6 fw-400 lh-tight" id="alternative-web-based-tool-gene-ontology-resource"> <a href="#alternative-web-based-tool-gene-ontology-resource" class="anchor-heading" aria-labelledby="alternative-web-based-tool-gene-ontology-resource"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Alternative Web-based Tool: Gene Ontology Resource</u></h2><p>The Gene Ontology Resource provides web-based tools for exploring GO annotations and performing enrichment analysis. This tools offers access to extensive GO annotations, tools for functional profiling, and interfaces for browsing and searching GO terms. Go to the <a href="https://david.ncifcrf.gov/" target="_blank">Gene Ontology Resource</a>.</p><div class="table-wrapper"><table><thead><tr><th>Advantage<th>Disadvantage<tbody><tr><td>- User-friendly and accessible without needing programming skills. <br /> - Comprehensive and detailed GO term information. <br /> - Regularly updated with the latest GO terms and annotations.<td>- Limited to GO analysis without extensive pathway mapping like KEGG. <br /> - Customization and advanced analysis options are less flexible than in command-line tools. <br /> - Relies on internet access and web interface performance.</table></div><h1 class="fs-7 fw-400 lh-tight" id="depicting-a-bubble-chart-from-gokegg-analysis-results"> <a href="#depicting-a-bubble-chart-from-gokegg-analysis-results" class="anchor-heading" aria-labelledby="depicting-a-bubble-chart-from-gokegg-analysis-results"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Depicting a Bubble Chart from GO/KEGG Analysis Results</h1><p>A bubble chart is an exceptionally effective visual tool for presenting the results of GO and KEGG analyses. This type of chart enhances the visualization of the significance and impact of gene sets across different biological processes, molecular functions, and cellular components. In a bubble chart:</p><ul><li><strong>Each bubble represents a specific GO/KEGG term</strong>, providing a direct visual correlation to the biological aspect being analyzed.<li><strong>The size of the bubble</strong> is determined by the number of genes associated with each GO/KEGG term, visually emphasizing terms with a greater gene count.<li><strong>The color of each bubble</strong> varies according to the statistical significance (e.g., p-value), offering an immediate sense of the term relevance and the robustness of the associated data.</ul><p>In this section, I will construct a bubble chart using <code class="language-plaintext highlighter-rouge">ggplot2</code> to display the GO analysis results. The y-axis will list the GO terms, while the x-axis will represent the gene ratio, indicating the proportion of genes linked to each GO term relative to the total number of genes analyzed. The size of each bubble will be proportional to the gene count associated with each term, and the color of bubble will reflect the statistical significance, providing a comprehensive and intuitive visual summary of the data.</p><p><strong>Required Packages:</strong></p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w"> 
</span><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">gridExtra</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">egg</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">readxl</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-guide-for-depicting-bubble-chart-with-multiple-facets"> <a href="#step-by-step-guide-for-depicting-bubble-chart-with-multiple-facets" class="anchor-heading" aria-labelledby="step-by-step-guide-for-depicting-bubble-chart-with-multiple-facets"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Guide for Depicting Bubble Chart with Multiple Facets</h1><h3 id="1-data-preparation"> <a href="#1-data-preparation" class="anchor-heading" aria-labelledby="1-data-preparation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Data Preparation</h3><ul><li>Import the Gene Ontology results from text files into R, ensuring that the files have headers in the first row and do not contain row names. This example uses data from a GO analysis of genes upregulated and downregulated by differentiation stimuli in normal mesenchymal stem cells:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">UP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="s2">"upDEG.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
  </span><span class="n">DOWN</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="s2">"downDEG.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>Next, select the five GO terms with the lowest False Discovery Rate (FDR).<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">top_up</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">UP</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> 
      </span><span class="n">arrange</span><span class="p">(</span><span class="n">FDR</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> 
      </span><span class="n">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> 
      </span><span class="n">mutate</span><span class="p">(</span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Up-regulated"</span><span class="p">)</span><span class="w">

  </span><span class="n">top_down</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DOWN</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> 
      </span><span class="n">arrange</span><span class="p">(</span><span class="n">FDR</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> 
      </span><span class="n">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> 
      </span><span class="n">mutate</span><span class="p">(</span><span class="n">Type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Down-regulated"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><p class="note">Selecting the top 5-10 terms based on FDR is a common practice but not a strict rule. <strong>Researchers are encouraged to tailor the selection criteria based on the specific narrative and needs of their study.</strong></p><ul><li>Combine the top GO terms from both datasets into a single object and compute the <code class="language-plaintext highlighter-rouge">Ratio</code> of <code class="language-plaintext highlighter-rouge">Count</code> to <code class="language-plaintext highlighter-rouge">List.Total</code> for each term:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">combined</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">bind_rows</span><span class="p">(</span><span class="n">top_up</span><span class="p">,</span><span class="w"> </span><span class="n">top_down</span><span class="p">)</span><span class="w">
  </span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">combined</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
      </span><span class="n">mutate</span><span class="p">(</span><span class="n">Ratio</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Count</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">List.Total</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>Output:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; data
          Category                                  Term Count        X.       PValue List.Total Pop.Hits Pop.Total
  1  GOTERM_BP_DIRECT    GO:0006629~lipid metabolic process   119 6.0994362 6.263319e-12       1695      741     20199
  2  GOTERM_BP_DIRECT        GO:0001666~response to hypoxia    43 2.2039979 1.888764e-07       1695      215     20199
  3  GOTERM_BP_DIRECT              GO:0006811~ion transport    84 4.3054844 5.390971e-06       1695      605     20199
  4  GOTERM_BP_DIRECT      GO:0006956~complement activation    12 0.6150692 7.838427e-06       1695       28     20199
  5  GOTERM_BP_DIRECT GO:0001649~osteoblast differentiation    25 1.2813942 1.118882e-05       1695      110     20199
  6  GOTERM_BP_DIRECT                 GO:0007049~cell cycle   190 9.8752599 9.081317e-52       1798      649     20199
  7  GOTERM_BP_DIRECT              GO:0051301~cell division   126 6.5488565 2.885314e-37       1798      404     20199
  8  GOTERM_BP_DIRECT     GO:0007059~chromosome segregation    55 2.8586279 9.352744e-28       1798      111     20199
  9  GOTERM_BP_DIRECT            GO:0006260~DNA replication    55 2.8586279 7.029347e-26       1798      119     20199
  10 GOTERM_BP_DIRECT                 GO:0006281~DNA repair   100 5.1975052 5.008471e-22       1798      395     20199
  Fold.Enrichment   Bonferroni    Benjamini          FDR           Type       Ratio
  1         1.913766 3.367162e-08 3.367160e-08 3.352128e-08   Up-regulated 0.070206490
  2         2.383363 1.014884e-03 5.076997e-04 5.054332e-04   Up-regulated 0.025368732
  3         1.654566 2.856599e-02 9.660621e-03 9.617493e-03   Up-regulated 0.049557522
  4         5.107206 4.126402e-02 9.976222e-03 9.931685e-03   Up-regulated 0.007079646
  5         2.708367 5.837805e-02 9.976222e-03 9.931685e-03   Up-regulated 0.014749263
  6         3.288888 5.278062e-48 5.278062e-48 5.175443e-48 Down-regulated 0.105672970
  7         3.503720 1.676944e-33 8.384722e-34 8.221701e-34 Down-regulated 0.070077864
  8         5.566470 5.435815e-24 1.811938e-24 1.776710e-24 Down-regulated 0.030589544
  9         5.192254 4.085457e-22 1.021364e-22 1.001506e-22 Down-regulated 0.030589544
  10        2.844088 2.910923e-18 5.821847e-19 5.708655e-19 Down-regulated 0.055617353
</code></pre></div></div><p class="note">In this example, the ‘Genes’ column was excessively lengthy and has been remove to <code class="language-plaintext highlighter-rouge">NULL</code> to streamline the dataset.</p><li>When working with data not originating directly from <code class="language-plaintext highlighter-rouge">txt</code> files, such as arbitrarily selected GO terms for specific analyses, using Excel can be advantageous. Pre-calculate <code class="language-plaintext highlighter-rouge">Type</code> and <code class="language-plaintext highlighter-rouge">Ratio</code> in Excel using its functions. This preparation makes the data ready for immediate plotting upon import into R.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read_excel</span><span class="p">(</span><span class="s2">"PlotGO.xlsx"</span><span class="p">,</span><span class="w"> </span><span class="n">sheet</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"DOWN_Facet"</span><span class="p">)</span><span class="w"> 
</span></code></pre></div></div></ul><h3 id="2-plotting"> <a href="#2-plotting" class="anchor-heading" aria-labelledby="2-plotting"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Plotting</h3><ul><li>Draw a Basal Plot:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">S1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Ratio</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reorder</span><span class="p">(</span><span class="n">Term</span><span class="p">,</span><span class="w"> </span><span class="n">Ratio</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">PValue</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">facet_grid</span><span class="p">(</span><span class="n">factor</span><span class="p">(</span><span class="n">Type</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Up-regulated"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Down-regulated"</span><span class="p">))</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"free"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Gene Ratio"</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">""</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'white'</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'black'</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'solid'</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w">
          </span><span class="n">panel.grid.major</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_line</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey"</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span><span class="p">),</span><span class="w">
          </span><span class="n">strip.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey80"</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'solid'</span><span class="p">),</span><span class="w">
          </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">),</span><span class="w">
          </span><span class="n">legend.key</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">  
          </span><span class="n">legend.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">),</span><span class="w">  
          </span><span class="n">legend.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"bold"</span><span class="p">),</span><span class="w"> 
          </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">guides</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">guide_colourbar</span><span class="p">(</span><span class="n">barwidth</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span><span class="w"> </span><span class="n">barheight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li>Change graphical components:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Color space</span><span class="w">
  </span><span class="n">S1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">S1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"red2"</span><span class="p">,</span><span class="w">  </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"mediumblue"</span><span class="p">,</span><span class="w"> </span><span class="n">space</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Lab"</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Buble size range can be variable as follow:</span><span class="w">
  </span><span class="n">S2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">S1</span><span class="o">+</span><span class="n">scale_size</span><span class="p">(</span><span class="n">range</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">5</span><span class="p">))</span><span class="w">

  </span><span class="c1"># Adjust the x-axis</span><span class="w">
  </span><span class="n">S3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">S2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">xlim</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.12</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Adjust chart size for publication, 26.5/25 (width) and 30/33/40 (height) combination</span><span class="w">
  </span><span class="n">S4</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">set_panel_size</span><span class="p">(</span><span class="n">S3</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">26.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">))</span><span class="w"> 
  </span><span class="n">grid.arrange</span><span class="p">(</span><span class="n">S4</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li><strong>Output:</strong><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Bubble_multi.png" alt="BubbleMulti" style="width: 35vh; height: auto;" /></p></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-guide-for-depicting-bubble-chart-with-a-single-column"> <a href="#step-by-step-guide-for-depicting-bubble-chart-with-a-single-column" class="anchor-heading" aria-labelledby="step-by-step-guide-for-depicting-bubble-chart-with-a-single-column"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Guide for Depicting Bubble Chart with a Single Column</h1><h3 id="1-data-preparation-1"> <a href="#1-data-preparation-1" class="anchor-heading" aria-labelledby="1-data-preparation-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Data Preparation</h3><ul><li>Import the Gene Ontology results from text files into R. This example uses data from a GO analysis of genes upregulated by differentiation stimuli in normal mesenchymal stem cells:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">UP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.table</span><span class="p">(</span><span class="s2">"upDEG.txt"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>Next, select the 10 GO terms with the lowest FDR. <strong>Once again, selecting 10 GO terms as FDR criteria is not mandatory.</strong> Then, calculate the ‘Ratio’ of ‘Count’ to ‘List.Total’.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">top_up</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">UP</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> 
      </span><span class="n">arrange</span><span class="p">(</span><span class="n">FDR</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> 
      </span><span class="n">slice_head</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span><span class="w">
    
  </span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">top_up</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
      </span><span class="n">mutate</span><span class="p">(</span><span class="n">Ratio</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Count</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">List.Total</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><h3 id="2-plotting-1"> <a href="#2-plotting-1" class="anchor-heading" aria-labelledby="2-plotting-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Plotting</h3><ul><li>Draw a Basal Plot:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">S1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Ratio</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reorder</span><span class="p">(</span><span class="n">Term</span><span class="p">,</span><span class="w"> </span><span class="n">Ratio</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">PValue</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Gene Ratio"</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">""</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'white'</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'black'</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">linetype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'solid'</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w">
          </span><span class="n">panel.grid.major</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_line</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey"</span><span class="p">,</span><span class="w"> 
                                          </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span><span class="p">),</span><span class="w">
          </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">),</span><span class="w">
          </span><span class="n">legend.key</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">  
          </span><span class="n">legend.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">),</span><span class="w">  
          </span><span class="n">legend.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"bold"</span><span class="p">),</span><span class="w"> 
          </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">vjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">guides</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">guide_colourbar</span><span class="p">(</span><span class="n">barwidth</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span><span class="w"> </span><span class="n">barheight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li>Change graphical components:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Color space</span><span class="w">
  </span><span class="n">S1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">S1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"red2"</span><span class="p">,</span><span class="w">  </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"mediumblue"</span><span class="p">,</span><span class="w"> </span><span class="n">space</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Lab"</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Buble size range can be variable as follow:</span><span class="w">
  </span><span class="n">S2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">S1</span><span class="o">+</span><span class="n">scale_size</span><span class="p">(</span><span class="n">range</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">5</span><span class="p">))</span><span class="w">

  </span><span class="c1"># Adjust the x-axis</span><span class="w">
  </span><span class="n">S3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">S2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">xlim</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.09</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Adjust chart size for publication</span><span class="w">
  </span><span class="n">S4</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">set_panel_size</span><span class="p">(</span><span class="n">S3</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">26.5</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">),</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">48</span><span class="p">,</span><span class="w"> </span><span class="s2">"mm"</span><span class="p">))</span><span class="w"> 
  </span><span class="n">grid.arrange</span><span class="p">(</span><span class="n">S4</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li><strong>Output:</strong><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/Bubble_single.png" alt="BubbleSingle" style="width: 35vh; height: auto;" /></p></ul><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; width: 95%; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <label for="codeInput" style="display: block; margin-bottom: 5px;">Enter your code:</label> <textarea id="codeInput" rows="10" style="width: 100%; padding: 10px; border: 1px solid #ccc; border-radius: 5px; font-family: monospace;"></textarea><div style="margin-top: 10px;"> <button onclick="copyCode()" style="padding: 10px 15px; margin-right: 10px; background-color: #7253ec; color: white; border: none; border-radius: 5px; cursor: pointer;">Copy</button> <button onclick="clearCode()" style="padding: 10px 15px; background-color: #fafafa; color: black; border: none; border-radius: 5px; cursor: pointer;">Clear</button></div></div><script> function copyCode() { var codeInput = document.getElementById("codeInput"); codeInput.select(); document.execCommand("copy"); alert("Code copied to clipboard!"); } function clearCode() { document.getElementById("codeInput").value = ""; } </script><h1 class="fs-7 fw-400 lh-tight" id="terminal-based-go-tools-enrichgo-and-enrichkegg"> <a href="#terminal-based-go-tools-enrichgo-and-enrichkegg" class="anchor-heading" aria-labelledby="terminal-based-go-tools-enrichgo-and-enrichkegg"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Terminal-based GO Tools: enrichGO and enrichKEGG</h1><p>The enrichGO and enrichKEGG are powerful R package part of the <code class="language-plaintext highlighter-rouge">clusterProfiler</code> suite, designed for the functional annotation and enrichment analysis of gene lists. They provide robust tools for GO and pathway enrichment analyses, allowing researchers to interpret the biological significance of large gene sets through comprehensive statistical analyses.</p><div class="table-wrapper"><table><thead><tr><th>Advantage<th>Disadvantage<tbody><tr><td>- High level of customization and flexibility. <br /> - Capable of handling large datasets efficiently. <br /> - Integrates well with other R packages for further analysis and visualization. <br /> - Provides reproducible results as part of R scripts or workflows.<td>- Requires knowledge of R and command-line operations. <br /> - Steeper learning curve for users unfamiliar with programming. <br /> - Lack of a graphical user interface, which may be less intuitive for some users.</table></div><p><strong>Required Packages:</strong></p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">org.Mm.eg.db</span><span class="p">)</span><span class="w"> </span><span class="c1"># Organism-specific genome data package, refer to bellow.</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">enrichplot</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w"> 
</span><span class="n">library</span><span class="p">(</span><span class="n">gridExtra</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">egg</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">readxl</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><h2 class="fs-6 fw-400 lh-tight" id="choosing-the-appropriate-organism-database"> <a href="#choosing-the-appropriate-organism-database" class="anchor-heading" aria-labelledby="choosing-the-appropriate-organism-database"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <u>Choosing the Appropriate Organism Database</u></h2><ul><li>For organisms other than mouse, you need to use the appropriate organism-specific database. To find the appropriate database for your organism, you can search the Bioconductor website or use the BiocManager package’s search functionality:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">available</span><span class="p">(</span><span class="s2">"org."</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; BiocManager::available("org.")
  'getOption("repos")' replaces Bioconductor standard repositories, see 'help("repositories", package = "BiocManager")' for
  details.
  Replacement repositories:
      CRAN: http://cran.rstudio.com/
  [1] "AnnotationForge"     "BarBorGradient"      "BSgenomeForge"       "facebookorganicR"    "forge"              
  [6] "IsoCorrectoRGUI"     "KTensorGraphs"       "org.Ag.eg.db"        "org.At.tair.db"      "org.Bt.eg.db"       
  [11] "org.Ce.eg.db"        "org.Cf.eg.db"        "org.Dm.eg.db"        "org.Dr.eg.db"        "org.EcK12.eg.db"    
  [16] "org.EcSakai.eg.db"   "org.Gg.eg.db"        "org.Hs.eg.db"        "org.Mm.eg.db"        "org.Mmu.eg.db"      
  [21] "org.Mxanthus.db"     "org.Pf.plasmo.db"    "org.Pt.eg.db"        "org.Rn.eg.db"        "org.Sc.sgd.db"      
  [26] "org.Ss.eg.db"        "org.Xl.eg.db"        "Organism.dplyr"      "OrganismDbi"         "organizr"           
  [31] "OrgMassSpecR"        "orgR"                "orgutils"            "OscillatorGenerator" "pd.porgene.1.0.st"  
  [36] "pd.porgene.1.1.st"   "PriorGen"            "rNeighborGWAS"       "RNGforGPD"           "sectorgap"          
</code></pre></div></div><li>Then you can install the required organism database using Bioconductor:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"org.*.eg.db"</span><span class="p">)</span><span class="w"> </span><span class="c1"># '*' is organism abb.</span><span class="w">
  </span><span class="n">library</span><span class="p">(</span><span class="n">org.</span><span class="o">*</span><span class="n">.eg.db</span><span class="p">)</span><span class="w">
</span></code></pre></div></div></ul><h1 class="fs-7 fw-400 lh-tight" id="step-by-step-enrichgo-guide"> <a href="#step-by-step-enrichgo-guide" class="anchor-heading" aria-labelledby="step-by-step-enrichgo-guide"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step enrichGO Guide</h1><h3 id="1-data-preparation-2"> <a href="#1-data-preparation-2" class="anchor-heading" aria-labelledby="1-data-preparation-2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Data Preparation</h3><ul><li><p>Load raw DEG data obtained from <code class="language-plaintext highlighter-rouge">edgeR</code> or user-modified DEG data into R, ensuring that the files have headers in the first row and do not contain row names. In this section, I will load the raw data of gene list (result of <code class="language-plaintext highlighter-rouge">edgeR</code>) that change on osteogenic stimuli in mouse stem cells, filtering out only the DEGs that are upregulated:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Load data</span><span class="w">
  </span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"DEG.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Filter data</span><span class="w">
  </span><span class="n">UP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
      </span><span class="n">filter</span><span class="p">(</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">FDR</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; head(UP)
  Symbol    ensembl_gene_id    logFC     logCPM       PValue          FDR osteoMarker
  1    Tbx2 ENSMUSG00000000093 3.044420  0.1584658 8.460000e-09 1.020000e-07            
  2   Axin2 ENSMUSG00000000142 1.695402 -0.9306477 8.412736e-03 2.965543e-02       Osteo
  3   Itgb2 ENSMUSG00000000290 2.506077  0.8950346 2.250000e-11 3.870000e-10            
  4     Fap ENSMUSG00000000392 2.422669  0.4124939 1.695242e-03 7.284350e-03            
  5 Septin1 ENSMUSG00000000486 1.792568  2.3315912 8.820000e-12 1.600000e-10            
  6  Acvr1b ENSMUSG00000000532 1.094280  5.3096726 3.170000e-07 2.980000e-06            
</code></pre></div></div><li>Next, check the resources of the inputs available to enrichGO.<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">keytypes</span><span class="p">(</span><span class="n">org.Mm.eg.db</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; keytypes(org.Mm.eg.db)
  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS" "ENTREZID"     "ENZYME"       "EVIDENCE"    
  [9] "EVIDENCEALL"  "GENENAME"     "GENETYPE"     "GO"           "GOALL"        "IPI"          "MGI"          "ONTOLOGY"    
  [17] "ONTOLOGYALL"  "PATH"         "PFAM"         "PMID"         "PROSITE"      "REFSEQ"       "SYMBOL"       "UNIPROT"     
</code></pre></div></div><li>In this case, I can use the <code class="language-plaintext highlighter-rouge">Symbol</code> row and the <code class="language-plaintext highlighter-rouge">ensembl_gene_id</code> row as input from the <code class="language-plaintext highlighter-rouge">UP</code> object. This corresponds to <code class="language-plaintext highlighter-rouge">SYMBOL</code> and <code class="language-plaintext highlighter-rouge">ENSEMBL</code> in the keytype information (<code class="language-plaintext highlighter-rouge">keytypes(org.Mm.eg.db)</code>), which means that enrichGO supports both gene symbol and gene ID row as resource keytypes. For this example, I will use <code class="language-plaintext highlighter-rouge">ENSEMBL</code> as the keytype.</ul><h3 id="2-run-enrichgo-and-enrichkegg"> <a href="#2-run-enrichgo-and-enrichkegg" class="anchor-heading" aria-labelledby="2-run-enrichgo-and-enrichkegg"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Run enrichGO and enrichKEGG</h3><ul><li>Although a filter has been applied, the example output remains extensive due to the substantial number of genes under analysis. Therefore, in this section, I will focus on filtering only the osteogenic markers among the upregulated DEGs:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"DEG.csv"</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">
  </span><span class="n">Filter_UP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
      </span><span class="n">filter</span><span class="p">(</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">FDR</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span><span class="w"> </span><span class="n">osteoMarker</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"Osteo"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><li>Run <code class="language-plaintext highlighter-rouge">enrichGO</code>:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">ego</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Filter_UP</span><span class="o">$</span><span class="n">ensembl_gene_id</span><span class="p">,</span><span class="w">
                  </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"ENSEMBL"</span><span class="p">,</span><span class="w">
                  </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Mm.eg.db</span><span class="p">,</span><span class="w">
                  </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"BP"</span><span class="p">,</span><span class="w">
                  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"fdr"</span><span class="p">,</span><span class="w">
                  </span><span class="n">qvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span><span class="w">
                  </span><span class="n">readable</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
  </span><span class="c1"># Save GO results</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span><span class="w"> </span><span class="s2">"GOenrich.csv"</span><span class="p">)</span><span class="w">  
</span></code></pre></div></div><ul><li>in these codes,</ul></ul><div style="padding-left: 40px;"><ol><li><code><b>gene = Filter_UP$ensembl_gene_id</b></code>: Specifies the list of genes to analyze. In this case, it uses the <code>ensembl_gene_id</code> column from the <code>Filter_UP</code> dataframe.<li><code><b>keyType = "ENSEMBL"</b></code>: Defines the type of gene identifier.<li><code><b>OrgDb = org.Mm.eg.db</b></code>: Specifies the gene annotation database.<li><code><b>ont = "BP"</b></code>: Specifies the type of GO terms to analyze. Possible terms are <code>BP</code>, <code>MP</code>, and <code>CC</code>.<li><code><b>pAdjustMethod = "fdr"</b></code>: Defines the method for p-value adjustment. Possible methods are <code>holm</code>, <code>hochberg</code>, <code>hommel</code>, <code>bonferroni</code>, <code>BH</code>, <code>BY</code>, <code>fdr</code>, <code>none</code>.<li><code><b>qvalueCutoff = 0.05</b></code>: Specifies the cutoff for q-value.<li><code><b>readable = TRUE</b></code>: If set to <code>TRUE</code>, converts gene IDs in the results to a more readable form (e.g., gene symbols).</ol></div><ul><li>Before running <code class="language-plaintext highlighter-rouge">enrichKEGG</code>, check the KEGG organism code:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">search_kegg_organism</span><span class="p">(</span><span class="s1">'Mus musculus'</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="o">=</span><span class="s1">'scientific_name'</span><span class="p">)</span><span class="w"> </span><span class="c1"># 'Mus musculus' specifies your organism.</span><span class="w">
</span></code></pre></div></div><li>Output:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; search_kegg_organism('Mus musculus', by='scientific_name')
  kegg_code scientific_name common_name
  30       mmu    Mus musculus house mouse
</code></pre></div></div><li>Run <code class="language-plaintext highlighter-rouge">enrichKEGG</code>:<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">ekegg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichKEGG</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Filter_UP</span><span class="o">$</span><span class="n">ensembl_gene_id</span><span class="p">,</span><span class="w">
                      </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"ENSEMBL"</span><span class="p">,</span><span class="w">
                      </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"mmu"</span><span class="p">,</span><span class="w"> </span><span class="c1"># This value (mmu) can be obtained from above 'search_kegg' code.</span><span class="w">
                      </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"fdr"</span><span class="p">,</span><span class="w">
                      </span><span class="n">qvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Save KEGG results</span><span class="w">
  </span><span class="n">write.csv</span><span class="p">(</span><span class="n">ekegg</span><span class="p">,</span><span class="w"> </span><span class="s2">"KEGGenrich.csv"</span><span class="p">)</span><span class="w">  
</span></code></pre></div></div></ul><h3 id="3-visualization"> <a href="#3-visualization" class="anchor-heading" aria-labelledby="3-visualization"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. Visualization</h3><ul><li>The <code class="language-plaintext highlighter-rouge">enrichplot</code> package offers a variety of visualization methods for the GO and KEGG analyses obtained from the <code class="language-plaintext highlighter-rouge">enrichGO</code> and <code class="language-plaintext highlighter-rouge">enrichKEGG</code> results. In addition to <code class="language-plaintext highlighter-rouge">enrichplot</code>, bubble charts can be created using the <code class="language-plaintext highlighter-rouge">ggplot2</code> package by selecting the desired GO/KEGG terms from the saved csv file (refer to the <strong>Depicting a Bubble Chart from GO Analysis Results</strong> section). In this section, I will present only some of the most frequently used features of enrichplot based on the GO results.<li>When using <code class="language-plaintext highlighter-rouge">enrichGO</code> and <code class="language-plaintext highlighter-rouge">enrichKEGG</code> consecutively and then visualizing the results, it is not necessary to load the data separately. Adjusting the graphical parameters is beyond the scope of this lesson. For further details, please refer to the enrichplot manual.</ul><h4 class="fs-4 fw-400 lh-tight" id="3-1-bar-plot"> <a href="#3-1-bar-plot" class="anchor-heading" aria-labelledby="3-1-bar-plot"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3-1. Bar plot:</h4><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">barplot</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="o">=</span><span class="m">15</span><span class="p">)</span><span class="w">
</span><span class="n">barplot</span><span class="p">(</span><span class="n">ekegg</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="o">=</span><span class="m">10</span><span class="p">)</span><span class="w"> 
</span></code></pre></div></div><h4 class="fs-4 fw-400 lh-tight" id="3-2-bubble-chart"> <a href="#3-2-bubble-chart" class="anchor-heading" aria-labelledby="3-2-bubble-chart"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3-2. Bubble chart:</h4><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">dotplot</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="o">=</span><span class="m">15</span><span class="p">)</span><span class="w">
</span><span class="n">dotplot</span><span class="p">(</span><span class="n">ekegg</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="o">=</span><span class="m">10</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li><p><strong>Output (Left; Barplot, Right; Dotplot):</strong> <i>(Clicking on the image will open it in a larger view.)</i></p><link rel="stylesheet" href="/assets/css/modal.css" /><p align="center" style="display: flex; justify-content: center;"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Barplot.png" alt="pheat1" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Barplot.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Barplot.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Dot.png'])" /> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Dot.png" alt="pheat2" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Dot.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Barplot.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Dot.png'])" /></p><div id="myModal" class="modal"> <span class="close" onclick="closeModal()">&times;</span> <span class="arrow left" onclick="changeImage(-1)">&#10094;</span> <img class="modal-content" id="img01" /> <span class="arrow right" onclick="changeImage(1)">&#10095;</span><div class="modal-list" id="imgList"></div></div><script src="/assets/js/modal.js"></script></ul><h4 class="fs-4 fw-400 lh-tight" id="3-3-gene-concept-network"> <a href="#3-3-gene-concept-network" class="anchor-heading" aria-labelledby="3-3-gene-concept-network"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3-3. Gene-concept network:</h4><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">setNames</span><span class="p">(</span><span class="n">Filter_UP</span><span class="o">$</span><span class="n">logFC</span><span class="p">,</span><span class="w"> </span><span class="n">Filter_UP</span><span class="o">$</span><span class="n">ensembl_gene_id</span><span class="p">)</span><span class="w">
</span><span class="n">ego2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">setReadable</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span><span class="w"> </span><span class="s2">"org.Mm.eg.db"</span><span class="p">,</span><span class="w"> </span><span class="s2">"ENSEMBL"</span><span class="p">)</span><span class="w">
</span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cnetplot</span><span class="p">(</span><span class="n">ego2</span><span class="p">,</span><span class="w"> </span><span class="n">color.params</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">foldChange</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">),</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
            </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"logFC"</span><span class="p">,</span><span class="w"> </span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"red"</span><span class="p">)</span><span class="w">
</span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cnetplot</span><span class="p">(</span><span class="n">ego2</span><span class="p">,</span><span class="w"> </span><span class="n">color.params</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">foldChange</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">),</span><span class="w"> 
            </span><span class="n">circular</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">colorEdge</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
            </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"logFC"</span><span class="p">,</span><span class="w"> </span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"red"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li><p><strong>Output:</strong> <i>(Clicking on the image will open it in a larger view.)</i></p><link rel="stylesheet" href="/assets/css/modal.css" /><p align="center" style="display: flex; justify-content: center;"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_network1.png" alt="pheat1" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_network1.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_network1.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_network2.png'])" /> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_network2.png" alt="pheat2" style="flex: 1; max-width: 30vh; height: auto; cursor: pointer;" onclick="openModal('https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_network2.png', ['https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_network1.png', 'https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_network2.png'])" /></p><div id="myModal" class="modal"> <span class="close" onclick="closeModal()">&times;</span> <span class="arrow left" onclick="changeImage(-1)">&#10094;</span> <img class="modal-content" id="img01" /> <span class="arrow right" onclick="changeImage(1)">&#10095;</span><div class="modal-list" id="imgList"></div></div><script src="/assets/js/modal.js"></script></ul><h4 class="fs-4 fw-400 lh-tight" id="3-4-heatmap-like-functional-classification"> <a href="#3-4-heatmap-like-functional-classification" class="anchor-heading" aria-labelledby="3-4-heatmap-like-functional-classification"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3-4. Heatmap-like functional classification:</h4><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">heatplot</span><span class="p">(</span><span class="n">ego2</span><span class="p">,</span><span class="w"> </span><span class="n">foldChange</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w">
</span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">p1</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">scale_fill_gradient</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"logFC"</span><span class="p">,</span><span class="w"> </span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"red"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">coord_fixed</span><span class="p">(</span><span class="n">ratio</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">guides</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">guide_colorbar</span><span class="p">(</span><span class="n">barwidth</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span><span class="w"> </span><span class="n">barheight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><ul><li><strong>Output:</strong><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Heatpl.png" alt="enrich_Heatpl" style="width: 80vh; height: auto;" /></p></ul><h4 class="fs-4 fw-400 lh-tight" id="3-5-tree-plot"> <a href="#3-5-tree-plot" class="anchor-heading" aria-labelledby="3-5-tree-plot"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3-5. Tree plot</h4><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">egoTree</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_termsim</span><span class="p">(</span><span class="n">ego2</span><span class="p">)</span><span class="w">
</span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">treeplot</span><span class="p">(</span><span class="n">egoTree</span><span class="p">,</span><span class="w"> </span><span class="n">hclust_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"average"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><ul><li><strong>Output:</strong><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_Treeplot.png" alt="enrich_Treeplot" style="width: 40vh; height: auto;" /></p></ul><h4 class="fs-4 fw-400 lh-tight" id="3-6-gene-set-enrichment-analysis-gsea"> <a href="#3-6-gene-set-enrichment-analysis-gsea" class="anchor-heading" aria-labelledby="3-6-gene-set-enrichment-analysis-gsea"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3-6. Gene Set Enrichment Analysis (GSEA)</h4><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 10px; padding-bottom: 5px; margin-top: 1px; margin-bottom: 1px; margin-left: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 1px; position: relative; left: -10px;">Note</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span>GSEA is a computational method that determines if a predefined set of genes shows statistically significant differences between two biological states by comparing gene ranks within the set to those in the entire genome.</span><li style="margin-bottom: 10px;"> <span>GSEA is crucial in RNA-seq analyses as it reveals differentially regulated biological pathways and processes, offering a comprehensive understanding beyond individual gene expression changes.</span><li style="margin-bottom: 10px;"> <span>GSEA effectively analyzes the human transcriptome/genome using the <u>Molecular Signatures Database</u> and tools from the <b>Broad Institute</b>. In this section, I will provide code to plot GSEA correlation plot, and create a separate section and explain more about GSEA later.</span></ol></div><p><br /></p><ul><li><p>Data selection:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"DEG.csv"</span><span class="p">)</span><span class="w"> </span><span class="c1"># Load raw DEG data obtained from edgeR result</span><span class="w">

  </span><span class="n">filtered_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
      </span><span class="n">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">FDR</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="o">!</span><span class="p">(</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">-1</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">FDR</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li>Output:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; head(filtered_data)
  Symbol    ensembl_gene_id     logFC     logCPM      PValue      FDR osteoMarker
  1  Gnai3 ENSMUSG00000000001 -1.028899  7.5123714 7.94000e-07 6.96e-06            
  2  Cdc45 ENSMUSG00000000028 -2.245032  4.7831577 1.61000e-22 8.34e-21            
  3  Scml2 ENSMUSG00000000037 -1.842400  2.3288016 4.06000e-10 5.96e-09            
  4   Apoh ENSMUSG00000000049 -0.161434 -0.1108714 8.26452e-01 1.00e+00            
  5   Narf ENSMUSG00000000056 -1.124729  5.6864476 1.27000e-07 1.28e-06            
  6   Cav2 ENSMUSG00000000058 -1.028229  6.7405093 1.12000e-06 9.54e-06   
</code></pre></div></div><li><p>Run GSEA:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Make samll varience (based on normal distribution) to reduce ties in the preranked stats</span><span class="w">
  </span><span class="n">set.seed</span><span class="p">(</span><span class="m">42</span><span class="p">)</span><span class="w">
  </span><span class="n">filtered_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">filtered_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">rnorm</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">nrow</span><span class="p">(</span><span class="n">filtered_data</span><span class="p">),</span><span class="w"> </span><span class="n">mean</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1e-5</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Make compare list</span><span class="w">
  </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">setNames</span><span class="p">(</span><span class="n">filtered_data</span><span class="o">$</span><span class="n">logFC</span><span class="p">,</span><span class="w"> </span><span class="n">filtered_data</span><span class="o">$</span><span class="n">ensembl_gene_id</span><span class="p">)</span><span class="w">
  </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sort</span><span class="p">(</span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Run GSEA</span><span class="w">
  </span><span class="n">gsea_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gseGO</span><span class="p">(</span><span class="n">geneList</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">,</span><span class="w">
                      </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Mm.eg.db</span><span class="p">,</span><span class="w">
                      </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"BP"</span><span class="p">,</span><span class="w">
                      </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"ENSEMBL"</span><span class="p">,</span><span class="w">
                      </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"fdr"</span><span class="p">,</span><span class="w">
                      </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.25</span><span class="p">,</span><span class="w">
                      </span><span class="n">eps</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w">
                      </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

  </span><span class="c1"># Call GO terms from the GSEA results</span><span class="w">
  </span><span class="n">go_terms</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gsea_results</span><span class="o">@</span><span class="n">result</span><span class="p">[,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"ID"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Description"</span><span class="p">,</span><span class="w"> </span><span class="s2">"NES"</span><span class="p">,</span><span class="w"> </span><span class="s2">"p.adjust"</span><span class="p">,</span><span class="w"> </span><span class="s2">"qvalue"</span><span class="p">)]</span><span class="w">
  </span><span class="n">osteo_terms</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">go_terms</span><span class="p">[</span><span class="n">grep</span><span class="p">(</span><span class="s2">"osteoblast"</span><span class="p">,</span><span class="w"> </span><span class="n">go_terms</span><span class="o">$</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="n">ignore.case</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">
</span></code></pre></div></div><li>Output:<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  &gt; osteo_terms &lt;- go_terms[grep("osteoblast", go_terms$Description, ignore.case = TRUE), ]
  &gt; osteo_terms
                  ID                              Description       NES  p.adjust    qvalue
  GO:0045667 GO:0045667 regulation of osteoblast differentiation -1.314391 0.2478609 0.2221023
</code></pre></div></div><li><p>Plotting:</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">  </span><span class="c1"># Make a correlation plot</span><span class="w">
  </span><span class="n">go_id</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"GO:0045667"</span><span class="w">
  </span><span class="n">gseaplot2</span><span class="p">(</span><span class="n">gsea_results</span><span class="p">,</span><span class="w"> </span><span class="n">geneSetID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">go_id</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">paste</span><span class="p">(</span><span class="s2">"GSEA Plot for"</span><span class="p">,</span><span class="w"> </span><span class="n">go_id</span><span class="p">))</span><span class="w">

  </span><span class="c1"># Make an information table</span><span class="w">
  </span><span class="n">nes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gsea_results</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">NES</span><span class="p">[</span><span class="n">gsea_results</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">ID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">go_id</span><span class="p">]</span><span class="w">
  </span><span class="n">pvalue</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gsea_results</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">pvalue</span><span class="p">[</span><span class="n">gsea_results</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">ID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">go_id</span><span class="p">]</span><span class="w">
  </span><span class="n">padj</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gsea_results</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">p.adjust</span><span class="p">[</span><span class="n">gsea_results</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">ID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">go_id</span><span class="p">]</span><span class="w">
  </span><span class="n">lab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="s2">"NES = "</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">nes</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span><span class="s2">"\nFDR = "</span><span class="p">,</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="n">padj</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">))</span><span class="w">

  </span><span class="c1">## The 'annotate' function is currently not operational; therefore, I will implement a workaround by overlaying the table on the plot.</span><span class="w">
  </span><span class="c1"># Plot with the table</span><span class="w">
  </span><span class="n">grid.text</span><span class="p">(</span><span class="n">lab</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.95</span><span class="p">,</span><span class="w"> </span><span class="s2">"npc"</span><span class="p">),</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">unit</span><span class="p">(</span><span class="m">0.95</span><span class="p">,</span><span class="w"> </span><span class="s2">"npc"</span><span class="p">),</span><span class="w"> </span><span class="n">just</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"right"</span><span class="p">,</span><span class="w"> </span><span class="s2">"top"</span><span class="p">),</span><span class="w"> </span><span class="n">gp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gpar</span><span class="p">(</span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">fontsize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">))</span><span class="w">
</span></code></pre></div></div><li><strong>Output:</strong><p align="center"> <img src="https://raw.githubusercontent.com/seq-jchoi-bio/seq-portfolio/main/DataAnalysis_RNASeq/Images/enrich_GSEA.png" alt="enrich_GSEA" style="width: 35vh; height: auto;" /></p></ul><h1 class="fs-7 fw-400 lh-tight" id="citations"> <a href="#citations" class="anchor-heading" aria-labelledby="citations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Citations</h1><div style="border: 2px solid #b19cd9; border-radius: 15px; padding: 15px; margin: 20px; text-align: left; font-size: 1em; line-height: 1.5; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);"> <span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">GO analysis</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Huang, D. W.</b>, Sherman, B. T., Tan, Q., Kir, J., Liu, D., Bryant, D., ... &amp; Lempicki, R. A. (2007). DAVID Bioinformatics Resources: expanded annotation database and novel algorithms to better extract biology from large gene lists. <i>Nucleic acids research</i>, 35(suppl_2), W169-W175. <a href="https://doi.org/10.1093/nar/gkm415" target="_blank">DOI</a></span><li style="margin-bottom: 10px;"> <span><b>Gene Ontology Consortium.</b> (2001). Creating the gene ontology resource: design and implementation. <i>Genome research</i>, 11(8), 1425-1433. <a href="https://doi.org/10.1101/gr.180801" target="_blank">DOI</a></span></ol><span class="label label-purple" style="display: inline-block; margin-bottom: 10px; position: relative; left: -10px;">enrichGO/GSEA</span><ol style="padding-left: 40px;"><li style="margin-bottom: 10px;"> <span><b>Yu, G.</b>, Wang, L. G., Han, Y., &amp; He, Q. Y. (2012). clusterProfiler: an R package for comparing biological themes among gene clusters. <i>Omics: a journal of integrative biology</i>, 16(5), 284-287. <a href="https://doi.org/10.1089/omi.2011.0118" target="_blank">DOI</a></span><li style="margin-bottom: 10px;"> <span><b>Subramanian, A.</b>, Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., ... &amp; Mesirov, J. P. (2005). Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. <i>Proceedings of the National Academy of Sciences*, 102(43), 15545-15550. <a href="https://doi.org/10.1073/pnas.0506580102" target="_blank">DOI</a>&lt;/span&gt; &lt;/li&gt; &lt;/ol&gt; &lt;/div&gt; </i></span></ol></div></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
