<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"><link rel="stylesheet" href="/assets/css/just-the-docs-head-nav.css" id="jtd-head-nav-stylesheet"><link rel="stylesheet" href="/assets/css/custom.css"><style id="jtd-nav-activation"> .site-nav > ul.nav-list:first-child > li:not(:nth-child(2)) > a, .site-nav > ul.nav-list:first-child > li > ul > li > a, .site-nav > ul.nav-list:first-child > li > ul > li > ul > li > a { background-image: none; } .site-nav > ul.nav-list:not(:first-child) a, .site-nav li.external a { background-image: none; } .site-nav > ul.nav-list:first-child > li:nth-child(2) > a { font-weight: 600; text-decoration: none; }.site-nav > ul.nav-list:first-child > li:nth-child(2) > button svg { transform: rotate(-90deg); }.site-nav > ul.nav-list:first-child > li.nav-list-item:nth-child(2) > ul.nav-list { display: block; }</style><script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="icon" href="/favicon.ico" type="image/x-icon"><title>About | Sequencing Portfolio</title><meta name="generator" content="Jekyll v4.3.3" /><meta property="og:title" content="About" /><meta property="og:locale" content="en_US" /><meta name="description" content="Sequencing Techniques Portfolio" /><meta property="og:description" content="Sequencing Techniques Portfolio" /><link rel="canonical" href="http://localhost:4000/docs/About/" /><meta property="og:url" content="http://localhost:4000/docs/About/" /><meta property="og:site_name" content="Sequencing Portfolio" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="About" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Sequencing Techniques Portfolio","headline":"About","url":"http://localhost:4000/docs/About/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header" role="banner"> <a href="/" class="site-title lh-tight"> Sequencing Portfolio </a> <button id="menu-button" class="site-button btn-reset" aria-label="Toggle menu" aria-pressed="false"> <svg viewBox="0 0 24 24" class="icon" aria-hidden="true"><use xlink:href="#svg-menu"></use></svg> </button></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/docs/About/" class="nav-list-link">About</a><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Process category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/DataProcess" class="nav-list-link">Data Process</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/DataProcess/FastQC_manual/" class="nav-list-link">Basal Quality Control with FastQC</a><li class="nav-list-item"><a href="/docs/DataProcess/Trimmmomatic_manual/" class="nav-list-link">Trimmed reads with Trimmomatic</a><li class="nav-list-item"><a href="/docs/DataProcess/HISAT2_MAUNAL/" class="nav-list-link">Mapping reads with HISAT2 for RNA-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/Bowtie2_MANUAL/" class="nav-list-link">Mapping reads with Bowtie2 for ChIP-seq</a><li class="nav-list-item"><a href="/docs/DataProcess/SAMtools_MANUAL/" class="nav-list-link">File Conversion and Management with SAMtools</a><li class="nav-list-item"><a href="/docs/DataProcess/htseqCount_MANUAL/" class="nav-list-link">Feature counts with htseqCount</a><li class="nav-list-item"><a href="/docs/DataProcess/Genozip_MANUAL/" class="nav-list-link">File storage with Genozip</a><li class="nav-list-item"><a href="/docs/DataProcess/Tr/" class="nav-list-link">Basal setting and troubleshooting</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - RNA-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/RNASeq" class="nav-list-link">Data Analysis - RNA-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/RNASeq/DEG_edgeR_MANUAL/" class="nav-list-link">Normalization and DEG analysis with edgeR</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/" class="nav-list-link">DEG - Correlation plot</a><li class="nav-list-item"><a href="/docs/RNASeq/DEG_plotting_MANUAL/" class="nav-list-link">DEG - Visualization</a><li class="nav-list-item"><a href="/docs/RNASeq/GO_KEGG_MANUAL/" class="nav-list-link">DEG - Gene ontology and KEGG pathway analysis</a><li class="nav-list-item"><a href="/docs/RNASeq/HEATMAP_MANUAL/" class="nav-list-link">DEG - Heatmap</a></ul><li class="nav-list-item"><button class="nav-list-expander btn-reset" aria-label="toggle items in Data Analysis - ChIP-seq category" aria-pressed="false"> <svg viewBox="0 0 24 24" aria-hidden="true"><use xlink:href="#svg-arrow-right"></use></svg> </button><a href="/docs/ChIPSeq" class="nav-list-link">Data Analysis - ChIP-seq</a><ul class="nav-list"><li class="nav-list-item"><a href="/docs/ChIPSeq/deepTools_MANUAL/" class="nav-list-link">Exploring Sequencing Data with deepTools</a><li class="nav-list-item"><a href="/docs/ChIPSeq/ngsplot_MANUAL/" class="nav-list-link">Efficient Analysis and Visualization with ngs.plot</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MACS_MANUAL/" class="nav-list-link">Peak Detection - MACS</a><li class="nav-list-item"><a href="/docs/ChIPSeq/SICER2_MANUAL/" class="nav-list-link">Peak Detection - SICER2</a><li class="nav-list-item"><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/" class="nav-list-link">Exploring ChIP-seq Data with ChIPseeker</a><li class="nav-list-item"><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/" class="nav-list-link">Motif Enrichment Analysis</a></ul><li class="nav-list-item"><a href="/docs/CV/" class="nav-list-link">CV</a></ul></nav><footer class="site-footer"> This site uses <a href="https://github.com/just-the-docs/just-the-docs">Just the Docs</a>, a documentation theme for Jekyll.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search" role="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Sequencing Portfolio" aria-label="Search Sequencing Portfolio" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/seq-jchoi-bio/seq-jchoi-bio.github.io" class="site-button" > Janghyun's Portfolio </a></ul></nav></div><div class="main-content-wrap"><div id="main-content" class="main-content"><main><h1 class="fs-9" id="about"> <a href="#about" class="anchor-heading" aria-labelledby="about"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> About</h1><p class="fs-6 fw-300">Sequencing Techniques Portfolio</p><hr /><p>This portfolio showcases a variety of approaches to sequencing technology, focusing on the data processing and analysis of different samples. The purpose of this repository is not to present a single tool, but to display the range of techniques and coding strategies I have employed in my research. Each directory or section of this repository corresponds to different sequencing projects, including but not limited to RNA-seq and ChIP-seq analyses. These projects demonstrate how I handle diverse data sets and apply unique analysis methodologies. This portfolio is broadly organized into two main categories: <strong>Data Processing</strong> and <strong>Data Analysis (RNA/ChIP-seq)</strong>.</p><hr /><h2 id="table-of-contents"> <a href="#table-of-contents" class="anchor-heading" aria-labelledby="table-of-contents"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Table of Contents</h2><details> <summary style="padding-left: 20px;"> <b style="font-size: 18px;">Data Processing</b> </summary><ul style="padding-left: 60px;"><li><a href="/docs/DataProcess/FastQC_manual/">Basal QC with FastQC</a><li><a href="/docs/DataProcess/Trimmmomatic_manual/">Trimmed reads with Trimmomatic</a><li><a href="/docs/DataProcess/HISAT2_MAUNAL/">Mapping reads with HISAT2 for RNA-seq</a><li><a href="/docs/DataProcess/Bowtie2_MANUAL/">Mapping reads with Bowtie2 for ChIP-seq</a><li><a href="/docs/DataProcess/SAMtools_MANUAL/">File Conversion and Management with SAMtools</a><li><a href="/docs/DataProcess/htseqCount_MANUAL/">Feature counts with htseqCount</a><li><a href="/docs/DataProcess/Genozip_MANUAL/">File storage with Genozip</a><li><a href="/docs/DataProcess/Tr/">Basal setting and troubleshooting</a></ul></details> <details> <summary style="padding-left: 20px;"> <b style="font-size: 18px;">Data Analysis - RNA-seq</b> </summary><ul style="padding-left: 60px;"><li><a href="/docs/RNASeq/DEG_edgeR_MANUAL/">Normalization and DEG analysis with edgeR</a><li><a href="/docs/RNASeq/DEG_CorPlot_MANUAL/">DEG: Correlation plot</a><li><a href="/docs/RNASeq/DEG_plotting_MANUAL/">DEG: Visualization</a><li><a href="/docs/RNASeq/GO_KEGG_MANUAL/">DEG: Gene ontology and KEGG pathway analysis</a><li><a href="/docs/RNASeq/HEATMAP_MANUAL/">DEG: Heatmap</a></ul></details> <details> <summary style="padding-left: 20px;"> <b style="font-size: 18px;">Data Analysis - ChIP-seq</b> </summary><ul style="padding-left: 60px;"><li><a href="/docs/ChIPSeq/deepTools_MANUAL/">Exploring Sequencing Data with deepTools</a><li><a href="/docs/ChIPSeq/ngsplot_MANUAL/">Efficient Analysis and Visualization of ChIP-seq Data with ngs.plot</a><li><a href="/docs/ChIPSeq/MACS_MANUAL/">Peak detection: MACS3</a><li><a href="/docs/ChIPSeq/SICER2_MANUAL/">Peak detection: SICER2</a><li><a href="/docs/ChIPSeq/PeakAnnotation_MANUAL/">Peak Annotation with ChIPseeker</a><li><a href="/docs/ChIPSeq/MotifAnalysis_MANUAL/">Motif analysis with HOMER and MEME</a></ul></details> <details> <summary style="padding-left: 20px;"> <b style="font-size: 18px;">Data Science - Animal Locomotion Analysis</b> <span class="label label-purple">WIP</span> </summary><ul style="padding-left: 60px;"><li>Tracking Animal Movement with R<li>Converting Tracking Data to Coordinates and Calculating Physical Properties Using Python<li>Plotting the Results with R</ul></details><hr /><h2 id="featured-projects"> <a href="#featured-projects" class="anchor-heading" aria-labelledby="featured-projects"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Featured Projects</h2><h3 class="fs-5 fw-300 text-purple-000 lh-default" id="project-1-identification-of-gene-expression-and-chromatin-dynamics-changes-during-oxidative-stress"> <a href="#project-1-identification-of-gene-expression-and-chromatin-dynamics-changes-during-oxidative-stress" class="anchor-heading" aria-labelledby="project-1-identification-of-gene-expression-and-chromatin-dynamics-changes-during-oxidative-stress"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <strong>Project 1</strong>: Identification of Gene Expression and Chromatin Dynamics Changes During Oxidative Stress</h3><ul><li><strong>Description</strong>: In this project, we explored the effects of hydrogen peroxide-induced oxidative stress in a normal human keratinocyte cell line and three knockdown cell lines. We identified crucial changes in gene expression and chromatin dynamics of H3K4/H3K27me3, shedding light on the cellular mechanisms that activate NRF2-defense programs under oxidative stress.<li><strong>Techniques Used</strong>: RNA-seq and ChIP-seq<li><strong>Key Finding</strong>: <strong>Choi J</strong> &amp; Lee H. <em>Free Radic. Biol. Med.</em> 217 (2024): 48-59. <a href="https://pubmed.ncbi.nlm.nih.gov/38527695/">Pubmed</a><li><strong>Dataset</strong>: Access the detailed datasets from GEO SuperSeries <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE250128">GSE250128</a>, including SubSeries GSE250126 and GSE250127.</ul><h3 class="fs-5 fw-300 text-purple-000 lh-default" id="project-2-establishing-the-chromatin-landscape-regulating-stemness-in-mesenchymal-stem-cells"> <a href="#project-2-establishing-the-chromatin-landscape-regulating-stemness-in-mesenchymal-stem-cells" class="anchor-heading" aria-labelledby="project-2-establishing-the-chromatin-landscape-regulating-stemness-in-mesenchymal-stem-cells"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <strong>Project 2</strong>: Establishing the Chromatin Landscape Regulating Stemness in Mesenchymal Stem Cells</h3><ul><li><strong>Description</strong>: We examined gene expression and chromatin alterations in normal and knockdown mesenchymal stem cell (MSC) lines exposed to differentiation stimuli. Our advanced bioinformatics analysis showed the critical roles of lineage-specific transcription factors and histone-modifying enzymes in maintaining MSC stemness and identity.<li><strong>Techniques Used</strong>: RNA-seq and ChIP-seq<li><strong>Key Findings</strong>: <strong>Choi J</strong> &amp; Lee H. <em>J. Biol. Chem.</em> 299.10 (2023). <a href="https://pubmed.ncbi.nlm.nih.gov/37633334/">Pubmed</a>, <strong>Choi J</strong> &amp; Lee H. <em>Sci. Rep.</em> 10.1 (2020): 3050. <a href="https://pubmed.ncbi.nlm.nih.gov/32080306/">Pubmed</a><li><strong>Datasets</strong>: Further explore our findings with GEO SuperSeries <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE227538">GSE227538</a> and <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE131369">GSE131369</a>.</ul><h3 class="fs-5 fw-300 text-purple-000 lh-default" id="project-3-comparative-transcriptome-analysis-in-young-vs-middle-aged-zebrafish-brains"> <a href="#project-3-comparative-transcriptome-analysis-in-young-vs-middle-aged-zebrafish-brains" class="anchor-heading" aria-labelledby="project-3-comparative-transcriptome-analysis-in-young-vs-middle-aged-zebrafish-brains"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <strong>Project 3</strong>: Comparative Transcriptome Analysis in Young vs. Middle-Aged Zebrafish Brains</h3><ul><li><strong>Description</strong>: We compared whole brain transcriptomes from 3-month-old and 12-month-old zebrafish to identify significant gene expression changes related to aging.<li><strong>Techniques Used</strong>: RNA-seq<li><strong>Status</strong>: The research findings are currently being prepared for submission. Upon acceptance of the paper, the dataset will be released to the public.</ul><h3 class="fs-5 fw-300 text-purple-000 lh-default" id="project-4-robust-link-between-gene-expression-and-histone-dynamics-during-short-day-treatment-in-plants"> <a href="#project-4-robust-link-between-gene-expression-and-histone-dynamics-during-short-day-treatment-in-plants" class="anchor-heading" aria-labelledby="project-4-robust-link-between-gene-expression-and-histone-dynamics-during-short-day-treatment-in-plants"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> <strong>Project 4</strong>: Robust Link Between Gene Expression and Histone Dynamics During Short-Day Treatment in Plants</h3><ul><li><strong>Description</strong>: We studied gene expression and histone acetylation in normal and knockout plants under short-day (SD) treatment to reveal the epigenetic mechanisms that trigger flowering.<li><strong>Techniques Used</strong>: RNA-seq and ChIP-seq<li><strong>Status</strong>: The research findings are currently being prepared for submission. Upon acceptance of the paper, the dataset will be released to the public.</ul><h2 id="implications-of-this-portfolio"> <a href="#implications-of-this-portfolio" class="anchor-heading" aria-labelledby="implications-of-this-portfolio"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Implications of This Portfolio</h2><p>This portfolio not only showcases a range of sophisticated sequencing techniques but also serves as a testament to my commitment to rigorous data analysis and innovative problem-solving in the field of genomics. The methodologies and strategies demonstrated here reflect my ability to adapt and optimize processes to meet the challenges presented by diverse datasets.</p><p><strong>For newcomers to sequencing,</strong> this portfolio will act as a valuable learning resource. It offers practical examples and workflows that clarify RNA-seq and ChIP-seq analysis complexities. Sharing these insights allows me to support the broader community and help others master the latest sequencing technologies.</p><p><strong>Furthermore,</strong> my work highlights the importance of transparency and reproducibility in scientific research. Each project within this repository is documented in a way that not only facilitates understanding but also enables others to replicate the results, fostering a culture of open science.</p><p><strong>In conclusion,</strong> through this portfolio, I aim to demonstrate not just the technical skills necessary for high-quality genomic analysis but also my enthusiasm for advancing the field and educating others. I believe that the tools and techniques provided here will be invaluable for anyone looking to deepen their understanding of sequencing and its applications in modern biology.</p><h2 id="about-this-portfolio"> <a href="#about-this-portfolio" class="anchor-heading" aria-labelledby="about-this-portfolio"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> About This portfolio</h2><ul><li><strong>Content License</strong> <br /> <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><img src="https://img.shields.io/badge/License-CC%20BY--NC--SA%204.0-green?labelColor=black" alt="CC BY-NC-SA 4.0" /></a> <a href="https://github.com/seq-jchoi-bio/seq-portfolio"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a><br /> Copyright © 2020-2024 Janghyun Choi, Licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a><li><strong>Design License</strong> <br /> <a href="https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt"><img src="https://img.shields.io/badge/License-MIT-yellow.svg?labelColor=black" alt="License: MIT" /></a> <a href="https://github.com/just-the-docs/just-the-docs"><img src="https://img.shields.io/badge/GitHub-181717?style=flat&amp;logo=github&amp;logoColor=white" alt="gith" /></a><br /> Copyright © 2017-2024 Patrick Marsceill, Distributed under an <a href="https://opensource.org/licenses/MIT">MIT license</a>.</ul></main><hr><footer><p><a href="#top" id="back-to-top">Back to top</a></p><p class="text-small text-grey-dk-100 mb-0">Content: Copyright &copy; 2020-2024 Janghyun Choi. Licensed under <a href='https://creativecommons.org/licenses/by-nc-sa/4.0/'>CC BY-NC-SA 4.0.</a><br> Design: Copyright &copy; 2017-2024 Patrick Marsceill. Distributed by an <a href='https://github.com/just-the-docs/just-the-docs/tree/main/LICENSE.txt'>MIT license.</a> <a href='https://www.netlify.com/'>This site is powered by Netlify.</a></p><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div><script src="https://cdn.jsdelivr.net/npm/mermaid@9.1.6/dist/mermaid.min.js"></script> <script> var config = {} ; mermaid.initialize(config); window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid')); </script>
